// pipedrive/dist/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ActivitiesApi", {
  enumerable: true,
  get: function get() {
    return _ActivitiesApi["default"];
  }
});
Object.defineProperty(exports, "ActivityDistributionData", {
  enumerable: true,
  get: function get() {
    return _ActivityDistributionData["default"];
  }
});
Object.defineProperty(exports, "ActivityDistributionDataActivityDistribution", {
  enumerable: true,
  get: function get() {
    return _ActivityDistributionDataActivityDistribution["default"];
  }
});
Object.defineProperty(exports, "ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID", {
  enumerable: true,
  get: function get() {
    return _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID["default"];
  }
});
Object.defineProperty(exports, "ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities", {
  enumerable: true,
  get: function get() {
    return _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities["default"];
  }
});
Object.defineProperty(exports, "ActivityDistributionDataWithAdditionalData", {
  enumerable: true,
  get: function get() {
    return _ActivityDistributionDataWithAdditionalData["default"];
  }
});
Object.defineProperty(exports, "ActivityFieldsApi", {
  enumerable: true,
  get: function get() {
    return _ActivityFieldsApi["default"];
  }
});
Object.defineProperty(exports, "ActivityInfo", {
  enumerable: true,
  get: function get() {
    return _ActivityInfo["default"];
  }
});
Object.defineProperty(exports, "ActivityObjectFragment", {
  enumerable: true,
  get: function get() {
    return _ActivityObjectFragment["default"];
  }
});
Object.defineProperty(exports, "ActivityPostObject", {
  enumerable: true,
  get: function get() {
    return _ActivityPostObject["default"];
  }
});
Object.defineProperty(exports, "ActivityPostObjectAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityPostObjectAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityPutObject", {
  enumerable: true,
  get: function get() {
    return _ActivityPutObject["default"];
  }
});
Object.defineProperty(exports, "ActivityPutObjectAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityPutObjectAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityRecordAdditionalData", {
  enumerable: true,
  get: function get() {
    return _ActivityRecordAdditionalData["default"];
  }
});
Object.defineProperty(exports, "ActivityResponseObject", {
  enumerable: true,
  get: function get() {
    return _ActivityResponseObject["default"];
  }
});
Object.defineProperty(exports, "ActivityResponseObjectAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityResponseObjectAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeBulkDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeBulkDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeBulkDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeBulkDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeBulkDeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeBulkDeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeCreateUpdateDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeCreateUpdateDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeCreateUpdateDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeCreateUpdateDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeListResponse", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeListResponse["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeListResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeListResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityTypeObjectResponse", {
  enumerable: true,
  get: function get() {
    return _ActivityTypeObjectResponse["default"];
  }
});
Object.defineProperty(exports, "ActivityTypesApi", {
  enumerable: true,
  get: function get() {
    return _ActivityTypesApi["default"];
  }
});
Object.defineProperty(exports, "AddActivityResponse200", {
  enumerable: true,
  get: function get() {
    return _AddActivityResponse["default"];
  }
});
Object.defineProperty(exports, "AddActivityResponse200RelatedObjects", {
  enumerable: true,
  get: function get() {
    return _AddActivityResponse200RelatedObjects["default"];
  }
});
Object.defineProperty(exports, "AddDealFollowerRequest", {
  enumerable: true,
  get: function get() {
    return _AddDealFollowerRequest["default"];
  }
});
Object.defineProperty(exports, "AddDealParticipantRequest", {
  enumerable: true,
  get: function get() {
    return _AddDealParticipantRequest["default"];
  }
});
Object.defineProperty(exports, "AddFile", {
  enumerable: true,
  get: function get() {
    return _AddFile["default"];
  }
});
Object.defineProperty(exports, "AddFilterRequest", {
  enumerable: true,
  get: function get() {
    return _AddFilterRequest["default"];
  }
});
Object.defineProperty(exports, "AddFollowerToPersonResponse", {
  enumerable: true,
  get: function get() {
    return _AddFollowerToPersonResponse["default"];
  }
});
Object.defineProperty(exports, "AddFollowerToPersonResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _AddFollowerToPersonResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "AddFollowerToPersonResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _AddFollowerToPersonResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "AddLeadLabelRequest", {
  enumerable: true,
  get: function get() {
    return _AddLeadLabelRequest["default"];
  }
});
Object.defineProperty(exports, "AddLeadRequest", {
  enumerable: true,
  get: function get() {
    return _AddLeadRequest["default"];
  }
});
Object.defineProperty(exports, "AddNewPipeline", {
  enumerable: true,
  get: function get() {
    return _AddNewPipeline["default"];
  }
});
Object.defineProperty(exports, "AddNewPipelineAllOf", {
  enumerable: true,
  get: function get() {
    return _AddNewPipelineAllOf["default"];
  }
});
Object.defineProperty(exports, "AddNoteRequest", {
  enumerable: true,
  get: function get() {
    return _AddNoteRequest["default"];
  }
});
Object.defineProperty(exports, "AddNoteRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _AddNoteRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "AddOrUpdateGoalResponse200", {
  enumerable: true,
  get: function get() {
    return _AddOrUpdateGoalResponse["default"];
  }
});
Object.defineProperty(exports, "AddOrUpdateLeadLabelResponse200", {
  enumerable: true,
  get: function get() {
    return _AddOrUpdateLeadLabelResponse["default"];
  }
});
Object.defineProperty(exports, "AddOrganizationFollowerRequest", {
  enumerable: true,
  get: function get() {
    return _AddOrganizationFollowerRequest["default"];
  }
});
Object.defineProperty(exports, "AddOrganizationRelationshipRequest", {
  enumerable: true,
  get: function get() {
    return _AddOrganizationRelationshipRequest["default"];
  }
});
Object.defineProperty(exports, "AddPersonFollowerRequest", {
  enumerable: true,
  get: function get() {
    return _AddPersonFollowerRequest["default"];
  }
});
Object.defineProperty(exports, "AddPersonPictureResponse", {
  enumerable: true,
  get: function get() {
    return _AddPersonPictureResponse["default"];
  }
});
Object.defineProperty(exports, "AddPersonPictureResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _AddPersonPictureResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "AddPersonResponse", {
  enumerable: true,
  get: function get() {
    return _AddPersonResponse["default"];
  }
});
Object.defineProperty(exports, "AddPersonResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _AddPersonResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "AddProductAttachmentDetails", {
  enumerable: true,
  get: function get() {
    return _AddProductAttachmentDetails["default"];
  }
});
Object.defineProperty(exports, "AddProductAttachmentDetailsAllOf", {
  enumerable: true,
  get: function get() {
    return _AddProductAttachmentDetailsAllOf["default"];
  }
});
Object.defineProperty(exports, "AddProductFollowerRequest", {
  enumerable: true,
  get: function get() {
    return _AddProductFollowerRequest["default"];
  }
});
Object.defineProperty(exports, "AddProductRequestBody", {
  enumerable: true,
  get: function get() {
    return _AddProductRequestBody["default"];
  }
});
Object.defineProperty(exports, "AddTeamUserRequest", {
  enumerable: true,
  get: function get() {
    return _AddTeamUserRequest["default"];
  }
});
Object.defineProperty(exports, "AddWebhookRequest", {
  enumerable: true,
  get: function get() {
    return _AddWebhookRequest["default"];
  }
});
Object.defineProperty(exports, "AddedDealFollower", {
  enumerable: true,
  get: function get() {
    return _AddedDealFollower["default"];
  }
});
Object.defineProperty(exports, "AddedDealFollowerData", {
  enumerable: true,
  get: function get() {
    return _AddedDealFollowerData["default"];
  }
});
Object.defineProperty(exports, "AdditionalBaseOrganizationItemInfo", {
  enumerable: true,
  get: function get() {
    return _AdditionalBaseOrganizationItemInfo["default"];
  }
});
Object.defineProperty(exports, "AdditionalData", {
  enumerable: true,
  get: function get() {
    return _AdditionalData["default"];
  }
});
Object.defineProperty(exports, "AdditionalDataWithPagination", {
  enumerable: true,
  get: function get() {
    return _AdditionalDataWithPagination["default"];
  }
});
Object.defineProperty(exports, "AdditionalDataWithPaginationDetails", {
  enumerable: true,
  get: function get() {
    return _AdditionalDataWithPaginationDetails["default"];
  }
});
Object.defineProperty(exports, "AdditionalMergePersonInfo", {
  enumerable: true,
  get: function get() {
    return _AdditionalMergePersonInfo["default"];
  }
});
Object.defineProperty(exports, "AdditionalPersonInfo", {
  enumerable: true,
  get: function get() {
    return _AdditionalPersonInfo["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationRelationshipsGetResponse", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationRelationshipsGetResponse["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationRelationshipsGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationRelationshipsGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationRelationshipsGetResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationsGetResponse", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationsGetResponse["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationsGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationsGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "AllOrganizationsGetResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _AllOrganizationsGetResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ArrayPrices", {
  enumerable: true,
  get: function get() {
    return _ArrayPrices["default"];
  }
});
Object.defineProperty(exports, "Assignee", {
  enumerable: true,
  get: function get() {
    return _Assignee["default"];
  }
});
Object.defineProperty(exports, "BaseComment", {
  enumerable: true,
  get: function get() {
    return _BaseComment["default"];
  }
});
Object.defineProperty(exports, "BaseCurrency", {
  enumerable: true,
  get: function get() {
    return _BaseCurrency["default"];
  }
});
Object.defineProperty(exports, "BaseDeal", {
  enumerable: true,
  get: function get() {
    return _BaseDeal["default"];
  }
});
Object.defineProperty(exports, "BaseFollowerItem", {
  enumerable: true,
  get: function get() {
    return _BaseFollowerItem["default"];
  }
});
Object.defineProperty(exports, "BaseMailThread", {
  enumerable: true,
  get: function get() {
    return _BaseMailThread["default"];
  }
});
Object.defineProperty(exports, "BaseMailThreadAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseMailThreadAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseMailThreadAllOfParties", {
  enumerable: true,
  get: function get() {
    return _BaseMailThreadAllOfParties["default"];
  }
});
Object.defineProperty(exports, "BaseMailThreadMessages", {
  enumerable: true,
  get: function get() {
    return _BaseMailThreadMessages["default"];
  }
});
Object.defineProperty(exports, "BaseMailThreadMessagesAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseMailThreadMessagesAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseNote", {
  enumerable: true,
  get: function get() {
    return _BaseNote["default"];
  }
});
Object.defineProperty(exports, "BaseNoteDealTitle", {
  enumerable: true,
  get: function get() {
    return _BaseNoteDealTitle["default"];
  }
});
Object.defineProperty(exports, "BaseNoteOrganization", {
  enumerable: true,
  get: function get() {
    return _BaseNoteOrganization["default"];
  }
});
Object.defineProperty(exports, "BaseNotePerson", {
  enumerable: true,
  get: function get() {
    return _BaseNotePerson["default"];
  }
});
Object.defineProperty(exports, "BaseOrganizationItem", {
  enumerable: true,
  get: function get() {
    return _BaseOrganizationItem["default"];
  }
});
Object.defineProperty(exports, "BaseOrganizationItemFields", {
  enumerable: true,
  get: function get() {
    return _BaseOrganizationItemFields["default"];
  }
});
Object.defineProperty(exports, "BaseOrganizationItemWithEditNameFlag", {
  enumerable: true,
  get: function get() {
    return _BaseOrganizationItemWithEditNameFlag["default"];
  }
});
Object.defineProperty(exports, "BaseOrganizationItemWithEditNameFlagAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseOrganizationItemWithEditNameFlagAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseOrganizationRelationshipItem", {
  enumerable: true,
  get: function get() {
    return _BaseOrganizationRelationshipItem["default"];
  }
});
Object.defineProperty(exports, "BasePersonItem", {
  enumerable: true,
  get: function get() {
    return _BasePersonItem["default"];
  }
});
Object.defineProperty(exports, "BasePersonItemEmail", {
  enumerable: true,
  get: function get() {
    return _BasePersonItemEmail["default"];
  }
});
Object.defineProperty(exports, "BasePersonItemPhone", {
  enumerable: true,
  get: function get() {
    return _BasePersonItemPhone["default"];
  }
});
Object.defineProperty(exports, "BasePipeline", {
  enumerable: true,
  get: function get() {
    return _BasePipeline["default"];
  }
});
Object.defineProperty(exports, "BasePipelineWithSelectedFlag", {
  enumerable: true,
  get: function get() {
    return _BasePipelineWithSelectedFlag["default"];
  }
});
Object.defineProperty(exports, "BasePipelineWithSelectedFlagAllOf", {
  enumerable: true,
  get: function get() {
    return _BasePipelineWithSelectedFlagAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseProduct", {
  enumerable: true,
  get: function get() {
    return _BaseProduct["default"];
  }
});
Object.defineProperty(exports, "BaseResponse", {
  enumerable: true,
  get: function get() {
    return _BaseResponse["default"];
  }
});
Object.defineProperty(exports, "BaseResponseWithStatus", {
  enumerable: true,
  get: function get() {
    return _BaseResponseWithStatus["default"];
  }
});
Object.defineProperty(exports, "BaseResponseWithStatusAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseResponseWithStatusAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseRole", {
  enumerable: true,
  get: function get() {
    return _BaseRole["default"];
  }
});
Object.defineProperty(exports, "BaseStage", {
  enumerable: true,
  get: function get() {
    return _BaseStage["default"];
  }
});
Object.defineProperty(exports, "BaseTeam", {
  enumerable: true,
  get: function get() {
    return _BaseTeam["default"];
  }
});
Object.defineProperty(exports, "BaseTeamAdditionalProperties", {
  enumerable: true,
  get: function get() {
    return _BaseTeamAdditionalProperties["default"];
  }
});
Object.defineProperty(exports, "BaseUser", {
  enumerable: true,
  get: function get() {
    return _BaseUser["default"];
  }
});
Object.defineProperty(exports, "BaseUserMe", {
  enumerable: true,
  get: function get() {
    return _BaseUserMe["default"];
  }
});
Object.defineProperty(exports, "BaseUserMeAllOf", {
  enumerable: true,
  get: function get() {
    return _BaseUserMeAllOf["default"];
  }
});
Object.defineProperty(exports, "BaseUserMeAllOfLanguage", {
  enumerable: true,
  get: function get() {
    return _BaseUserMeAllOfLanguage["default"];
  }
});
Object.defineProperty(exports, "BaseWebhook", {
  enumerable: true,
  get: function get() {
    return _BaseWebhook["default"];
  }
});
Object.defineProperty(exports, "BasicDeal", {
  enumerable: true,
  get: function get() {
    return _BasicDeal["default"];
  }
});
Object.defineProperty(exports, "BasicDealProduct", {
  enumerable: true,
  get: function get() {
    return _BasicDealProduct["default"];
  }
});
Object.defineProperty(exports, "BasicGoal", {
  enumerable: true,
  get: function get() {
    return _BasicGoal["default"];
  }
});
Object.defineProperty(exports, "BasicOrganization", {
  enumerable: true,
  get: function get() {
    return _BasicOrganization["default"];
  }
});
Object.defineProperty(exports, "BasicPerson", {
  enumerable: true,
  get: function get() {
    return _BasicPerson["default"];
  }
});
Object.defineProperty(exports, "BasicPersonEmail", {
  enumerable: true,
  get: function get() {
    return _BasicPersonEmail["default"];
  }
});
Object.defineProperty(exports, "BasicProductField", {
  enumerable: true,
  get: function get() {
    return _BasicProductField["default"];
  }
});
Object.defineProperty(exports, "BulkDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _BulkDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "BulkDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _BulkDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "BulkDeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _BulkDeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "CalculatedFields", {
  enumerable: true,
  get: function get() {
    return _CalculatedFields["default"];
  }
});
Object.defineProperty(exports, "CallLogObject", {
  enumerable: true,
  get: function get() {
    return _CallLogObject["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse400", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse403", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse2["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse404", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse3["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse409", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse4["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse410", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse5["default"];
  }
});
Object.defineProperty(exports, "CallLogResponse500", {
  enumerable: true,
  get: function get() {
    return _CallLogResponse6["default"];
  }
});
Object.defineProperty(exports, "CallLogsApi", {
  enumerable: true,
  get: function get() {
    return _CallLogsApi["default"];
  }
});
Object.defineProperty(exports, "CommentPostPutObject", {
  enumerable: true,
  get: function get() {
    return _CommentPostPutObject["default"];
  }
});
Object.defineProperty(exports, "CommonMailThread", {
  enumerable: true,
  get: function get() {
    return _CommonMailThread["default"];
  }
});
Object.defineProperty(exports, "CreateRemoteFileAndLinkItToItem", {
  enumerable: true,
  get: function get() {
    return _CreateRemoteFileAndLinkItToItem["default"];
  }
});
Object.defineProperty(exports, "CreateTeam", {
  enumerable: true,
  get: function get() {
    return _CreateTeam["default"];
  }
});
Object.defineProperty(exports, "Currencies", {
  enumerable: true,
  get: function get() {
    return _Currencies["default"];
  }
});
Object.defineProperty(exports, "CurrenciesApi", {
  enumerable: true,
  get: function get() {
    return _CurrenciesApi["default"];
  }
});
Object.defineProperty(exports, "DealCountAndActivityInfo", {
  enumerable: true,
  get: function get() {
    return _DealCountAndActivityInfo["default"];
  }
});
Object.defineProperty(exports, "DealFieldsApi", {
  enumerable: true,
  get: function get() {
    return _DealFieldsApi["default"];
  }
});
Object.defineProperty(exports, "DealFlowResponse", {
  enumerable: true,
  get: function get() {
    return _DealFlowResponse["default"];
  }
});
Object.defineProperty(exports, "DealFlowResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DealFlowResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DealFlowResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _DealFlowResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "DealFlowResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _DealFlowResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "DealListActivitiesResponse", {
  enumerable: true,
  get: function get() {
    return _DealListActivitiesResponse["default"];
  }
});
Object.defineProperty(exports, "DealListActivitiesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DealListActivitiesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DealListActivitiesResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _DealListActivitiesResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "DealNonStrict", {
  enumerable: true,
  get: function get() {
    return _DealNonStrict["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictModeFields", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictModeFields["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictModeFieldsCreatorUserId", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictModeFieldsCreatorUserId["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictWithDetails", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictWithDetails["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictWithDetailsAllOf", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictWithDetailsAllOf["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictWithDetailsAllOfAge", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictWithDetailsAllOfAge["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictWithDetailsAllOfAverageTimeToWon", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictWithDetailsAllOfAverageTimeToWon["default"];
  }
});
Object.defineProperty(exports, "DealNonStrictWithDetailsAllOfStayInPipelineStages", {
  enumerable: true,
  get: function get() {
    return _DealNonStrictWithDetailsAllOfStayInPipelineStages["default"];
  }
});
Object.defineProperty(exports, "DealOrganizationData", {
  enumerable: true,
  get: function get() {
    return _DealOrganizationData["default"];
  }
});
Object.defineProperty(exports, "DealOrganizationDataWithId", {
  enumerable: true,
  get: function get() {
    return _DealOrganizationDataWithId["default"];
  }
});
Object.defineProperty(exports, "DealOrganizationDataWithIdAllOf", {
  enumerable: true,
  get: function get() {
    return _DealOrganizationDataWithIdAllOf["default"];
  }
});
Object.defineProperty(exports, "DealParticipantCountInfo", {
  enumerable: true,
  get: function get() {
    return _DealParticipantCountInfo["default"];
  }
});
Object.defineProperty(exports, "DealParticipants", {
  enumerable: true,
  get: function get() {
    return _DealParticipants["default"];
  }
});
Object.defineProperty(exports, "DealPersonData", {
  enumerable: true,
  get: function get() {
    return _DealPersonData["default"];
  }
});
Object.defineProperty(exports, "DealPersonDataEmail", {
  enumerable: true,
  get: function get() {
    return _DealPersonDataEmail["default"];
  }
});
Object.defineProperty(exports, "DealPersonDataPhone", {
  enumerable: true,
  get: function get() {
    return _DealPersonDataPhone["default"];
  }
});
Object.defineProperty(exports, "DealPersonDataWithId", {
  enumerable: true,
  get: function get() {
    return _DealPersonDataWithId["default"];
  }
});
Object.defineProperty(exports, "DealPersonDataWithIdAllOf", {
  enumerable: true,
  get: function get() {
    return _DealPersonDataWithIdAllOf["default"];
  }
});
Object.defineProperty(exports, "DealSearchItem", {
  enumerable: true,
  get: function get() {
    return _DealSearchItem["default"];
  }
});
Object.defineProperty(exports, "DealSearchItemItem", {
  enumerable: true,
  get: function get() {
    return _DealSearchItemItem["default"];
  }
});
Object.defineProperty(exports, "DealSearchItemItemOrganization", {
  enumerable: true,
  get: function get() {
    return _DealSearchItemItemOrganization["default"];
  }
});
Object.defineProperty(exports, "DealSearchItemItemOwner", {
  enumerable: true,
  get: function get() {
    return _DealSearchItemItemOwner["default"];
  }
});
Object.defineProperty(exports, "DealSearchItemItemPerson", {
  enumerable: true,
  get: function get() {
    return _DealSearchItemItemPerson["default"];
  }
});
Object.defineProperty(exports, "DealSearchItemItemStage", {
  enumerable: true,
  get: function get() {
    return _DealSearchItemItemStage["default"];
  }
});
Object.defineProperty(exports, "DealSearchResponse", {
  enumerable: true,
  get: function get() {
    return _DealSearchResponse["default"];
  }
});
Object.defineProperty(exports, "DealSearchResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DealSearchResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DealSearchResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _DealSearchResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "DealStrict", {
  enumerable: true,
  get: function get() {
    return _DealStrict["default"];
  }
});
Object.defineProperty(exports, "DealStrictModeFields", {
  enumerable: true,
  get: function get() {
    return _DealStrictModeFields["default"];
  }
});
Object.defineProperty(exports, "DealStrictWithMergeId", {
  enumerable: true,
  get: function get() {
    return _DealStrictWithMergeId["default"];
  }
});
Object.defineProperty(exports, "DealStrictWithMergeIdAllOf", {
  enumerable: true,
  get: function get() {
    return _DealStrictWithMergeIdAllOf["default"];
  }
});
Object.defineProperty(exports, "DealSummary", {
  enumerable: true,
  get: function get() {
    return _DealSummary["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerCurrency", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerCurrency["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerCurrencyFull", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerCurrencyFull["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerCurrencyFullCURRENCYID", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerCurrencyFullCURRENCYID["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerStages", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerStages["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerStagesSTAGEID", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerStagesSTAGEID["default"];
  }
});
Object.defineProperty(exports, "DealSummaryPerStagesSTAGEIDCURRENCYID", {
  enumerable: true,
  get: function get() {
    return _DealSummaryPerStagesSTAGEIDCURRENCYID["default"];
  }
});
Object.defineProperty(exports, "DealTitleParameter", {
  enumerable: true,
  get: function get() {
    return _DealTitleParameter["default"];
  }
});
Object.defineProperty(exports, "DealUserData", {
  enumerable: true,
  get: function get() {
    return _DealUserData["default"];
  }
});
Object.defineProperty(exports, "DealUserDataWithId", {
  enumerable: true,
  get: function get() {
    return _DealUserDataWithId["default"];
  }
});
Object.defineProperty(exports, "DealUserDataWithIdAllOf", {
  enumerable: true,
  get: function get() {
    return _DealUserDataWithIdAllOf["default"];
  }
});
Object.defineProperty(exports, "DealsApi", {
  enumerable: true,
  get: function get() {
    return _DealsApi["default"];
  }
});
Object.defineProperty(exports, "DealsCountAndActivityInfo", {
  enumerable: true,
  get: function get() {
    return _DealsCountAndActivityInfo["default"];
  }
});
Object.defineProperty(exports, "DealsCountInfo", {
  enumerable: true,
  get: function get() {
    return _DealsCountInfo["default"];
  }
});
Object.defineProperty(exports, "DealsMovementsInfo", {
  enumerable: true,
  get: function get() {
    return _DealsMovementsInfo["default"];
  }
});
Object.defineProperty(exports, "DealsMovementsInfoFormattedValues", {
  enumerable: true,
  get: function get() {
    return _DealsMovementsInfoFormattedValues["default"];
  }
});
Object.defineProperty(exports, "DealsMovementsInfoValues", {
  enumerable: true,
  get: function get() {
    return _DealsMovementsInfoValues["default"];
  }
});
Object.defineProperty(exports, "DeleteActivitiesResponse200", {
  enumerable: true,
  get: function get() {
    return _DeleteActivitiesResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteActivitiesResponse200Data", {
  enumerable: true,
  get: function get() {
    return _DeleteActivitiesResponse200Data["default"];
  }
});
Object.defineProperty(exports, "DeleteActivityResponse200", {
  enumerable: true,
  get: function get() {
    return _DeleteActivityResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteActivityResponse200Data", {
  enumerable: true,
  get: function get() {
    return _DeleteActivityResponse200Data["default"];
  }
});
Object.defineProperty(exports, "DeleteComment", {
  enumerable: true,
  get: function get() {
    return _DeleteComment["default"];
  }
});
Object.defineProperty(exports, "DeleteDeal", {
  enumerable: true,
  get: function get() {
    return _DeleteDeal["default"];
  }
});
Object.defineProperty(exports, "DeleteDealData", {
  enumerable: true,
  get: function get() {
    return _DeleteDealData["default"];
  }
});
Object.defineProperty(exports, "DeleteDealFollower", {
  enumerable: true,
  get: function get() {
    return _DeleteDealFollower["default"];
  }
});
Object.defineProperty(exports, "DeleteDealFollowerData", {
  enumerable: true,
  get: function get() {
    return _DeleteDealFollowerData["default"];
  }
});
Object.defineProperty(exports, "DeleteDealParticipant", {
  enumerable: true,
  get: function get() {
    return _DeleteDealParticipant["default"];
  }
});
Object.defineProperty(exports, "DeleteDealParticipantData", {
  enumerable: true,
  get: function get() {
    return _DeleteDealParticipantData["default"];
  }
});
Object.defineProperty(exports, "DeleteDealProduct", {
  enumerable: true,
  get: function get() {
    return _DeleteDealProduct["default"];
  }
});
Object.defineProperty(exports, "DeleteDealProductData", {
  enumerable: true,
  get: function get() {
    return _DeleteDealProductData["default"];
  }
});
Object.defineProperty(exports, "DeleteFile", {
  enumerable: true,
  get: function get() {
    return _DeleteFile["default"];
  }
});
Object.defineProperty(exports, "DeleteFileData", {
  enumerable: true,
  get: function get() {
    return _DeleteFileData["default"];
  }
});
Object.defineProperty(exports, "DeleteGoalResponse200", {
  enumerable: true,
  get: function get() {
    return _DeleteGoalResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteMultipleDeals", {
  enumerable: true,
  get: function get() {
    return _DeleteMultipleDeals["default"];
  }
});
Object.defineProperty(exports, "DeleteMultipleDealsData", {
  enumerable: true,
  get: function get() {
    return _DeleteMultipleDealsData["default"];
  }
});
Object.defineProperty(exports, "DeleteMultipleProductFieldsResponse", {
  enumerable: true,
  get: function get() {
    return _DeleteMultipleProductFieldsResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteMultipleProductFieldsResponseData", {
  enumerable: true,
  get: function get() {
    return _DeleteMultipleProductFieldsResponseData["default"];
  }
});
Object.defineProperty(exports, "DeleteNote", {
  enumerable: true,
  get: function get() {
    return _DeleteNote["default"];
  }
});
Object.defineProperty(exports, "DeletePersonResponse", {
  enumerable: true,
  get: function get() {
    return _DeletePersonResponse["default"];
  }
});
Object.defineProperty(exports, "DeletePersonResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DeletePersonResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DeletePersonResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _DeletePersonResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "DeletePersonsInBulkResponse", {
  enumerable: true,
  get: function get() {
    return _DeletePersonsInBulkResponse["default"];
  }
});
Object.defineProperty(exports, "DeletePersonsInBulkResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DeletePersonsInBulkResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DeletePersonsInBulkResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _DeletePersonsInBulkResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "DeletePipelineResponse200", {
  enumerable: true,
  get: function get() {
    return _DeletePipelineResponse["default"];
  }
});
Object.defineProperty(exports, "DeletePipelineResponse200Data", {
  enumerable: true,
  get: function get() {
    return _DeletePipelineResponse200Data["default"];
  }
});
Object.defineProperty(exports, "DeleteProductFieldResponse", {
  enumerable: true,
  get: function get() {
    return _DeleteProductFieldResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteProductFieldResponseData", {
  enumerable: true,
  get: function get() {
    return _DeleteProductFieldResponseData["default"];
  }
});
Object.defineProperty(exports, "DeleteProductFollowerResponse", {
  enumerable: true,
  get: function get() {
    return _DeleteProductFollowerResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteProductFollowerResponseData", {
  enumerable: true,
  get: function get() {
    return _DeleteProductFollowerResponseData["default"];
  }
});
Object.defineProperty(exports, "DeleteProductResponse", {
  enumerable: true,
  get: function get() {
    return _DeleteProductResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteProductResponseData", {
  enumerable: true,
  get: function get() {
    return _DeleteProductResponseData["default"];
  }
});
Object.defineProperty(exports, "DeleteResponse", {
  enumerable: true,
  get: function get() {
    return _DeleteResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _DeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "DeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _DeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "DeleteRole", {
  enumerable: true,
  get: function get() {
    return _DeleteRole["default"];
  }
});
Object.defineProperty(exports, "DeleteRoleAllOf", {
  enumerable: true,
  get: function get() {
    return _DeleteRoleAllOf["default"];
  }
});
Object.defineProperty(exports, "DeleteRoleAllOfData", {
  enumerable: true,
  get: function get() {
    return _DeleteRoleAllOfData["default"];
  }
});
Object.defineProperty(exports, "DeleteRoleAssignment", {
  enumerable: true,
  get: function get() {
    return _DeleteRoleAssignment["default"];
  }
});
Object.defineProperty(exports, "DeleteRoleAssignmentAllOf", {
  enumerable: true,
  get: function get() {
    return _DeleteRoleAssignmentAllOf["default"];
  }
});
Object.defineProperty(exports, "DeleteRoleAssignmentAllOfData", {
  enumerable: true,
  get: function get() {
    return _DeleteRoleAssignmentAllOfData["default"];
  }
});
Object.defineProperty(exports, "DeleteStageResponse200", {
  enumerable: true,
  get: function get() {
    return _DeleteStageResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteStageResponse200Data", {
  enumerable: true,
  get: function get() {
    return _DeleteStageResponse200Data["default"];
  }
});
Object.defineProperty(exports, "DeleteStagesResponse200", {
  enumerable: true,
  get: function get() {
    return _DeleteStagesResponse["default"];
  }
});
Object.defineProperty(exports, "DeleteStagesResponse200Data", {
  enumerable: true,
  get: function get() {
    return _DeleteStagesResponse200Data["default"];
  }
});
Object.defineProperty(exports, "DeleteTeamUserRequest", {
  enumerable: true,
  get: function get() {
    return _DeleteTeamUserRequest["default"];
  }
});
Object.defineProperty(exports, "Duration", {
  enumerable: true,
  get: function get() {
    return _Duration["default"];
  }
});
Object.defineProperty(exports, "EditPipeline", {
  enumerable: true,
  get: function get() {
    return _EditPipeline["default"];
  }
});
Object.defineProperty(exports, "EditPipelineAllOf", {
  enumerable: true,
  get: function get() {
    return _EditPipelineAllOf["default"];
  }
});
Object.defineProperty(exports, "EmailInfo", {
  enumerable: true,
  get: function get() {
    return _EmailInfo["default"];
  }
});
Object.defineProperty(exports, "ExpectedOutcome", {
  enumerable: true,
  get: function get() {
    return _ExpectedOutcome["default"];
  }
});
Object.defineProperty(exports, "FailResponse", {
  enumerable: true,
  get: function get() {
    return _FailResponse["default"];
  }
});
Object.defineProperty(exports, "Field", {
  enumerable: true,
  get: function get() {
    return _Field["default"];
  }
});
Object.defineProperty(exports, "FieldCreateRequest", {
  enumerable: true,
  get: function get() {
    return _FieldCreateRequest["default"];
  }
});
Object.defineProperty(exports, "FieldCreateRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _FieldCreateRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "FieldResponse", {
  enumerable: true,
  get: function get() {
    return _FieldResponse["default"];
  }
});
Object.defineProperty(exports, "FieldResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FieldResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FieldType", {
  enumerable: true,
  get: function get() {
    return _FieldType["default"];
  }
});
Object.defineProperty(exports, "FieldTypeAsString", {
  enumerable: true,
  get: function get() {
    return _FieldTypeAsString["default"];
  }
});
Object.defineProperty(exports, "FieldUpdateRequest", {
  enumerable: true,
  get: function get() {
    return _FieldUpdateRequest["default"];
  }
});
Object.defineProperty(exports, "FieldsResponse", {
  enumerable: true,
  get: function get() {
    return _FieldsResponse["default"];
  }
});
Object.defineProperty(exports, "FieldsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FieldsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FileData", {
  enumerable: true,
  get: function get() {
    return _FileData["default"];
  }
});
Object.defineProperty(exports, "FileItem", {
  enumerable: true,
  get: function get() {
    return _FileItem["default"];
  }
});
Object.defineProperty(exports, "FilesApi", {
  enumerable: true,
  get: function get() {
    return _FilesApi["default"];
  }
});
Object.defineProperty(exports, "FilterGetItem", {
  enumerable: true,
  get: function get() {
    return _FilterGetItem["default"];
  }
});
Object.defineProperty(exports, "FilterType", {
  enumerable: true,
  get: function get() {
    return _FilterType["default"];
  }
});
Object.defineProperty(exports, "FiltersApi", {
  enumerable: true,
  get: function get() {
    return _FiltersApi["default"];
  }
});
Object.defineProperty(exports, "FiltersBulkDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _FiltersBulkDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "FiltersBulkDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FiltersBulkDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FiltersBulkDeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _FiltersBulkDeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "FiltersBulkGetResponse", {
  enumerable: true,
  get: function get() {
    return _FiltersBulkGetResponse["default"];
  }
});
Object.defineProperty(exports, "FiltersBulkGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FiltersBulkGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FiltersDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _FiltersDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "FiltersDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FiltersDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FiltersDeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _FiltersDeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "FiltersGetResponse", {
  enumerable: true,
  get: function get() {
    return _FiltersGetResponse["default"];
  }
});
Object.defineProperty(exports, "FiltersGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FiltersGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FiltersPostResponse", {
  enumerable: true,
  get: function get() {
    return _FiltersPostResponse["default"];
  }
});
Object.defineProperty(exports, "FiltersPostResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _FiltersPostResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "FiltersPostResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _FiltersPostResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "FindGoalResponse", {
  enumerable: true,
  get: function get() {
    return _FindGoalResponse["default"];
  }
});
Object.defineProperty(exports, "FollowerData", {
  enumerable: true,
  get: function get() {
    return _FollowerData["default"];
  }
});
Object.defineProperty(exports, "FollowerDataWithID", {
  enumerable: true,
  get: function get() {
    return _FollowerDataWithID["default"];
  }
});
Object.defineProperty(exports, "FollowerDataWithIDAllOf", {
  enumerable: true,
  get: function get() {
    return _FollowerDataWithIDAllOf["default"];
  }
});
Object.defineProperty(exports, "FullRole", {
  enumerable: true,
  get: function get() {
    return _FullRole["default"];
  }
});
Object.defineProperty(exports, "FullRoleAllOf", {
  enumerable: true,
  get: function get() {
    return _FullRoleAllOf["default"];
  }
});
Object.defineProperty(exports, "GetActivitiesResponse200", {
  enumerable: true,
  get: function get() {
    return _GetActivitiesResponse["default"];
  }
});
Object.defineProperty(exports, "GetActivitiesResponse200RelatedObjects", {
  enumerable: true,
  get: function get() {
    return _GetActivitiesResponse200RelatedObjects["default"];
  }
});
Object.defineProperty(exports, "GetActivityResponse200", {
  enumerable: true,
  get: function get() {
    return _GetActivityResponse["default"];
  }
});
Object.defineProperty(exports, "GetAddProductAttachementDetails", {
  enumerable: true,
  get: function get() {
    return _GetAddProductAttachementDetails["default"];
  }
});
Object.defineProperty(exports, "GetAddUpdateStage", {
  enumerable: true,
  get: function get() {
    return _GetAddUpdateStage["default"];
  }
});
Object.defineProperty(exports, "GetAddedDeal", {
  enumerable: true,
  get: function get() {
    return _GetAddedDeal["default"];
  }
});
Object.defineProperty(exports, "GetAddedDealAdditionalData", {
  enumerable: true,
  get: function get() {
    return _GetAddedDealAdditionalData["default"];
  }
});
Object.defineProperty(exports, "GetAllFiles", {
  enumerable: true,
  get: function get() {
    return _GetAllFiles["default"];
  }
});
Object.defineProperty(exports, "GetAllPersonsResponse", {
  enumerable: true,
  get: function get() {
    return _GetAllPersonsResponse["default"];
  }
});
Object.defineProperty(exports, "GetAllPersonsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _GetAllPersonsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "GetAllPipelines", {
  enumerable: true,
  get: function get() {
    return _GetAllPipelines["default"];
  }
});
Object.defineProperty(exports, "GetAllPipelinesAllOf", {
  enumerable: true,
  get: function get() {
    return _GetAllPipelinesAllOf["default"];
  }
});
Object.defineProperty(exports, "GetAllProductFieldsResponse", {
  enumerable: true,
  get: function get() {
    return _GetAllProductFieldsResponse["default"];
  }
});
Object.defineProperty(exports, "GetComments", {
  enumerable: true,
  get: function get() {
    return _GetComments["default"];
  }
});
Object.defineProperty(exports, "GetDeal", {
  enumerable: true,
  get: function get() {
    return _GetDeal["default"];
  }
});
Object.defineProperty(exports, "GetDealAdditionalData", {
  enumerable: true,
  get: function get() {
    return _GetDealAdditionalData["default"];
  }
});
Object.defineProperty(exports, "GetDeals", {
  enumerable: true,
  get: function get() {
    return _GetDeals["default"];
  }
});
Object.defineProperty(exports, "GetDealsConversionRatesInPipeline", {
  enumerable: true,
  get: function get() {
    return _GetDealsConversionRatesInPipeline["default"];
  }
});
Object.defineProperty(exports, "GetDealsConversionRatesInPipelineAllOf", {
  enumerable: true,
  get: function get() {
    return _GetDealsConversionRatesInPipelineAllOf["default"];
  }
});
Object.defineProperty(exports, "GetDealsConversionRatesInPipelineAllOfData", {
  enumerable: true,
  get: function get() {
    return _GetDealsConversionRatesInPipelineAllOfData["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipeline", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipeline["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipelineAllOf", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipelineAllOf["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipelineAllOfData", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipelineAllOfData["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipelineAllOfDataAverageAgeInDays", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipelineAllOfDataAverageAgeInDays["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages["default"];
  }
});
Object.defineProperty(exports, "GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages", {
  enumerable: true,
  get: function get() {
    return _GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages["default"];
  }
});
Object.defineProperty(exports, "GetDealsRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _GetDealsRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "GetDealsSummary", {
  enumerable: true,
  get: function get() {
    return _GetDealsSummary["default"];
  }
});
Object.defineProperty(exports, "GetDealsSummaryData", {
  enumerable: true,
  get: function get() {
    return _GetDealsSummaryData["default"];
  }
});
Object.defineProperty(exports, "GetDealsSummaryDataValuesTotal", {
  enumerable: true,
  get: function get() {
    return _GetDealsSummaryDataValuesTotal["default"];
  }
});
Object.defineProperty(exports, "GetDealsSummaryDataWeightedValuesTotal", {
  enumerable: true,
  get: function get() {
    return _GetDealsSummaryDataWeightedValuesTotal["default"];
  }
});
Object.defineProperty(exports, "GetDealsTimeline", {
  enumerable: true,
  get: function get() {
    return _GetDealsTimeline["default"];
  }
});
Object.defineProperty(exports, "GetDealsTimelineData", {
  enumerable: true,
  get: function get() {
    return _GetDealsTimelineData["default"];
  }
});
Object.defineProperty(exports, "GetDealsTimelineDataTotals", {
  enumerable: true,
  get: function get() {
    return _GetDealsTimelineDataTotals["default"];
  }
});
Object.defineProperty(exports, "GetDuplicatedDeal", {
  enumerable: true,
  get: function get() {
    return _GetDuplicatedDeal["default"];
  }
});
Object.defineProperty(exports, "GetGoalResultResponse200", {
  enumerable: true,
  get: function get() {
    return _GetGoalResultResponse["default"];
  }
});
Object.defineProperty(exports, "GetGoalsResponse200", {
  enumerable: true,
  get: function get() {
    return _GetGoalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetLeadLabelsResponse200", {
  enumerable: true,
  get: function get() {
    return _GetLeadLabelsResponse["default"];
  }
});
Object.defineProperty(exports, "GetLeadSourcesResponse200", {
  enumerable: true,
  get: function get() {
    return _GetLeadSourcesResponse["default"];
  }
});
Object.defineProperty(exports, "GetLeadSourcesResponse200Data", {
  enumerable: true,
  get: function get() {
    return _GetLeadSourcesResponse200Data["default"];
  }
});
Object.defineProperty(exports, "GetLeadsResponse200", {
  enumerable: true,
  get: function get() {
    return _GetLeadsResponse["default"];
  }
});
Object.defineProperty(exports, "GetMergedDeal", {
  enumerable: true,
  get: function get() {
    return _GetMergedDeal["default"];
  }
});
Object.defineProperty(exports, "GetNotes", {
  enumerable: true,
  get: function get() {
    return _GetNotes["default"];
  }
});
Object.defineProperty(exports, "GetOneFile", {
  enumerable: true,
  get: function get() {
    return _GetOneFile["default"];
  }
});
Object.defineProperty(exports, "GetOnePipeline", {
  enumerable: true,
  get: function get() {
    return _GetOnePipeline["default"];
  }
});
Object.defineProperty(exports, "GetOnePipelineAllOf", {
  enumerable: true,
  get: function get() {
    return _GetOnePipelineAllOf["default"];
  }
});
Object.defineProperty(exports, "GetOneStage", {
  enumerable: true,
  get: function get() {
    return _GetOneStage["default"];
  }
});
Object.defineProperty(exports, "GetPersonDetailsResponse", {
  enumerable: true,
  get: function get() {
    return _GetPersonDetailsResponse["default"];
  }
});
Object.defineProperty(exports, "GetPersonDetailsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _GetPersonDetailsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "GetPersonDetailsResponseAllOfAdditionalData", {
  enumerable: true,
  get: function get() {
    return _GetPersonDetailsResponseAllOfAdditionalData["default"];
  }
});
Object.defineProperty(exports, "GetProductAttachementDetails", {
  enumerable: true,
  get: function get() {
    return _GetProductAttachementDetails["default"];
  }
});
Object.defineProperty(exports, "GetProductFieldResponse", {
  enumerable: true,
  get: function get() {
    return _GetProductFieldResponse["default"];
  }
});
Object.defineProperty(exports, "GetRecents", {
  enumerable: true,
  get: function get() {
    return _GetRecents["default"];
  }
});
Object.defineProperty(exports, "GetRecentsAdditionalData", {
  enumerable: true,
  get: function get() {
    return _GetRecentsAdditionalData["default"];
  }
});
Object.defineProperty(exports, "GetRole", {
  enumerable: true,
  get: function get() {
    return _GetRole["default"];
  }
});
Object.defineProperty(exports, "GetRoleAllOf", {
  enumerable: true,
  get: function get() {
    return _GetRoleAllOf["default"];
  }
});
Object.defineProperty(exports, "GetRoleAllOfAdditionalData", {
  enumerable: true,
  get: function get() {
    return _GetRoleAllOfAdditionalData["default"];
  }
});
Object.defineProperty(exports, "GetRoleAssignments", {
  enumerable: true,
  get: function get() {
    return _GetRoleAssignments["default"];
  }
});
Object.defineProperty(exports, "GetRoleAssignmentsAllOf", {
  enumerable: true,
  get: function get() {
    return _GetRoleAssignmentsAllOf["default"];
  }
});
Object.defineProperty(exports, "GetRoleSettings", {
  enumerable: true,
  get: function get() {
    return _GetRoleSettings["default"];
  }
});
Object.defineProperty(exports, "GetRoleSettingsAllOf", {
  enumerable: true,
  get: function get() {
    return _GetRoleSettingsAllOf["default"];
  }
});
Object.defineProperty(exports, "GetRoleSubroles", {
  enumerable: true,
  get: function get() {
    return _GetRoleSubroles["default"];
  }
});
Object.defineProperty(exports, "GetRoleSubrolesAllOf", {
  enumerable: true,
  get: function get() {
    return _GetRoleSubrolesAllOf["default"];
  }
});
Object.defineProperty(exports, "GetRoles", {
  enumerable: true,
  get: function get() {
    return _GetRoles["default"];
  }
});
Object.defineProperty(exports, "GetRolesAllOf", {
  enumerable: true,
  get: function get() {
    return _GetRolesAllOf["default"];
  }
});
Object.defineProperty(exports, "GetStageDeals", {
  enumerable: true,
  get: function get() {
    return _GetStageDeals["default"];
  }
});
Object.defineProperty(exports, "GetStages", {
  enumerable: true,
  get: function get() {
    return _GetStages["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageBaseResponse", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageBaseResponse["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageData", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageData["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageDelete", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageDelete["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageDeleteAllOf", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageDeleteAllOf["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageGet", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageGet["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageGetAllOf", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageGetAllOf["default"];
  }
});
Object.defineProperty(exports, "GlobalMessageUserData", {
  enumerable: true,
  get: function get() {
    return _GlobalMessageUserData["default"];
  }
});
Object.defineProperty(exports, "GlobalMessagesApi", {
  enumerable: true,
  get: function get() {
    return _GlobalMessagesApi["default"];
  }
});
Object.defineProperty(exports, "GoalResults", {
  enumerable: true,
  get: function get() {
    return _GoalResults["default"];
  }
});
Object.defineProperty(exports, "GoalType", {
  enumerable: true,
  get: function get() {
    return _GoalType["default"];
  }
});
Object.defineProperty(exports, "GoalsApi", {
  enumerable: true,
  get: function get() {
    return _GoalsApi["default"];
  }
});
Object.defineProperty(exports, "GoalsResponseComponent", {
  enumerable: true,
  get: function get() {
    return _GoalsResponseComponent["default"];
  }
});
Object.defineProperty(exports, "IconKey", {
  enumerable: true,
  get: function get() {
    return _IconKey["default"];
  }
});
Object.defineProperty(exports, "ItemSearchAdditionalData", {
  enumerable: true,
  get: function get() {
    return _ItemSearchAdditionalData["default"];
  }
});
Object.defineProperty(exports, "ItemSearchAdditionalDataPagination", {
  enumerable: true,
  get: function get() {
    return _ItemSearchAdditionalDataPagination["default"];
  }
});
Object.defineProperty(exports, "ItemSearchApi", {
  enumerable: true,
  get: function get() {
    return _ItemSearchApi["default"];
  }
});
Object.defineProperty(exports, "ItemSearchFieldResponse", {
  enumerable: true,
  get: function get() {
    return _ItemSearchFieldResponse["default"];
  }
});
Object.defineProperty(exports, "ItemSearchFieldResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ItemSearchFieldResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ItemSearchFieldResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ItemSearchFieldResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ItemSearchItem", {
  enumerable: true,
  get: function get() {
    return _ItemSearchItem["default"];
  }
});
Object.defineProperty(exports, "ItemSearchResponse", {
  enumerable: true,
  get: function get() {
    return _ItemSearchResponse["default"];
  }
});
Object.defineProperty(exports, "ItemSearchResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ItemSearchResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ItemSearchResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ItemSearchResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "LeadIdResponse200", {
  enumerable: true,
  get: function get() {
    return _LeadIdResponse["default"];
  }
});
Object.defineProperty(exports, "LeadIdResponse200Data", {
  enumerable: true,
  get: function get() {
    return _LeadIdResponse200Data["default"];
  }
});
Object.defineProperty(exports, "LeadLabelColor", {
  enumerable: true,
  get: function get() {
    return _LeadLabelColor["default"];
  }
});
Object.defineProperty(exports, "LeadLabelResponse", {
  enumerable: true,
  get: function get() {
    return _LeadLabelResponse["default"];
  }
});
Object.defineProperty(exports, "LeadLabelsApi", {
  enumerable: true,
  get: function get() {
    return _LeadLabelsApi["default"];
  }
});
Object.defineProperty(exports, "LeadResponse", {
  enumerable: true,
  get: function get() {
    return _LeadResponse["default"];
  }
});
Object.defineProperty(exports, "LeadResponse404", {
  enumerable: true,
  get: function get() {
    return _LeadResponse2["default"];
  }
});
Object.defineProperty(exports, "LeadSourcesApi", {
  enumerable: true,
  get: function get() {
    return _LeadSourcesApi["default"];
  }
});
Object.defineProperty(exports, "LeadValue", {
  enumerable: true,
  get: function get() {
    return _LeadValue["default"];
  }
});
Object.defineProperty(exports, "LeadsApi", {
  enumerable: true,
  get: function get() {
    return _LeadsApi["default"];
  }
});
Object.defineProperty(exports, "LinkRemoteFileToItem", {
  enumerable: true,
  get: function get() {
    return _LinkRemoteFileToItem["default"];
  }
});
Object.defineProperty(exports, "ListActivitiesResponse", {
  enumerable: true,
  get: function get() {
    return _ListActivitiesResponse["default"];
  }
});
Object.defineProperty(exports, "ListActivitiesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListActivitiesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListDealsResponse", {
  enumerable: true,
  get: function get() {
    return _ListDealsResponse["default"];
  }
});
Object.defineProperty(exports, "ListDealsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListDealsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListDealsResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _ListDealsResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "ListFilesResponse", {
  enumerable: true,
  get: function get() {
    return _ListFilesResponse["default"];
  }
});
Object.defineProperty(exports, "ListFilesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListFilesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListFollowersResponse", {
  enumerable: true,
  get: function get() {
    return _ListFollowersResponse["default"];
  }
});
Object.defineProperty(exports, "ListFollowersResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListFollowersResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListFollowersResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListFollowersResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListMailMessagesResponse", {
  enumerable: true,
  get: function get() {
    return _ListMailMessagesResponse["default"];
  }
});
Object.defineProperty(exports, "ListMailMessagesResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListMailMessagesResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListMailMessagesResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListMailMessagesResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListPermittedUsersResponse", {
  enumerable: true,
  get: function get() {
    return _ListPermittedUsersResponse["default"];
  }
});
Object.defineProperty(exports, "ListPermittedUsersResponse1", {
  enumerable: true,
  get: function get() {
    return _ListPermittedUsersResponse2["default"];
  }
});
Object.defineProperty(exports, "ListPermittedUsersResponse1AllOf", {
  enumerable: true,
  get: function get() {
    return _ListPermittedUsersResponse1AllOf["default"];
  }
});
Object.defineProperty(exports, "ListPermittedUsersResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListPermittedUsersResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListPermittedUsersResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListPermittedUsersResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListPersonProductsResponse", {
  enumerable: true,
  get: function get() {
    return _ListPersonProductsResponse["default"];
  }
});
Object.defineProperty(exports, "ListPersonProductsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListPersonProductsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListPersonProductsResponseAllOfDEALID", {
  enumerable: true,
  get: function get() {
    return _ListPersonProductsResponseAllOfDEALID["default"];
  }
});
Object.defineProperty(exports, "ListPersonProductsResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListPersonProductsResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListPersonsResponse", {
  enumerable: true,
  get: function get() {
    return _ListPersonsResponse["default"];
  }
});
Object.defineProperty(exports, "ListPersonsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListPersonsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListPersonsResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _ListPersonsResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "ListProductAdditionalData", {
  enumerable: true,
  get: function get() {
    return _ListProductAdditionalData["default"];
  }
});
Object.defineProperty(exports, "ListProductAdditionalDataAllOf", {
  enumerable: true,
  get: function get() {
    return _ListProductAdditionalDataAllOf["default"];
  }
});
Object.defineProperty(exports, "ListProductFollowersResponse", {
  enumerable: true,
  get: function get() {
    return _ListProductFollowersResponse["default"];
  }
});
Object.defineProperty(exports, "ListProductFollowersResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListProductFollowersResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListProductFollowersResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListProductFollowersResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponse", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponse["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "MailMessage", {
  enumerable: true,
  get: function get() {
    return _MailMessage["default"];
  }
});
Object.defineProperty(exports, "MailMessageAllOf", {
  enumerable: true,
  get: function get() {
    return _MailMessageAllOf["default"];
  }
});
Object.defineProperty(exports, "MailMessageData", {
  enumerable: true,
  get: function get() {
    return _MailMessageData["default"];
  }
});
Object.defineProperty(exports, "MailMessageItemForList", {
  enumerable: true,
  get: function get() {
    return _MailMessageItemForList["default"];
  }
});
Object.defineProperty(exports, "MailMessageItemForListAllOf", {
  enumerable: true,
  get: function get() {
    return _MailMessageItemForListAllOf["default"];
  }
});
Object.defineProperty(exports, "MailParticipant", {
  enumerable: true,
  get: function get() {
    return _MailParticipant["default"];
  }
});
Object.defineProperty(exports, "MailServiceBaseResponse", {
  enumerable: true,
  get: function get() {
    return _MailServiceBaseResponse["default"];
  }
});
Object.defineProperty(exports, "MailThread", {
  enumerable: true,
  get: function get() {
    return _MailThread["default"];
  }
});
Object.defineProperty(exports, "MailThreadAllOf", {
  enumerable: true,
  get: function get() {
    return _MailThreadAllOf["default"];
  }
});
Object.defineProperty(exports, "MailThreadDelete", {
  enumerable: true,
  get: function get() {
    return _MailThreadDelete["default"];
  }
});
Object.defineProperty(exports, "MailThreadDeleteAllOf", {
  enumerable: true,
  get: function get() {
    return _MailThreadDeleteAllOf["default"];
  }
});
Object.defineProperty(exports, "MailThreadMessages", {
  enumerable: true,
  get: function get() {
    return _MailThreadMessages["default"];
  }
});
Object.defineProperty(exports, "MailThreadMessagesAllOf", {
  enumerable: true,
  get: function get() {
    return _MailThreadMessagesAllOf["default"];
  }
});
Object.defineProperty(exports, "MailThreadOne", {
  enumerable: true,
  get: function get() {
    return _MailThreadOne["default"];
  }
});
Object.defineProperty(exports, "MailThreadOneAllOf", {
  enumerable: true,
  get: function get() {
    return _MailThreadOneAllOf["default"];
  }
});
Object.defineProperty(exports, "MailThreadParticipant", {
  enumerable: true,
  get: function get() {
    return _MailThreadParticipant["default"];
  }
});
Object.defineProperty(exports, "MailThreadPut", {
  enumerable: true,
  get: function get() {
    return _MailThreadPut["default"];
  }
});
Object.defineProperty(exports, "MailThreadPutAllOf", {
  enumerable: true,
  get: function get() {
    return _MailThreadPutAllOf["default"];
  }
});
Object.defineProperty(exports, "MailboxApi", {
  enumerable: true,
  get: function get() {
    return _MailboxApi["default"];
  }
});
Object.defineProperty(exports, "MarketingStatus", {
  enumerable: true,
  get: function get() {
    return _MarketingStatus["default"];
  }
});
Object.defineProperty(exports, "MergeDealsRequest", {
  enumerable: true,
  get: function get() {
    return _MergeDealsRequest["default"];
  }
});
Object.defineProperty(exports, "MergeOrganizationsRequest", {
  enumerable: true,
  get: function get() {
    return _MergeOrganizationsRequest["default"];
  }
});
Object.defineProperty(exports, "MergePersonDealRelatedInfo", {
  enumerable: true,
  get: function get() {
    return _MergePersonDealRelatedInfo["default"];
  }
});
Object.defineProperty(exports, "MergePersonItem", {
  enumerable: true,
  get: function get() {
    return _MergePersonItem["default"];
  }
});
Object.defineProperty(exports, "MergePersonsRequest", {
  enumerable: true,
  get: function get() {
    return _MergePersonsRequest["default"];
  }
});
Object.defineProperty(exports, "MergePersonsResponse", {
  enumerable: true,
  get: function get() {
    return _MergePersonsResponse["default"];
  }
});
Object.defineProperty(exports, "MergePersonsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _MergePersonsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "NewDeal", {
  enumerable: true,
  get: function get() {
    return _NewDeal["default"];
  }
});
Object.defineProperty(exports, "NewDealAllOf", {
  enumerable: true,
  get: function get() {
    return _NewDealAllOf["default"];
  }
});
Object.defineProperty(exports, "NewDealProduct", {
  enumerable: true,
  get: function get() {
    return _NewDealProduct["default"];
  }
});
Object.defineProperty(exports, "NewDealProductAllOf", {
  enumerable: true,
  get: function get() {
    return _NewDealProductAllOf["default"];
  }
});
Object.defineProperty(exports, "NewFollowerResponse", {
  enumerable: true,
  get: function get() {
    return _NewFollowerResponse["default"];
  }
});
Object.defineProperty(exports, "NewFollowerResponseData", {
  enumerable: true,
  get: function get() {
    return _NewFollowerResponseData["default"];
  }
});
Object.defineProperty(exports, "NewGoal", {
  enumerable: true,
  get: function get() {
    return _NewGoal["default"];
  }
});
Object.defineProperty(exports, "NewOrganization", {
  enumerable: true,
  get: function get() {
    return _NewOrganization["default"];
  }
});
Object.defineProperty(exports, "NewOrganizationAllOf", {
  enumerable: true,
  get: function get() {
    return _NewOrganizationAllOf["default"];
  }
});
Object.defineProperty(exports, "NewPerson", {
  enumerable: true,
  get: function get() {
    return _NewPerson["default"];
  }
});
Object.defineProperty(exports, "NewPersonAllOf", {
  enumerable: true,
  get: function get() {
    return _NewPersonAllOf["default"];
  }
});
Object.defineProperty(exports, "NewProductField", {
  enumerable: true,
  get: function get() {
    return _NewProductField["default"];
  }
});
Object.defineProperty(exports, "Note", {
  enumerable: true,
  get: function get() {
    return _Note["default"];
  }
});
Object.defineProperty(exports, "NoteConnectToParams", {
  enumerable: true,
  get: function get() {
    return _NoteConnectToParams["default"];
  }
});
Object.defineProperty(exports, "NoteCreatorUser", {
  enumerable: true,
  get: function get() {
    return _NoteCreatorUser["default"];
  }
});
Object.defineProperty(exports, "NoteField", {
  enumerable: true,
  get: function get() {
    return _NoteField["default"];
  }
});
Object.defineProperty(exports, "NoteFieldsApi", {
  enumerable: true,
  get: function get() {
    return _NoteFieldsApi["default"];
  }
});
Object.defineProperty(exports, "NoteFieldsResponse", {
  enumerable: true,
  get: function get() {
    return _NoteFieldsResponse["default"];
  }
});
Object.defineProperty(exports, "NoteFieldsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _NoteFieldsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "NoteParams", {
  enumerable: true,
  get: function get() {
    return _NoteParams["default"];
  }
});
Object.defineProperty(exports, "NotesApi", {
  enumerable: true,
  get: function get() {
    return _NotesApi["default"];
  }
});
Object.defineProperty(exports, "NumberBoolean", {
  enumerable: true,
  get: function get() {
    return _NumberBoolean["default"];
  }
});
Object.defineProperty(exports, "NumberBooleanDefault0", {
  enumerable: true,
  get: function get() {
    return _NumberBooleanDefault["default"];
  }
});
Object.defineProperty(exports, "NumberBooleanDefault1", {
  enumerable: true,
  get: function get() {
    return _NumberBooleanDefault2["default"];
  }
});
Object.defineProperty(exports, "ObjectPrices", {
  enumerable: true,
  get: function get() {
    return _ObjectPrices["default"];
  }
});
Object.defineProperty(exports, "OneLeadResponse200", {
  enumerable: true,
  get: function get() {
    return _OneLeadResponse["default"];
  }
});
Object.defineProperty(exports, "OrgAndOwnerId", {
  enumerable: true,
  get: function get() {
    return _OrgAndOwnerId["default"];
  }
});
Object.defineProperty(exports, "OrganizationAddressInfo", {
  enumerable: true,
  get: function get() {
    return _OrganizationAddressInfo["default"];
  }
});
Object.defineProperty(exports, "OrganizationCountAndAddressInfo", {
  enumerable: true,
  get: function get() {
    return _OrganizationCountAndAddressInfo["default"];
  }
});
Object.defineProperty(exports, "OrganizationCountInfo", {
  enumerable: true,
  get: function get() {
    return _OrganizationCountInfo["default"];
  }
});
Object.defineProperty(exports, "OrganizationData", {
  enumerable: true,
  get: function get() {
    return _OrganizationData["default"];
  }
});
Object.defineProperty(exports, "OrganizationDataWithId", {
  enumerable: true,
  get: function get() {
    return _OrganizationDataWithId["default"];
  }
});
Object.defineProperty(exports, "OrganizationDataWithIdAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationDataWithIdAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationDataWithIdAndActiveFlag", {
  enumerable: true,
  get: function get() {
    return _OrganizationDataWithIdAndActiveFlag["default"];
  }
});
Object.defineProperty(exports, "OrganizationDataWithIdAndActiveFlagAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationDataWithIdAndActiveFlagAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationDeleteResponseData", {
  enumerable: true,
  get: function get() {
    return _OrganizationDeleteResponseData["default"];
  }
});
Object.defineProperty(exports, "OrganizationDetailsGetResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationDetailsGetResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationDetailsGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationDetailsGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationDetailsGetResponseAllOfAdditionalData", {
  enumerable: true,
  get: function get() {
    return _OrganizationDetailsGetResponseAllOfAdditionalData["default"];
  }
});
Object.defineProperty(exports, "OrganizationFieldsApi", {
  enumerable: true,
  get: function get() {
    return _OrganizationFieldsApi["default"];
  }
});
Object.defineProperty(exports, "OrganizationFlowResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationFlowResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationFlowResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationFlowResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationFlowResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _OrganizationFlowResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "OrganizationFlowResponseAllOfRelatedObjects", {
  enumerable: true,
  get: function get() {
    return _OrganizationFlowResponseAllOfRelatedObjects["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowerDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowerDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowerDeleteResponseData", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowerDeleteResponseData["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowerItem", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowerItem["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowerItemAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowerItemAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowerPostResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowerPostResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationFollowersListResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationFollowersListResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationItem", {
  enumerable: true,
  get: function get() {
    return _OrganizationItem["default"];
  }
});
Object.defineProperty(exports, "OrganizationItemAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationItemAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationPostResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationPostResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationPostResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationPostResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationship", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationship["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipDeleteResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipDeleteResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipDeleteResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipDeleteResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipDetails", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipDetails["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipGetResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipGetResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipGetResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipGetResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipPostResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipPostResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipPostResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipPostResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipUpdateResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipUpdateResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipWithCalculatedFields", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipWithCalculatedFields["default"];
  }
});
Object.defineProperty(exports, "OrganizationRelationshipsApi", {
  enumerable: true,
  get: function get() {
    return _OrganizationRelationshipsApi["default"];
  }
});
Object.defineProperty(exports, "OrganizationSearchItem", {
  enumerable: true,
  get: function get() {
    return _OrganizationSearchItem["default"];
  }
});
Object.defineProperty(exports, "OrganizationSearchItemItem", {
  enumerable: true,
  get: function get() {
    return _OrganizationSearchItemItem["default"];
  }
});
Object.defineProperty(exports, "OrganizationSearchResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationSearchResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationSearchResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationSearchResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationSearchResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _OrganizationSearchResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "OrganizationUpdateResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationUpdateResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationUpdateResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _OrganizationUpdateResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "OrganizationsApi", {
  enumerable: true,
  get: function get() {
    return _OrganizationsApi["default"];
  }
});
Object.defineProperty(exports, "OrganizationsDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationsDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationsDeleteResponseData", {
  enumerable: true,
  get: function get() {
    return _OrganizationsDeleteResponseData["default"];
  }
});
Object.defineProperty(exports, "OrganizationsMergeResponse", {
  enumerable: true,
  get: function get() {
    return _OrganizationsMergeResponse["default"];
  }
});
Object.defineProperty(exports, "OrganizationsMergeResponseData", {
  enumerable: true,
  get: function get() {
    return _OrganizationsMergeResponseData["default"];
  }
});
Object.defineProperty(exports, "Owner", {
  enumerable: true,
  get: function get() {
    return _Owner["default"];
  }
});
Object.defineProperty(exports, "OwnerAllOf", {
  enumerable: true,
  get: function get() {
    return _OwnerAllOf["default"];
  }
});
Object.defineProperty(exports, "PaginationDetails", {
  enumerable: true,
  get: function get() {
    return _PaginationDetails["default"];
  }
});
Object.defineProperty(exports, "PaginationDetailsAllOf", {
  enumerable: true,
  get: function get() {
    return _PaginationDetailsAllOf["default"];
  }
});
Object.defineProperty(exports, "Params", {
  enumerable: true,
  get: function get() {
    return _Params["default"];
  }
});
Object.defineProperty(exports, "PaymentItem", {
  enumerable: true,
  get: function get() {
    return _PaymentItem["default"];
  }
});
Object.defineProperty(exports, "PaymentsResponse", {
  enumerable: true,
  get: function get() {
    return _PaymentsResponse["default"];
  }
});
Object.defineProperty(exports, "PaymentsResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _PaymentsResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "PaymentsResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _PaymentsResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "PermissionSets", {
  enumerable: true,
  get: function get() {
    return _PermissionSets["default"];
  }
});
Object.defineProperty(exports, "PermissionSetsAllOf", {
  enumerable: true,
  get: function get() {
    return _PermissionSetsAllOf["default"];
  }
});
Object.defineProperty(exports, "PermissionSetsApi", {
  enumerable: true,
  get: function get() {
    return _PermissionSetsApi["default"];
  }
});
Object.defineProperty(exports, "PermissionSetsItem", {
  enumerable: true,
  get: function get() {
    return _PermissionSetsItem["default"];
  }
});
Object.defineProperty(exports, "PersonCountAndEmailInfo", {
  enumerable: true,
  get: function get() {
    return _PersonCountAndEmailInfo["default"];
  }
});
Object.defineProperty(exports, "PersonCountEmailDealAndActivityInfo", {
  enumerable: true,
  get: function get() {
    return _PersonCountEmailDealAndActivityInfo["default"];
  }
});
Object.defineProperty(exports, "PersonCountInfo", {
  enumerable: true,
  get: function get() {
    return _PersonCountInfo["default"];
  }
});
Object.defineProperty(exports, "PersonData", {
  enumerable: true,
  get: function get() {
    return _PersonData["default"];
  }
});
Object.defineProperty(exports, "PersonDataEmail", {
  enumerable: true,
  get: function get() {
    return _PersonDataEmail["default"];
  }
});
Object.defineProperty(exports, "PersonDataPhone", {
  enumerable: true,
  get: function get() {
    return _PersonDataPhone["default"];
  }
});
Object.defineProperty(exports, "PersonDataWithActiveFlag", {
  enumerable: true,
  get: function get() {
    return _PersonDataWithActiveFlag["default"];
  }
});
Object.defineProperty(exports, "PersonDataWithActiveFlagAllOf", {
  enumerable: true,
  get: function get() {
    return _PersonDataWithActiveFlagAllOf["default"];
  }
});
Object.defineProperty(exports, "PersonFieldsApi", {
  enumerable: true,
  get: function get() {
    return _PersonFieldsApi["default"];
  }
});
Object.defineProperty(exports, "PersonFlowResponse", {
  enumerable: true,
  get: function get() {
    return _PersonFlowResponse["default"];
  }
});
Object.defineProperty(exports, "PersonFlowResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _PersonFlowResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "PersonFlowResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _PersonFlowResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "PersonItem", {
  enumerable: true,
  get: function get() {
    return _PersonItem["default"];
  }
});
Object.defineProperty(exports, "PersonListProduct", {
  enumerable: true,
  get: function get() {
    return _PersonListProduct["default"];
  }
});
Object.defineProperty(exports, "PersonNameCountAndEmailInfo", {
  enumerable: true,
  get: function get() {
    return _PersonNameCountAndEmailInfo["default"];
  }
});
Object.defineProperty(exports, "PersonNameCountAndEmailInfoWithIds", {
  enumerable: true,
  get: function get() {
    return _PersonNameCountAndEmailInfoWithIds["default"];
  }
});
Object.defineProperty(exports, "PersonNameCountAndEmailInfoWithIdsAllOf", {
  enumerable: true,
  get: function get() {
    return _PersonNameCountAndEmailInfoWithIdsAllOf["default"];
  }
});
Object.defineProperty(exports, "PersonNameInfo", {
  enumerable: true,
  get: function get() {
    return _PersonNameInfo["default"];
  }
});
Object.defineProperty(exports, "PersonNameInfoWithOrgAndOwnerId", {
  enumerable: true,
  get: function get() {
    return _PersonNameInfoWithOrgAndOwnerId["default"];
  }
});
Object.defineProperty(exports, "PersonSearchItem", {
  enumerable: true,
  get: function get() {
    return _PersonSearchItem["default"];
  }
});
Object.defineProperty(exports, "PersonSearchItemItem", {
  enumerable: true,
  get: function get() {
    return _PersonSearchItemItem["default"];
  }
});
Object.defineProperty(exports, "PersonSearchItemItemOrganization", {
  enumerable: true,
  get: function get() {
    return _PersonSearchItemItemOrganization["default"];
  }
});
Object.defineProperty(exports, "PersonSearchItemItemOwner", {
  enumerable: true,
  get: function get() {
    return _PersonSearchItemItemOwner["default"];
  }
});
Object.defineProperty(exports, "PersonSearchResponse", {
  enumerable: true,
  get: function get() {
    return _PersonSearchResponse["default"];
  }
});
Object.defineProperty(exports, "PersonSearchResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _PersonSearchResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "PersonSearchResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _PersonSearchResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "PersonsApi", {
  enumerable: true,
  get: function get() {
    return _PersonsApi["default"];
  }
});
Object.defineProperty(exports, "PictureData", {
  enumerable: true,
  get: function get() {
    return _PictureData["default"];
  }
});
Object.defineProperty(exports, "PictureDataPictures", {
  enumerable: true,
  get: function get() {
    return _PictureDataPictures["default"];
  }
});
Object.defineProperty(exports, "PictureDataWithID", {
  enumerable: true,
  get: function get() {
    return _PictureDataWithID["default"];
  }
});
Object.defineProperty(exports, "PictureDataWithIDAllOf", {
  enumerable: true,
  get: function get() {
    return _PictureDataWithIDAllOf["default"];
  }
});
Object.defineProperty(exports, "PictureDataWithValue", {
  enumerable: true,
  get: function get() {
    return _PictureDataWithValue["default"];
  }
});
Object.defineProperty(exports, "PictureDataWithValueAllOf", {
  enumerable: true,
  get: function get() {
    return _PictureDataWithValueAllOf["default"];
  }
});
Object.defineProperty(exports, "Pipeline", {
  enumerable: true,
  get: function get() {
    return _Pipeline["default"];
  }
});
Object.defineProperty(exports, "PipelineDetails", {
  enumerable: true,
  get: function get() {
    return _PipelineDetails["default"];
  }
});
Object.defineProperty(exports, "PipelineDetailsAllOf", {
  enumerable: true,
  get: function get() {
    return _PipelineDetailsAllOf["default"];
  }
});
Object.defineProperty(exports, "PipelinesApi", {
  enumerable: true,
  get: function get() {
    return _PipelinesApi["default"];
  }
});
Object.defineProperty(exports, "PostComment", {
  enumerable: true,
  get: function get() {
    return _PostComment["default"];
  }
});
Object.defineProperty(exports, "PostDealParticipants", {
  enumerable: true,
  get: function get() {
    return _PostDealParticipants["default"];
  }
});
Object.defineProperty(exports, "PostGoalResponse", {
  enumerable: true,
  get: function get() {
    return _PostGoalResponse["default"];
  }
});
Object.defineProperty(exports, "PostNote", {
  enumerable: true,
  get: function get() {
    return _PostNote["default"];
  }
});
Object.defineProperty(exports, "PostRoleAssignment", {
  enumerable: true,
  get: function get() {
    return _PostRoleAssignment["default"];
  }
});
Object.defineProperty(exports, "PostRoleAssignmentAllOf", {
  enumerable: true,
  get: function get() {
    return _PostRoleAssignmentAllOf["default"];
  }
});
Object.defineProperty(exports, "PostRoleAssignmentAllOfData", {
  enumerable: true,
  get: function get() {
    return _PostRoleAssignmentAllOfData["default"];
  }
});
Object.defineProperty(exports, "PostRoleSettings", {
  enumerable: true,
  get: function get() {
    return _PostRoleSettings["default"];
  }
});
Object.defineProperty(exports, "PostRoleSettingsAllOf", {
  enumerable: true,
  get: function get() {
    return _PostRoleSettingsAllOf["default"];
  }
});
Object.defineProperty(exports, "PostRoleSettingsAllOfData", {
  enumerable: true,
  get: function get() {
    return _PostRoleSettingsAllOfData["default"];
  }
});
Object.defineProperty(exports, "PostRoles", {
  enumerable: true,
  get: function get() {
    return _PostRoles["default"];
  }
});
Object.defineProperty(exports, "PostRolesAllOf", {
  enumerable: true,
  get: function get() {
    return _PostRolesAllOf["default"];
  }
});
Object.defineProperty(exports, "PostRolesAllOfData", {
  enumerable: true,
  get: function get() {
    return _PostRolesAllOfData["default"];
  }
});
Object.defineProperty(exports, "ProductAttachementFields", {
  enumerable: true,
  get: function get() {
    return _ProductAttachementFields["default"];
  }
});
Object.defineProperty(exports, "ProductAttachmentDetails", {
  enumerable: true,
  get: function get() {
    return _ProductAttachmentDetails["default"];
  }
});
Object.defineProperty(exports, "ProductBaseDeal", {
  enumerable: true,
  get: function get() {
    return _ProductBaseDeal["default"];
  }
});
Object.defineProperty(exports, "ProductField", {
  enumerable: true,
  get: function get() {
    return _ProductField["default"];
  }
});
Object.defineProperty(exports, "ProductFieldAllOf", {
  enumerable: true,
  get: function get() {
    return _ProductFieldAllOf["default"];
  }
});
Object.defineProperty(exports, "ProductFieldsApi", {
  enumerable: true,
  get: function get() {
    return _ProductFieldsApi["default"];
  }
});
Object.defineProperty(exports, "ProductListItem", {
  enumerable: true,
  get: function get() {
    return _ProductListItem["default"];
  }
});
Object.defineProperty(exports, "ProductRequest", {
  enumerable: true,
  get: function get() {
    return _ProductRequest["default"];
  }
});
Object.defineProperty(exports, "ProductResponse", {
  enumerable: true,
  get: function get() {
    return _ProductResponse["default"];
  }
});
Object.defineProperty(exports, "ProductSearchItem", {
  enumerable: true,
  get: function get() {
    return _ProductSearchItem["default"];
  }
});
Object.defineProperty(exports, "ProductSearchItemItem", {
  enumerable: true,
  get: function get() {
    return _ProductSearchItemItem["default"];
  }
});
Object.defineProperty(exports, "ProductSearchItemItemOwner", {
  enumerable: true,
  get: function get() {
    return _ProductSearchItemItemOwner["default"];
  }
});
Object.defineProperty(exports, "ProductSearchResponse", {
  enumerable: true,
  get: function get() {
    return _ProductSearchResponse["default"];
  }
});
Object.defineProperty(exports, "ProductSearchResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _ProductSearchResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "ProductSearchResponseAllOfData", {
  enumerable: true,
  get: function get() {
    return _ProductSearchResponseAllOfData["default"];
  }
});
Object.defineProperty(exports, "ProductWithArrayPrices", {
  enumerable: true,
  get: function get() {
    return _ProductWithArrayPrices["default"];
  }
});
Object.defineProperty(exports, "ProductWithObjectPrices", {
  enumerable: true,
  get: function get() {
    return _ProductWithObjectPrices["default"];
  }
});
Object.defineProperty(exports, "ProductsApi", {
  enumerable: true,
  get: function get() {
    return _ProductsApi["default"];
  }
});
Object.defineProperty(exports, "ProductsResponse", {
  enumerable: true,
  get: function get() {
    return _ProductsResponse["default"];
  }
});
Object.defineProperty(exports, "PutRole", {
  enumerable: true,
  get: function get() {
    return _PutRole["default"];
  }
});
Object.defineProperty(exports, "PutRoleAllOf", {
  enumerable: true,
  get: function get() {
    return _PutRoleAllOf["default"];
  }
});
Object.defineProperty(exports, "PutRoleAllOfData", {
  enumerable: true,
  get: function get() {
    return _PutRoleAllOfData["default"];
  }
});
Object.defineProperty(exports, "RecentDataProduct", {
  enumerable: true,
  get: function get() {
    return _RecentDataProduct["default"];
  }
});
Object.defineProperty(exports, "RecentsActivity", {
  enumerable: true,
  get: function get() {
    return _RecentsActivity["default"];
  }
});
Object.defineProperty(exports, "RecentsActivityType", {
  enumerable: true,
  get: function get() {
    return _RecentsActivityType["default"];
  }
});
Object.defineProperty(exports, "RecentsApi", {
  enumerable: true,
  get: function get() {
    return _RecentsApi["default"];
  }
});
Object.defineProperty(exports, "RecentsDeal", {
  enumerable: true,
  get: function get() {
    return _RecentsDeal["default"];
  }
});
Object.defineProperty(exports, "RecentsFile", {
  enumerable: true,
  get: function get() {
    return _RecentsFile["default"];
  }
});
Object.defineProperty(exports, "RecentsFilter", {
  enumerable: true,
  get: function get() {
    return _RecentsFilter["default"];
  }
});
Object.defineProperty(exports, "RecentsNote", {
  enumerable: true,
  get: function get() {
    return _RecentsNote["default"];
  }
});
Object.defineProperty(exports, "RecentsOrganization", {
  enumerable: true,
  get: function get() {
    return _RecentsOrganization["default"];
  }
});
Object.defineProperty(exports, "RecentsPerson", {
  enumerable: true,
  get: function get() {
    return _RecentsPerson["default"];
  }
});
Object.defineProperty(exports, "RecentsPipeline", {
  enumerable: true,
  get: function get() {
    return _RecentsPipeline["default"];
  }
});
Object.defineProperty(exports, "RecentsProduct", {
  enumerable: true,
  get: function get() {
    return _RecentsProduct["default"];
  }
});
Object.defineProperty(exports, "RecentsStage", {
  enumerable: true,
  get: function get() {
    return _RecentsStage["default"];
  }
});
Object.defineProperty(exports, "RecentsUser", {
  enumerable: true,
  get: function get() {
    return _RecentsUser["default"];
  }
});
Object.defineProperty(exports, "RelatedDealData", {
  enumerable: true,
  get: function get() {
    return _RelatedDealData["default"];
  }
});
Object.defineProperty(exports, "RelatedDealDataDEALID", {
  enumerable: true,
  get: function get() {
    return _RelatedDealDataDEALID["default"];
  }
});
Object.defineProperty(exports, "RelatedFollowerData", {
  enumerable: true,
  get: function get() {
    return _RelatedFollowerData["default"];
  }
});
Object.defineProperty(exports, "RelatedOrganizationData", {
  enumerable: true,
  get: function get() {
    return _RelatedOrganizationData["default"];
  }
});
Object.defineProperty(exports, "RelatedOrganizationDataWithActiveFlag", {
  enumerable: true,
  get: function get() {
    return _RelatedOrganizationDataWithActiveFlag["default"];
  }
});
Object.defineProperty(exports, "RelatedOrganizationName", {
  enumerable: true,
  get: function get() {
    return _RelatedOrganizationName["default"];
  }
});
Object.defineProperty(exports, "RelatedPersonData", {
  enumerable: true,
  get: function get() {
    return _RelatedPersonData["default"];
  }
});
Object.defineProperty(exports, "RelatedPersonDataWithActiveFlag", {
  enumerable: true,
  get: function get() {
    return _RelatedPersonDataWithActiveFlag["default"];
  }
});
Object.defineProperty(exports, "RelatedPictureData", {
  enumerable: true,
  get: function get() {
    return _RelatedPictureData["default"];
  }
});
Object.defineProperty(exports, "RelatedUserData", {
  enumerable: true,
  get: function get() {
    return _RelatedUserData["default"];
  }
});
Object.defineProperty(exports, "RelationshipOrganizationInfoItem", {
  enumerable: true,
  get: function get() {
    return _RelationshipOrganizationInfoItem["default"];
  }
});
Object.defineProperty(exports, "RelationshipOrganizationInfoItemAllOf", {
  enumerable: true,
  get: function get() {
    return _RelationshipOrganizationInfoItemAllOf["default"];
  }
});
Object.defineProperty(exports, "RelationshipOrganizationInfoItemWithActiveFlag", {
  enumerable: true,
  get: function get() {
    return _RelationshipOrganizationInfoItemWithActiveFlag["default"];
  }
});
Object.defineProperty(exports, "RequredTitleParameter", {
  enumerable: true,
  get: function get() {
    return _RequredTitleParameter["default"];
  }
});
Object.defineProperty(exports, "ResponseCallLogObject", {
  enumerable: true,
  get: function get() {
    return _ResponseCallLogObject["default"];
  }
});
Object.defineProperty(exports, "ResponseCallLogObjectAllOf", {
  enumerable: true,
  get: function get() {
    return _ResponseCallLogObjectAllOf["default"];
  }
});
Object.defineProperty(exports, "RoleAssignment", {
  enumerable: true,
  get: function get() {
    return _RoleAssignment["default"];
  }
});
Object.defineProperty(exports, "RoleAssignmentAllOf", {
  enumerable: true,
  get: function get() {
    return _RoleAssignmentAllOf["default"];
  }
});
Object.defineProperty(exports, "RoleSettings", {
  enumerable: true,
  get: function get() {
    return _RoleSettings["default"];
  }
});
Object.defineProperty(exports, "RolesAdditionalData", {
  enumerable: true,
  get: function get() {
    return _RolesAdditionalData["default"];
  }
});
Object.defineProperty(exports, "RolesAdditionalDataPagination", {
  enumerable: true,
  get: function get() {
    return _RolesAdditionalDataPagination["default"];
  }
});
Object.defineProperty(exports, "RolesApi", {
  enumerable: true,
  get: function get() {
    return _RolesApi["default"];
  }
});
Object.defineProperty(exports, "SinglePermissionSetsItem", {
  enumerable: true,
  get: function get() {
    return _SinglePermissionSetsItem["default"];
  }
});
Object.defineProperty(exports, "SinglePermissionSetsItemAllOf", {
  enumerable: true,
  get: function get() {
    return _SinglePermissionSetsItemAllOf["default"];
  }
});
Object.defineProperty(exports, "Stage", {
  enumerable: true,
  get: function get() {
    return _Stage["default"];
  }
});
Object.defineProperty(exports, "StageConversions", {
  enumerable: true,
  get: function get() {
    return _StageConversions["default"];
  }
});
Object.defineProperty(exports, "StageDetails", {
  enumerable: true,
  get: function get() {
    return _StageDetails["default"];
  }
});
Object.defineProperty(exports, "StageWithPipelineInfo", {
  enumerable: true,
  get: function get() {
    return _StageWithPipelineInfo["default"];
  }
});
Object.defineProperty(exports, "StageWithPipelineInfoAllOf", {
  enumerable: true,
  get: function get() {
    return _StageWithPipelineInfoAllOf["default"];
  }
});
Object.defineProperty(exports, "StagesApi", {
  enumerable: true,
  get: function get() {
    return _StagesApi["default"];
  }
});
Object.defineProperty(exports, "SubRole", {
  enumerable: true,
  get: function get() {
    return _SubRole["default"];
  }
});
Object.defineProperty(exports, "SubRoleAllOf", {
  enumerable: true,
  get: function get() {
    return _SubRoleAllOf["default"];
  }
});
Object.defineProperty(exports, "SubscriptionInstallmentCreateRequest", {
  enumerable: true,
  get: function get() {
    return _SubscriptionInstallmentCreateRequest["default"];
  }
});
Object.defineProperty(exports, "SubscriptionInstallmentUpdateRequest", {
  enumerable: true,
  get: function get() {
    return _SubscriptionInstallmentUpdateRequest["default"];
  }
});
Object.defineProperty(exports, "SubscriptionItem", {
  enumerable: true,
  get: function get() {
    return _SubscriptionItem["default"];
  }
});
Object.defineProperty(exports, "SubscriptionRecurringCancelRequest", {
  enumerable: true,
  get: function get() {
    return _SubscriptionRecurringCancelRequest["default"];
  }
});
Object.defineProperty(exports, "SubscriptionRecurringCreateRequest", {
  enumerable: true,
  get: function get() {
    return _SubscriptionRecurringCreateRequest["default"];
  }
});
Object.defineProperty(exports, "SubscriptionRecurringUpdateRequest", {
  enumerable: true,
  get: function get() {
    return _SubscriptionRecurringUpdateRequest["default"];
  }
});
Object.defineProperty(exports, "SubscriptionsApi", {
  enumerable: true,
  get: function get() {
    return _SubscriptionsApi["default"];
  }
});
Object.defineProperty(exports, "SubscriptionsIdResponse", {
  enumerable: true,
  get: function get() {
    return _SubscriptionsIdResponse["default"];
  }
});
Object.defineProperty(exports, "SubscriptionsIdResponseAllOf", {
  enumerable: true,
  get: function get() {
    return _SubscriptionsIdResponseAllOf["default"];
  }
});
Object.defineProperty(exports, "Team", {
  enumerable: true,
  get: function get() {
    return _Team["default"];
  }
});
Object.defineProperty(exports, "TeamAllOf", {
  enumerable: true,
  get: function get() {
    return _TeamAllOf["default"];
  }
});
Object.defineProperty(exports, "TeamId", {
  enumerable: true,
  get: function get() {
    return _TeamId["default"];
  }
});
Object.defineProperty(exports, "Teams", {
  enumerable: true,
  get: function get() {
    return _Teams["default"];
  }
});
Object.defineProperty(exports, "TeamsAllOf", {
  enumerable: true,
  get: function get() {
    return _TeamsAllOf["default"];
  }
});
Object.defineProperty(exports, "TeamsApi", {
  enumerable: true,
  get: function get() {
    return _TeamsApi["default"];
  }
});
Object.defineProperty(exports, "Unauthorized", {
  enumerable: true,
  get: function get() {
    return _Unauthorized["default"];
  }
});
Object.defineProperty(exports, "UpdateActivityResponse200", {
  enumerable: true,
  get: function get() {
    return _UpdateActivityResponse["default"];
  }
});
Object.defineProperty(exports, "UpdateDealRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateDealRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateFile", {
  enumerable: true,
  get: function get() {
    return _UpdateFile["default"];
  }
});
Object.defineProperty(exports, "UpdateFilterRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateFilterRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateLeadLabelRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateLeadLabelRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateLeadRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateLeadRequest["default"];
  }
});
Object.defineProperty(exports, "UpdatePersonResponse", {
  enumerable: true,
  get: function get() {
    return _UpdatePersonResponse["default"];
  }
});
Object.defineProperty(exports, "UpdateProductResponse", {
  enumerable: true,
  get: function get() {
    return _UpdateProductResponse["default"];
  }
});
Object.defineProperty(exports, "UpdateStageRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateStageRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateStageRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateStageRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateTeam", {
  enumerable: true,
  get: function get() {
    return _UpdateTeam["default"];
  }
});
Object.defineProperty(exports, "UpdateTeamAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateTeamAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateTeamWithAdditionalProperties", {
  enumerable: true,
  get: function get() {
    return _UpdateTeamWithAdditionalProperties["default"];
  }
});
Object.defineProperty(exports, "User", {
  enumerable: true,
  get: function get() {
    return _User["default"];
  }
});
Object.defineProperty(exports, "UserAllOf", {
  enumerable: true,
  get: function get() {
    return _UserAllOf["default"];
  }
});
Object.defineProperty(exports, "UserAssignmentToPermissionSet", {
  enumerable: true,
  get: function get() {
    return _UserAssignmentToPermissionSet["default"];
  }
});
Object.defineProperty(exports, "UserAssignmentsToPermissionSet", {
  enumerable: true,
  get: function get() {
    return _UserAssignmentsToPermissionSet["default"];
  }
});
Object.defineProperty(exports, "UserAssignmentsToPermissionSetAllOf", {
  enumerable: true,
  get: function get() {
    return _UserAssignmentsToPermissionSetAllOf["default"];
  }
});
Object.defineProperty(exports, "UserConnections", {
  enumerable: true,
  get: function get() {
    return _UserConnections["default"];
  }
});
Object.defineProperty(exports, "UserConnectionsAllOf", {
  enumerable: true,
  get: function get() {
    return _UserConnectionsAllOf["default"];
  }
});
Object.defineProperty(exports, "UserConnectionsAllOfData", {
  enumerable: true,
  get: function get() {
    return _UserConnectionsAllOfData["default"];
  }
});
Object.defineProperty(exports, "UserConnectionsApi", {
  enumerable: true,
  get: function get() {
    return _UserConnectionsApi["default"];
  }
});
Object.defineProperty(exports, "UserData", {
  enumerable: true,
  get: function get() {
    return _UserData["default"];
  }
});
Object.defineProperty(exports, "UserDataWithId", {
  enumerable: true,
  get: function get() {
    return _UserDataWithId["default"];
  }
});
Object.defineProperty(exports, "UserIDs", {
  enumerable: true,
  get: function get() {
    return _UserIDs["default"];
  }
});
Object.defineProperty(exports, "UserIDsAllOf", {
  enumerable: true,
  get: function get() {
    return _UserIDsAllOf["default"];
  }
});
Object.defineProperty(exports, "UserMe", {
  enumerable: true,
  get: function get() {
    return _UserMe["default"];
  }
});
Object.defineProperty(exports, "UserMeAllOf", {
  enumerable: true,
  get: function get() {
    return _UserMeAllOf["default"];
  }
});
Object.defineProperty(exports, "UserPermissions", {
  enumerable: true,
  get: function get() {
    return _UserPermissions["default"];
  }
});
Object.defineProperty(exports, "UserPermissionsAllOf", {
  enumerable: true,
  get: function get() {
    return _UserPermissionsAllOf["default"];
  }
});
Object.defineProperty(exports, "UserPermissionsItem", {
  enumerable: true,
  get: function get() {
    return _UserPermissionsItem["default"];
  }
});
Object.defineProperty(exports, "UserSettings", {
  enumerable: true,
  get: function get() {
    return _UserSettings["default"];
  }
});
Object.defineProperty(exports, "UserSettingsAllOf", {
  enumerable: true,
  get: function get() {
    return _UserSettingsAllOf["default"];
  }
});
Object.defineProperty(exports, "UserSettingsApi", {
  enumerable: true,
  get: function get() {
    return _UserSettingsApi["default"];
  }
});
Object.defineProperty(exports, "UserSettingsItem", {
  enumerable: true,
  get: function get() {
    return _UserSettingsItem["default"];
  }
});
Object.defineProperty(exports, "Users", {
  enumerable: true,
  get: function get() {
    return _Users["default"];
  }
});
Object.defineProperty(exports, "UsersAllOf", {
  enumerable: true,
  get: function get() {
    return _UsersAllOf["default"];
  }
});
Object.defineProperty(exports, "UsersApi", {
  enumerable: true,
  get: function get() {
    return _UsersApi["default"];
  }
});
Object.defineProperty(exports, "VisibleTo", {
  enumerable: true,
  get: function get() {
    return _VisibleTo["default"];
  }
});
Object.defineProperty(exports, "Webhook", {
  enumerable: true,
  get: function get() {
    return _Webhook["default"];
  }
});
Object.defineProperty(exports, "WebhookAllOf", {
  enumerable: true,
  get: function get() {
    return _WebhookAllOf["default"];
  }
});
Object.defineProperty(exports, "WebhookBadRequest", {
  enumerable: true,
  get: function get() {
    return _WebhookBadRequest["default"];
  }
});
Object.defineProperty(exports, "WebhookBadRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _WebhookBadRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "Webhooks", {
  enumerable: true,
  get: function get() {
    return _Webhooks["default"];
  }
});
Object.defineProperty(exports, "WebhooksAllOf", {
  enumerable: true,
  get: function get() {
    return _WebhooksAllOf["default"];
  }
});
Object.defineProperty(exports, "WebhooksApi", {
  enumerable: true,
  get: function get() {
    return _WebhooksApi["default"];
  }
});
Object.defineProperty(exports, "WebhooksDeleteForbiddenSchema", {
  enumerable: true,
  get: function get() {
    return _WebhooksDeleteForbiddenSchema["default"];
  }
});
Object.defineProperty(exports, "WebhooksDeleteForbiddenSchemaAllOf", {
  enumerable: true,
  get: function get() {
    return _WebhooksDeleteForbiddenSchemaAllOf["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _ActivityDistributionData = _interopRequireDefault(require("./model/ActivityDistributionData"));

var _ActivityDistributionDataActivityDistribution = _interopRequireDefault(require("./model/ActivityDistributionDataActivityDistribution"));

var _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID = _interopRequireDefault(require("./model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID"));

var _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities = _interopRequireDefault(require("./model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities"));

var _ActivityDistributionDataWithAdditionalData = _interopRequireDefault(require("./model/ActivityDistributionDataWithAdditionalData"));

var _ActivityInfo = _interopRequireDefault(require("./model/ActivityInfo"));

var _ActivityObjectFragment = _interopRequireDefault(require("./model/ActivityObjectFragment"));

var _ActivityPostObject = _interopRequireDefault(require("./model/ActivityPostObject"));

var _ActivityPostObjectAllOf = _interopRequireDefault(require("./model/ActivityPostObjectAllOf"));

var _ActivityPutObject = _interopRequireDefault(require("./model/ActivityPutObject"));

var _ActivityPutObjectAllOf = _interopRequireDefault(require("./model/ActivityPutObjectAllOf"));

var _ActivityRecordAdditionalData = _interopRequireDefault(require("./model/ActivityRecordAdditionalData"));

var _ActivityResponseObject = _interopRequireDefault(require("./model/ActivityResponseObject"));

var _ActivityResponseObjectAllOf = _interopRequireDefault(require("./model/ActivityResponseObjectAllOf"));

var _ActivityTypeBulkDeleteResponse = _interopRequireDefault(require("./model/ActivityTypeBulkDeleteResponse"));

var _ActivityTypeBulkDeleteResponseAllOf = _interopRequireDefault(require("./model/ActivityTypeBulkDeleteResponseAllOf"));

var _ActivityTypeBulkDeleteResponseAllOfData = _interopRequireDefault(require("./model/ActivityTypeBulkDeleteResponseAllOfData"));

var _ActivityTypeCreateUpdateDeleteResponse = _interopRequireDefault(require("./model/ActivityTypeCreateUpdateDeleteResponse"));

var _ActivityTypeCreateUpdateDeleteResponseAllOf = _interopRequireDefault(require("./model/ActivityTypeCreateUpdateDeleteResponseAllOf"));

var _ActivityTypeListResponse = _interopRequireDefault(require("./model/ActivityTypeListResponse"));

var _ActivityTypeListResponseAllOf = _interopRequireDefault(require("./model/ActivityTypeListResponseAllOf"));

var _ActivityTypeObjectResponse = _interopRequireDefault(require("./model/ActivityTypeObjectResponse"));

var _AddActivityResponse = _interopRequireDefault(require("./model/AddActivityResponse200"));

var _AddActivityResponse200RelatedObjects = _interopRequireDefault(require("./model/AddActivityResponse200RelatedObjects"));

var _AddDealFollowerRequest = _interopRequireDefault(require("./model/AddDealFollowerRequest"));

var _AddDealParticipantRequest = _interopRequireDefault(require("./model/AddDealParticipantRequest"));

var _AddFile = _interopRequireDefault(require("./model/AddFile"));

var _AddFilterRequest = _interopRequireDefault(require("./model/AddFilterRequest"));

var _AddFollowerToPersonResponse = _interopRequireDefault(require("./model/AddFollowerToPersonResponse"));

var _AddFollowerToPersonResponseAllOf = _interopRequireDefault(require("./model/AddFollowerToPersonResponseAllOf"));

var _AddFollowerToPersonResponseAllOfData = _interopRequireDefault(require("./model/AddFollowerToPersonResponseAllOfData"));

var _AddLeadLabelRequest = _interopRequireDefault(require("./model/AddLeadLabelRequest"));

var _AddLeadRequest = _interopRequireDefault(require("./model/AddLeadRequest"));

var _AddNewPipeline = _interopRequireDefault(require("./model/AddNewPipeline"));

var _AddNewPipelineAllOf = _interopRequireDefault(require("./model/AddNewPipelineAllOf"));

var _AddNoteRequest = _interopRequireDefault(require("./model/AddNoteRequest"));

var _AddNoteRequestAllOf = _interopRequireDefault(require("./model/AddNoteRequestAllOf"));

var _AddOrUpdateGoalResponse = _interopRequireDefault(require("./model/AddOrUpdateGoalResponse200"));

var _AddOrUpdateLeadLabelResponse = _interopRequireDefault(require("./model/AddOrUpdateLeadLabelResponse200"));

var _AddOrganizationFollowerRequest = _interopRequireDefault(require("./model/AddOrganizationFollowerRequest"));

var _AddOrganizationRelationshipRequest = _interopRequireDefault(require("./model/AddOrganizationRelationshipRequest"));

var _AddPersonFollowerRequest = _interopRequireDefault(require("./model/AddPersonFollowerRequest"));

var _AddPersonPictureResponse = _interopRequireDefault(require("./model/AddPersonPictureResponse"));

var _AddPersonPictureResponseAllOf = _interopRequireDefault(require("./model/AddPersonPictureResponseAllOf"));

var _AddPersonResponse = _interopRequireDefault(require("./model/AddPersonResponse"));

var _AddPersonResponseAllOf = _interopRequireDefault(require("./model/AddPersonResponseAllOf"));

var _AddProductAttachmentDetails = _interopRequireDefault(require("./model/AddProductAttachmentDetails"));

var _AddProductAttachmentDetailsAllOf = _interopRequireDefault(require("./model/AddProductAttachmentDetailsAllOf"));

var _AddProductFollowerRequest = _interopRequireDefault(require("./model/AddProductFollowerRequest"));

var _AddProductRequestBody = _interopRequireDefault(require("./model/AddProductRequestBody"));

var _AddTeamUserRequest = _interopRequireDefault(require("./model/AddTeamUserRequest"));

var _AddWebhookRequest = _interopRequireDefault(require("./model/AddWebhookRequest"));

var _AddedDealFollower = _interopRequireDefault(require("./model/AddedDealFollower"));

var _AddedDealFollowerData = _interopRequireDefault(require("./model/AddedDealFollowerData"));

var _AdditionalBaseOrganizationItemInfo = _interopRequireDefault(require("./model/AdditionalBaseOrganizationItemInfo"));

var _AdditionalData = _interopRequireDefault(require("./model/AdditionalData"));

var _AdditionalDataWithPagination = _interopRequireDefault(require("./model/AdditionalDataWithPagination"));

var _AdditionalDataWithPaginationDetails = _interopRequireDefault(require("./model/AdditionalDataWithPaginationDetails"));

var _AdditionalMergePersonInfo = _interopRequireDefault(require("./model/AdditionalMergePersonInfo"));

var _AdditionalPersonInfo = _interopRequireDefault(require("./model/AdditionalPersonInfo"));

var _AllOrganizationRelationshipsGetResponse = _interopRequireDefault(require("./model/AllOrganizationRelationshipsGetResponse"));

var _AllOrganizationRelationshipsGetResponseAllOf = _interopRequireDefault(require("./model/AllOrganizationRelationshipsGetResponseAllOf"));

var _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects"));

var _AllOrganizationsGetResponse = _interopRequireDefault(require("./model/AllOrganizationsGetResponse"));

var _AllOrganizationsGetResponseAllOf = _interopRequireDefault(require("./model/AllOrganizationsGetResponseAllOf"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/AllOrganizationsGetResponseAllOfRelatedObjects"));

var _ArrayPrices = _interopRequireDefault(require("./model/ArrayPrices"));

var _Assignee = _interopRequireDefault(require("./model/Assignee"));

var _BaseComment = _interopRequireDefault(require("./model/BaseComment"));

var _BaseCurrency = _interopRequireDefault(require("./model/BaseCurrency"));

var _BaseDeal = _interopRequireDefault(require("./model/BaseDeal"));

var _BaseFollowerItem = _interopRequireDefault(require("./model/BaseFollowerItem"));

var _BaseMailThread = _interopRequireDefault(require("./model/BaseMailThread"));

var _BaseMailThreadAllOf = _interopRequireDefault(require("./model/BaseMailThreadAllOf"));

var _BaseMailThreadAllOfParties = _interopRequireDefault(require("./model/BaseMailThreadAllOfParties"));

var _BaseMailThreadMessages = _interopRequireDefault(require("./model/BaseMailThreadMessages"));

var _BaseMailThreadMessagesAllOf = _interopRequireDefault(require("./model/BaseMailThreadMessagesAllOf"));

var _BaseNote = _interopRequireDefault(require("./model/BaseNote"));

var _BaseNoteDealTitle = _interopRequireDefault(require("./model/BaseNoteDealTitle"));

var _BaseNoteOrganization = _interopRequireDefault(require("./model/BaseNoteOrganization"));

var _BaseNotePerson = _interopRequireDefault(require("./model/BaseNotePerson"));

var _BaseOrganizationItem = _interopRequireDefault(require("./model/BaseOrganizationItem"));

var _BaseOrganizationItemFields = _interopRequireDefault(require("./model/BaseOrganizationItemFields"));

var _BaseOrganizationItemWithEditNameFlag = _interopRequireDefault(require("./model/BaseOrganizationItemWithEditNameFlag"));

var _BaseOrganizationItemWithEditNameFlagAllOf = _interopRequireDefault(require("./model/BaseOrganizationItemWithEditNameFlagAllOf"));

var _BaseOrganizationRelationshipItem = _interopRequireDefault(require("./model/BaseOrganizationRelationshipItem"));

var _BasePersonItem = _interopRequireDefault(require("./model/BasePersonItem"));

var _BasePersonItemEmail = _interopRequireDefault(require("./model/BasePersonItemEmail"));

var _BasePersonItemPhone = _interopRequireDefault(require("./model/BasePersonItemPhone"));

var _BasePipeline = _interopRequireDefault(require("./model/BasePipeline"));

var _BasePipelineWithSelectedFlag = _interopRequireDefault(require("./model/BasePipelineWithSelectedFlag"));

var _BasePipelineWithSelectedFlagAllOf = _interopRequireDefault(require("./model/BasePipelineWithSelectedFlagAllOf"));

var _BaseProduct = _interopRequireDefault(require("./model/BaseProduct"));

var _BaseResponse = _interopRequireDefault(require("./model/BaseResponse"));

var _BaseResponseWithStatus = _interopRequireDefault(require("./model/BaseResponseWithStatus"));

var _BaseResponseWithStatusAllOf = _interopRequireDefault(require("./model/BaseResponseWithStatusAllOf"));

var _BaseRole = _interopRequireDefault(require("./model/BaseRole"));

var _BaseStage = _interopRequireDefault(require("./model/BaseStage"));

var _BaseTeam = _interopRequireDefault(require("./model/BaseTeam"));

var _BaseTeamAdditionalProperties = _interopRequireDefault(require("./model/BaseTeamAdditionalProperties"));

var _BaseUser = _interopRequireDefault(require("./model/BaseUser"));

var _BaseUserMe = _interopRequireDefault(require("./model/BaseUserMe"));

var _BaseUserMeAllOf = _interopRequireDefault(require("./model/BaseUserMeAllOf"));

var _BaseUserMeAllOfLanguage = _interopRequireDefault(require("./model/BaseUserMeAllOfLanguage"));

var _BaseWebhook = _interopRequireDefault(require("./model/BaseWebhook"));

var _BasicDeal = _interopRequireDefault(require("./model/BasicDeal"));

var _BasicDealProduct = _interopRequireDefault(require("./model/BasicDealProduct"));

var _BasicGoal = _interopRequireDefault(require("./model/BasicGoal"));

var _BasicOrganization = _interopRequireDefault(require("./model/BasicOrganization"));

var _BasicPerson = _interopRequireDefault(require("./model/BasicPerson"));

var _BasicPersonEmail = _interopRequireDefault(require("./model/BasicPersonEmail"));

var _BasicProductField = _interopRequireDefault(require("./model/BasicProductField"));

var _BulkDeleteResponse = _interopRequireDefault(require("./model/BulkDeleteResponse"));

var _BulkDeleteResponseAllOf = _interopRequireDefault(require("./model/BulkDeleteResponseAllOf"));

var _BulkDeleteResponseAllOfData = _interopRequireDefault(require("./model/BulkDeleteResponseAllOfData"));

var _CalculatedFields = _interopRequireDefault(require("./model/CalculatedFields"));

var _CallLogObject = _interopRequireDefault(require("./model/CallLogObject"));

var _CallLogResponse = _interopRequireDefault(require("./model/CallLogResponse400"));

var _CallLogResponse2 = _interopRequireDefault(require("./model/CallLogResponse403"));

var _CallLogResponse3 = _interopRequireDefault(require("./model/CallLogResponse404"));

var _CallLogResponse4 = _interopRequireDefault(require("./model/CallLogResponse409"));

var _CallLogResponse5 = _interopRequireDefault(require("./model/CallLogResponse410"));

var _CallLogResponse6 = _interopRequireDefault(require("./model/CallLogResponse500"));

var _CommentPostPutObject = _interopRequireDefault(require("./model/CommentPostPutObject"));

var _CommonMailThread = _interopRequireDefault(require("./model/CommonMailThread"));

var _CreateRemoteFileAndLinkItToItem = _interopRequireDefault(require("./model/CreateRemoteFileAndLinkItToItem"));

var _CreateTeam = _interopRequireDefault(require("./model/CreateTeam"));

var _Currencies = _interopRequireDefault(require("./model/Currencies"));

var _DealCountAndActivityInfo = _interopRequireDefault(require("./model/DealCountAndActivityInfo"));

var _DealFlowResponse = _interopRequireDefault(require("./model/DealFlowResponse"));

var _DealFlowResponseAllOf = _interopRequireDefault(require("./model/DealFlowResponseAllOf"));

var _DealFlowResponseAllOfData = _interopRequireDefault(require("./model/DealFlowResponseAllOfData"));

var _DealFlowResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/DealFlowResponseAllOfRelatedObjects"));

var _DealListActivitiesResponse = _interopRequireDefault(require("./model/DealListActivitiesResponse"));

var _DealListActivitiesResponseAllOf = _interopRequireDefault(require("./model/DealListActivitiesResponseAllOf"));

var _DealListActivitiesResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/DealListActivitiesResponseAllOfRelatedObjects"));

var _DealNonStrict = _interopRequireDefault(require("./model/DealNonStrict"));

var _DealNonStrictModeFields = _interopRequireDefault(require("./model/DealNonStrictModeFields"));

var _DealNonStrictModeFieldsCreatorUserId = _interopRequireDefault(require("./model/DealNonStrictModeFieldsCreatorUserId"));

var _DealNonStrictWithDetails = _interopRequireDefault(require("./model/DealNonStrictWithDetails"));

var _DealNonStrictWithDetailsAllOf = _interopRequireDefault(require("./model/DealNonStrictWithDetailsAllOf"));

var _DealNonStrictWithDetailsAllOfAge = _interopRequireDefault(require("./model/DealNonStrictWithDetailsAllOfAge"));

var _DealNonStrictWithDetailsAllOfAverageTimeToWon = _interopRequireDefault(require("./model/DealNonStrictWithDetailsAllOfAverageTimeToWon"));

var _DealNonStrictWithDetailsAllOfStayInPipelineStages = _interopRequireDefault(require("./model/DealNonStrictWithDetailsAllOfStayInPipelineStages"));

var _DealOrganizationData = _interopRequireDefault(require("./model/DealOrganizationData"));

var _DealOrganizationDataWithId = _interopRequireDefault(require("./model/DealOrganizationDataWithId"));

var _DealOrganizationDataWithIdAllOf = _interopRequireDefault(require("./model/DealOrganizationDataWithIdAllOf"));

var _DealParticipantCountInfo = _interopRequireDefault(require("./model/DealParticipantCountInfo"));

var _DealParticipants = _interopRequireDefault(require("./model/DealParticipants"));

var _DealPersonData = _interopRequireDefault(require("./model/DealPersonData"));

var _DealPersonDataEmail = _interopRequireDefault(require("./model/DealPersonDataEmail"));

var _DealPersonDataPhone = _interopRequireDefault(require("./model/DealPersonDataPhone"));

var _DealPersonDataWithId = _interopRequireDefault(require("./model/DealPersonDataWithId"));

var _DealPersonDataWithIdAllOf = _interopRequireDefault(require("./model/DealPersonDataWithIdAllOf"));

var _DealSearchItem = _interopRequireDefault(require("./model/DealSearchItem"));

var _DealSearchItemItem = _interopRequireDefault(require("./model/DealSearchItemItem"));

var _DealSearchItemItemOrganization = _interopRequireDefault(require("./model/DealSearchItemItemOrganization"));

var _DealSearchItemItemOwner = _interopRequireDefault(require("./model/DealSearchItemItemOwner"));

var _DealSearchItemItemPerson = _interopRequireDefault(require("./model/DealSearchItemItemPerson"));

var _DealSearchItemItemStage = _interopRequireDefault(require("./model/DealSearchItemItemStage"));

var _DealSearchResponse = _interopRequireDefault(require("./model/DealSearchResponse"));

var _DealSearchResponseAllOf = _interopRequireDefault(require("./model/DealSearchResponseAllOf"));

var _DealSearchResponseAllOfData = _interopRequireDefault(require("./model/DealSearchResponseAllOfData"));

var _DealStrict = _interopRequireDefault(require("./model/DealStrict"));

var _DealStrictModeFields = _interopRequireDefault(require("./model/DealStrictModeFields"));

var _DealStrictWithMergeId = _interopRequireDefault(require("./model/DealStrictWithMergeId"));

var _DealStrictWithMergeIdAllOf = _interopRequireDefault(require("./model/DealStrictWithMergeIdAllOf"));

var _DealSummary = _interopRequireDefault(require("./model/DealSummary"));

var _DealSummaryPerCurrency = _interopRequireDefault(require("./model/DealSummaryPerCurrency"));

var _DealSummaryPerCurrencyFull = _interopRequireDefault(require("./model/DealSummaryPerCurrencyFull"));

var _DealSummaryPerCurrencyFullCURRENCYID = _interopRequireDefault(require("./model/DealSummaryPerCurrencyFullCURRENCYID"));

var _DealSummaryPerStages = _interopRequireDefault(require("./model/DealSummaryPerStages"));

var _DealSummaryPerStagesSTAGEID = _interopRequireDefault(require("./model/DealSummaryPerStagesSTAGEID"));

var _DealSummaryPerStagesSTAGEIDCURRENCYID = _interopRequireDefault(require("./model/DealSummaryPerStagesSTAGEIDCURRENCYID"));

var _DealTitleParameter = _interopRequireDefault(require("./model/DealTitleParameter"));

var _DealUserData = _interopRequireDefault(require("./model/DealUserData"));

var _DealUserDataWithId = _interopRequireDefault(require("./model/DealUserDataWithId"));

var _DealUserDataWithIdAllOf = _interopRequireDefault(require("./model/DealUserDataWithIdAllOf"));

var _DealsCountAndActivityInfo = _interopRequireDefault(require("./model/DealsCountAndActivityInfo"));

var _DealsCountInfo = _interopRequireDefault(require("./model/DealsCountInfo"));

var _DealsMovementsInfo = _interopRequireDefault(require("./model/DealsMovementsInfo"));

var _DealsMovementsInfoFormattedValues = _interopRequireDefault(require("./model/DealsMovementsInfoFormattedValues"));

var _DealsMovementsInfoValues = _interopRequireDefault(require("./model/DealsMovementsInfoValues"));

var _DeleteActivitiesResponse = _interopRequireDefault(require("./model/DeleteActivitiesResponse200"));

var _DeleteActivitiesResponse200Data = _interopRequireDefault(require("./model/DeleteActivitiesResponse200Data"));

var _DeleteActivityResponse = _interopRequireDefault(require("./model/DeleteActivityResponse200"));

var _DeleteActivityResponse200Data = _interopRequireDefault(require("./model/DeleteActivityResponse200Data"));

var _DeleteComment = _interopRequireDefault(require("./model/DeleteComment"));

var _DeleteDeal = _interopRequireDefault(require("./model/DeleteDeal"));

var _DeleteDealData = _interopRequireDefault(require("./model/DeleteDealData"));

var _DeleteDealFollower = _interopRequireDefault(require("./model/DeleteDealFollower"));

var _DeleteDealFollowerData = _interopRequireDefault(require("./model/DeleteDealFollowerData"));

var _DeleteDealParticipant = _interopRequireDefault(require("./model/DeleteDealParticipant"));

var _DeleteDealParticipantData = _interopRequireDefault(require("./model/DeleteDealParticipantData"));

var _DeleteDealProduct = _interopRequireDefault(require("./model/DeleteDealProduct"));

var _DeleteDealProductData = _interopRequireDefault(require("./model/DeleteDealProductData"));

var _DeleteFile = _interopRequireDefault(require("./model/DeleteFile"));

var _DeleteFileData = _interopRequireDefault(require("./model/DeleteFileData"));

var _DeleteGoalResponse = _interopRequireDefault(require("./model/DeleteGoalResponse200"));

var _DeleteMultipleDeals = _interopRequireDefault(require("./model/DeleteMultipleDeals"));

var _DeleteMultipleDealsData = _interopRequireDefault(require("./model/DeleteMultipleDealsData"));

var _DeleteMultipleProductFieldsResponse = _interopRequireDefault(require("./model/DeleteMultipleProductFieldsResponse"));

var _DeleteMultipleProductFieldsResponseData = _interopRequireDefault(require("./model/DeleteMultipleProductFieldsResponseData"));

var _DeleteNote = _interopRequireDefault(require("./model/DeleteNote"));

var _DeletePersonResponse = _interopRequireDefault(require("./model/DeletePersonResponse"));

var _DeletePersonResponseAllOf = _interopRequireDefault(require("./model/DeletePersonResponseAllOf"));

var _DeletePersonResponseAllOfData = _interopRequireDefault(require("./model/DeletePersonResponseAllOfData"));

var _DeletePersonsInBulkResponse = _interopRequireDefault(require("./model/DeletePersonsInBulkResponse"));

var _DeletePersonsInBulkResponseAllOf = _interopRequireDefault(require("./model/DeletePersonsInBulkResponseAllOf"));

var _DeletePersonsInBulkResponseAllOfData = _interopRequireDefault(require("./model/DeletePersonsInBulkResponseAllOfData"));

var _DeletePipelineResponse = _interopRequireDefault(require("./model/DeletePipelineResponse200"));

var _DeletePipelineResponse200Data = _interopRequireDefault(require("./model/DeletePipelineResponse200Data"));

var _DeleteProductFieldResponse = _interopRequireDefault(require("./model/DeleteProductFieldResponse"));

var _DeleteProductFieldResponseData = _interopRequireDefault(require("./model/DeleteProductFieldResponseData"));

var _DeleteProductFollowerResponse = _interopRequireDefault(require("./model/DeleteProductFollowerResponse"));

var _DeleteProductFollowerResponseData = _interopRequireDefault(require("./model/DeleteProductFollowerResponseData"));

var _DeleteProductResponse = _interopRequireDefault(require("./model/DeleteProductResponse"));

var _DeleteProductResponseData = _interopRequireDefault(require("./model/DeleteProductResponseData"));

var _DeleteResponse = _interopRequireDefault(require("./model/DeleteResponse"));

var _DeleteResponseAllOf = _interopRequireDefault(require("./model/DeleteResponseAllOf"));

var _DeleteResponseAllOfData = _interopRequireDefault(require("./model/DeleteResponseAllOfData"));

var _DeleteRole = _interopRequireDefault(require("./model/DeleteRole"));

var _DeleteRoleAllOf = _interopRequireDefault(require("./model/DeleteRoleAllOf"));

var _DeleteRoleAllOfData = _interopRequireDefault(require("./model/DeleteRoleAllOfData"));

var _DeleteRoleAssignment = _interopRequireDefault(require("./model/DeleteRoleAssignment"));

var _DeleteRoleAssignmentAllOf = _interopRequireDefault(require("./model/DeleteRoleAssignmentAllOf"));

var _DeleteRoleAssignmentAllOfData = _interopRequireDefault(require("./model/DeleteRoleAssignmentAllOfData"));

var _DeleteStageResponse = _interopRequireDefault(require("./model/DeleteStageResponse200"));

var _DeleteStageResponse200Data = _interopRequireDefault(require("./model/DeleteStageResponse200Data"));

var _DeleteStagesResponse = _interopRequireDefault(require("./model/DeleteStagesResponse200"));

var _DeleteStagesResponse200Data = _interopRequireDefault(require("./model/DeleteStagesResponse200Data"));

var _DeleteTeamUserRequest = _interopRequireDefault(require("./model/DeleteTeamUserRequest"));

var _Duration = _interopRequireDefault(require("./model/Duration"));

var _EditPipeline = _interopRequireDefault(require("./model/EditPipeline"));

var _EditPipelineAllOf = _interopRequireDefault(require("./model/EditPipelineAllOf"));

var _EmailInfo = _interopRequireDefault(require("./model/EmailInfo"));

var _ExpectedOutcome = _interopRequireDefault(require("./model/ExpectedOutcome"));

var _FailResponse = _interopRequireDefault(require("./model/FailResponse"));

var _Field = _interopRequireDefault(require("./model/Field"));

var _FieldCreateRequest = _interopRequireDefault(require("./model/FieldCreateRequest"));

var _FieldCreateRequestAllOf = _interopRequireDefault(require("./model/FieldCreateRequestAllOf"));

var _FieldResponse = _interopRequireDefault(require("./model/FieldResponse"));

var _FieldResponseAllOf = _interopRequireDefault(require("./model/FieldResponseAllOf"));

var _FieldType = _interopRequireDefault(require("./model/FieldType"));

var _FieldTypeAsString = _interopRequireDefault(require("./model/FieldTypeAsString"));

var _FieldUpdateRequest = _interopRequireDefault(require("./model/FieldUpdateRequest"));

var _FieldsResponse = _interopRequireDefault(require("./model/FieldsResponse"));

var _FieldsResponseAllOf = _interopRequireDefault(require("./model/FieldsResponseAllOf"));

var _FileData = _interopRequireDefault(require("./model/FileData"));

var _FileItem = _interopRequireDefault(require("./model/FileItem"));

var _FilterGetItem = _interopRequireDefault(require("./model/FilterGetItem"));

var _FilterType = _interopRequireDefault(require("./model/FilterType"));

var _FiltersBulkDeleteResponse = _interopRequireDefault(require("./model/FiltersBulkDeleteResponse"));

var _FiltersBulkDeleteResponseAllOf = _interopRequireDefault(require("./model/FiltersBulkDeleteResponseAllOf"));

var _FiltersBulkDeleteResponseAllOfData = _interopRequireDefault(require("./model/FiltersBulkDeleteResponseAllOfData"));

var _FiltersBulkGetResponse = _interopRequireDefault(require("./model/FiltersBulkGetResponse"));

var _FiltersBulkGetResponseAllOf = _interopRequireDefault(require("./model/FiltersBulkGetResponseAllOf"));

var _FiltersDeleteResponse = _interopRequireDefault(require("./model/FiltersDeleteResponse"));

var _FiltersDeleteResponseAllOf = _interopRequireDefault(require("./model/FiltersDeleteResponseAllOf"));

var _FiltersDeleteResponseAllOfData = _interopRequireDefault(require("./model/FiltersDeleteResponseAllOfData"));

var _FiltersGetResponse = _interopRequireDefault(require("./model/FiltersGetResponse"));

var _FiltersGetResponseAllOf = _interopRequireDefault(require("./model/FiltersGetResponseAllOf"));

var _FiltersPostResponse = _interopRequireDefault(require("./model/FiltersPostResponse"));

var _FiltersPostResponseAllOf = _interopRequireDefault(require("./model/FiltersPostResponseAllOf"));

var _FiltersPostResponseAllOfData = _interopRequireDefault(require("./model/FiltersPostResponseAllOfData"));

var _FindGoalResponse = _interopRequireDefault(require("./model/FindGoalResponse"));

var _FollowerData = _interopRequireDefault(require("./model/FollowerData"));

var _FollowerDataWithID = _interopRequireDefault(require("./model/FollowerDataWithID"));

var _FollowerDataWithIDAllOf = _interopRequireDefault(require("./model/FollowerDataWithIDAllOf"));

var _FullRole = _interopRequireDefault(require("./model/FullRole"));

var _FullRoleAllOf = _interopRequireDefault(require("./model/FullRoleAllOf"));

var _GetActivitiesResponse = _interopRequireDefault(require("./model/GetActivitiesResponse200"));

var _GetActivitiesResponse200RelatedObjects = _interopRequireDefault(require("./model/GetActivitiesResponse200RelatedObjects"));

var _GetActivityResponse = _interopRequireDefault(require("./model/GetActivityResponse200"));

var _GetAddProductAttachementDetails = _interopRequireDefault(require("./model/GetAddProductAttachementDetails"));

var _GetAddUpdateStage = _interopRequireDefault(require("./model/GetAddUpdateStage"));

var _GetAddedDeal = _interopRequireDefault(require("./model/GetAddedDeal"));

var _GetAddedDealAdditionalData = _interopRequireDefault(require("./model/GetAddedDealAdditionalData"));

var _GetAllFiles = _interopRequireDefault(require("./model/GetAllFiles"));

var _GetAllPersonsResponse = _interopRequireDefault(require("./model/GetAllPersonsResponse"));

var _GetAllPersonsResponseAllOf = _interopRequireDefault(require("./model/GetAllPersonsResponseAllOf"));

var _GetAllPipelines = _interopRequireDefault(require("./model/GetAllPipelines"));

var _GetAllPipelinesAllOf = _interopRequireDefault(require("./model/GetAllPipelinesAllOf"));

var _GetAllProductFieldsResponse = _interopRequireDefault(require("./model/GetAllProductFieldsResponse"));

var _GetComments = _interopRequireDefault(require("./model/GetComments"));

var _GetDeal = _interopRequireDefault(require("./model/GetDeal"));

var _GetDealAdditionalData = _interopRequireDefault(require("./model/GetDealAdditionalData"));

var _GetDeals = _interopRequireDefault(require("./model/GetDeals"));

var _GetDealsConversionRatesInPipeline = _interopRequireDefault(require("./model/GetDealsConversionRatesInPipeline"));

var _GetDealsConversionRatesInPipelineAllOf = _interopRequireDefault(require("./model/GetDealsConversionRatesInPipelineAllOf"));

var _GetDealsConversionRatesInPipelineAllOfData = _interopRequireDefault(require("./model/GetDealsConversionRatesInPipelineAllOfData"));

var _GetDealsMovementsInPipeline = _interopRequireDefault(require("./model/GetDealsMovementsInPipeline"));

var _GetDealsMovementsInPipelineAllOf = _interopRequireDefault(require("./model/GetDealsMovementsInPipelineAllOf"));

var _GetDealsMovementsInPipelineAllOfData = _interopRequireDefault(require("./model/GetDealsMovementsInPipelineAllOfData"));

var _GetDealsMovementsInPipelineAllOfDataAverageAgeInDays = _interopRequireDefault(require("./model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDays"));

var _GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages = _interopRequireDefault(require("./model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages"));

var _GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages = _interopRequireDefault(require("./model/GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages"));

var _GetDealsRelatedObjects = _interopRequireDefault(require("./model/GetDealsRelatedObjects"));

var _GetDealsSummary = _interopRequireDefault(require("./model/GetDealsSummary"));

var _GetDealsSummaryData = _interopRequireDefault(require("./model/GetDealsSummaryData"));

var _GetDealsSummaryDataValuesTotal = _interopRequireDefault(require("./model/GetDealsSummaryDataValuesTotal"));

var _GetDealsSummaryDataWeightedValuesTotal = _interopRequireDefault(require("./model/GetDealsSummaryDataWeightedValuesTotal"));

var _GetDealsTimeline = _interopRequireDefault(require("./model/GetDealsTimeline"));

var _GetDealsTimelineData = _interopRequireDefault(require("./model/GetDealsTimelineData"));

var _GetDealsTimelineDataTotals = _interopRequireDefault(require("./model/GetDealsTimelineDataTotals"));

var _GetDuplicatedDeal = _interopRequireDefault(require("./model/GetDuplicatedDeal"));

var _GetGoalResultResponse = _interopRequireDefault(require("./model/GetGoalResultResponse200"));

var _GetGoalsResponse = _interopRequireDefault(require("./model/GetGoalsResponse200"));

var _GetLeadLabelsResponse = _interopRequireDefault(require("./model/GetLeadLabelsResponse200"));

var _GetLeadSourcesResponse = _interopRequireDefault(require("./model/GetLeadSourcesResponse200"));

var _GetLeadSourcesResponse200Data = _interopRequireDefault(require("./model/GetLeadSourcesResponse200Data"));

var _GetLeadsResponse = _interopRequireDefault(require("./model/GetLeadsResponse200"));

var _GetMergedDeal = _interopRequireDefault(require("./model/GetMergedDeal"));

var _GetNotes = _interopRequireDefault(require("./model/GetNotes"));

var _GetOneFile = _interopRequireDefault(require("./model/GetOneFile"));

var _GetOnePipeline = _interopRequireDefault(require("./model/GetOnePipeline"));

var _GetOnePipelineAllOf = _interopRequireDefault(require("./model/GetOnePipelineAllOf"));

var _GetOneStage = _interopRequireDefault(require("./model/GetOneStage"));

var _GetPersonDetailsResponse = _interopRequireDefault(require("./model/GetPersonDetailsResponse"));

var _GetPersonDetailsResponseAllOf = _interopRequireDefault(require("./model/GetPersonDetailsResponseAllOf"));

var _GetPersonDetailsResponseAllOfAdditionalData = _interopRequireDefault(require("./model/GetPersonDetailsResponseAllOfAdditionalData"));

var _GetProductAttachementDetails = _interopRequireDefault(require("./model/GetProductAttachementDetails"));

var _GetProductFieldResponse = _interopRequireDefault(require("./model/GetProductFieldResponse"));

var _GetRecents = _interopRequireDefault(require("./model/GetRecents"));

var _GetRecentsAdditionalData = _interopRequireDefault(require("./model/GetRecentsAdditionalData"));

var _GetRole = _interopRequireDefault(require("./model/GetRole"));

var _GetRoleAllOf = _interopRequireDefault(require("./model/GetRoleAllOf"));

var _GetRoleAllOfAdditionalData = _interopRequireDefault(require("./model/GetRoleAllOfAdditionalData"));

var _GetRoleAssignments = _interopRequireDefault(require("./model/GetRoleAssignments"));

var _GetRoleAssignmentsAllOf = _interopRequireDefault(require("./model/GetRoleAssignmentsAllOf"));

var _GetRoleSettings = _interopRequireDefault(require("./model/GetRoleSettings"));

var _GetRoleSettingsAllOf = _interopRequireDefault(require("./model/GetRoleSettingsAllOf"));

var _GetRoleSubroles = _interopRequireDefault(require("./model/GetRoleSubroles"));

var _GetRoleSubrolesAllOf = _interopRequireDefault(require("./model/GetRoleSubrolesAllOf"));

var _GetRoles = _interopRequireDefault(require("./model/GetRoles"));

var _GetRolesAllOf = _interopRequireDefault(require("./model/GetRolesAllOf"));

var _GetStageDeals = _interopRequireDefault(require("./model/GetStageDeals"));

var _GetStages = _interopRequireDefault(require("./model/GetStages"));

var _GlobalMessageBaseResponse = _interopRequireDefault(require("./model/GlobalMessageBaseResponse"));

var _GlobalMessageData = _interopRequireDefault(require("./model/GlobalMessageData"));

var _GlobalMessageDelete = _interopRequireDefault(require("./model/GlobalMessageDelete"));

var _GlobalMessageDeleteAllOf = _interopRequireDefault(require("./model/GlobalMessageDeleteAllOf"));

var _GlobalMessageGet = _interopRequireDefault(require("./model/GlobalMessageGet"));

var _GlobalMessageGetAllOf = _interopRequireDefault(require("./model/GlobalMessageGetAllOf"));

var _GlobalMessageUserData = _interopRequireDefault(require("./model/GlobalMessageUserData"));

var _GoalResults = _interopRequireDefault(require("./model/GoalResults"));

var _GoalType = _interopRequireDefault(require("./model/GoalType"));

var _GoalsResponseComponent = _interopRequireDefault(require("./model/GoalsResponseComponent"));

var _IconKey = _interopRequireDefault(require("./model/IconKey"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./model/ItemSearchAdditionalData"));

var _ItemSearchAdditionalDataPagination = _interopRequireDefault(require("./model/ItemSearchAdditionalDataPagination"));

var _ItemSearchFieldResponse = _interopRequireDefault(require("./model/ItemSearchFieldResponse"));

var _ItemSearchFieldResponseAllOf = _interopRequireDefault(require("./model/ItemSearchFieldResponseAllOf"));

var _ItemSearchFieldResponseAllOfData = _interopRequireDefault(require("./model/ItemSearchFieldResponseAllOfData"));

var _ItemSearchItem = _interopRequireDefault(require("./model/ItemSearchItem"));

var _ItemSearchResponse = _interopRequireDefault(require("./model/ItemSearchResponse"));

var _ItemSearchResponseAllOf = _interopRequireDefault(require("./model/ItemSearchResponseAllOf"));

var _ItemSearchResponseAllOfData = _interopRequireDefault(require("./model/ItemSearchResponseAllOfData"));

var _LeadIdResponse = _interopRequireDefault(require("./model/LeadIdResponse200"));

var _LeadIdResponse200Data = _interopRequireDefault(require("./model/LeadIdResponse200Data"));

var _LeadLabelColor = _interopRequireDefault(require("./model/LeadLabelColor"));

var _LeadLabelResponse = _interopRequireDefault(require("./model/LeadLabelResponse"));

var _LeadResponse = _interopRequireDefault(require("./model/LeadResponse"));

var _LeadResponse2 = _interopRequireDefault(require("./model/LeadResponse404"));

var _LeadValue = _interopRequireDefault(require("./model/LeadValue"));

var _LinkRemoteFileToItem = _interopRequireDefault(require("./model/LinkRemoteFileToItem"));

var _ListActivitiesResponse = _interopRequireDefault(require("./model/ListActivitiesResponse"));

var _ListActivitiesResponseAllOf = _interopRequireDefault(require("./model/ListActivitiesResponseAllOf"));

var _ListDealsResponse = _interopRequireDefault(require("./model/ListDealsResponse"));

var _ListDealsResponseAllOf = _interopRequireDefault(require("./model/ListDealsResponseAllOf"));

var _ListDealsResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/ListDealsResponseAllOfRelatedObjects"));

var _ListFilesResponse = _interopRequireDefault(require("./model/ListFilesResponse"));

var _ListFilesResponseAllOf = _interopRequireDefault(require("./model/ListFilesResponseAllOf"));

var _ListFollowersResponse = _interopRequireDefault(require("./model/ListFollowersResponse"));

var _ListFollowersResponseAllOf = _interopRequireDefault(require("./model/ListFollowersResponseAllOf"));

var _ListFollowersResponseAllOfData = _interopRequireDefault(require("./model/ListFollowersResponseAllOfData"));

var _ListMailMessagesResponse = _interopRequireDefault(require("./model/ListMailMessagesResponse"));

var _ListMailMessagesResponseAllOf = _interopRequireDefault(require("./model/ListMailMessagesResponseAllOf"));

var _ListMailMessagesResponseAllOfData = _interopRequireDefault(require("./model/ListMailMessagesResponseAllOfData"));

var _ListPermittedUsersResponse = _interopRequireDefault(require("./model/ListPermittedUsersResponse"));

var _ListPermittedUsersResponse2 = _interopRequireDefault(require("./model/ListPermittedUsersResponse1"));

var _ListPermittedUsersResponse1AllOf = _interopRequireDefault(require("./model/ListPermittedUsersResponse1AllOf"));

var _ListPermittedUsersResponseAllOf = _interopRequireDefault(require("./model/ListPermittedUsersResponseAllOf"));

var _ListPermittedUsersResponseAllOfData = _interopRequireDefault(require("./model/ListPermittedUsersResponseAllOfData"));

var _ListPersonProductsResponse = _interopRequireDefault(require("./model/ListPersonProductsResponse"));

var _ListPersonProductsResponseAllOf = _interopRequireDefault(require("./model/ListPersonProductsResponseAllOf"));

var _ListPersonProductsResponseAllOfDEALID = _interopRequireDefault(require("./model/ListPersonProductsResponseAllOfDEALID"));

var _ListPersonProductsResponseAllOfData = _interopRequireDefault(require("./model/ListPersonProductsResponseAllOfData"));

var _ListPersonsResponse = _interopRequireDefault(require("./model/ListPersonsResponse"));

var _ListPersonsResponseAllOf = _interopRequireDefault(require("./model/ListPersonsResponseAllOf"));

var _ListPersonsResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/ListPersonsResponseAllOfRelatedObjects"));

var _ListProductAdditionalData = _interopRequireDefault(require("./model/ListProductAdditionalData"));

var _ListProductAdditionalDataAllOf = _interopRequireDefault(require("./model/ListProductAdditionalDataAllOf"));

var _ListProductFollowersResponse = _interopRequireDefault(require("./model/ListProductFollowersResponse"));

var _ListProductFollowersResponseAllOf = _interopRequireDefault(require("./model/ListProductFollowersResponseAllOf"));

var _ListProductFollowersResponseAllOfData = _interopRequireDefault(require("./model/ListProductFollowersResponseAllOfData"));

var _ListProductsResponse = _interopRequireDefault(require("./model/ListProductsResponse"));

var _ListProductsResponseAllOf = _interopRequireDefault(require("./model/ListProductsResponseAllOf"));

var _ListProductsResponseAllOfData = _interopRequireDefault(require("./model/ListProductsResponseAllOfData"));

var _ListProductsResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/ListProductsResponseAllOfRelatedObjects"));

var _MailMessage = _interopRequireDefault(require("./model/MailMessage"));

var _MailMessageAllOf = _interopRequireDefault(require("./model/MailMessageAllOf"));

var _MailMessageData = _interopRequireDefault(require("./model/MailMessageData"));

var _MailMessageItemForList = _interopRequireDefault(require("./model/MailMessageItemForList"));

var _MailMessageItemForListAllOf = _interopRequireDefault(require("./model/MailMessageItemForListAllOf"));

var _MailParticipant = _interopRequireDefault(require("./model/MailParticipant"));

var _MailServiceBaseResponse = _interopRequireDefault(require("./model/MailServiceBaseResponse"));

var _MailThread = _interopRequireDefault(require("./model/MailThread"));

var _MailThreadAllOf = _interopRequireDefault(require("./model/MailThreadAllOf"));

var _MailThreadDelete = _interopRequireDefault(require("./model/MailThreadDelete"));

var _MailThreadDeleteAllOf = _interopRequireDefault(require("./model/MailThreadDeleteAllOf"));

var _MailThreadMessages = _interopRequireDefault(require("./model/MailThreadMessages"));

var _MailThreadMessagesAllOf = _interopRequireDefault(require("./model/MailThreadMessagesAllOf"));

var _MailThreadOne = _interopRequireDefault(require("./model/MailThreadOne"));

var _MailThreadOneAllOf = _interopRequireDefault(require("./model/MailThreadOneAllOf"));

var _MailThreadParticipant = _interopRequireDefault(require("./model/MailThreadParticipant"));

var _MailThreadPut = _interopRequireDefault(require("./model/MailThreadPut"));

var _MailThreadPutAllOf = _interopRequireDefault(require("./model/MailThreadPutAllOf"));

var _MarketingStatus = _interopRequireDefault(require("./model/MarketingStatus"));

var _MergeDealsRequest = _interopRequireDefault(require("./model/MergeDealsRequest"));

var _MergeOrganizationsRequest = _interopRequireDefault(require("./model/MergeOrganizationsRequest"));

var _MergePersonDealRelatedInfo = _interopRequireDefault(require("./model/MergePersonDealRelatedInfo"));

var _MergePersonItem = _interopRequireDefault(require("./model/MergePersonItem"));

var _MergePersonsRequest = _interopRequireDefault(require("./model/MergePersonsRequest"));

var _MergePersonsResponse = _interopRequireDefault(require("./model/MergePersonsResponse"));

var _MergePersonsResponseAllOf = _interopRequireDefault(require("./model/MergePersonsResponseAllOf"));

var _NewDeal = _interopRequireDefault(require("./model/NewDeal"));

var _NewDealAllOf = _interopRequireDefault(require("./model/NewDealAllOf"));

var _NewDealProduct = _interopRequireDefault(require("./model/NewDealProduct"));

var _NewDealProductAllOf = _interopRequireDefault(require("./model/NewDealProductAllOf"));

var _NewFollowerResponse = _interopRequireDefault(require("./model/NewFollowerResponse"));

var _NewFollowerResponseData = _interopRequireDefault(require("./model/NewFollowerResponseData"));

var _NewGoal = _interopRequireDefault(require("./model/NewGoal"));

var _NewOrganization = _interopRequireDefault(require("./model/NewOrganization"));

var _NewOrganizationAllOf = _interopRequireDefault(require("./model/NewOrganizationAllOf"));

var _NewPerson = _interopRequireDefault(require("./model/NewPerson"));

var _NewPersonAllOf = _interopRequireDefault(require("./model/NewPersonAllOf"));

var _NewProductField = _interopRequireDefault(require("./model/NewProductField"));

var _Note = _interopRequireDefault(require("./model/Note"));

var _NoteConnectToParams = _interopRequireDefault(require("./model/NoteConnectToParams"));

var _NoteCreatorUser = _interopRequireDefault(require("./model/NoteCreatorUser"));

var _NoteField = _interopRequireDefault(require("./model/NoteField"));

var _NoteFieldsResponse = _interopRequireDefault(require("./model/NoteFieldsResponse"));

var _NoteFieldsResponseAllOf = _interopRequireDefault(require("./model/NoteFieldsResponseAllOf"));

var _NoteParams = _interopRequireDefault(require("./model/NoteParams"));

var _NumberBoolean = _interopRequireDefault(require("./model/NumberBoolean"));

var _NumberBooleanDefault = _interopRequireDefault(require("./model/NumberBooleanDefault0"));

var _NumberBooleanDefault2 = _interopRequireDefault(require("./model/NumberBooleanDefault1"));

var _ObjectPrices = _interopRequireDefault(require("./model/ObjectPrices"));

var _OneLeadResponse = _interopRequireDefault(require("./model/OneLeadResponse200"));

var _OrgAndOwnerId = _interopRequireDefault(require("./model/OrgAndOwnerId"));

var _OrganizationAddressInfo = _interopRequireDefault(require("./model/OrganizationAddressInfo"));

var _OrganizationCountAndAddressInfo = _interopRequireDefault(require("./model/OrganizationCountAndAddressInfo"));

var _OrganizationCountInfo = _interopRequireDefault(require("./model/OrganizationCountInfo"));

var _OrganizationData = _interopRequireDefault(require("./model/OrganizationData"));

var _OrganizationDataWithId = _interopRequireDefault(require("./model/OrganizationDataWithId"));

var _OrganizationDataWithIdAllOf = _interopRequireDefault(require("./model/OrganizationDataWithIdAllOf"));

var _OrganizationDataWithIdAndActiveFlag = _interopRequireDefault(require("./model/OrganizationDataWithIdAndActiveFlag"));

var _OrganizationDataWithIdAndActiveFlagAllOf = _interopRequireDefault(require("./model/OrganizationDataWithIdAndActiveFlagAllOf"));

var _OrganizationDeleteResponse = _interopRequireDefault(require("./model/OrganizationDeleteResponse"));

var _OrganizationDeleteResponseData = _interopRequireDefault(require("./model/OrganizationDeleteResponseData"));

var _OrganizationDetailsGetResponse = _interopRequireDefault(require("./model/OrganizationDetailsGetResponse"));

var _OrganizationDetailsGetResponseAllOf = _interopRequireDefault(require("./model/OrganizationDetailsGetResponseAllOf"));

var _OrganizationDetailsGetResponseAllOfAdditionalData = _interopRequireDefault(require("./model/OrganizationDetailsGetResponseAllOfAdditionalData"));

var _OrganizationFlowResponse = _interopRequireDefault(require("./model/OrganizationFlowResponse"));

var _OrganizationFlowResponseAllOf = _interopRequireDefault(require("./model/OrganizationFlowResponseAllOf"));

var _OrganizationFlowResponseAllOfData = _interopRequireDefault(require("./model/OrganizationFlowResponseAllOfData"));

var _OrganizationFlowResponseAllOfRelatedObjects = _interopRequireDefault(require("./model/OrganizationFlowResponseAllOfRelatedObjects"));

var _OrganizationFollowerDeleteResponse = _interopRequireDefault(require("./model/OrganizationFollowerDeleteResponse"));

var _OrganizationFollowerDeleteResponseData = _interopRequireDefault(require("./model/OrganizationFollowerDeleteResponseData"));

var _OrganizationFollowerItem = _interopRequireDefault(require("./model/OrganizationFollowerItem"));

var _OrganizationFollowerItemAllOf = _interopRequireDefault(require("./model/OrganizationFollowerItemAllOf"));

var _OrganizationFollowerPostResponse = _interopRequireDefault(require("./model/OrganizationFollowerPostResponse"));

var _OrganizationFollowersListResponse = _interopRequireDefault(require("./model/OrganizationFollowersListResponse"));

var _OrganizationItem = _interopRequireDefault(require("./model/OrganizationItem"));

var _OrganizationItemAllOf = _interopRequireDefault(require("./model/OrganizationItemAllOf"));

var _OrganizationPostResponse = _interopRequireDefault(require("./model/OrganizationPostResponse"));

var _OrganizationPostResponseAllOf = _interopRequireDefault(require("./model/OrganizationPostResponseAllOf"));

var _OrganizationRelationship = _interopRequireDefault(require("./model/OrganizationRelationship"));

var _OrganizationRelationshipDeleteResponse = _interopRequireDefault(require("./model/OrganizationRelationshipDeleteResponse"));

var _OrganizationRelationshipDeleteResponseAllOf = _interopRequireDefault(require("./model/OrganizationRelationshipDeleteResponseAllOf"));

var _OrganizationRelationshipDeleteResponseAllOfData = _interopRequireDefault(require("./model/OrganizationRelationshipDeleteResponseAllOfData"));

var _OrganizationRelationshipDetails = _interopRequireDefault(require("./model/OrganizationRelationshipDetails"));

var _OrganizationRelationshipGetResponse = _interopRequireDefault(require("./model/OrganizationRelationshipGetResponse"));

var _OrganizationRelationshipGetResponseAllOf = _interopRequireDefault(require("./model/OrganizationRelationshipGetResponseAllOf"));

var _OrganizationRelationshipPostResponse = _interopRequireDefault(require("./model/OrganizationRelationshipPostResponse"));

var _OrganizationRelationshipPostResponseAllOf = _interopRequireDefault(require("./model/OrganizationRelationshipPostResponseAllOf"));

var _OrganizationRelationshipUpdateResponse = _interopRequireDefault(require("./model/OrganizationRelationshipUpdateResponse"));

var _OrganizationRelationshipWithCalculatedFields = _interopRequireDefault(require("./model/OrganizationRelationshipWithCalculatedFields"));

var _OrganizationSearchItem = _interopRequireDefault(require("./model/OrganizationSearchItem"));

var _OrganizationSearchItemItem = _interopRequireDefault(require("./model/OrganizationSearchItemItem"));

var _OrganizationSearchResponse = _interopRequireDefault(require("./model/OrganizationSearchResponse"));

var _OrganizationSearchResponseAllOf = _interopRequireDefault(require("./model/OrganizationSearchResponseAllOf"));

var _OrganizationSearchResponseAllOfData = _interopRequireDefault(require("./model/OrganizationSearchResponseAllOfData"));

var _OrganizationUpdateResponse = _interopRequireDefault(require("./model/OrganizationUpdateResponse"));

var _OrganizationUpdateResponseAllOf = _interopRequireDefault(require("./model/OrganizationUpdateResponseAllOf"));

var _OrganizationsDeleteResponse = _interopRequireDefault(require("./model/OrganizationsDeleteResponse"));

var _OrganizationsDeleteResponseData = _interopRequireDefault(require("./model/OrganizationsDeleteResponseData"));

var _OrganizationsMergeResponse = _interopRequireDefault(require("./model/OrganizationsMergeResponse"));

var _OrganizationsMergeResponseData = _interopRequireDefault(require("./model/OrganizationsMergeResponseData"));

var _Owner = _interopRequireDefault(require("./model/Owner"));

var _OwnerAllOf = _interopRequireDefault(require("./model/OwnerAllOf"));

var _PaginationDetails = _interopRequireDefault(require("./model/PaginationDetails"));

var _PaginationDetailsAllOf = _interopRequireDefault(require("./model/PaginationDetailsAllOf"));

var _Params = _interopRequireDefault(require("./model/Params"));

var _PaymentItem = _interopRequireDefault(require("./model/PaymentItem"));

var _PaymentsResponse = _interopRequireDefault(require("./model/PaymentsResponse"));

var _PaymentsResponseAllOf = _interopRequireDefault(require("./model/PaymentsResponseAllOf"));

var _PaymentsResponseAllOfData = _interopRequireDefault(require("./model/PaymentsResponseAllOfData"));

var _PermissionSets = _interopRequireDefault(require("./model/PermissionSets"));

var _PermissionSetsAllOf = _interopRequireDefault(require("./model/PermissionSetsAllOf"));

var _PermissionSetsItem = _interopRequireDefault(require("./model/PermissionSetsItem"));

var _PersonCountAndEmailInfo = _interopRequireDefault(require("./model/PersonCountAndEmailInfo"));

var _PersonCountEmailDealAndActivityInfo = _interopRequireDefault(require("./model/PersonCountEmailDealAndActivityInfo"));

var _PersonCountInfo = _interopRequireDefault(require("./model/PersonCountInfo"));

var _PersonData = _interopRequireDefault(require("./model/PersonData"));

var _PersonDataEmail = _interopRequireDefault(require("./model/PersonDataEmail"));

var _PersonDataPhone = _interopRequireDefault(require("./model/PersonDataPhone"));

var _PersonDataWithActiveFlag = _interopRequireDefault(require("./model/PersonDataWithActiveFlag"));

var _PersonDataWithActiveFlagAllOf = _interopRequireDefault(require("./model/PersonDataWithActiveFlagAllOf"));

var _PersonFlowResponse = _interopRequireDefault(require("./model/PersonFlowResponse"));

var _PersonFlowResponseAllOf = _interopRequireDefault(require("./model/PersonFlowResponseAllOf"));

var _PersonFlowResponseAllOfData = _interopRequireDefault(require("./model/PersonFlowResponseAllOfData"));

var _PersonItem = _interopRequireDefault(require("./model/PersonItem"));

var _PersonListProduct = _interopRequireDefault(require("./model/PersonListProduct"));

var _PersonNameCountAndEmailInfo = _interopRequireDefault(require("./model/PersonNameCountAndEmailInfo"));

var _PersonNameCountAndEmailInfoWithIds = _interopRequireDefault(require("./model/PersonNameCountAndEmailInfoWithIds"));

var _PersonNameCountAndEmailInfoWithIdsAllOf = _interopRequireDefault(require("./model/PersonNameCountAndEmailInfoWithIdsAllOf"));

var _PersonNameInfo = _interopRequireDefault(require("./model/PersonNameInfo"));

var _PersonNameInfoWithOrgAndOwnerId = _interopRequireDefault(require("./model/PersonNameInfoWithOrgAndOwnerId"));

var _PersonSearchItem = _interopRequireDefault(require("./model/PersonSearchItem"));

var _PersonSearchItemItem = _interopRequireDefault(require("./model/PersonSearchItemItem"));

var _PersonSearchItemItemOrganization = _interopRequireDefault(require("./model/PersonSearchItemItemOrganization"));

var _PersonSearchItemItemOwner = _interopRequireDefault(require("./model/PersonSearchItemItemOwner"));

var _PersonSearchResponse = _interopRequireDefault(require("./model/PersonSearchResponse"));

var _PersonSearchResponseAllOf = _interopRequireDefault(require("./model/PersonSearchResponseAllOf"));

var _PersonSearchResponseAllOfData = _interopRequireDefault(require("./model/PersonSearchResponseAllOfData"));

var _PictureData = _interopRequireDefault(require("./model/PictureData"));

var _PictureDataPictures = _interopRequireDefault(require("./model/PictureDataPictures"));

var _PictureDataWithID = _interopRequireDefault(require("./model/PictureDataWithID"));

var _PictureDataWithIDAllOf = _interopRequireDefault(require("./model/PictureDataWithIDAllOf"));

var _PictureDataWithValue = _interopRequireDefault(require("./model/PictureDataWithValue"));

var _PictureDataWithValueAllOf = _interopRequireDefault(require("./model/PictureDataWithValueAllOf"));

var _Pipeline = _interopRequireDefault(require("./model/Pipeline"));

var _PipelineDetails = _interopRequireDefault(require("./model/PipelineDetails"));

var _PipelineDetailsAllOf = _interopRequireDefault(require("./model/PipelineDetailsAllOf"));

var _PostComment = _interopRequireDefault(require("./model/PostComment"));

var _PostDealParticipants = _interopRequireDefault(require("./model/PostDealParticipants"));

var _PostGoalResponse = _interopRequireDefault(require("./model/PostGoalResponse"));

var _PostNote = _interopRequireDefault(require("./model/PostNote"));

var _PostRoleAssignment = _interopRequireDefault(require("./model/PostRoleAssignment"));

var _PostRoleAssignmentAllOf = _interopRequireDefault(require("./model/PostRoleAssignmentAllOf"));

var _PostRoleAssignmentAllOfData = _interopRequireDefault(require("./model/PostRoleAssignmentAllOfData"));

var _PostRoleSettings = _interopRequireDefault(require("./model/PostRoleSettings"));

var _PostRoleSettingsAllOf = _interopRequireDefault(require("./model/PostRoleSettingsAllOf"));

var _PostRoleSettingsAllOfData = _interopRequireDefault(require("./model/PostRoleSettingsAllOfData"));

var _PostRoles = _interopRequireDefault(require("./model/PostRoles"));

var _PostRolesAllOf = _interopRequireDefault(require("./model/PostRolesAllOf"));

var _PostRolesAllOfData = _interopRequireDefault(require("./model/PostRolesAllOfData"));

var _ProductAttachementFields = _interopRequireDefault(require("./model/ProductAttachementFields"));

var _ProductAttachmentDetails = _interopRequireDefault(require("./model/ProductAttachmentDetails"));

var _ProductBaseDeal = _interopRequireDefault(require("./model/ProductBaseDeal"));

var _ProductField = _interopRequireDefault(require("./model/ProductField"));

var _ProductFieldAllOf = _interopRequireDefault(require("./model/ProductFieldAllOf"));

var _ProductListItem = _interopRequireDefault(require("./model/ProductListItem"));

var _ProductRequest = _interopRequireDefault(require("./model/ProductRequest"));

var _ProductResponse = _interopRequireDefault(require("./model/ProductResponse"));

var _ProductSearchItem = _interopRequireDefault(require("./model/ProductSearchItem"));

var _ProductSearchItemItem = _interopRequireDefault(require("./model/ProductSearchItemItem"));

var _ProductSearchItemItemOwner = _interopRequireDefault(require("./model/ProductSearchItemItemOwner"));

var _ProductSearchResponse = _interopRequireDefault(require("./model/ProductSearchResponse"));

var _ProductSearchResponseAllOf = _interopRequireDefault(require("./model/ProductSearchResponseAllOf"));

var _ProductSearchResponseAllOfData = _interopRequireDefault(require("./model/ProductSearchResponseAllOfData"));

var _ProductWithArrayPrices = _interopRequireDefault(require("./model/ProductWithArrayPrices"));

var _ProductWithObjectPrices = _interopRequireDefault(require("./model/ProductWithObjectPrices"));

var _ProductsResponse = _interopRequireDefault(require("./model/ProductsResponse"));

var _PutRole = _interopRequireDefault(require("./model/PutRole"));

var _PutRoleAllOf = _interopRequireDefault(require("./model/PutRoleAllOf"));

var _PutRoleAllOfData = _interopRequireDefault(require("./model/PutRoleAllOfData"));

var _RecentDataProduct = _interopRequireDefault(require("./model/RecentDataProduct"));

var _RecentsActivity = _interopRequireDefault(require("./model/RecentsActivity"));

var _RecentsActivityType = _interopRequireDefault(require("./model/RecentsActivityType"));

var _RecentsDeal = _interopRequireDefault(require("./model/RecentsDeal"));

var _RecentsFile = _interopRequireDefault(require("./model/RecentsFile"));

var _RecentsFilter = _interopRequireDefault(require("./model/RecentsFilter"));

var _RecentsNote = _interopRequireDefault(require("./model/RecentsNote"));

var _RecentsOrganization = _interopRequireDefault(require("./model/RecentsOrganization"));

var _RecentsPerson = _interopRequireDefault(require("./model/RecentsPerson"));

var _RecentsPipeline = _interopRequireDefault(require("./model/RecentsPipeline"));

var _RecentsProduct = _interopRequireDefault(require("./model/RecentsProduct"));

var _RecentsStage = _interopRequireDefault(require("./model/RecentsStage"));

var _RecentsUser = _interopRequireDefault(require("./model/RecentsUser"));

var _RelatedDealData = _interopRequireDefault(require("./model/RelatedDealData"));

var _RelatedDealDataDEALID = _interopRequireDefault(require("./model/RelatedDealDataDEALID"));

var _RelatedFollowerData = _interopRequireDefault(require("./model/RelatedFollowerData"));

var _RelatedOrganizationData = _interopRequireDefault(require("./model/RelatedOrganizationData"));

var _RelatedOrganizationDataWithActiveFlag = _interopRequireDefault(require("./model/RelatedOrganizationDataWithActiveFlag"));

var _RelatedOrganizationName = _interopRequireDefault(require("./model/RelatedOrganizationName"));

var _RelatedPersonData = _interopRequireDefault(require("./model/RelatedPersonData"));

var _RelatedPersonDataWithActiveFlag = _interopRequireDefault(require("./model/RelatedPersonDataWithActiveFlag"));

var _RelatedPictureData = _interopRequireDefault(require("./model/RelatedPictureData"));

var _RelatedUserData = _interopRequireDefault(require("./model/RelatedUserData"));

var _RelationshipOrganizationInfoItem = _interopRequireDefault(require("./model/RelationshipOrganizationInfoItem"));

var _RelationshipOrganizationInfoItemAllOf = _interopRequireDefault(require("./model/RelationshipOrganizationInfoItemAllOf"));

var _RelationshipOrganizationInfoItemWithActiveFlag = _interopRequireDefault(require("./model/RelationshipOrganizationInfoItemWithActiveFlag"));

var _RequredTitleParameter = _interopRequireDefault(require("./model/RequredTitleParameter"));

var _ResponseCallLogObject = _interopRequireDefault(require("./model/ResponseCallLogObject"));

var _ResponseCallLogObjectAllOf = _interopRequireDefault(require("./model/ResponseCallLogObjectAllOf"));

var _RoleAssignment = _interopRequireDefault(require("./model/RoleAssignment"));

var _RoleAssignmentAllOf = _interopRequireDefault(require("./model/RoleAssignmentAllOf"));

var _RoleSettings = _interopRequireDefault(require("./model/RoleSettings"));

var _RolesAdditionalData = _interopRequireDefault(require("./model/RolesAdditionalData"));

var _RolesAdditionalDataPagination = _interopRequireDefault(require("./model/RolesAdditionalDataPagination"));

var _SinglePermissionSetsItem = _interopRequireDefault(require("./model/SinglePermissionSetsItem"));

var _SinglePermissionSetsItemAllOf = _interopRequireDefault(require("./model/SinglePermissionSetsItemAllOf"));

var _Stage = _interopRequireDefault(require("./model/Stage"));

var _StageConversions = _interopRequireDefault(require("./model/StageConversions"));

var _StageDetails = _interopRequireDefault(require("./model/StageDetails"));

var _StageWithPipelineInfo = _interopRequireDefault(require("./model/StageWithPipelineInfo"));

var _StageWithPipelineInfoAllOf = _interopRequireDefault(require("./model/StageWithPipelineInfoAllOf"));

var _SubRole = _interopRequireDefault(require("./model/SubRole"));

var _SubRoleAllOf = _interopRequireDefault(require("./model/SubRoleAllOf"));

var _SubscriptionInstallmentCreateRequest = _interopRequireDefault(require("./model/SubscriptionInstallmentCreateRequest"));

var _SubscriptionInstallmentUpdateRequest = _interopRequireDefault(require("./model/SubscriptionInstallmentUpdateRequest"));

var _SubscriptionItem = _interopRequireDefault(require("./model/SubscriptionItem"));

var _SubscriptionRecurringCancelRequest = _interopRequireDefault(require("./model/SubscriptionRecurringCancelRequest"));

var _SubscriptionRecurringCreateRequest = _interopRequireDefault(require("./model/SubscriptionRecurringCreateRequest"));

var _SubscriptionRecurringUpdateRequest = _interopRequireDefault(require("./model/SubscriptionRecurringUpdateRequest"));

var _SubscriptionsIdResponse = _interopRequireDefault(require("./model/SubscriptionsIdResponse"));

var _SubscriptionsIdResponseAllOf = _interopRequireDefault(require("./model/SubscriptionsIdResponseAllOf"));

var _Team = _interopRequireDefault(require("./model/Team"));

var _TeamAllOf = _interopRequireDefault(require("./model/TeamAllOf"));

var _TeamId = _interopRequireDefault(require("./model/TeamId"));

var _Teams = _interopRequireDefault(require("./model/Teams"));

var _TeamsAllOf = _interopRequireDefault(require("./model/TeamsAllOf"));

var _Unauthorized = _interopRequireDefault(require("./model/Unauthorized"));

var _UpdateActivityResponse = _interopRequireDefault(require("./model/UpdateActivityResponse200"));

var _UpdateDealRequest = _interopRequireDefault(require("./model/UpdateDealRequest"));

var _UpdateFile = _interopRequireDefault(require("./model/UpdateFile"));

var _UpdateFilterRequest = _interopRequireDefault(require("./model/UpdateFilterRequest"));

var _UpdateLeadLabelRequest = _interopRequireDefault(require("./model/UpdateLeadLabelRequest"));

var _UpdateLeadRequest = _interopRequireDefault(require("./model/UpdateLeadRequest"));

var _UpdatePersonResponse = _interopRequireDefault(require("./model/UpdatePersonResponse"));

var _UpdateProductResponse = _interopRequireDefault(require("./model/UpdateProductResponse"));

var _UpdateStageRequest = _interopRequireDefault(require("./model/UpdateStageRequest"));

var _UpdateStageRequestAllOf = _interopRequireDefault(require("./model/UpdateStageRequestAllOf"));

var _UpdateTeam = _interopRequireDefault(require("./model/UpdateTeam"));

var _UpdateTeamAllOf = _interopRequireDefault(require("./model/UpdateTeamAllOf"));

var _UpdateTeamWithAdditionalProperties = _interopRequireDefault(require("./model/UpdateTeamWithAdditionalProperties"));

var _User = _interopRequireDefault(require("./model/User"));

var _UserAllOf = _interopRequireDefault(require("./model/UserAllOf"));

var _UserAssignmentToPermissionSet = _interopRequireDefault(require("./model/UserAssignmentToPermissionSet"));

var _UserAssignmentsToPermissionSet = _interopRequireDefault(require("./model/UserAssignmentsToPermissionSet"));

var _UserAssignmentsToPermissionSetAllOf = _interopRequireDefault(require("./model/UserAssignmentsToPermissionSetAllOf"));

var _UserConnections = _interopRequireDefault(require("./model/UserConnections"));

var _UserConnectionsAllOf = _interopRequireDefault(require("./model/UserConnectionsAllOf"));

var _UserConnectionsAllOfData = _interopRequireDefault(require("./model/UserConnectionsAllOfData"));

var _UserData = _interopRequireDefault(require("./model/UserData"));

var _UserDataWithId = _interopRequireDefault(require("./model/UserDataWithId"));

var _UserIDs = _interopRequireDefault(require("./model/UserIDs"));

var _UserIDsAllOf = _interopRequireDefault(require("./model/UserIDsAllOf"));

var _UserMe = _interopRequireDefault(require("./model/UserMe"));

var _UserMeAllOf = _interopRequireDefault(require("./model/UserMeAllOf"));

var _UserPermissions = _interopRequireDefault(require("./model/UserPermissions"));

var _UserPermissionsAllOf = _interopRequireDefault(require("./model/UserPermissionsAllOf"));

var _UserPermissionsItem = _interopRequireDefault(require("./model/UserPermissionsItem"));

var _UserSettings = _interopRequireDefault(require("./model/UserSettings"));

var _UserSettingsAllOf = _interopRequireDefault(require("./model/UserSettingsAllOf"));

var _UserSettingsItem = _interopRequireDefault(require("./model/UserSettingsItem"));

var _Users = _interopRequireDefault(require("./model/Users"));

var _UsersAllOf = _interopRequireDefault(require("./model/UsersAllOf"));

var _VisibleTo = _interopRequireDefault(require("./model/VisibleTo"));

var _Webhook = _interopRequireDefault(require("./model/Webhook"));

var _WebhookAllOf = _interopRequireDefault(require("./model/WebhookAllOf"));

var _WebhookBadRequest = _interopRequireDefault(require("./model/WebhookBadRequest"));

var _WebhookBadRequestAllOf = _interopRequireDefault(require("./model/WebhookBadRequestAllOf"));

var _Webhooks = _interopRequireDefault(require("./model/Webhooks"));

var _WebhooksAllOf = _interopRequireDefault(require("./model/WebhooksAllOf"));

var _WebhooksDeleteForbiddenSchema = _interopRequireDefault(require("./model/WebhooksDeleteForbiddenSchema"));

var _WebhooksDeleteForbiddenSchemaAllOf = _interopRequireDefault(require("./model/WebhooksDeleteForbiddenSchemaAllOf"));

var _ActivitiesApi = _interopRequireDefault(require("./api/ActivitiesApi"));

var _ActivityFieldsApi = _interopRequireDefault(require("./api/ActivityFieldsApi"));

var _ActivityTypesApi = _interopRequireDefault(require("./api/ActivityTypesApi"));

var _CallLogsApi = _interopRequireDefault(require("./api/CallLogsApi"));

var _CurrenciesApi = _interopRequireDefault(require("./api/CurrenciesApi"));

var _DealFieldsApi = _interopRequireDefault(require("./api/DealFieldsApi"));

var _DealsApi = _interopRequireDefault(require("./api/DealsApi"));

var _FilesApi = _interopRequireDefault(require("./api/FilesApi"));

var _FiltersApi = _interopRequireDefault(require("./api/FiltersApi"));

var _GlobalMessagesApi = _interopRequireDefault(require("./api/GlobalMessagesApi"));

var _GoalsApi = _interopRequireDefault(require("./api/GoalsApi"));

var _ItemSearchApi = _interopRequireDefault(require("./api/ItemSearchApi"));

var _LeadLabelsApi = _interopRequireDefault(require("./api/LeadLabelsApi"));

var _LeadSourcesApi = _interopRequireDefault(require("./api/LeadSourcesApi"));

var _LeadsApi = _interopRequireDefault(require("./api/LeadsApi"));

var _MailboxApi = _interopRequireDefault(require("./api/MailboxApi"));

var _NoteFieldsApi = _interopRequireDefault(require("./api/NoteFieldsApi"));

var _NotesApi = _interopRequireDefault(require("./api/NotesApi"));

var _OrganizationFieldsApi = _interopRequireDefault(require("./api/OrganizationFieldsApi"));

var _OrganizationRelationshipsApi = _interopRequireDefault(require("./api/OrganizationRelationshipsApi"));

var _OrganizationsApi = _interopRequireDefault(require("./api/OrganizationsApi"));

var _PermissionSetsApi = _interopRequireDefault(require("./api/PermissionSetsApi"));

var _PersonFieldsApi = _interopRequireDefault(require("./api/PersonFieldsApi"));

var _PersonsApi = _interopRequireDefault(require("./api/PersonsApi"));

var _PipelinesApi = _interopRequireDefault(require("./api/PipelinesApi"));

var _ProductFieldsApi = _interopRequireDefault(require("./api/ProductFieldsApi"));

var _ProductsApi = _interopRequireDefault(require("./api/ProductsApi"));

var _RecentsApi = _interopRequireDefault(require("./api/RecentsApi"));

var _RolesApi = _interopRequireDefault(require("./api/RolesApi"));

var _StagesApi = _interopRequireDefault(require("./api/StagesApi"));

var _SubscriptionsApi = _interopRequireDefault(require("./api/SubscriptionsApi"));

var _TeamsApi = _interopRequireDefault(require("./api/TeamsApi"));

var _UserConnectionsApi = _interopRequireDefault(require("./api/UserConnectionsApi"));

var _UserSettingsApi = _interopRequireDefault(require("./api/UserSettingsApi"));

var _UsersApi = _interopRequireDefault(require("./api/UsersApi"));

var _WebhooksApi = _interopRequireDefault(require("./api/WebhooksApi"));
