// pipedrive/dist/model/ActivityTypeUpdateRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _IconKey = _interopRequireDefault(require("./IconKey"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityTypeUpdateRequest model module.
 * @module model/ActivityTypeUpdateRequest
 * @version 1.0.0
 */
var ActivityTypeUpdateRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityTypeUpdateRequest</code>.
   * @alias module:model/ActivityTypeUpdateRequest
   */
  function ActivityTypeUpdateRequest() {
    (0, _classCallCheck2["default"])(this, ActivityTypeUpdateRequest);
    ActivityTypeUpdateRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityTypeUpdateRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityTypeUpdateRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityTypeUpdateRequest} obj Optional instance to populate.
     * @return {module:model/ActivityTypeUpdateRequest} The populated <code>ActivityTypeUpdateRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityTypeUpdateRequest();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('icon_key')) {
          obj['icon_key'] = _IconKey["default"].constructFromObject(data['icon_key']);
          delete data['icon_key'];
        }

        if (data.hasOwnProperty('color')) {
          obj['color'] = _ApiClient["default"].convertToType(data['color'], 'String');
          delete data['color'];
        }

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return ActivityTypeUpdateRequest;
}();
/**
 * The name of the ActivityType
 * @member {String} name
 */


ActivityTypeUpdateRequest.prototype['name'] = undefined;
/**
 * @member {module:model/IconKey} icon_key
 */

ActivityTypeUpdateRequest.prototype['icon_key'] = undefined;
/**
 * A designated color for the ActivityType in 6-character HEX format (e.g. `FFFFFF` for white, `000000` for black)
 * @member {String} color
 */

ActivityTypeUpdateRequest.prototype['color'] = undefined;
/**
 * An order number for this ActivityType. Order numbers should be used to order the types in the ActivityType selections.
 * @member {Number} order_nr
 */

ActivityTypeUpdateRequest.prototype['order_nr'] = undefined;
var _default = ActivityTypeUpdateRequest;
exports["default"] = _default;
