// pipedrive/dist/model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities = _interopRequireDefault(require("./ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID model module.
 * @module model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID
 * @version 1.0.0
 */
var ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID</code>.
   * The ID of the user
   * @alias module:model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID
   */
  function ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID() {
    (0, _classCallCheck2["default"])(this, ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID);
    ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID} obj Optional instance to populate.
     * @return {module:model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID} The populated <code>ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID();

        if (data.hasOwnProperty('activities')) {
          obj['activities'] = _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities["default"].constructFromObject(data['activities']);
          delete data['activities'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('activity_count')) {
          obj['activity_count'] = _ApiClient["default"].convertToType(data['activity_count'], 'Number');
          delete data['activity_count'];
        }

        if (data.hasOwnProperty('share')) {
          obj['share'] = _ApiClient["default"].convertToType(data['share'], 'Number');
          delete data['share'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID;
}();
/**
 * @member {module:model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERIDActivities} activities
 */


ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.prototype['activities'] = undefined;
/**
 * The name of the user
 * @member {String} name
 */

ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.prototype['name'] = undefined;
/**
 * The overall count of activities for the user
 * @member {Number} activity_count
 */

ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.prototype['activity_count'] = undefined;
/**
 * The percentage of activities belongs to the user
 * @member {Number} share
 */

ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID.prototype['share'] = undefined;
var _default = ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID;
exports["default"] = _default;
