// pipedrive/dist/model/ActivityDistributionDataWithAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionData = _interopRequireDefault(require("./ActivityDistributionData"));

var _ActivityDistributionDataActivityDistribution = _interopRequireDefault(require("./ActivityDistributionDataActivityDistribution"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityDistributionDataWithAdditionalData model module.
 * @module model/ActivityDistributionDataWithAdditionalData
 * @version 1.0.0
 */
var ActivityDistributionDataWithAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityDistributionDataWithAdditionalData</code>.
   * @alias module:model/ActivityDistributionDataWithAdditionalData
   * @implements module:model/ActivityDistributionData
   * @implements module:model/AdditionalData
   */
  function ActivityDistributionDataWithAdditionalData() {
    (0, _classCallCheck2["default"])(this, ActivityDistributionDataWithAdditionalData);

    _ActivityDistributionData["default"].initialize(this);

    _AdditionalData["default"].initialize(this);

    ActivityDistributionDataWithAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityDistributionDataWithAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityDistributionDataWithAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityDistributionDataWithAdditionalData} obj Optional instance to populate.
     * @return {module:model/ActivityDistributionDataWithAdditionalData} The populated <code>ActivityDistributionDataWithAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityDistributionDataWithAdditionalData();

        _ActivityDistributionData["default"].constructFromObject(data, obj);

        _AdditionalData["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('activity_distribution')) {
          obj['activity_distribution'] = _ActivityDistributionDataActivityDistribution["default"].constructFromObject(data['activity_distribution']);
          delete data['activity_distribution'];
        }

        if (data.hasOwnProperty('start')) {
          obj['start'] = _ApiClient["default"].convertToType(data['start'], 'Number');
          delete data['start'];
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
          delete data['limit'];
        }

        if (data.hasOwnProperty('more_items_in_collection')) {
          obj['more_items_in_collection'] = _ApiClient["default"].convertToType(data['more_items_in_collection'], 'Boolean');
          delete data['more_items_in_collection'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ActivityDistributionDataWithAdditionalData;
}();
/**
 * @member {module:model/ActivityDistributionDataActivityDistribution} activity_distribution
 */


ActivityDistributionDataWithAdditionalData.prototype['activity_distribution'] = undefined;
/**
 * Pagination start
 * @member {Number} start
 */

ActivityDistributionDataWithAdditionalData.prototype['start'] = undefined;
/**
 * Items shown per page
 * @member {Number} limit
 */

ActivityDistributionDataWithAdditionalData.prototype['limit'] = undefined;
/**
 * If there are more list items in the collection than displayed or not
 * @member {Boolean} more_items_in_collection
 */

ActivityDistributionDataWithAdditionalData.prototype['more_items_in_collection'] = undefined; // Implement ActivityDistributionData interface:

/**
 * @member {module:model/ActivityDistributionDataActivityDistribution} activity_distribution
 */

_ActivityDistributionData["default"].prototype['activity_distribution'] = undefined; // Implement AdditionalData interface:

/**
 * Pagination start
 * @member {Number} start
 */

_AdditionalData["default"].prototype['start'] = undefined;
/**
 * Items shown per page
 * @member {Number} limit
 */

_AdditionalData["default"].prototype['limit'] = undefined;
/**
 * If there are more list items in the collection than displayed or not
 * @member {Boolean} more_items_in_collection
 */

_AdditionalData["default"].prototype['more_items_in_collection'] = undefined;
var _default = ActivityDistributionDataWithAdditionalData;
exports["default"] = _default;
