// pipedrive/dist/model/ActivityDistributionDataActivityDistribution.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID = _interopRequireDefault(require("./ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityDistributionDataActivityDistribution model module.
 * @module model/ActivityDistributionDataActivityDistribution
 * @version 1.0.0
 */
var ActivityDistributionDataActivityDistribution = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityDistributionDataActivityDistribution</code>.
   * The distribution of activities related to the organization grouped by the user ID
   * @alias module:model/ActivityDistributionDataActivityDistribution
   */
  function ActivityDistributionDataActivityDistribution() {
    (0, _classCallCheck2["default"])(this, ActivityDistributionDataActivityDistribution);
    ActivityDistributionDataActivityDistribution.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityDistributionDataActivityDistribution, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityDistributionDataActivityDistribution</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityDistributionDataActivityDistribution} obj Optional instance to populate.
     * @return {module:model/ActivityDistributionDataActivityDistribution} The populated <code>ActivityDistributionDataActivityDistribution</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityDistributionDataActivityDistribution();

        if (data.hasOwnProperty('ASSIGNED_TO_USER_ID')) {
          obj['ASSIGNED_TO_USER_ID'] = _ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID["default"].constructFromObject(data['ASSIGNED_TO_USER_ID']);
          delete data['ASSIGNED_TO_USER_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ActivityDistributionDataActivityDistribution;
}();
/**
 * @member {module:model/ActivityDistributionDataActivityDistributionASSIGNEDTOUSERID} ASSIGNED_TO_USER_ID
 */


ActivityDistributionDataActivityDistribution.prototype['ASSIGNED_TO_USER_ID'] = undefined;
var _default = ActivityDistributionDataActivityDistribution;
exports["default"] = _default;
