// pipedrive/dist/model/ActivityRecordAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityRecordAdditionalData model module.
 * @module model/ActivityRecordAdditionalData
 * @version 1.0.0
 */
var ActivityRecordAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityRecordAdditionalData</code>.
   * @alias module:model/ActivityRecordAdditionalData
   */
  function ActivityRecordAdditionalData() {
    (0, _classCallCheck2["default"])(this, ActivityRecordAdditionalData);
    ActivityRecordAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityRecordAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityRecordAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityRecordAdditionalData} obj Optional instance to populate.
     * @return {module:model/ActivityRecordAdditionalData} The populated <code>ActivityRecordAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityRecordAdditionalData();

        if (data.hasOwnProperty('updates_story_id')) {
          obj['updates_story_id'] = _ApiClient["default"].convertToType(data['updates_story_id'], 'Number');
          delete data['updates_story_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ActivityRecordAdditionalData;
}();
/**
 * This field will be deprecated
 * @member {Number} updates_story_id
 */


ActivityRecordAdditionalData.prototype['updates_story_id'] = undefined;
var _default = ActivityRecordAdditionalData;
exports["default"] = _default;
