// pipedrive/dist/model/ActivityDistributionData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataActivityDistribution = _interopRequireDefault(require("./ActivityDistributionDataActivityDistribution"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityDistributionData model module.
 * @module model/ActivityDistributionData
 * @version 1.0.0
 */
var ActivityDistributionData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityDistributionData</code>.
   * @alias module:model/ActivityDistributionData
   */
  function ActivityDistributionData() {
    (0, _classCallCheck2["default"])(this, ActivityDistributionData);
    ActivityDistributionData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityDistributionData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityDistributionData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityDistributionData} obj Optional instance to populate.
     * @return {module:model/ActivityDistributionData} The populated <code>ActivityDistributionData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityDistributionData();

        if (data.hasOwnProperty('activity_distribution')) {
          obj['activity_distribution'] = _ActivityDistributionDataActivityDistribution["default"].constructFromObject(data['activity_distribution']);
          delete data['activity_distribution'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ActivityDistributionData;
}();
/**
 * @member {module:model/ActivityDistributionDataActivityDistribution} activity_distribution
 */


ActivityDistributionData.prototype['activity_distribution'] = undefined;
var _default = ActivityDistributionData;
exports["default"] = _default;
