// pipedrive/dist/model/ActivityTypeCreateRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _IconKey = _interopRequireDefault(require("./IconKey"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ActivityTypeCreateRequest model module.
 * @module model/ActivityTypeCreateRequest
 * @version 1.0.0
 */
var ActivityTypeCreateRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityTypeCreateRequest</code>.
   * @alias module:model/ActivityTypeCreateRequest
   * @param name {String} The name of the ActivityType
   * @param iconKey {module:model/IconKey} 
   */
  function ActivityTypeCreateRequest(name, iconKey) {
    (0, _classCallCheck2["default"])(this, ActivityTypeCreateRequest);
    ActivityTypeCreateRequest.initialize(this, name, iconKey);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ActivityTypeCreateRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, iconKey) {
      obj['name'] = name;
      obj['icon_key'] = iconKey;
    }
    /**
     * Constructs a <code>ActivityTypeCreateRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActivityTypeCreateRequest} obj Optional instance to populate.
     * @return {module:model/ActivityTypeCreateRequest} The populated <code>ActivityTypeCreateRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityTypeCreateRequest();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('icon_key')) {
          obj['icon_key'] = _IconKey["default"].constructFromObject(data['icon_key']);
          delete data['icon_key'];
        }

        if (data.hasOwnProperty('color')) {
          obj['color'] = _ApiClient["default"].convertToType(data['color'], 'String');
          delete data['color'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return ActivityTypeCreateRequest;
}();
/**
 * The name of the ActivityType
 * @member {String} name
 */


ActivityTypeCreateRequest.prototype['name'] = undefined;
/**
 * @member {module:model/IconKey} icon_key
 */

ActivityTypeCreateRequest.prototype['icon_key'] = undefined;
/**
 * A designated color for the ActivityType in 6-character HEX format (e.g. `FFFFFF` for white, `000000` for black)
 * @member {String} color
 */

ActivityTypeCreateRequest.prototype['color'] = undefined;
var _default = ActivityTypeCreateRequest;
exports["default"] = _default;
