// pipedrive/dist/model/AddCallLogAudioFileRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddCallLogAudioFileRequest model module.
 * @module model/AddCallLogAudioFileRequest
 * @version 1.0.0
 */
var AddCallLogAudioFileRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddCallLogAudioFileRequest</code>.
   * @alias module:model/AddCallLogAudioFileRequest
   * @param file {File} Audio file supported by the HTML5 specification
   */
  function AddCallLogAudioFileRequest(file) {
    (0, _classCallCheck2["default"])(this, AddCallLogAudioFileRequest);
    AddCallLogAudioFileRequest.initialize(this, file);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddCallLogAudioFileRequest, null, [{
    key: "initialize",
    value: function initialize(obj, file) {
      obj['file'] = file;
    }
    /**
     * Constructs a <code>AddCallLogAudioFileRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddCallLogAudioFileRequest} obj Optional instance to populate.
     * @return {module:model/AddCallLogAudioFileRequest} The populated <code>AddCallLogAudioFileRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddCallLogAudioFileRequest();

        if (data.hasOwnProperty('file')) {
          obj['file'] = _ApiClient["default"].convertToType(data['file'], File);
          delete data['file'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return AddCallLogAudioFileRequest;
}();
/**
 * Audio file supported by the HTML5 specification
 * @member {File} file
 */


AddCallLogAudioFileRequest.prototype['file'] = undefined;
var _default = AddCallLogAudioFileRequest;
exports["default"] = _default;
