// pipedrive/dist/model/AddDealParticipantRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddDealParticipantRequest model module.
 * @module model/AddDealParticipantRequest
 * @version 1.0.0
 */
var AddDealParticipantRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddDealParticipantRequest</code>.
   * @alias module:model/AddDealParticipantRequest
   * @param personId {Number} The ID of the person
   */
  function AddDealParticipantRequest(personId) {
    (0, _classCallCheck2["default"])(this, AddDealParticipantRequest);
    AddDealParticipantRequest.initialize(this, personId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddDealParticipantRequest, null, [{
    key: "initialize",
    value: function initialize(obj, personId) {
      obj['person_id'] = personId;
    }
    /**
     * Constructs a <code>AddDealParticipantRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddDealParticipantRequest} obj Optional instance to populate.
     * @return {module:model/AddDealParticipantRequest} The populated <code>AddDealParticipantRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddDealParticipantRequest();

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddDealParticipantRequest;
}();
/**
 * The ID of the person
 * @member {Number} person_id
 */


AddDealParticipantRequest.prototype['person_id'] = undefined;
var _default = AddDealParticipantRequest;
exports["default"] = _default;
