// pipedrive/dist/model/AddedDealFollowerData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddedDealFollowerData model module.
 * @module model/AddedDealFollowerData
 * @version 1.0.0
 */
var AddedDealFollowerData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddedDealFollowerData</code>.
   * @alias module:model/AddedDealFollowerData
   */
  function AddedDealFollowerData() {
    (0, _classCallCheck2["default"])(this, AddedDealFollowerData);
    AddedDealFollowerData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddedDealFollowerData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddedDealFollowerData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddedDealFollowerData} obj Optional instance to populate.
     * @return {module:model/AddedDealFollowerData} The populated <code>AddedDealFollowerData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddedDealFollowerData();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddedDealFollowerData;
}();
/**
 * The user ID who added the follower
 * @member {Number} user_id
 */


AddedDealFollowerData.prototype['user_id'] = undefined;
/**
 * The follower ID
 * @member {Number} id
 */

AddedDealFollowerData.prototype['id'] = undefined;
/**
 * The ID of the deal which the follower was added to
 * @member {Number} deal_id
 */

AddedDealFollowerData.prototype['deal_id'] = undefined;
/**
 * The date and time when the deal follower was added
 * @member {String} add_time
 */

AddedDealFollowerData.prototype['add_time'] = undefined;
var _default = AddedDealFollowerData;
exports["default"] = _default;
