// pipedrive/dist/model/AddFileRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddFileRequest model module.
 * @module model/AddFileRequest
 * @version 1.0.0
 */
var AddFileRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddFileRequest</code>.
   * @alias module:model/AddFileRequest
   * @param file {File} A single file, supplied in the multipart/form-data encoding and contained within the given boundaries.
   */
  function AddFileRequest(file) {
    (0, _classCallCheck2["default"])(this, AddFileRequest);
    AddFileRequest.initialize(this, file);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddFileRequest, null, [{
    key: "initialize",
    value: function initialize(obj, file) {
      obj['file'] = file;
    }
    /**
     * Constructs a <code>AddFileRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddFileRequest} obj Optional instance to populate.
     * @return {module:model/AddFileRequest} The populated <code>AddFileRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddFileRequest();

        if (data.hasOwnProperty('file')) {
          obj['file'] = _ApiClient["default"].convertToType(data['file'], File);
          delete data['file'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (data.hasOwnProperty('product_id')) {
          obj['product_id'] = _ApiClient["default"].convertToType(data['product_id'], 'Number');
          delete data['product_id'];
        }

        if (data.hasOwnProperty('activity_id')) {
          obj['activity_id'] = _ApiClient["default"].convertToType(data['activity_id'], 'Number');
          delete data['activity_id'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return AddFileRequest;
}();
/**
 * A single file, supplied in the multipart/form-data encoding and contained within the given boundaries.
 * @member {File} file
 */


AddFileRequest.prototype['file'] = undefined;
/**
 * ID of the deal to associate file(s) with
 * @member {Number} deal_id
 */

AddFileRequest.prototype['deal_id'] = undefined;
/**
 * ID of the person to associate file(s) with
 * @member {Number} person_id
 */

AddFileRequest.prototype['person_id'] = undefined;
/**
 * ID of the organization to associate file(s) with
 * @member {Number} org_id
 */

AddFileRequest.prototype['org_id'] = undefined;
/**
 * ID of the product to associate file(s) with
 * @member {Number} product_id
 */

AddFileRequest.prototype['product_id'] = undefined;
/**
 * ID of the activity to associate file(s) with
 * @member {Number} activity_id
 */

AddFileRequest.prototype['activity_id'] = undefined;
var _default = AddFileRequest;
exports["default"] = _default;
