// pipedrive/dist/model/AdditionalDataWithPagination.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AdditionalDataWithPagination model module.
 * @module model/AdditionalDataWithPagination
 * @version 1.0.0
 */
var AdditionalDataWithPagination = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AdditionalDataWithPagination</code>.
   * @alias module:model/AdditionalDataWithPagination
   */
  function AdditionalDataWithPagination() {
    (0, _classCallCheck2["default"])(this, AdditionalDataWithPagination);
    AdditionalDataWithPagination.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AdditionalDataWithPagination, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AdditionalDataWithPagination</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AdditionalDataWithPagination} obj Optional instance to populate.
     * @return {module:model/AdditionalDataWithPagination} The populated <code>AdditionalDataWithPagination</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AdditionalDataWithPagination();

        if (data.hasOwnProperty('pagination')) {
          obj['pagination'] = _ApiClient["default"].convertToType(data['pagination'], _AdditionalData["default"]);
          delete data['pagination'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AdditionalDataWithPagination;
}();
/**
 * The pagination details of the list
 * @member {module:model/AdditionalData} pagination
 */


AdditionalDataWithPagination.prototype['pagination'] = undefined;
var _default = AdditionalDataWithPagination;
exports["default"] = _default;
