// pipedrive/dist/model/AdditionalPersonInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Owner = _interopRequireDefault(require("./Owner"));

var _PersonCountEmailDealAndActivityInfo = _interopRequireDefault(require("./PersonCountEmailDealAndActivityInfo"));

var _PersonNameInfoWithOrgAndOwnerId = _interopRequireDefault(require("./PersonNameInfoWithOrgAndOwnerId"));

var _RelationshipOrganizationInfoItemWithActiveFlag = _interopRequireDefault(require("./RelationshipOrganizationInfoItemWithActiveFlag"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AdditionalPersonInfo model module.
 * @module model/AdditionalPersonInfo
 * @version 1.0.0
 */
var AdditionalPersonInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AdditionalPersonInfo</code>.
   * @alias module:model/AdditionalPersonInfo
   * @implements module:model/PersonNameInfoWithOrgAndOwnerId
   * @implements module:model/PersonCountEmailDealAndActivityInfo
   */
  function AdditionalPersonInfo() {
    (0, _classCallCheck2["default"])(this, AdditionalPersonInfo);

    _PersonNameInfoWithOrgAndOwnerId["default"].initialize(this);

    _PersonCountEmailDealAndActivityInfo["default"].initialize(this);

    AdditionalPersonInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AdditionalPersonInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AdditionalPersonInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AdditionalPersonInfo} obj Optional instance to populate.
     * @return {module:model/AdditionalPersonInfo} The populated <code>AdditionalPersonInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AdditionalPersonInfo();

        _PersonNameInfoWithOrgAndOwnerId["default"].constructFromObject(data, obj);

        _PersonCountEmailDealAndActivityInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _Owner["default"].constructFromObject(data['owner_id']);
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _RelationshipOrganizationInfoItemWithActiveFlag["default"].constructFromObject(data['org_id']);
          delete data['org_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
          delete data['first_name'];
        }

        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
          delete data['last_name'];
        }

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('last_incoming_mail_time')) {
          obj['last_incoming_mail_time'] = _ApiClient["default"].convertToType(data['last_incoming_mail_time'], 'String');
          delete data['last_incoming_mail_time'];
        }

        if (data.hasOwnProperty('last_outgoing_mail_time')) {
          obj['last_outgoing_mail_time'] = _ApiClient["default"].convertToType(data['last_outgoing_mail_time'], 'String');
          delete data['last_outgoing_mail_time'];
        }

        if (data.hasOwnProperty('open_deals_count')) {
          obj['open_deals_count'] = _ApiClient["default"].convertToType(data['open_deals_count'], 'Number');
          delete data['open_deals_count'];
        }

        if (data.hasOwnProperty('related_open_deals_count')) {
          obj['related_open_deals_count'] = _ApiClient["default"].convertToType(data['related_open_deals_count'], 'Number');
          delete data['related_open_deals_count'];
        }

        if (data.hasOwnProperty('closed_deals_count')) {
          obj['closed_deals_count'] = _ApiClient["default"].convertToType(data['closed_deals_count'], 'Number');
          delete data['closed_deals_count'];
        }

        if (data.hasOwnProperty('related_closed_deals_count')) {
          obj['related_closed_deals_count'] = _ApiClient["default"].convertToType(data['related_closed_deals_count'], 'Number');
          delete data['related_closed_deals_count'];
        }

        if (data.hasOwnProperty('won_deals_count')) {
          obj['won_deals_count'] = _ApiClient["default"].convertToType(data['won_deals_count'], 'Number');
          delete data['won_deals_count'];
        }

        if (data.hasOwnProperty('related_won_deals_count')) {
          obj['related_won_deals_count'] = _ApiClient["default"].convertToType(data['related_won_deals_count'], 'Number');
          delete data['related_won_deals_count'];
        }

        if (data.hasOwnProperty('lost_deals_count')) {
          obj['lost_deals_count'] = _ApiClient["default"].convertToType(data['lost_deals_count'], 'Number');
          delete data['lost_deals_count'];
        }

        if (data.hasOwnProperty('related_lost_deals_count')) {
          obj['related_lost_deals_count'] = _ApiClient["default"].convertToType(data['related_lost_deals_count'], 'Number');
          delete data['related_lost_deals_count'];
        }

        if (data.hasOwnProperty('next_activity_date')) {
          obj['next_activity_date'] = _ApiClient["default"].convertToType(data['next_activity_date'], 'String');
          delete data['next_activity_date'];
        }

        if (data.hasOwnProperty('next_activity_time')) {
          obj['next_activity_time'] = _ApiClient["default"].convertToType(data['next_activity_time'], 'String');
          delete data['next_activity_time'];
        }

        if (data.hasOwnProperty('next_activity_id')) {
          obj['next_activity_id'] = _ApiClient["default"].convertToType(data['next_activity_id'], 'Number');
          delete data['next_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_id')) {
          obj['last_activity_id'] = _ApiClient["default"].convertToType(data['last_activity_id'], 'Number');
          delete data['last_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_date')) {
          obj['last_activity_date'] = _ApiClient["default"].convertToType(data['last_activity_date'], 'String');
          delete data['last_activity_date'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AdditionalPersonInfo;
}();
/**
 * @member {module:model/Owner} owner_id
 */


AdditionalPersonInfo.prototype['owner_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItemWithActiveFlag} org_id
 */

AdditionalPersonInfo.prototype['org_id'] = undefined;
/**
 * The name of the person
 * @member {String} name
 */

AdditionalPersonInfo.prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

AdditionalPersonInfo.prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

AdditionalPersonInfo.prototype['last_name'] = undefined;
/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

AdditionalPersonInfo.prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

AdditionalPersonInfo.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

AdditionalPersonInfo.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

AdditionalPersonInfo.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

AdditionalPersonInfo.prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

AdditionalPersonInfo.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

AdditionalPersonInfo.prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

AdditionalPersonInfo.prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

AdditionalPersonInfo.prototype['last_outgoing_mail_time'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

AdditionalPersonInfo.prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

AdditionalPersonInfo.prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

AdditionalPersonInfo.prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

AdditionalPersonInfo.prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

AdditionalPersonInfo.prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

AdditionalPersonInfo.prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

AdditionalPersonInfo.prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

AdditionalPersonInfo.prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

AdditionalPersonInfo.prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

AdditionalPersonInfo.prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

AdditionalPersonInfo.prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

AdditionalPersonInfo.prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

AdditionalPersonInfo.prototype['last_activity_date'] = undefined; // Implement PersonNameInfoWithOrgAndOwnerId interface:

/**
 * @member {module:model/Owner} owner_id
 */

_PersonNameInfoWithOrgAndOwnerId["default"].prototype['owner_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItemWithActiveFlag} org_id
 */

_PersonNameInfoWithOrgAndOwnerId["default"].prototype['org_id'] = undefined;
/**
 * The name of the person
 * @member {String} name
 */

_PersonNameInfoWithOrgAndOwnerId["default"].prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

_PersonNameInfoWithOrgAndOwnerId["default"].prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

_PersonNameInfoWithOrgAndOwnerId["default"].prototype['last_name'] = undefined; // Implement PersonCountEmailDealAndActivityInfo interface:

/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['last_outgoing_mail_time'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

_PersonCountEmailDealAndActivityInfo["default"].prototype['last_activity_date'] = undefined;
var _default = AdditionalPersonInfo;
exports["default"] = _default;
