// pipedrive/dist/model/AdditionalBaseOrganizationItemInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealsCountAndActivityInfo = _interopRequireDefault(require("./DealsCountAndActivityInfo"));

var _OrganizationCountAndAddressInfo = _interopRequireDefault(require("./OrganizationCountAndAddressInfo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AdditionalBaseOrganizationItemInfo model module.
 * @module model/AdditionalBaseOrganizationItemInfo
 * @version 1.0.0
 */
var AdditionalBaseOrganizationItemInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AdditionalBaseOrganizationItemInfo</code>.
   * @alias module:model/AdditionalBaseOrganizationItemInfo
   * @implements module:model/OrganizationCountAndAddressInfo
   * @implements module:model/DealsCountAndActivityInfo
   */
  function AdditionalBaseOrganizationItemInfo() {
    (0, _classCallCheck2["default"])(this, AdditionalBaseOrganizationItemInfo);

    _OrganizationCountAndAddressInfo["default"].initialize(this);

    _DealsCountAndActivityInfo["default"].initialize(this);

    AdditionalBaseOrganizationItemInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AdditionalBaseOrganizationItemInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AdditionalBaseOrganizationItemInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AdditionalBaseOrganizationItemInfo} obj Optional instance to populate.
     * @return {module:model/AdditionalBaseOrganizationItemInfo} The populated <code>AdditionalBaseOrganizationItemInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AdditionalBaseOrganizationItemInfo();

        _OrganizationCountAndAddressInfo["default"].constructFromObject(data, obj);

        _DealsCountAndActivityInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('people_count')) {
          obj['people_count'] = _ApiClient["default"].convertToType(data['people_count'], 'Number');
          delete data['people_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (data.hasOwnProperty('address_subpremise')) {
          obj['address_subpremise'] = _ApiClient["default"].convertToType(data['address_subpremise'], 'String');
          delete data['address_subpremise'];
        }

        if (data.hasOwnProperty('address_street_number')) {
          obj['address_street_number'] = _ApiClient["default"].convertToType(data['address_street_number'], 'String');
          delete data['address_street_number'];
        }

        if (data.hasOwnProperty('address_route')) {
          obj['address_route'] = _ApiClient["default"].convertToType(data['address_route'], 'String');
          delete data['address_route'];
        }

        if (data.hasOwnProperty('address_sublocality')) {
          obj['address_sublocality'] = _ApiClient["default"].convertToType(data['address_sublocality'], 'String');
          delete data['address_sublocality'];
        }

        if (data.hasOwnProperty('address_locality')) {
          obj['address_locality'] = _ApiClient["default"].convertToType(data['address_locality'], 'String');
          delete data['address_locality'];
        }

        if (data.hasOwnProperty('address_admin_area_level_1')) {
          obj['address_admin_area_level_1'] = _ApiClient["default"].convertToType(data['address_admin_area_level_1'], 'String');
          delete data['address_admin_area_level_1'];
        }

        if (data.hasOwnProperty('address_admin_area_level_2')) {
          obj['address_admin_area_level_2'] = _ApiClient["default"].convertToType(data['address_admin_area_level_2'], 'String');
          delete data['address_admin_area_level_2'];
        }

        if (data.hasOwnProperty('address_country')) {
          obj['address_country'] = _ApiClient["default"].convertToType(data['address_country'], 'String');
          delete data['address_country'];
        }

        if (data.hasOwnProperty('address_postal_code')) {
          obj['address_postal_code'] = _ApiClient["default"].convertToType(data['address_postal_code'], 'String');
          delete data['address_postal_code'];
        }

        if (data.hasOwnProperty('address_formatted_address')) {
          obj['address_formatted_address'] = _ApiClient["default"].convertToType(data['address_formatted_address'], 'String');
          delete data['address_formatted_address'];
        }

        if (data.hasOwnProperty('open_deals_count')) {
          obj['open_deals_count'] = _ApiClient["default"].convertToType(data['open_deals_count'], 'Number');
          delete data['open_deals_count'];
        }

        if (data.hasOwnProperty('related_open_deals_count')) {
          obj['related_open_deals_count'] = _ApiClient["default"].convertToType(data['related_open_deals_count'], 'Number');
          delete data['related_open_deals_count'];
        }

        if (data.hasOwnProperty('closed_deals_count')) {
          obj['closed_deals_count'] = _ApiClient["default"].convertToType(data['closed_deals_count'], 'Number');
          delete data['closed_deals_count'];
        }

        if (data.hasOwnProperty('related_closed_deals_count')) {
          obj['related_closed_deals_count'] = _ApiClient["default"].convertToType(data['related_closed_deals_count'], 'Number');
          delete data['related_closed_deals_count'];
        }

        if (data.hasOwnProperty('won_deals_count')) {
          obj['won_deals_count'] = _ApiClient["default"].convertToType(data['won_deals_count'], 'Number');
          delete data['won_deals_count'];
        }

        if (data.hasOwnProperty('related_won_deals_count')) {
          obj['related_won_deals_count'] = _ApiClient["default"].convertToType(data['related_won_deals_count'], 'Number');
          delete data['related_won_deals_count'];
        }

        if (data.hasOwnProperty('lost_deals_count')) {
          obj['lost_deals_count'] = _ApiClient["default"].convertToType(data['lost_deals_count'], 'Number');
          delete data['lost_deals_count'];
        }

        if (data.hasOwnProperty('related_lost_deals_count')) {
          obj['related_lost_deals_count'] = _ApiClient["default"].convertToType(data['related_lost_deals_count'], 'Number');
          delete data['related_lost_deals_count'];
        }

        if (data.hasOwnProperty('next_activity_date')) {
          obj['next_activity_date'] = _ApiClient["default"].convertToType(data['next_activity_date'], 'String');
          delete data['next_activity_date'];
        }

        if (data.hasOwnProperty('next_activity_time')) {
          obj['next_activity_time'] = _ApiClient["default"].convertToType(data['next_activity_time'], 'String');
          delete data['next_activity_time'];
        }

        if (data.hasOwnProperty('next_activity_id')) {
          obj['next_activity_id'] = _ApiClient["default"].convertToType(data['next_activity_id'], 'Number');
          delete data['next_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_id')) {
          obj['last_activity_id'] = _ApiClient["default"].convertToType(data['last_activity_id'], 'Number');
          delete data['last_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_date')) {
          obj['last_activity_date'] = _ApiClient["default"].convertToType(data['last_activity_date'], 'String');
          delete data['last_activity_date'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AdditionalBaseOrganizationItemInfo;
}();
/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */


AdditionalBaseOrganizationItemInfo.prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

AdditionalBaseOrganizationItemInfo.prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

AdditionalBaseOrganizationItemInfo.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

AdditionalBaseOrganizationItemInfo.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

AdditionalBaseOrganizationItemInfo.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

AdditionalBaseOrganizationItemInfo.prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

AdditionalBaseOrganizationItemInfo.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

AdditionalBaseOrganizationItemInfo.prototype['followers_count'] = undefined;
/**
 * The full address of the organization
 * @member {String} address
 */

AdditionalBaseOrganizationItemInfo.prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

AdditionalBaseOrganizationItemInfo.prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

AdditionalBaseOrganizationItemInfo.prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

AdditionalBaseOrganizationItemInfo.prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

AdditionalBaseOrganizationItemInfo.prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

AdditionalBaseOrganizationItemInfo.prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

AdditionalBaseOrganizationItemInfo.prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

AdditionalBaseOrganizationItemInfo.prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

AdditionalBaseOrganizationItemInfo.prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

AdditionalBaseOrganizationItemInfo.prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

AdditionalBaseOrganizationItemInfo.prototype['address_formatted_address'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

AdditionalBaseOrganizationItemInfo.prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

AdditionalBaseOrganizationItemInfo.prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

AdditionalBaseOrganizationItemInfo.prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

AdditionalBaseOrganizationItemInfo.prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

AdditionalBaseOrganizationItemInfo.prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

AdditionalBaseOrganizationItemInfo.prototype['last_activity_date'] = undefined; // Implement OrganizationCountAndAddressInfo interface:

/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

_OrganizationCountAndAddressInfo["default"].prototype['followers_count'] = undefined;
/**
 * The full address of the organization
 * @member {String} address
 */

_OrganizationCountAndAddressInfo["default"].prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

_OrganizationCountAndAddressInfo["default"].prototype['address_formatted_address'] = undefined; // Implement DealsCountAndActivityInfo interface:

/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

_DealsCountAndActivityInfo["default"].prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

_DealsCountAndActivityInfo["default"].prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

_DealsCountAndActivityInfo["default"].prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

_DealsCountAndActivityInfo["default"].prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

_DealsCountAndActivityInfo["default"].prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

_DealsCountAndActivityInfo["default"].prototype['last_activity_date'] = undefined;
var _default = AdditionalBaseOrganizationItemInfo;
exports["default"] = _default;
