// pipedrive/dist/model/AddLeadLabelRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LeadLabelColor = _interopRequireDefault(require("./LeadLabelColor"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddLeadLabelRequest model module.
 * @module model/AddLeadLabelRequest
 * @version 1.0.0
 */
var AddLeadLabelRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddLeadLabelRequest</code>.
   * @alias module:model/AddLeadLabelRequest
   * @param name {String} The name of the lead label
   * @param color {module:model/LeadLabelColor} 
   */
  function AddLeadLabelRequest(name, color) {
    (0, _classCallCheck2["default"])(this, AddLeadLabelRequest);
    AddLeadLabelRequest.initialize(this, name, color);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddLeadLabelRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, color) {
      obj['name'] = name;
      obj['color'] = color;
    }
    /**
     * Constructs a <code>AddLeadLabelRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddLeadLabelRequest} obj Optional instance to populate.
     * @return {module:model/AddLeadLabelRequest} The populated <code>AddLeadLabelRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddLeadLabelRequest();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('color')) {
          obj['color'] = _LeadLabelColor["default"].constructFromObject(data['color']);
          delete data['color'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddLeadLabelRequest;
}();
/**
 * The name of the lead label
 * @member {String} name
 */


AddLeadLabelRequest.prototype['name'] = undefined;
/**
 * @member {module:model/LeadLabelColor} color
 */

AddLeadLabelRequest.prototype['color'] = undefined;
var _default = AddLeadLabelRequest;
exports["default"] = _default;
