// pipedrive/dist/model/AddNewPipelineAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasePipeline = _interopRequireDefault(require("./BasePipeline"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddNewPipelineAllOf model module.
 * @module model/AddNewPipelineAllOf
 * @version 1.0.0
 */
var AddNewPipelineAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddNewPipelineAllOf</code>.
   * @alias module:model/AddNewPipelineAllOf
   */
  function AddNewPipelineAllOf() {
    (0, _classCallCheck2["default"])(this, AddNewPipelineAllOf);
    AddNewPipelineAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddNewPipelineAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddNewPipelineAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddNewPipelineAllOf} obj Optional instance to populate.
     * @return {module:model/AddNewPipelineAllOf} The populated <code>AddNewPipelineAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddNewPipelineAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BasePipeline["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddNewPipelineAllOf;
}();
/**
 * @member {module:model/BasePipeline} data
 */


AddNewPipelineAllOf.prototype['data'] = undefined;
var _default = AddNewPipelineAllOf;
exports["default"] = _default;
