// pipedrive/dist/model/AddNewPipeline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AddNewPipelineAllOf = _interopRequireDefault(require("./AddNewPipelineAllOf"));

var _BasePipeline = _interopRequireDefault(require("./BasePipeline"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddNewPipeline model module.
 * @module model/AddNewPipeline
 * @version 1.0.0
 */
var AddNewPipeline = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddNewPipeline</code>.
   * @alias module:model/AddNewPipeline
   * @implements module:model/BaseResponse
   * @implements module:model/AddNewPipelineAllOf
   */
  function AddNewPipeline() {
    (0, _classCallCheck2["default"])(this, AddNewPipeline);

    _BaseResponse["default"].initialize(this);

    _AddNewPipelineAllOf["default"].initialize(this);

    AddNewPipeline.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddNewPipeline, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddNewPipeline</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddNewPipeline} obj Optional instance to populate.
     * @return {module:model/AddNewPipeline} The populated <code>AddNewPipeline</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddNewPipeline();

        _BaseResponse["default"].constructFromObject(data, obj);

        _AddNewPipelineAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BasePipeline["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddNewPipeline;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


AddNewPipeline.prototype['success'] = undefined;
/**
 * @member {module:model/BasePipeline} data
 */

AddNewPipeline.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement AddNewPipelineAllOf interface:

/**
 * @member {module:model/BasePipeline} data
 */

_AddNewPipelineAllOf["default"].prototype['data'] = undefined;
var _default = AddNewPipeline;
exports["default"] = _default;
