// pipedrive/dist/model/AddOrUpdateRoleSettingRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddOrUpdateRoleSettingRequest model module.
 * @module model/AddOrUpdateRoleSettingRequest
 * @version 1.0.0
 */
var AddOrUpdateRoleSettingRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddOrUpdateRoleSettingRequest</code>.
   * @alias module:model/AddOrUpdateRoleSettingRequest
   * @param settingKey {module:model/AddOrUpdateRoleSettingRequest.SettingKeyEnum} 
   * @param value {module:model/NumberBooleanDefault0} Possible values for `default_visibility` settings: `0` or `1`
   */
  function AddOrUpdateRoleSettingRequest(settingKey, value) {
    (0, _classCallCheck2["default"])(this, AddOrUpdateRoleSettingRequest);
    AddOrUpdateRoleSettingRequest.initialize(this, settingKey, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddOrUpdateRoleSettingRequest, null, [{
    key: "initialize",
    value: function initialize(obj, settingKey, value) {
      obj['setting_key'] = settingKey;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>AddOrUpdateRoleSettingRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddOrUpdateRoleSettingRequest} obj Optional instance to populate.
     * @return {module:model/AddOrUpdateRoleSettingRequest} The populated <code>AddOrUpdateRoleSettingRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddOrUpdateRoleSettingRequest();

        if (data.hasOwnProperty('setting_key')) {
          obj['setting_key'] = _ApiClient["default"].convertToType(data['setting_key'], 'String');
          delete data['setting_key'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], _NumberBooleanDefault["default"]);
          delete data['value'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return AddOrUpdateRoleSettingRequest;
}();
/**
 * @member {module:model/AddOrUpdateRoleSettingRequest.SettingKeyEnum} setting_key
 */


AddOrUpdateRoleSettingRequest.prototype['setting_key'] = undefined;
/**
 * Possible values for `default_visibility` settings: `0` or `1`
 * @member {module:model/NumberBooleanDefault0} value
 */

AddOrUpdateRoleSettingRequest.prototype['value'] = undefined;
/**
 * Allowed values for the <code>setting_key</code> property.
 * @enum {String}
 * @readonly
 */

AddOrUpdateRoleSettingRequest['SettingKeyEnum'] = {
  /**
   * value: "deal_default_visibility"
   * @const
   */
  "deal_default_visibility": "deal_default_visibility",

  /**
   * value: "org_default_visibility"
   * @const
   */
  "org_default_visibility": "org_default_visibility",

  /**
   * value: "person_default_visibility"
   * @const
   */
  "person_default_visibility": "person_default_visibility",

  /**
   * value: "product_default_visibility"
   * @const
   */
  "product_default_visibility": "product_default_visibility"
};
var _default = AddOrUpdateRoleSettingRequest;
exports["default"] = _default;
