// pipedrive/dist/model/AddPersonFollowerRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddPersonFollowerRequest model module.
 * @module model/AddPersonFollowerRequest
 * @version 1.0.0
 */
var AddPersonFollowerRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddPersonFollowerRequest</code>.
   * @alias module:model/AddPersonFollowerRequest
   * @param userId {Number} The ID of the user
   */
  function AddPersonFollowerRequest(userId) {
    (0, _classCallCheck2["default"])(this, AddPersonFollowerRequest);
    AddPersonFollowerRequest.initialize(this, userId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddPersonFollowerRequest, null, [{
    key: "initialize",
    value: function initialize(obj, userId) {
      obj['user_id'] = userId;
    }
    /**
     * Constructs a <code>AddPersonFollowerRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddPersonFollowerRequest} obj Optional instance to populate.
     * @return {module:model/AddPersonFollowerRequest} The populated <code>AddPersonFollowerRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddPersonFollowerRequest();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddPersonFollowerRequest;
}();
/**
 * The ID of the user
 * @member {Number} user_id
 */


AddPersonFollowerRequest.prototype['user_id'] = undefined;
var _default = AddPersonFollowerRequest;
exports["default"] = _default;
