// pipedrive/dist/model/AddPersonPictureResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AddPersonPictureResponseAllOf = _interopRequireDefault(require("./AddPersonPictureResponseAllOf"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _RelatedPictureData = _interopRequireDefault(require("./RelatedPictureData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddPersonPictureResponse model module.
 * @module model/AddPersonPictureResponse
 * @version 1.0.0
 */
var AddPersonPictureResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddPersonPictureResponse</code>.
   * @alias module:model/AddPersonPictureResponse
   * @implements module:model/BaseResponse
   * @implements module:model/AddPersonPictureResponseAllOf
   */
  function AddPersonPictureResponse() {
    (0, _classCallCheck2["default"])(this, AddPersonPictureResponse);

    _BaseResponse["default"].initialize(this);

    _AddPersonPictureResponseAllOf["default"].initialize(this);

    AddPersonPictureResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddPersonPictureResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddPersonPictureResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddPersonPictureResponse} obj Optional instance to populate.
     * @return {module:model/AddPersonPictureResponse} The populated <code>AddPersonPictureResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddPersonPictureResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _AddPersonPictureResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _RelatedPictureData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddPersonPictureResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


AddPersonPictureResponse.prototype['success'] = undefined;
/**
 * @member {module:model/RelatedPictureData} data
 */

AddPersonPictureResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement AddPersonPictureResponseAllOf interface:

/**
 * @member {module:model/RelatedPictureData} data
 */

_AddPersonPictureResponseAllOf["default"].prototype['data'] = undefined;
var _default = AddPersonPictureResponse;
exports["default"] = _default;
