// pipedrive/dist/model/AddPersonResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AddPersonResponseAllOf = _interopRequireDefault(require("./AddPersonResponseAllOf"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListProductsResponseAllOfRelatedObjects = _interopRequireDefault(require("./ListProductsResponseAllOfRelatedObjects"));

var _PersonItem = _interopRequireDefault(require("./PersonItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddPersonResponse model module.
 * @module model/AddPersonResponse
 * @version 1.0.0
 */
var AddPersonResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddPersonResponse</code>.
   * @alias module:model/AddPersonResponse
   * @implements module:model/BaseResponse
   * @implements module:model/AddPersonResponseAllOf
   */
  function AddPersonResponse() {
    (0, _classCallCheck2["default"])(this, AddPersonResponse);

    _BaseResponse["default"].initialize(this);

    _AddPersonResponseAllOf["default"].initialize(this);

    AddPersonResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddPersonResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddPersonResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddPersonResponse} obj Optional instance to populate.
     * @return {module:model/AddPersonResponse} The populated <code>AddPersonResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddPersonResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _AddPersonResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _PersonItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _ListProductsResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddPersonResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


AddPersonResponse.prototype['success'] = undefined;
/**
 * @member {module:model/PersonItem} data
 */

AddPersonResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ListProductsResponseAllOfRelatedObjects} related_objects
 */

AddPersonResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement AddPersonResponseAllOf interface:

/**
 * @member {module:model/PersonItem} data
 */

_AddPersonResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ListProductsResponseAllOfRelatedObjects} related_objects
 */

_AddPersonResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = AddPersonResponse;
exports["default"] = _default;
