// pipedrive/dist/model/AddProductAttachmentDetailsAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddProductAttachmentDetailsAllOf model module.
 * @module model/AddProductAttachmentDetailsAllOf
 * @version 1.0.0
 */
var AddProductAttachmentDetailsAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddProductAttachmentDetailsAllOf</code>.
   * @alias module:model/AddProductAttachmentDetailsAllOf
   */
  function AddProductAttachmentDetailsAllOf() {
    (0, _classCallCheck2["default"])(this, AddProductAttachmentDetailsAllOf);
    AddProductAttachmentDetailsAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddProductAttachmentDetailsAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AddProductAttachmentDetailsAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddProductAttachmentDetailsAllOf} obj Optional instance to populate.
     * @return {module:model/AddProductAttachmentDetailsAllOf} The populated <code>AddProductAttachmentDetailsAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddProductAttachmentDetailsAllOf();

        if (data.hasOwnProperty('product_attachment_id')) {
          obj['product_attachment_id'] = _ApiClient["default"].convertToType(data['product_attachment_id'], 'Number');
          delete data['product_attachment_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AddProductAttachmentDetailsAllOf;
}();
/**
 * The ID of the deal-product (the ID of the product attached to the deal)
 * @member {Number} product_attachment_id
 */


AddProductAttachmentDetailsAllOf.prototype['product_attachment_id'] = undefined;
var _default = AddProductAttachmentDetailsAllOf;
exports["default"] = _default;
