// pipedrive/dist/model/AddUserRoleAssignmentRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddUserRoleAssignmentRequest model module.
 * @module model/AddUserRoleAssignmentRequest
 * @version 1.0.0
 */
var AddUserRoleAssignmentRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddUserRoleAssignmentRequest</code>.
   * @alias module:model/AddUserRoleAssignmentRequest
   * @param roleId {Number} ID of the role
   */
  function AddUserRoleAssignmentRequest(roleId) {
    (0, _classCallCheck2["default"])(this, AddUserRoleAssignmentRequest);
    AddUserRoleAssignmentRequest.initialize(this, roleId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddUserRoleAssignmentRequest, null, [{
    key: "initialize",
    value: function initialize(obj, roleId) {
      obj['role_id'] = roleId;
    }
    /**
     * Constructs a <code>AddUserRoleAssignmentRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddUserRoleAssignmentRequest} obj Optional instance to populate.
     * @return {module:model/AddUserRoleAssignmentRequest} The populated <code>AddUserRoleAssignmentRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddUserRoleAssignmentRequest();

        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], 'Number');
          delete data['role_id'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return AddUserRoleAssignmentRequest;
}();
/**
 * ID of the role
 * @member {Number} role_id
 */


AddUserRoleAssignmentRequest.prototype['role_id'] = undefined;
var _default = AddUserRoleAssignmentRequest;
exports["default"] = _default;
