// pipedrive/dist/model/AddUserBlacklistedEmailRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AddUserBlacklistedEmailRequest model module.
 * @module model/AddUserBlacklistedEmailRequest
 * @version 1.0.0
 */
var AddUserBlacklistedEmailRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddUserBlacklistedEmailRequest</code>.
   * @alias module:model/AddUserBlacklistedEmailRequest
   * @param address {String} Email address to blacklist (can contain \\\\* for wildcards, e.g. \\\\*@example.com, or john\\\\*@ex\\\\*.com)
   */
  function AddUserBlacklistedEmailRequest(address) {
    (0, _classCallCheck2["default"])(this, AddUserBlacklistedEmailRequest);
    AddUserBlacklistedEmailRequest.initialize(this, address);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AddUserBlacklistedEmailRequest, null, [{
    key: "initialize",
    value: function initialize(obj, address) {
      obj['address'] = address;
    }
    /**
     * Constructs a <code>AddUserBlacklistedEmailRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddUserBlacklistedEmailRequest} obj Optional instance to populate.
     * @return {module:model/AddUserBlacklistedEmailRequest} The populated <code>AddUserBlacklistedEmailRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddUserBlacklistedEmailRequest();

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return AddUserBlacklistedEmailRequest;
}();
/**
 * Email address to blacklist (can contain \\\\* for wildcards, e.g. \\\\*@example.com, or john\\\\*@ex\\\\*.com)
 * @member {String} address
 */


AddUserBlacklistedEmailRequest.prototype['address'] = undefined;
var _default = AddUserBlacklistedEmailRequest;
exports["default"] = _default;
