// pipedrive/dist/model/AllOrganizationRelationshipsGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _AllOrganizationRelationshipsGetResponseAllOf = _interopRequireDefault(require("./AllOrganizationRelationshipsGetResponseAllOf"));

var _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationRelationshipsGetResponseAllOfRelatedObjects"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _OrganizationRelationshipDetails = _interopRequireDefault(require("./OrganizationRelationshipDetails"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AllOrganizationRelationshipsGetResponse model module.
 * @module model/AllOrganizationRelationshipsGetResponse
 * @version 1.0.0
 */
var AllOrganizationRelationshipsGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllOrganizationRelationshipsGetResponse</code>.
   * @alias module:model/AllOrganizationRelationshipsGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/AllOrganizationRelationshipsGetResponseAllOf
   */
  function AllOrganizationRelationshipsGetResponse() {
    (0, _classCallCheck2["default"])(this, AllOrganizationRelationshipsGetResponse);

    _BaseResponse["default"].initialize(this);

    _AllOrganizationRelationshipsGetResponseAllOf["default"].initialize(this);

    AllOrganizationRelationshipsGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AllOrganizationRelationshipsGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllOrganizationRelationshipsGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllOrganizationRelationshipsGetResponse} obj Optional instance to populate.
     * @return {module:model/AllOrganizationRelationshipsGetResponse} The populated <code>AllOrganizationRelationshipsGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllOrganizationRelationshipsGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _AllOrganizationRelationshipsGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_OrganizationRelationshipDetails["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AllOrganizationRelationshipsGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


AllOrganizationRelationshipsGetResponse.prototype['success'] = undefined;
/**
 * The array of organization relationships
 * @member {Array.<module:model/OrganizationRelationshipDetails>} data
 */

AllOrganizationRelationshipsGetResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

AllOrganizationRelationshipsGetResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} related_objects
 */

AllOrganizationRelationshipsGetResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement AllOrganizationRelationshipsGetResponseAllOf interface:

/**
 * The array of organization relationships
 * @member {Array.<module:model/OrganizationRelationshipDetails>} data
 */

_AllOrganizationRelationshipsGetResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_AllOrganizationRelationshipsGetResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} related_objects
 */

_AllOrganizationRelationshipsGetResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = AllOrganizationRelationshipsGetResponse;
exports["default"] = _default;
