// pipedrive/dist/model/AllOrganizationsGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOf = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOf"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _BaseOrganizationItem = _interopRequireDefault(require("./BaseOrganizationItem"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AllOrganizationsGetResponse model module.
 * @module model/AllOrganizationsGetResponse
 * @version 1.0.0
 */
var AllOrganizationsGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllOrganizationsGetResponse</code>.
   * @alias module:model/AllOrganizationsGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/AllOrganizationsGetResponseAllOf
   */
  function AllOrganizationsGetResponse() {
    (0, _classCallCheck2["default"])(this, AllOrganizationsGetResponse);

    _BaseResponse["default"].initialize(this);

    _AllOrganizationsGetResponseAllOf["default"].initialize(this);

    AllOrganizationsGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AllOrganizationsGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllOrganizationsGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllOrganizationsGetResponse} obj Optional instance to populate.
     * @return {module:model/AllOrganizationsGetResponse} The populated <code>AllOrganizationsGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllOrganizationsGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _AllOrganizationsGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseOrganizationItem["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AllOrganizationsGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


AllOrganizationsGetResponse.prototype['success'] = undefined;
/**
 * The array of organizations
 * @member {Array.<module:model/BaseOrganizationItem>} data
 */

AllOrganizationsGetResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

AllOrganizationsGetResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

AllOrganizationsGetResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement AllOrganizationsGetResponseAllOf interface:

/**
 * The array of organizations
 * @member {Array.<module:model/BaseOrganizationItem>} data
 */

_AllOrganizationsGetResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_AllOrganizationsGetResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

_AllOrganizationsGetResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = AllOrganizationsGetResponse;
exports["default"] = _default;
