// pipedrive/dist/model/AllOrganizationsGetResponseAllOfRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RelatedOrganizationData = _interopRequireDefault(require("./RelatedOrganizationData"));

var _RelatedPictureData = _interopRequireDefault(require("./RelatedPictureData"));

var _RelatedUserData = _interopRequireDefault(require("./RelatedUserData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AllOrganizationsGetResponseAllOfRelatedObjects model module.
 * @module model/AllOrganizationsGetResponseAllOfRelatedObjects
 * @version 1.0.0
 */
var AllOrganizationsGetResponseAllOfRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllOrganizationsGetResponseAllOfRelatedObjects</code>.
   * @alias module:model/AllOrganizationsGetResponseAllOfRelatedObjects
   */
  function AllOrganizationsGetResponseAllOfRelatedObjects() {
    (0, _classCallCheck2["default"])(this, AllOrganizationsGetResponseAllOfRelatedObjects);
    AllOrganizationsGetResponseAllOfRelatedObjects.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AllOrganizationsGetResponseAllOfRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllOrganizationsGetResponseAllOfRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} obj Optional instance to populate.
     * @return {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} The populated <code>AllOrganizationsGetResponseAllOfRelatedObjects</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllOrganizationsGetResponseAllOfRelatedObjects();

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _RelatedOrganizationData["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('user')) {
          obj['user'] = _RelatedUserData["default"].constructFromObject(data['user']);
          delete data['user'];
        }

        if (data.hasOwnProperty('picture')) {
          obj['picture'] = _RelatedPictureData["default"].constructFromObject(data['picture']);
          delete data['picture'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AllOrganizationsGetResponseAllOfRelatedObjects;
}();
/**
 * @member {module:model/RelatedOrganizationData} organization
 */


AllOrganizationsGetResponseAllOfRelatedObjects.prototype['organization'] = undefined;
/**
 * @member {module:model/RelatedUserData} user
 */

AllOrganizationsGetResponseAllOfRelatedObjects.prototype['user'] = undefined;
/**
 * @member {module:model/RelatedPictureData} picture
 */

AllOrganizationsGetResponseAllOfRelatedObjects.prototype['picture'] = undefined;
var _default = AllOrganizationsGetResponseAllOfRelatedObjects;
exports["default"] = _default;
