// pipedrive/dist/model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RelatedOrganizationData = _interopRequireDefault(require("./RelatedOrganizationData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The AllOrganizationRelationshipsGetResponseAllOfRelatedObjects model module.
 * @module model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects
 * @version 1.0.0
 */
var AllOrganizationRelationshipsGetResponseAllOfRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllOrganizationRelationshipsGetResponseAllOfRelatedObjects</code>.
   * @alias module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects
   */
  function AllOrganizationRelationshipsGetResponseAllOfRelatedObjects() {
    (0, _classCallCheck2["default"])(this, AllOrganizationRelationshipsGetResponseAllOfRelatedObjects);
    AllOrganizationRelationshipsGetResponseAllOfRelatedObjects.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(AllOrganizationRelationshipsGetResponseAllOfRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllOrganizationRelationshipsGetResponseAllOfRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} obj Optional instance to populate.
     * @return {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} The populated <code>AllOrganizationRelationshipsGetResponseAllOfRelatedObjects</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllOrganizationRelationshipsGetResponseAllOfRelatedObjects();

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _RelatedOrganizationData["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return AllOrganizationRelationshipsGetResponseAllOfRelatedObjects;
}();
/**
 * @member {module:model/RelatedOrganizationData} organization
 */


AllOrganizationRelationshipsGetResponseAllOfRelatedObjects.prototype['organization'] = undefined;
var _default = AllOrganizationRelationshipsGetResponseAllOfRelatedObjects;
exports["default"] = _default;
