// pipedrive/dist/model/BaseFollowerItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseFollowerItem model module.
 * @module model/BaseFollowerItem
 * @version 1.0.0
 */
var BaseFollowerItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseFollowerItem</code>.
   * @alias module:model/BaseFollowerItem
   */
  function BaseFollowerItem() {
    (0, _classCallCheck2["default"])(this, BaseFollowerItem);
    BaseFollowerItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseFollowerItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseFollowerItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseFollowerItem} obj Optional instance to populate.
     * @return {module:model/BaseFollowerItem} The populated <code>BaseFollowerItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseFollowerItem();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Number');
          delete data['add_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseFollowerItem;
}();
/**
 * The user ID of the follower related to the item
 * @member {Number} user_id
 */


BaseFollowerItem.prototype['user_id'] = undefined;
/**
 * The ID of the follower
 * @member {Number} id
 */

BaseFollowerItem.prototype['id'] = undefined;
/**
 * The date and time of adding the follower to the item
 * @member {Number} add_time
 */

BaseFollowerItem.prototype['add_time'] = undefined;
var _default = BaseFollowerItem;
exports["default"] = _default;
