// pipedrive/dist/model/BaseMailThreadMessages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThreadMessagesAllOf = _interopRequireDefault(require("./BaseMailThreadMessagesAllOf"));

var _CommonMailThread = _interopRequireDefault(require("./CommonMailThread"));

var _MailThreadParticipant = _interopRequireDefault(require("./MailThreadParticipant"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseMailThreadMessages model module.
 * @module model/BaseMailThreadMessages
 * @version 1.0.0
 */
var BaseMailThreadMessages = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseMailThreadMessages</code>.
   * @alias module:model/BaseMailThreadMessages
   * @implements module:model/CommonMailThread
   * @implements module:model/BaseMailThreadMessagesAllOf
   */
  function BaseMailThreadMessages() {
    (0, _classCallCheck2["default"])(this, BaseMailThreadMessages);

    _CommonMailThread["default"].initialize(this);

    _BaseMailThreadMessagesAllOf["default"].initialize(this);

    BaseMailThreadMessages.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseMailThreadMessages, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseMailThreadMessages</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseMailThreadMessages} obj Optional instance to populate.
     * @return {module:model/BaseMailThreadMessages} The populated <code>BaseMailThreadMessages</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseMailThreadMessages();

        _CommonMailThread["default"].constructFromObject(data, obj);

        _BaseMailThreadMessagesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
          delete data['account_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
          delete data['subject'];
        }

        if (data.hasOwnProperty('snippet')) {
          obj['snippet'] = _ApiClient["default"].convertToType(data['snippet'], 'String');
          delete data['snippet'];
        }

        if (data.hasOwnProperty('read_flag')) {
          obj['read_flag'] = _ApiClient["default"].convertToType(data['read_flag'], _NumberBooleanDefault["default"]);
          delete data['read_flag'];
        }

        if (data.hasOwnProperty('mail_tracking_status')) {
          obj['mail_tracking_status'] = _ApiClient["default"].convertToType(data['mail_tracking_status'], 'String');
          delete data['mail_tracking_status'];
        }

        if (data.hasOwnProperty('has_attachments_flag')) {
          obj['has_attachments_flag'] = _ApiClient["default"].convertToType(data['has_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_attachments_flag'];
        }

        if (data.hasOwnProperty('has_inline_attachments_flag')) {
          obj['has_inline_attachments_flag'] = _ApiClient["default"].convertToType(data['has_inline_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_inline_attachments_flag'];
        }

        if (data.hasOwnProperty('has_real_attachments_flag')) {
          obj['has_real_attachments_flag'] = _ApiClient["default"].convertToType(data['has_real_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_real_attachments_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBooleanDefault["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('synced_flag')) {
          obj['synced_flag'] = _ApiClient["default"].convertToType(data['synced_flag'], _NumberBooleanDefault["default"]);
          delete data['synced_flag'];
        }

        if (data.hasOwnProperty('smart_bcc_flag')) {
          obj['smart_bcc_flag'] = _ApiClient["default"].convertToType(data['smart_bcc_flag'], _NumberBooleanDefault["default"]);
          delete data['smart_bcc_flag'];
        }

        if (data.hasOwnProperty('mail_link_tracking_enabled_flag')) {
          obj['mail_link_tracking_enabled_flag'] = _ApiClient["default"].convertToType(data['mail_link_tracking_enabled_flag'], _NumberBooleanDefault["default"]);
          delete data['mail_link_tracking_enabled_flag'];
        }

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], [_MailThreadParticipant["default"]]);
          delete data['from'];
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], [_MailThreadParticipant["default"]]);
          delete data['to'];
        }

        if (data.hasOwnProperty('cc')) {
          obj['cc'] = _ApiClient["default"].convertToType(data['cc'], [_MailThreadParticipant["default"]]);
          delete data['cc'];
        }

        if (data.hasOwnProperty('bcc')) {
          obj['bcc'] = _ApiClient["default"].convertToType(data['bcc'], [_MailThreadParticipant["default"]]);
          delete data['bcc'];
        }

        if (data.hasOwnProperty('body_url')) {
          obj['body_url'] = _ApiClient["default"].convertToType(data['body_url'], 'String');
          delete data['body_url'];
        }

        if (data.hasOwnProperty('mail_thread_id')) {
          obj['mail_thread_id'] = _ApiClient["default"].convertToType(data['mail_thread_id'], 'Number');
          delete data['mail_thread_id'];
        }

        if (data.hasOwnProperty('draft')) {
          obj['draft'] = _ApiClient["default"].convertToType(data['draft'], 'String');
          delete data['draft'];
        }

        if (data.hasOwnProperty('has_body_flag')) {
          obj['has_body_flag'] = _ApiClient["default"].convertToType(data['has_body_flag'], _NumberBooleanDefault["default"]);
          delete data['has_body_flag'];
        }

        if (data.hasOwnProperty('sent_flag')) {
          obj['sent_flag'] = _ApiClient["default"].convertToType(data['sent_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_flag'];
        }

        if (data.hasOwnProperty('sent_from_pipedrive_flag')) {
          obj['sent_from_pipedrive_flag'] = _ApiClient["default"].convertToType(data['sent_from_pipedrive_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_from_pipedrive_flag'];
        }

        if (data.hasOwnProperty('message_time')) {
          obj['message_time'] = _ApiClient["default"].convertToType(data['message_time'], 'Date');
          delete data['message_time'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Date');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'Date');
          delete data['update_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseMailThreadMessages;
}();
/**
 * ID of the mail thread
 * @member {Number} id
 */


BaseMailThreadMessages.prototype['id'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

BaseMailThreadMessages.prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

BaseMailThreadMessages.prototype['user_id'] = undefined;
/**
 * The subject
 * @member {String} subject
 */

BaseMailThreadMessages.prototype['subject'] = undefined;
/**
 * A snippet
 * @member {String} snippet
 */

BaseMailThreadMessages.prototype['snippet'] = undefined;
/**
 * Whether the mail thread is read
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

BaseMailThreadMessages.prototype['read_flag'] = undefined;
/**
 * Mail tracking status
 * @member {String} mail_tracking_status
 */

BaseMailThreadMessages.prototype['mail_tracking_status'] = undefined;
/**
 * Whether the mail thread has an attachment
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

BaseMailThreadMessages.prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail thread has inline attachments
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

BaseMailThreadMessages.prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail thread has real attachments (which are not inline)
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

BaseMailThreadMessages.prototype['has_real_attachments_flag'] = undefined;
/**
 * Whether the mail thread is deleted
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

BaseMailThreadMessages.prototype['deleted_flag'] = undefined;
/**
 * Whether the mail thread is synced
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

BaseMailThreadMessages.prototype['synced_flag'] = undefined;
/**
 * Whether one of the parties of the mail thread is Bcc
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

BaseMailThreadMessages.prototype['smart_bcc_flag'] = undefined;
/**
 * Whether the link tracking of the mail thread is enabled
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

BaseMailThreadMessages.prototype['mail_link_tracking_enabled_flag'] = undefined;
/**
 * Senders of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} from
 */

BaseMailThreadMessages.prototype['from'] = undefined;
/**
 * Recipients of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} to
 */

BaseMailThreadMessages.prototype['to'] = undefined;
/**
 * Participants of the Cc
 * @member {Array.<module:model/MailThreadParticipant>} cc
 */

BaseMailThreadMessages.prototype['cc'] = undefined;
/**
 * Participants of the Bcc
 * @member {Array.<module:model/MailThreadParticipant>} bcc
 */

BaseMailThreadMessages.prototype['bcc'] = undefined;
/**
 * A link to the mail thread message
 * @member {String} body_url
 */

BaseMailThreadMessages.prototype['body_url'] = undefined;
/**
 * ID of the mail thread
 * @member {Number} mail_thread_id
 */

BaseMailThreadMessages.prototype['mail_thread_id'] = undefined;
/**
 * If the mail message has a draft status then the value is the mail message object as JSON formatted string, otherwise `null`.
 * @member {String} draft
 */

BaseMailThreadMessages.prototype['draft'] = undefined;
/**
 * Whether the mail thread message has a body
 * @member {module:model/NumberBooleanDefault0} has_body_flag
 */

BaseMailThreadMessages.prototype['has_body_flag'] = undefined;
/**
 * Whether the mail thread message is sent
 * @member {module:model/NumberBooleanDefault0} sent_flag
 */

BaseMailThreadMessages.prototype['sent_flag'] = undefined;
/**
 * Whether the mail thread message is sent from Pipedrive
 * @member {module:model/NumberBooleanDefault0} sent_from_pipedrive_flag
 */

BaseMailThreadMessages.prototype['sent_from_pipedrive_flag'] = undefined;
/**
 * The time when the mail message was received or created
 * @member {Date} message_time
 */

BaseMailThreadMessages.prototype['message_time'] = undefined;
/**
 * The time when the mail message was inserted to database
 * @member {Date} add_time
 */

BaseMailThreadMessages.prototype['add_time'] = undefined;
/**
 * The time when the mail message was updated in database received
 * @member {Date} update_time
 */

BaseMailThreadMessages.prototype['update_time'] = undefined; // Implement CommonMailThread interface:

/**
 * ID of the mail thread
 * @member {Number} id
 */

_CommonMailThread["default"].prototype['id'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

_CommonMailThread["default"].prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

_CommonMailThread["default"].prototype['user_id'] = undefined;
/**
 * The subject
 * @member {String} subject
 */

_CommonMailThread["default"].prototype['subject'] = undefined;
/**
 * A snippet
 * @member {String} snippet
 */

_CommonMailThread["default"].prototype['snippet'] = undefined;
/**
 * Whether the mail thread is read
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

_CommonMailThread["default"].prototype['read_flag'] = undefined;
/**
 * Mail tracking status
 * @member {String} mail_tracking_status
 */

_CommonMailThread["default"].prototype['mail_tracking_status'] = undefined;
/**
 * Whether the mail thread has an attachment
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

_CommonMailThread["default"].prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail thread has inline attachments
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

_CommonMailThread["default"].prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail thread has real attachments (which are not inline)
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

_CommonMailThread["default"].prototype['has_real_attachments_flag'] = undefined;
/**
 * Whether the mail thread is deleted
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

_CommonMailThread["default"].prototype['deleted_flag'] = undefined;
/**
 * Whether the mail thread is synced
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

_CommonMailThread["default"].prototype['synced_flag'] = undefined;
/**
 * Whether one of the parties of the mail thread is Bcc
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

_CommonMailThread["default"].prototype['smart_bcc_flag'] = undefined;
/**
 * Whether the link tracking of the mail thread is enabled
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

_CommonMailThread["default"].prototype['mail_link_tracking_enabled_flag'] = undefined; // Implement BaseMailThreadMessagesAllOf interface:

/**
 * Senders of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} from
 */

_BaseMailThreadMessagesAllOf["default"].prototype['from'] = undefined;
/**
 * Recipients of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} to
 */

_BaseMailThreadMessagesAllOf["default"].prototype['to'] = undefined;
/**
 * Participants of the Cc
 * @member {Array.<module:model/MailThreadParticipant>} cc
 */

_BaseMailThreadMessagesAllOf["default"].prototype['cc'] = undefined;
/**
 * Participants of the Bcc
 * @member {Array.<module:model/MailThreadParticipant>} bcc
 */

_BaseMailThreadMessagesAllOf["default"].prototype['bcc'] = undefined;
/**
 * A link to the mail thread message
 * @member {String} body_url
 */

_BaseMailThreadMessagesAllOf["default"].prototype['body_url'] = undefined;
/**
 * ID of the mail thread
 * @member {Number} mail_thread_id
 */

_BaseMailThreadMessagesAllOf["default"].prototype['mail_thread_id'] = undefined;
/**
 * If the mail message has a draft status then the value is the mail message object as JSON formatted string, otherwise `null`.
 * @member {String} draft
 */

_BaseMailThreadMessagesAllOf["default"].prototype['draft'] = undefined;
/**
 * Whether the mail thread message has a body
 * @member {module:model/NumberBooleanDefault0} has_body_flag
 */

_BaseMailThreadMessagesAllOf["default"].prototype['has_body_flag'] = undefined;
/**
 * Whether the mail thread message is sent
 * @member {module:model/NumberBooleanDefault0} sent_flag
 */

_BaseMailThreadMessagesAllOf["default"].prototype['sent_flag'] = undefined;
/**
 * Whether the mail thread message is sent from Pipedrive
 * @member {module:model/NumberBooleanDefault0} sent_from_pipedrive_flag
 */

_BaseMailThreadMessagesAllOf["default"].prototype['sent_from_pipedrive_flag'] = undefined;
/**
 * The time when the mail message was received or created
 * @member {Date} message_time
 */

_BaseMailThreadMessagesAllOf["default"].prototype['message_time'] = undefined;
/**
 * The time when the mail message was inserted to database
 * @member {Date} add_time
 */

_BaseMailThreadMessagesAllOf["default"].prototype['add_time'] = undefined;
/**
 * The time when the mail message was updated in database received
 * @member {Date} update_time
 */

_BaseMailThreadMessagesAllOf["default"].prototype['update_time'] = undefined;
var _default = BaseMailThreadMessages;
exports["default"] = _default;
