// pipedrive/dist/model/BaseMailThreadAllOfParties.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MailThreadParticipant = _interopRequireDefault(require("./MailThreadParticipant"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseMailThreadAllOfParties model module.
 * @module model/BaseMailThreadAllOfParties
 * @version 1.0.0
 */
var BaseMailThreadAllOfParties = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseMailThreadAllOfParties</code>.
   * Parties of the mail thread
   * @alias module:model/BaseMailThreadAllOfParties
   */
  function BaseMailThreadAllOfParties() {
    (0, _classCallCheck2["default"])(this, BaseMailThreadAllOfParties);
    BaseMailThreadAllOfParties.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseMailThreadAllOfParties, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseMailThreadAllOfParties</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseMailThreadAllOfParties} obj Optional instance to populate.
     * @return {module:model/BaseMailThreadAllOfParties} The populated <code>BaseMailThreadAllOfParties</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseMailThreadAllOfParties();

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], [_MailThreadParticipant["default"]]);
          delete data['to'];
        }

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], [_MailThreadParticipant["default"]]);
          delete data['from'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseMailThreadAllOfParties;
}();
/**
 * Recipients of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} to
 */


BaseMailThreadAllOfParties.prototype['to'] = undefined;
/**
 * Senders of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} from
 */

BaseMailThreadAllOfParties.prototype['from'] = undefined;
var _default = BaseMailThreadAllOfParties;
exports["default"] = _default;
