// pipedrive/dist/model/BaseOrganizationItemWithEditNameFlag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseOrganizationItem = _interopRequireDefault(require("./BaseOrganizationItem"));

var _BaseOrganizationItemWithEditNameFlagAllOf = _interopRequireDefault(require("./BaseOrganizationItemWithEditNameFlagAllOf"));

var _Owner = _interopRequireDefault(require("./Owner"));

var _PictureDataWithValue = _interopRequireDefault(require("./PictureDataWithValue"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseOrganizationItemWithEditNameFlag model module.
 * @module model/BaseOrganizationItemWithEditNameFlag
 * @version 1.0.0
 */
var BaseOrganizationItemWithEditNameFlag = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseOrganizationItemWithEditNameFlag</code>.
   * @alias module:model/BaseOrganizationItemWithEditNameFlag
   * @implements module:model/BaseOrganizationItem
   * @implements module:model/BaseOrganizationItemWithEditNameFlagAllOf
   */
  function BaseOrganizationItemWithEditNameFlag() {
    (0, _classCallCheck2["default"])(this, BaseOrganizationItemWithEditNameFlag);

    _BaseOrganizationItem["default"].initialize(this);

    _BaseOrganizationItemWithEditNameFlagAllOf["default"].initialize(this);

    BaseOrganizationItemWithEditNameFlag.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseOrganizationItemWithEditNameFlag, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseOrganizationItemWithEditNameFlag</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseOrganizationItemWithEditNameFlag} obj Optional instance to populate.
     * @return {module:model/BaseOrganizationItemWithEditNameFlag} The populated <code>BaseOrganizationItemWithEditNameFlag</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseOrganizationItemWithEditNameFlag();

        _BaseOrganizationItem["default"].constructFromObject(data, obj);

        _BaseOrganizationItemWithEditNameFlagAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _Owner["default"].constructFromObject(data['owner_id']);
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('picture_id')) {
          obj['picture_id'] = _PictureDataWithValue["default"].constructFromObject(data['picture_id']);
          delete data['picture_id'];
        }

        if (data.hasOwnProperty('country_code')) {
          obj['country_code'] = _ApiClient["default"].convertToType(data['country_code'], 'String');
          delete data['country_code'];
        }

        if (data.hasOwnProperty('first_char')) {
          obj['first_char'] = _ApiClient["default"].convertToType(data['first_char'], 'String');
          delete data['first_char'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'String');
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'Number');
          delete data['label'];
        }

        if (data.hasOwnProperty('owner_name')) {
          obj['owner_name'] = _ApiClient["default"].convertToType(data['owner_name'], 'String');
          delete data['owner_name'];
        }

        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('people_count')) {
          obj['people_count'] = _ApiClient["default"].convertToType(data['people_count'], 'Number');
          delete data['people_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (data.hasOwnProperty('address_subpremise')) {
          obj['address_subpremise'] = _ApiClient["default"].convertToType(data['address_subpremise'], 'String');
          delete data['address_subpremise'];
        }

        if (data.hasOwnProperty('address_street_number')) {
          obj['address_street_number'] = _ApiClient["default"].convertToType(data['address_street_number'], 'String');
          delete data['address_street_number'];
        }

        if (data.hasOwnProperty('address_route')) {
          obj['address_route'] = _ApiClient["default"].convertToType(data['address_route'], 'String');
          delete data['address_route'];
        }

        if (data.hasOwnProperty('address_sublocality')) {
          obj['address_sublocality'] = _ApiClient["default"].convertToType(data['address_sublocality'], 'String');
          delete data['address_sublocality'];
        }

        if (data.hasOwnProperty('address_locality')) {
          obj['address_locality'] = _ApiClient["default"].convertToType(data['address_locality'], 'String');
          delete data['address_locality'];
        }

        if (data.hasOwnProperty('address_admin_area_level_1')) {
          obj['address_admin_area_level_1'] = _ApiClient["default"].convertToType(data['address_admin_area_level_1'], 'String');
          delete data['address_admin_area_level_1'];
        }

        if (data.hasOwnProperty('address_admin_area_level_2')) {
          obj['address_admin_area_level_2'] = _ApiClient["default"].convertToType(data['address_admin_area_level_2'], 'String');
          delete data['address_admin_area_level_2'];
        }

        if (data.hasOwnProperty('address_country')) {
          obj['address_country'] = _ApiClient["default"].convertToType(data['address_country'], 'String');
          delete data['address_country'];
        }

        if (data.hasOwnProperty('address_postal_code')) {
          obj['address_postal_code'] = _ApiClient["default"].convertToType(data['address_postal_code'], 'String');
          delete data['address_postal_code'];
        }

        if (data.hasOwnProperty('address_formatted_address')) {
          obj['address_formatted_address'] = _ApiClient["default"].convertToType(data['address_formatted_address'], 'String');
          delete data['address_formatted_address'];
        }

        if (data.hasOwnProperty('open_deals_count')) {
          obj['open_deals_count'] = _ApiClient["default"].convertToType(data['open_deals_count'], 'Number');
          delete data['open_deals_count'];
        }

        if (data.hasOwnProperty('related_open_deals_count')) {
          obj['related_open_deals_count'] = _ApiClient["default"].convertToType(data['related_open_deals_count'], 'Number');
          delete data['related_open_deals_count'];
        }

        if (data.hasOwnProperty('closed_deals_count')) {
          obj['closed_deals_count'] = _ApiClient["default"].convertToType(data['closed_deals_count'], 'Number');
          delete data['closed_deals_count'];
        }

        if (data.hasOwnProperty('related_closed_deals_count')) {
          obj['related_closed_deals_count'] = _ApiClient["default"].convertToType(data['related_closed_deals_count'], 'Number');
          delete data['related_closed_deals_count'];
        }

        if (data.hasOwnProperty('won_deals_count')) {
          obj['won_deals_count'] = _ApiClient["default"].convertToType(data['won_deals_count'], 'Number');
          delete data['won_deals_count'];
        }

        if (data.hasOwnProperty('related_won_deals_count')) {
          obj['related_won_deals_count'] = _ApiClient["default"].convertToType(data['related_won_deals_count'], 'Number');
          delete data['related_won_deals_count'];
        }

        if (data.hasOwnProperty('lost_deals_count')) {
          obj['lost_deals_count'] = _ApiClient["default"].convertToType(data['lost_deals_count'], 'Number');
          delete data['lost_deals_count'];
        }

        if (data.hasOwnProperty('related_lost_deals_count')) {
          obj['related_lost_deals_count'] = _ApiClient["default"].convertToType(data['related_lost_deals_count'], 'Number');
          delete data['related_lost_deals_count'];
        }

        if (data.hasOwnProperty('next_activity_date')) {
          obj['next_activity_date'] = _ApiClient["default"].convertToType(data['next_activity_date'], 'String');
          delete data['next_activity_date'];
        }

        if (data.hasOwnProperty('next_activity_time')) {
          obj['next_activity_time'] = _ApiClient["default"].convertToType(data['next_activity_time'], 'String');
          delete data['next_activity_time'];
        }

        if (data.hasOwnProperty('next_activity_id')) {
          obj['next_activity_id'] = _ApiClient["default"].convertToType(data['next_activity_id'], 'Number');
          delete data['next_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_id')) {
          obj['last_activity_id'] = _ApiClient["default"].convertToType(data['last_activity_id'], 'Number');
          delete data['last_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_date')) {
          obj['last_activity_date'] = _ApiClient["default"].convertToType(data['last_activity_date'], 'String');
          delete data['last_activity_date'];
        }

        if (data.hasOwnProperty('edit_name')) {
          obj['edit_name'] = _ApiClient["default"].convertToType(data['edit_name'], 'Boolean');
          delete data['edit_name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseOrganizationItemWithEditNameFlag;
}();
/**
 * The ID of the organization
 * @member {Number} id
 */


BaseOrganizationItemWithEditNameFlag.prototype['id'] = undefined;
/**
 * The ID of the company related to the organization
 * @member {Number} company_id
 */

BaseOrganizationItemWithEditNameFlag.prototype['company_id'] = undefined;
/**
 * @member {module:model/Owner} owner_id
 */

BaseOrganizationItemWithEditNameFlag.prototype['owner_id'] = undefined;
/**
 * The name of the organization
 * @member {String} name
 */

BaseOrganizationItemWithEditNameFlag.prototype['name'] = undefined;
/**
 * Whether the organization is active or not
 * @member {Boolean} active_flag
 */

BaseOrganizationItemWithEditNameFlag.prototype['active_flag'] = undefined;
/**
 * @member {module:model/PictureDataWithValue} picture_id
 */

BaseOrganizationItemWithEditNameFlag.prototype['picture_id'] = undefined;
/**
 * The country code of the organization
 * @member {String} country_code
 */

BaseOrganizationItemWithEditNameFlag.prototype['country_code'] = undefined;
/**
 * The first character of the organization name
 * @member {String} first_char
 */

BaseOrganizationItemWithEditNameFlag.prototype['first_char'] = undefined;
/**
 * The creation date and time of the organization
 * @member {String} add_time
 */

BaseOrganizationItemWithEditNameFlag.prototype['add_time'] = undefined;
/**
 * The last updated date and time of the organization
 * @member {String} update_time
 */

BaseOrganizationItemWithEditNameFlag.prototype['update_time'] = undefined;
/**
 * The visibility group ID of who can see the organization
 * @member {String} visible_to
 */

BaseOrganizationItemWithEditNameFlag.prototype['visible_to'] = undefined;
/**
 * The ID of the label
 * @member {Number} label
 */

BaseOrganizationItemWithEditNameFlag.prototype['label'] = undefined;
/**
 * The name of the organization owner
 * @member {String} owner_name
 */

BaseOrganizationItemWithEditNameFlag.prototype['owner_name'] = undefined;
/**
 * The BCC email of the organization
 * @member {String} cc_email
 */

BaseOrganizationItemWithEditNameFlag.prototype['cc_email'] = undefined;
/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['followers_count'] = undefined;
/**
 * The full address of the organization
 * @member {String} address
 */

BaseOrganizationItemWithEditNameFlag.prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

BaseOrganizationItemWithEditNameFlag.prototype['address_formatted_address'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

BaseOrganizationItemWithEditNameFlag.prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

BaseOrganizationItemWithEditNameFlag.prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

BaseOrganizationItemWithEditNameFlag.prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

BaseOrganizationItemWithEditNameFlag.prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

BaseOrganizationItemWithEditNameFlag.prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

BaseOrganizationItemWithEditNameFlag.prototype['last_activity_date'] = undefined;
/**
 * If the company ID of the organization and company ID of the request is same or not
 * @member {Boolean} edit_name
 */

BaseOrganizationItemWithEditNameFlag.prototype['edit_name'] = undefined; // Implement BaseOrganizationItem interface:

/**
 * The ID of the organization
 * @member {Number} id
 */

_BaseOrganizationItem["default"].prototype['id'] = undefined;
/**
 * The ID of the company related to the organization
 * @member {Number} company_id
 */

_BaseOrganizationItem["default"].prototype['company_id'] = undefined;
/**
 * @member {module:model/Owner} owner_id
 */

_BaseOrganizationItem["default"].prototype['owner_id'] = undefined;
/**
 * The name of the organization
 * @member {String} name
 */

_BaseOrganizationItem["default"].prototype['name'] = undefined;
/**
 * Whether the organization is active or not
 * @member {Boolean} active_flag
 */

_BaseOrganizationItem["default"].prototype['active_flag'] = undefined;
/**
 * @member {module:model/PictureDataWithValue} picture_id
 */

_BaseOrganizationItem["default"].prototype['picture_id'] = undefined;
/**
 * The country code of the organization
 * @member {String} country_code
 */

_BaseOrganizationItem["default"].prototype['country_code'] = undefined;
/**
 * The first character of the organization name
 * @member {String} first_char
 */

_BaseOrganizationItem["default"].prototype['first_char'] = undefined;
/**
 * The creation date and time of the organization
 * @member {String} add_time
 */

_BaseOrganizationItem["default"].prototype['add_time'] = undefined;
/**
 * The last updated date and time of the organization
 * @member {String} update_time
 */

_BaseOrganizationItem["default"].prototype['update_time'] = undefined;
/**
 * The visibility group ID of who can see the organization
 * @member {String} visible_to
 */

_BaseOrganizationItem["default"].prototype['visible_to'] = undefined;
/**
 * The ID of the label
 * @member {Number} label
 */

_BaseOrganizationItem["default"].prototype['label'] = undefined;
/**
 * The name of the organization owner
 * @member {String} owner_name
 */

_BaseOrganizationItem["default"].prototype['owner_name'] = undefined;
/**
 * The BCC email of the organization
 * @member {String} cc_email
 */

_BaseOrganizationItem["default"].prototype['cc_email'] = undefined;
/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */

_BaseOrganizationItem["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

_BaseOrganizationItem["default"].prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

_BaseOrganizationItem["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

_BaseOrganizationItem["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

_BaseOrganizationItem["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

_BaseOrganizationItem["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

_BaseOrganizationItem["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

_BaseOrganizationItem["default"].prototype['followers_count'] = undefined;
/**
 * The full address of the organization
 * @member {String} address
 */

_BaseOrganizationItem["default"].prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

_BaseOrganizationItem["default"].prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

_BaseOrganizationItem["default"].prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

_BaseOrganizationItem["default"].prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

_BaseOrganizationItem["default"].prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

_BaseOrganizationItem["default"].prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

_BaseOrganizationItem["default"].prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

_BaseOrganizationItem["default"].prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

_BaseOrganizationItem["default"].prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

_BaseOrganizationItem["default"].prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

_BaseOrganizationItem["default"].prototype['address_formatted_address'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

_BaseOrganizationItem["default"].prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

_BaseOrganizationItem["default"].prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

_BaseOrganizationItem["default"].prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

_BaseOrganizationItem["default"].prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

_BaseOrganizationItem["default"].prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

_BaseOrganizationItem["default"].prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

_BaseOrganizationItem["default"].prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

_BaseOrganizationItem["default"].prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

_BaseOrganizationItem["default"].prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

_BaseOrganizationItem["default"].prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

_BaseOrganizationItem["default"].prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

_BaseOrganizationItem["default"].prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

_BaseOrganizationItem["default"].prototype['last_activity_date'] = undefined; // Implement BaseOrganizationItemWithEditNameFlagAllOf interface:

/**
 * If the company ID of the organization and company ID of the request is same or not
 * @member {Boolean} edit_name
 */

_BaseOrganizationItemWithEditNameFlagAllOf["default"].prototype['edit_name'] = undefined;
var _default = BaseOrganizationItemWithEditNameFlag;
exports["default"] = _default;
