// pipedrive/dist/model/BaseTeamAdditionalProperties.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseTeamAdditionalProperties model module.
 * @module model/BaseTeamAdditionalProperties
 * @version 1.0.0
 */
var BaseTeamAdditionalProperties = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseTeamAdditionalProperties</code>.
   * @alias module:model/BaseTeamAdditionalProperties
   */
  function BaseTeamAdditionalProperties() {
    (0, _classCallCheck2["default"])(this, BaseTeamAdditionalProperties);
    BaseTeamAdditionalProperties.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseTeamAdditionalProperties, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseTeamAdditionalProperties</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseTeamAdditionalProperties} obj Optional instance to populate.
     * @return {module:model/BaseTeamAdditionalProperties} The populated <code>BaseTeamAdditionalProperties</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseTeamAdditionalProperties();

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('created_by_user_id')) {
          obj['created_by_user_id'] = _ApiClient["default"].convertToType(data['created_by_user_id'], 'Number');
          delete data['created_by_user_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseTeamAdditionalProperties;
}();
/**
 * The team creation time. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */


BaseTeamAdditionalProperties.prototype['add_time'] = undefined;
/**
 * The ID of the user who created the team
 * @member {Number} created_by_user_id
 */

BaseTeamAdditionalProperties.prototype['created_by_user_id'] = undefined;
var _default = BaseTeamAdditionalProperties;
exports["default"] = _default;
