// pipedrive/dist/model/BaseCurrency.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseCurrency model module.
 * @module model/BaseCurrency
 * @version 1.0.0
 */
var BaseCurrency = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseCurrency</code>.
   * @alias module:model/BaseCurrency
   */
  function BaseCurrency() {
    (0, _classCallCheck2["default"])(this, BaseCurrency);
    BaseCurrency.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseCurrency, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseCurrency</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseCurrency} obj Optional instance to populate.
     * @return {module:model/BaseCurrency} The populated <code>BaseCurrency</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseCurrency();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
          delete data['code'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('decimal_points')) {
          obj['decimal_points'] = _ApiClient["default"].convertToType(data['decimal_points'], 'Number');
          delete data['decimal_points'];
        }

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
          delete data['symbol'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('is_custom_flag')) {
          obj['is_custom_flag'] = _ApiClient["default"].convertToType(data['is_custom_flag'], 'Boolean');
          delete data['is_custom_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseCurrency;
}();
/**
 * The ID of the currency
 * @member {Number} id
 */


BaseCurrency.prototype['id'] = undefined;
/**
 * The code of the currency
 * @member {String} code
 */

BaseCurrency.prototype['code'] = undefined;
/**
 * The name of the currency
 * @member {String} name
 */

BaseCurrency.prototype['name'] = undefined;
/**
 * The amount of decimal points of the currency
 * @member {Number} decimal_points
 */

BaseCurrency.prototype['decimal_points'] = undefined;
/**
 * The symbol of the currency
 * @member {String} symbol
 */

BaseCurrency.prototype['symbol'] = undefined;
/**
 * Whether the currency is active or not
 * @member {Boolean} active_flag
 */

BaseCurrency.prototype['active_flag'] = undefined;
/**
 * Whether the currency is a custom one or not
 * @member {Boolean} is_custom_flag
 */

BaseCurrency.prototype['is_custom_flag'] = undefined;
var _default = BaseCurrency;
exports["default"] = _default;
