// pipedrive/dist/model/BasePipeline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BasePipeline model module.
 * @module model/BasePipeline
 * @version 1.0.0
 */
var BasePipeline = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BasePipeline</code>.
   * @alias module:model/BasePipeline
   */
  function BasePipeline() {
    (0, _classCallCheck2["default"])(this, BasePipeline);
    BasePipeline.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BasePipeline, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BasePipeline</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BasePipeline} obj Optional instance to populate.
     * @return {module:model/BasePipeline} The populated <code>BasePipeline</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BasePipeline();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('url_title')) {
          obj['url_title'] = _ApiClient["default"].convertToType(data['url_title'], 'String');
          delete data['url_title'];
        }

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
          delete data['active'];
        }

        if (data.hasOwnProperty('deal_probability')) {
          obj['deal_probability'] = _ApiClient["default"].convertToType(data['deal_probability'], 'Boolean');
          delete data['deal_probability'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BasePipeline;
}();
/**
 * The ID of the pipeline
 * @member {Number} id
 */


BasePipeline.prototype['id'] = undefined;
/**
 * The name of the pipeline
 * @member {String} name
 */

BasePipeline.prototype['name'] = undefined;
/**
 * The pipeline title displayed in the URL
 * @member {String} url_title
 */

BasePipeline.prototype['url_title'] = undefined;
/**
 * Defines the order of pipelines. First order (`order_nr=0`) is the default pipeline.
 * @member {Number} order_nr
 */

BasePipeline.prototype['order_nr'] = undefined;
/**
 * Whether this pipeline will be made inactive (hidden) or active
 * @member {Boolean} active
 */

BasePipeline.prototype['active'] = undefined;
/**
 * Whether deal probability is disabled or enabled for this pipeline
 * @member {Boolean} deal_probability
 */

BasePipeline.prototype['deal_probability'] = undefined;
/**
 * The pipeline creation time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} add_time
 */

BasePipeline.prototype['add_time'] = undefined;
/**
 * The pipeline update time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} update_time
 */

BasePipeline.prototype['update_time'] = undefined;
var _default = BasePipeline;
exports["default"] = _default;
