// pipedrive/dist/model/BaseNote.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseNoteDealTitle = _interopRequireDefault(require("./BaseNoteDealTitle"));

var _BaseNoteOrganization = _interopRequireDefault(require("./BaseNoteOrganization"));

var _BaseNotePerson = _interopRequireDefault(require("./BaseNotePerson"));

var _NoteCreatorUser = _interopRequireDefault(require("./NoteCreatorUser"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseNote model module.
 * @module model/BaseNote
 * @version 1.0.0
 */
var BaseNote = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseNote</code>.
   * @alias module:model/BaseNote
   */
  function BaseNote() {
    (0, _classCallCheck2["default"])(this, BaseNote);
    BaseNote.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseNote, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseNote</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseNote} obj Optional instance to populate.
     * @return {module:model/BaseNote} The populated <code>BaseNote</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseNote();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('content')) {
          obj['content'] = _ApiClient["default"].convertToType(data['content'], 'String');
          delete data['content'];
        }

        if (data.hasOwnProperty('deal')) {
          obj['deal'] = _BaseNoteDealTitle["default"].constructFromObject(data['deal']);
          delete data['deal'];
        }

        if (data.hasOwnProperty('lead_id')) {
          obj['lead_id'] = _ApiClient["default"].convertToType(data['lead_id'], 'String');
          delete data['lead_id'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('last_update_user_id')) {
          obj['last_update_user_id'] = _ApiClient["default"].convertToType(data['last_update_user_id'], 'Number');
          delete data['last_update_user_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _BaseNoteOrganization["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('person')) {
          obj['person'] = _BaseNotePerson["default"].constructFromObject(data['person']);
          delete data['person'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('pinned_to_deal_flag')) {
          obj['pinned_to_deal_flag'] = _ApiClient["default"].convertToType(data['pinned_to_deal_flag'], 'Boolean');
          delete data['pinned_to_deal_flag'];
        }

        if (data.hasOwnProperty('pinned_to_organization_flag')) {
          obj['pinned_to_organization_flag'] = _ApiClient["default"].convertToType(data['pinned_to_organization_flag'], 'Boolean');
          delete data['pinned_to_organization_flag'];
        }

        if (data.hasOwnProperty('pinned_to_person_flag')) {
          obj['pinned_to_person_flag'] = _ApiClient["default"].convertToType(data['pinned_to_person_flag'], 'Boolean');
          delete data['pinned_to_person_flag'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('user')) {
          obj['user'] = _NoteCreatorUser["default"].constructFromObject(data['user']);
          delete data['user'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseNote;
}();
/**
 * The ID of the note
 * @member {Number} id
 */


BaseNote.prototype['id'] = undefined;
/**
 * Whether the note is active or deleted
 * @member {Boolean} active_flag
 */

BaseNote.prototype['active_flag'] = undefined;
/**
 * The creation date and time of the note
 * @member {String} add_time
 */

BaseNote.prototype['add_time'] = undefined;
/**
 * The content of the note in HTML format. Subject to sanitization on the back-end.
 * @member {String} content
 */

BaseNote.prototype['content'] = undefined;
/**
 * @member {module:model/BaseNoteDealTitle} deal
 */

BaseNote.prototype['deal'] = undefined;
/**
 * The ID of the lead the note is attached to
 * @member {String} lead_id
 */

BaseNote.prototype['lead_id'] = undefined;
/**
 * The ID of the deal the note is attached to
 * @member {Number} deal_id
 */

BaseNote.prototype['deal_id'] = undefined;
/**
 * The ID of the user who last updated the note
 * @member {Number} last_update_user_id
 */

BaseNote.prototype['last_update_user_id'] = undefined;
/**
 * The ID of the organization the note is attached to
 * @member {Number} org_id
 */

BaseNote.prototype['org_id'] = undefined;
/**
 * @member {module:model/BaseNoteOrganization} organization
 */

BaseNote.prototype['organization'] = undefined;
/**
 * @member {module:model/BaseNotePerson} person
 */

BaseNote.prototype['person'] = undefined;
/**
 * The ID of the person the note is attached to
 * @member {Number} person_id
 */

BaseNote.prototype['person_id'] = undefined;
/**
 * If true, the results are filtered by note to deal pinning state
 * @member {Boolean} pinned_to_deal_flag
 */

BaseNote.prototype['pinned_to_deal_flag'] = undefined;
/**
 * If true, the results are filtered by note to organization pinning state
 * @member {Boolean} pinned_to_organization_flag
 */

BaseNote.prototype['pinned_to_organization_flag'] = undefined;
/**
 * If true, the results are filtered by note to person pinning state
 * @member {Boolean} pinned_to_person_flag
 */

BaseNote.prototype['pinned_to_person_flag'] = undefined;
/**
 * The last updated date and time of the note
 * @member {String} update_time
 */

BaseNote.prototype['update_time'] = undefined;
/**
 * @member {module:model/NoteCreatorUser} user
 */

BaseNote.prototype['user'] = undefined;
/**
 * The ID of the note creator
 * @member {Number} user_id
 */

BaseNote.prototype['user_id'] = undefined;
var _default = BaseNote;
exports["default"] = _default;
