// pipedrive/dist/model/BaseMailThread.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThreadAllOf = _interopRequireDefault(require("./BaseMailThreadAllOf"));

var _BaseMailThreadAllOfParties = _interopRequireDefault(require("./BaseMailThreadAllOfParties"));

var _CommonMailThread = _interopRequireDefault(require("./CommonMailThread"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseMailThread model module.
 * @module model/BaseMailThread
 * @version 1.0.0
 */
var BaseMailThread = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseMailThread</code>.
   * @alias module:model/BaseMailThread
   * @implements module:model/CommonMailThread
   * @implements module:model/BaseMailThreadAllOf
   */
  function BaseMailThread() {
    (0, _classCallCheck2["default"])(this, BaseMailThread);

    _CommonMailThread["default"].initialize(this);

    _BaseMailThreadAllOf["default"].initialize(this);

    BaseMailThread.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseMailThread, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseMailThread</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseMailThread} obj Optional instance to populate.
     * @return {module:model/BaseMailThread} The populated <code>BaseMailThread</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseMailThread();

        _CommonMailThread["default"].constructFromObject(data, obj);

        _BaseMailThreadAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
          delete data['account_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
          delete data['subject'];
        }

        if (data.hasOwnProperty('snippet')) {
          obj['snippet'] = _ApiClient["default"].convertToType(data['snippet'], 'String');
          delete data['snippet'];
        }

        if (data.hasOwnProperty('read_flag')) {
          obj['read_flag'] = _ApiClient["default"].convertToType(data['read_flag'], _NumberBooleanDefault["default"]);
          delete data['read_flag'];
        }

        if (data.hasOwnProperty('mail_tracking_status')) {
          obj['mail_tracking_status'] = _ApiClient["default"].convertToType(data['mail_tracking_status'], 'String');
          delete data['mail_tracking_status'];
        }

        if (data.hasOwnProperty('has_attachments_flag')) {
          obj['has_attachments_flag'] = _ApiClient["default"].convertToType(data['has_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_attachments_flag'];
        }

        if (data.hasOwnProperty('has_inline_attachments_flag')) {
          obj['has_inline_attachments_flag'] = _ApiClient["default"].convertToType(data['has_inline_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_inline_attachments_flag'];
        }

        if (data.hasOwnProperty('has_real_attachments_flag')) {
          obj['has_real_attachments_flag'] = _ApiClient["default"].convertToType(data['has_real_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_real_attachments_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBooleanDefault["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('synced_flag')) {
          obj['synced_flag'] = _ApiClient["default"].convertToType(data['synced_flag'], _NumberBooleanDefault["default"]);
          delete data['synced_flag'];
        }

        if (data.hasOwnProperty('smart_bcc_flag')) {
          obj['smart_bcc_flag'] = _ApiClient["default"].convertToType(data['smart_bcc_flag'], _NumberBooleanDefault["default"]);
          delete data['smart_bcc_flag'];
        }

        if (data.hasOwnProperty('mail_link_tracking_enabled_flag')) {
          obj['mail_link_tracking_enabled_flag'] = _ApiClient["default"].convertToType(data['mail_link_tracking_enabled_flag'], _NumberBooleanDefault["default"]);
          delete data['mail_link_tracking_enabled_flag'];
        }

        if (data.hasOwnProperty('parties')) {
          obj['parties'] = _BaseMailThreadAllOfParties["default"].constructFromObject(data['parties']);
          delete data['parties'];
        }

        if (data.hasOwnProperty('drafts_parties')) {
          obj['drafts_parties'] = _ApiClient["default"].convertToType(data['drafts_parties'], [Object]);
          delete data['drafts_parties'];
        }

        if (data.hasOwnProperty('folders')) {
          obj['folders'] = _ApiClient["default"].convertToType(data['folders'], ['String']);
          delete data['folders'];
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
          delete data['version'];
        }

        if (data.hasOwnProperty('snippet_draft')) {
          obj['snippet_draft'] = _ApiClient["default"].convertToType(data['snippet_draft'], 'String');
          delete data['snippet_draft'];
        }

        if (data.hasOwnProperty('snippet_sent')) {
          obj['snippet_sent'] = _ApiClient["default"].convertToType(data['snippet_sent'], 'String');
          delete data['snippet_sent'];
        }

        if (data.hasOwnProperty('message_count')) {
          obj['message_count'] = _ApiClient["default"].convertToType(data['message_count'], 'Number');
          delete data['message_count'];
        }

        if (data.hasOwnProperty('has_draft_flag')) {
          obj['has_draft_flag'] = _ApiClient["default"].convertToType(data['has_draft_flag'], _NumberBooleanDefault["default"]);
          delete data['has_draft_flag'];
        }

        if (data.hasOwnProperty('has_sent_flag')) {
          obj['has_sent_flag'] = _ApiClient["default"].convertToType(data['has_sent_flag'], _NumberBooleanDefault["default"]);
          delete data['has_sent_flag'];
        }

        if (data.hasOwnProperty('archived_flag')) {
          obj['archived_flag'] = _ApiClient["default"].convertToType(data['archived_flag'], _NumberBooleanDefault["default"]);
          delete data['archived_flag'];
        }

        if (data.hasOwnProperty('shared_flag')) {
          obj['shared_flag'] = _ApiClient["default"].convertToType(data['shared_flag'], _NumberBooleanDefault["default"]);
          delete data['shared_flag'];
        }

        if (data.hasOwnProperty('external_deleted_flag')) {
          obj['external_deleted_flag'] = _ApiClient["default"].convertToType(data['external_deleted_flag'], _NumberBooleanDefault["default"]);
          delete data['external_deleted_flag'];
        }

        if (data.hasOwnProperty('first_message_to_me_flag')) {
          obj['first_message_to_me_flag'] = _ApiClient["default"].convertToType(data['first_message_to_me_flag'], _NumberBooleanDefault["default"]);
          delete data['first_message_to_me_flag'];
        }

        if (data.hasOwnProperty('last_message_timestamp')) {
          obj['last_message_timestamp'] = _ApiClient["default"].convertToType(data['last_message_timestamp'], 'Date');
          delete data['last_message_timestamp'];
        }

        if (data.hasOwnProperty('first_message_timestamp')) {
          obj['first_message_timestamp'] = _ApiClient["default"].convertToType(data['first_message_timestamp'], 'Date');
          delete data['first_message_timestamp'];
        }

        if (data.hasOwnProperty('last_message_sent_timestamp')) {
          obj['last_message_sent_timestamp'] = _ApiClient["default"].convertToType(data['last_message_sent_timestamp'], 'Date');
          delete data['last_message_sent_timestamp'];
        }

        if (data.hasOwnProperty('last_message_received_timestamp')) {
          obj['last_message_received_timestamp'] = _ApiClient["default"].convertToType(data['last_message_received_timestamp'], 'Date');
          delete data['last_message_received_timestamp'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Date');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'Date');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('deal_status')) {
          obj['deal_status'] = _ApiClient["default"].convertToType(data['deal_status'], 'String');
          delete data['deal_status'];
        }

        if (data.hasOwnProperty('lead_id')) {
          obj['lead_id'] = _ApiClient["default"].convertToType(data['lead_id'], 'String');
          delete data['lead_id'];
        }

        if (data.hasOwnProperty('all_messages_sent_flag')) {
          obj['all_messages_sent_flag'] = _ApiClient["default"].convertToType(data['all_messages_sent_flag'], _NumberBooleanDefault["default"]);
          delete data['all_messages_sent_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseMailThread;
}();
/**
 * ID of the mail thread
 * @member {Number} id
 */


BaseMailThread.prototype['id'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

BaseMailThread.prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

BaseMailThread.prototype['user_id'] = undefined;
/**
 * The subject
 * @member {String} subject
 */

BaseMailThread.prototype['subject'] = undefined;
/**
 * A snippet
 * @member {String} snippet
 */

BaseMailThread.prototype['snippet'] = undefined;
/**
 * Whether the mail thread is read
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

BaseMailThread.prototype['read_flag'] = undefined;
/**
 * Mail tracking status
 * @member {String} mail_tracking_status
 */

BaseMailThread.prototype['mail_tracking_status'] = undefined;
/**
 * Whether the mail thread has an attachment
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

BaseMailThread.prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail thread has inline attachments
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

BaseMailThread.prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail thread has real attachments (which are not inline)
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

BaseMailThread.prototype['has_real_attachments_flag'] = undefined;
/**
 * Whether the mail thread is deleted
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

BaseMailThread.prototype['deleted_flag'] = undefined;
/**
 * Whether the mail thread is synced
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

BaseMailThread.prototype['synced_flag'] = undefined;
/**
 * Whether one of the parties of the mail thread is Bcc
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

BaseMailThread.prototype['smart_bcc_flag'] = undefined;
/**
 * Whether the link tracking of the mail thread is enabled
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

BaseMailThread.prototype['mail_link_tracking_enabled_flag'] = undefined;
/**
 * @member {module:model/BaseMailThreadAllOfParties} parties
 */

BaseMailThread.prototype['parties'] = undefined;
/**
 * Parties of the drafted mail thread
 * @member {Array.<Object>} drafts_parties
 */

BaseMailThread.prototype['drafts_parties'] = undefined;
/**
 * Folders in which messages from thread are being stored
 * @member {Array.<String>} folders
 */

BaseMailThread.prototype['folders'] = undefined;
/**
 * Version
 * @member {Number} version
 */

BaseMailThread.prototype['version'] = undefined;
/**
 * A snippet from a draft
 * @member {String} snippet_draft
 */

BaseMailThread.prototype['snippet_draft'] = undefined;
/**
 * A snippet from a message sent
 * @member {String} snippet_sent
 */

BaseMailThread.prototype['snippet_sent'] = undefined;
/**
 * An amount of messages
 * @member {Number} message_count
 */

BaseMailThread.prototype['message_count'] = undefined;
/**
 * Whether the mail thread has any drafts
 * @member {module:model/NumberBooleanDefault0} has_draft_flag
 */

BaseMailThread.prototype['has_draft_flag'] = undefined;
/**
 * Whether the mail thread has messages sent
 * @member {module:model/NumberBooleanDefault0} has_sent_flag
 */

BaseMailThread.prototype['has_sent_flag'] = undefined;
/**
 * Whether the mail thread is archived
 * @member {module:model/NumberBooleanDefault0} archived_flag
 */

BaseMailThread.prototype['archived_flag'] = undefined;
/**
 * Whether the mail thread is shared
 * @member {module:model/NumberBooleanDefault0} shared_flag
 */

BaseMailThread.prototype['shared_flag'] = undefined;
/**
 * Whether the mail thread has been deleted externally
 * @member {module:model/NumberBooleanDefault0} external_deleted_flag
 */

BaseMailThread.prototype['external_deleted_flag'] = undefined;
/**
 * Whether the mail thread was initialized by others
 * @member {module:model/NumberBooleanDefault0} first_message_to_me_flag
 */

BaseMailThread.prototype['first_message_to_me_flag'] = undefined;
/**
 * Last message timestamp
 * @member {Date} last_message_timestamp
 */

BaseMailThread.prototype['last_message_timestamp'] = undefined;
/**
 * The time when the mail thread has had the first message received or created
 * @member {Date} first_message_timestamp
 */

BaseMailThread.prototype['first_message_timestamp'] = undefined;
/**
 * The last time when the mail thread has had a message sent
 * @member {Date} last_message_sent_timestamp
 */

BaseMailThread.prototype['last_message_sent_timestamp'] = undefined;
/**
 * The last time when the mail thread has had a message received
 * @member {Date} last_message_received_timestamp
 */

BaseMailThread.prototype['last_message_received_timestamp'] = undefined;
/**
 * The time when the mail thread was inserted to database
 * @member {Date} add_time
 */

BaseMailThread.prototype['add_time'] = undefined;
/**
 * The time when the mail thread was updated in database received
 * @member {Date} update_time
 */

BaseMailThread.prototype['update_time'] = undefined;
/**
 * The ID of the deal
 * @member {Number} deal_id
 */

BaseMailThread.prototype['deal_id'] = undefined;
/**
 * Status of the deal
 * @member {String} deal_status
 */

BaseMailThread.prototype['deal_status'] = undefined;
/**
 * The ID of the lead
 * @member {String} lead_id
 */

BaseMailThread.prototype['lead_id'] = undefined;
/**
 * Whether all the mail thread messages have been sent
 * @member {module:model/NumberBooleanDefault0} all_messages_sent_flag
 */

BaseMailThread.prototype['all_messages_sent_flag'] = undefined; // Implement CommonMailThread interface:

/**
 * ID of the mail thread
 * @member {Number} id
 */

_CommonMailThread["default"].prototype['id'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

_CommonMailThread["default"].prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

_CommonMailThread["default"].prototype['user_id'] = undefined;
/**
 * The subject
 * @member {String} subject
 */

_CommonMailThread["default"].prototype['subject'] = undefined;
/**
 * A snippet
 * @member {String} snippet
 */

_CommonMailThread["default"].prototype['snippet'] = undefined;
/**
 * Whether the mail thread is read
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

_CommonMailThread["default"].prototype['read_flag'] = undefined;
/**
 * Mail tracking status
 * @member {String} mail_tracking_status
 */

_CommonMailThread["default"].prototype['mail_tracking_status'] = undefined;
/**
 * Whether the mail thread has an attachment
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

_CommonMailThread["default"].prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail thread has inline attachments
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

_CommonMailThread["default"].prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail thread has real attachments (which are not inline)
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

_CommonMailThread["default"].prototype['has_real_attachments_flag'] = undefined;
/**
 * Whether the mail thread is deleted
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

_CommonMailThread["default"].prototype['deleted_flag'] = undefined;
/**
 * Whether the mail thread is synced
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

_CommonMailThread["default"].prototype['synced_flag'] = undefined;
/**
 * Whether one of the parties of the mail thread is Bcc
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

_CommonMailThread["default"].prototype['smart_bcc_flag'] = undefined;
/**
 * Whether the link tracking of the mail thread is enabled
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

_CommonMailThread["default"].prototype['mail_link_tracking_enabled_flag'] = undefined; // Implement BaseMailThreadAllOf interface:

/**
 * @member {module:model/BaseMailThreadAllOfParties} parties
 */

_BaseMailThreadAllOf["default"].prototype['parties'] = undefined;
/**
 * Parties of the drafted mail thread
 * @member {Array.<Object>} drafts_parties
 */

_BaseMailThreadAllOf["default"].prototype['drafts_parties'] = undefined;
/**
 * Folders in which messages from thread are being stored
 * @member {Array.<String>} folders
 */

_BaseMailThreadAllOf["default"].prototype['folders'] = undefined;
/**
 * Version
 * @member {Number} version
 */

_BaseMailThreadAllOf["default"].prototype['version'] = undefined;
/**
 * A snippet from a draft
 * @member {String} snippet_draft
 */

_BaseMailThreadAllOf["default"].prototype['snippet_draft'] = undefined;
/**
 * A snippet from a message sent
 * @member {String} snippet_sent
 */

_BaseMailThreadAllOf["default"].prototype['snippet_sent'] = undefined;
/**
 * An amount of messages
 * @member {Number} message_count
 */

_BaseMailThreadAllOf["default"].prototype['message_count'] = undefined;
/**
 * Whether the mail thread has any drafts
 * @member {module:model/NumberBooleanDefault0} has_draft_flag
 */

_BaseMailThreadAllOf["default"].prototype['has_draft_flag'] = undefined;
/**
 * Whether the mail thread has messages sent
 * @member {module:model/NumberBooleanDefault0} has_sent_flag
 */

_BaseMailThreadAllOf["default"].prototype['has_sent_flag'] = undefined;
/**
 * Whether the mail thread is archived
 * @member {module:model/NumberBooleanDefault0} archived_flag
 */

_BaseMailThreadAllOf["default"].prototype['archived_flag'] = undefined;
/**
 * Whether the mail thread is shared
 * @member {module:model/NumberBooleanDefault0} shared_flag
 */

_BaseMailThreadAllOf["default"].prototype['shared_flag'] = undefined;
/**
 * Whether the mail thread has been deleted externally
 * @member {module:model/NumberBooleanDefault0} external_deleted_flag
 */

_BaseMailThreadAllOf["default"].prototype['external_deleted_flag'] = undefined;
/**
 * Whether the mail thread was initialized by others
 * @member {module:model/NumberBooleanDefault0} first_message_to_me_flag
 */

_BaseMailThreadAllOf["default"].prototype['first_message_to_me_flag'] = undefined;
/**
 * Last message timestamp
 * @member {Date} last_message_timestamp
 */

_BaseMailThreadAllOf["default"].prototype['last_message_timestamp'] = undefined;
/**
 * The time when the mail thread has had the first message received or created
 * @member {Date} first_message_timestamp
 */

_BaseMailThreadAllOf["default"].prototype['first_message_timestamp'] = undefined;
/**
 * The last time when the mail thread has had a message sent
 * @member {Date} last_message_sent_timestamp
 */

_BaseMailThreadAllOf["default"].prototype['last_message_sent_timestamp'] = undefined;
/**
 * The last time when the mail thread has had a message received
 * @member {Date} last_message_received_timestamp
 */

_BaseMailThreadAllOf["default"].prototype['last_message_received_timestamp'] = undefined;
/**
 * The time when the mail thread was inserted to database
 * @member {Date} add_time
 */

_BaseMailThreadAllOf["default"].prototype['add_time'] = undefined;
/**
 * The time when the mail thread was updated in database received
 * @member {Date} update_time
 */

_BaseMailThreadAllOf["default"].prototype['update_time'] = undefined;
/**
 * The ID of the deal
 * @member {Number} deal_id
 */

_BaseMailThreadAllOf["default"].prototype['deal_id'] = undefined;
/**
 * Status of the deal
 * @member {String} deal_status
 */

_BaseMailThreadAllOf["default"].prototype['deal_status'] = undefined;
/**
 * The ID of the lead
 * @member {String} lead_id
 */

_BaseMailThreadAllOf["default"].prototype['lead_id'] = undefined;
/**
 * Whether all the mail thread messages have been sent
 * @member {module:model/NumberBooleanDefault0} all_messages_sent_flag
 */

_BaseMailThreadAllOf["default"].prototype['all_messages_sent_flag'] = undefined;
var _default = BaseMailThread;
exports["default"] = _default;
