// pipedrive/dist/model/BaseUserMeAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseUserMeAllOfLanguage = _interopRequireDefault(require("./BaseUserMeAllOfLanguage"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseUserMeAllOf model module.
 * @module model/BaseUserMeAllOf
 * @version 1.0.0
 */
var BaseUserMeAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseUserMeAllOf</code>.
   * @alias module:model/BaseUserMeAllOf
   */
  function BaseUserMeAllOf() {
    (0, _classCallCheck2["default"])(this, BaseUserMeAllOf);
    BaseUserMeAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseUserMeAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseUserMeAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseUserMeAllOf} obj Optional instance to populate.
     * @return {module:model/BaseUserMeAllOf} The populated <code>BaseUserMeAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseUserMeAllOf();

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (data.hasOwnProperty('company_name')) {
          obj['company_name'] = _ApiClient["default"].convertToType(data['company_name'], 'String');
          delete data['company_name'];
        }

        if (data.hasOwnProperty('company_domain')) {
          obj['company_domain'] = _ApiClient["default"].convertToType(data['company_domain'], 'String');
          delete data['company_domain'];
        }

        if (data.hasOwnProperty('company_country')) {
          obj['company_country'] = _ApiClient["default"].convertToType(data['company_country'], 'String');
          delete data['company_country'];
        }

        if (data.hasOwnProperty('company_industry')) {
          obj['company_industry'] = _ApiClient["default"].convertToType(data['company_industry'], 'String');
          delete data['company_industry'];
        }

        if (data.hasOwnProperty('language')) {
          obj['language'] = _BaseUserMeAllOfLanguage["default"].constructFromObject(data['language']);
          delete data['language'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseUserMeAllOf;
}();
/**
 * The user company ID
 * @member {Number} company_id
 */


BaseUserMeAllOf.prototype['company_id'] = undefined;
/**
 * The user company name
 * @member {String} company_name
 */

BaseUserMeAllOf.prototype['company_name'] = undefined;
/**
 * The user company domain
 * @member {String} company_domain
 */

BaseUserMeAllOf.prototype['company_domain'] = undefined;
/**
 * The user company country
 * @member {String} company_country
 */

BaseUserMeAllOf.prototype['company_country'] = undefined;
/**
 * The user company industry
 * @member {String} company_industry
 */

BaseUserMeAllOf.prototype['company_industry'] = undefined;
/**
 * @member {module:model/BaseUserMeAllOfLanguage} language
 */

BaseUserMeAllOf.prototype['language'] = undefined;
var _default = BaseUserMeAllOf;
exports["default"] = _default;
