// pipedrive/dist/model/BasePipelineWithSelectedFlagAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BasePipelineWithSelectedFlagAllOf model module.
 * @module model/BasePipelineWithSelectedFlagAllOf
 * @version 1.0.0
 */
var BasePipelineWithSelectedFlagAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BasePipelineWithSelectedFlagAllOf</code>.
   * @alias module:model/BasePipelineWithSelectedFlagAllOf
   */
  function BasePipelineWithSelectedFlagAllOf() {
    (0, _classCallCheck2["default"])(this, BasePipelineWithSelectedFlagAllOf);
    BasePipelineWithSelectedFlagAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BasePipelineWithSelectedFlagAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BasePipelineWithSelectedFlagAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BasePipelineWithSelectedFlagAllOf} obj Optional instance to populate.
     * @return {module:model/BasePipelineWithSelectedFlagAllOf} The populated <code>BasePipelineWithSelectedFlagAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BasePipelineWithSelectedFlagAllOf();

        if (data.hasOwnProperty('selected')) {
          obj['selected'] = _ApiClient["default"].convertToType(data['selected'], 'Boolean');
          delete data['selected'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BasePipelineWithSelectedFlagAllOf;
}();
/**
 * A boolean that shows if the pipeline is selected from a filter or not
 * @member {Boolean} selected
 */


BasePipelineWithSelectedFlagAllOf.prototype['selected'] = undefined;
var _default = BasePipelineWithSelectedFlagAllOf;
exports["default"] = _default;
