// pipedrive/dist/model/BaseResponseWithStatus.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _BaseResponseWithStatusAllOf = _interopRequireDefault(require("./BaseResponseWithStatusAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseResponseWithStatus model module.
 * @module model/BaseResponseWithStatus
 * @version 1.0.0
 */
var BaseResponseWithStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseResponseWithStatus</code>.
   * @alias module:model/BaseResponseWithStatus
   * @implements module:model/BaseResponse
   * @implements module:model/BaseResponseWithStatusAllOf
   */
  function BaseResponseWithStatus() {
    (0, _classCallCheck2["default"])(this, BaseResponseWithStatus);

    _BaseResponse["default"].initialize(this);

    _BaseResponseWithStatusAllOf["default"].initialize(this);

    BaseResponseWithStatus.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseResponseWithStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseResponseWithStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseResponseWithStatus} obj Optional instance to populate.
     * @return {module:model/BaseResponseWithStatus} The populated <code>BaseResponseWithStatus</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseResponseWithStatus();

        _BaseResponse["default"].constructFromObject(data, obj);

        _BaseResponseWithStatusAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseResponseWithStatus;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


BaseResponseWithStatus.prototype['success'] = undefined;
/**
 * The status of the response
 * @member {String} status
 */

BaseResponseWithStatus.prototype['status'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement BaseResponseWithStatusAllOf interface:

/**
 * The status of the response
 * @member {String} status
 */

_BaseResponseWithStatusAllOf["default"].prototype['status'] = undefined;
var _default = BaseResponseWithStatus;
exports["default"] = _default;
