// pipedrive/dist/model/BaseProduct.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault1"));

var _VisibleTo = _interopRequireDefault(require("./VisibleTo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseProduct model module.
 * @module model/BaseProduct
 * @version 1.0.0
 */
var BaseProduct = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseProduct</code>.
   * @alias module:model/BaseProduct
   */
  function BaseProduct() {
    (0, _classCallCheck2["default"])(this, BaseProduct);
    BaseProduct.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseProduct, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseProduct</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseProduct} obj Optional instance to populate.
     * @return {module:model/BaseProduct} The populated <code>BaseProduct</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseProduct();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
          delete data['code'];
        }

        if (data.hasOwnProperty('unit')) {
          obj['unit'] = _ApiClient["default"].convertToType(data['unit'], 'String');
          delete data['unit'];
        }

        if (data.hasOwnProperty('tax')) {
          obj['tax'] = _ApiClient["default"].convertToType(data['tax'], 'Number');
          delete data['tax'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _NumberBooleanDefault["default"].constructFromObject(data['active_flag']);
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('selectable')) {
          obj['selectable'] = _NumberBooleanDefault["default"].constructFromObject(data['selectable']);
          delete data['selectable'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], _VisibleTo["default"]);
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], Object);
          delete data['owner_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseProduct;
}();
/**
 * The ID of the product
 * @member {Number} id
 */


BaseProduct.prototype['id'] = undefined;
/**
 * The name of the product
 * @member {String} name
 */

BaseProduct.prototype['name'] = undefined;
/**
 * The product code
 * @member {String} code
 */

BaseProduct.prototype['code'] = undefined;
/**
 * The unit in which this product is sold
 * @member {String} unit
 */

BaseProduct.prototype['unit'] = undefined;
/**
 * The ax percentage
 * @member {Number} tax
 * @default 0
 */

BaseProduct.prototype['tax'] = 0;
/**
 * @member {module:model/NumberBooleanDefault1} active_flag
 */

BaseProduct.prototype['active_flag'] = undefined;
/**
 * @member {module:model/NumberBooleanDefault1} selectable
 */

BaseProduct.prototype['selectable'] = undefined;
/**
 * Visibility of the product
 * @member {module:model/VisibleTo} visible_to
 */

BaseProduct.prototype['visible_to'] = undefined;
/**
 * Information about the Pipedrive user who owns the product
 * @member {Object} owner_id
 */

BaseProduct.prototype['owner_id'] = undefined;
var _default = BaseProduct;
exports["default"] = _default;
