// pipedrive/dist/model/BaseOrganizationItemWithEditNameFlagAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseOrganizationItemWithEditNameFlagAllOf model module.
 * @module model/BaseOrganizationItemWithEditNameFlagAllOf
 * @version 1.0.0
 */
var BaseOrganizationItemWithEditNameFlagAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseOrganizationItemWithEditNameFlagAllOf</code>.
   * @alias module:model/BaseOrganizationItemWithEditNameFlagAllOf
   */
  function BaseOrganizationItemWithEditNameFlagAllOf() {
    (0, _classCallCheck2["default"])(this, BaseOrganizationItemWithEditNameFlagAllOf);
    BaseOrganizationItemWithEditNameFlagAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseOrganizationItemWithEditNameFlagAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseOrganizationItemWithEditNameFlagAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseOrganizationItemWithEditNameFlagAllOf} obj Optional instance to populate.
     * @return {module:model/BaseOrganizationItemWithEditNameFlagAllOf} The populated <code>BaseOrganizationItemWithEditNameFlagAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseOrganizationItemWithEditNameFlagAllOf();

        if (data.hasOwnProperty('edit_name')) {
          obj['edit_name'] = _ApiClient["default"].convertToType(data['edit_name'], 'Boolean');
          delete data['edit_name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseOrganizationItemWithEditNameFlagAllOf;
}();
/**
 * If the company ID of the organization and company ID of the request is same or not
 * @member {Boolean} edit_name
 */


BaseOrganizationItemWithEditNameFlagAllOf.prototype['edit_name'] = undefined;
var _default = BaseOrganizationItemWithEditNameFlagAllOf;
exports["default"] = _default;
