// pipedrive/dist/model/BaseMailThreadMessagesAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MailThreadParticipant = _interopRequireDefault(require("./MailThreadParticipant"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseMailThreadMessagesAllOf model module.
 * @module model/BaseMailThreadMessagesAllOf
 * @version 1.0.0
 */
var BaseMailThreadMessagesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseMailThreadMessagesAllOf</code>.
   * @alias module:model/BaseMailThreadMessagesAllOf
   */
  function BaseMailThreadMessagesAllOf() {
    (0, _classCallCheck2["default"])(this, BaseMailThreadMessagesAllOf);
    BaseMailThreadMessagesAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseMailThreadMessagesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BaseMailThreadMessagesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseMailThreadMessagesAllOf} obj Optional instance to populate.
     * @return {module:model/BaseMailThreadMessagesAllOf} The populated <code>BaseMailThreadMessagesAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseMailThreadMessagesAllOf();

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], [_MailThreadParticipant["default"]]);
          delete data['from'];
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], [_MailThreadParticipant["default"]]);
          delete data['to'];
        }

        if (data.hasOwnProperty('cc')) {
          obj['cc'] = _ApiClient["default"].convertToType(data['cc'], [_MailThreadParticipant["default"]]);
          delete data['cc'];
        }

        if (data.hasOwnProperty('bcc')) {
          obj['bcc'] = _ApiClient["default"].convertToType(data['bcc'], [_MailThreadParticipant["default"]]);
          delete data['bcc'];
        }

        if (data.hasOwnProperty('body_url')) {
          obj['body_url'] = _ApiClient["default"].convertToType(data['body_url'], 'String');
          delete data['body_url'];
        }

        if (data.hasOwnProperty('mail_thread_id')) {
          obj['mail_thread_id'] = _ApiClient["default"].convertToType(data['mail_thread_id'], 'Number');
          delete data['mail_thread_id'];
        }

        if (data.hasOwnProperty('draft')) {
          obj['draft'] = _ApiClient["default"].convertToType(data['draft'], 'String');
          delete data['draft'];
        }

        if (data.hasOwnProperty('has_body_flag')) {
          obj['has_body_flag'] = _ApiClient["default"].convertToType(data['has_body_flag'], _NumberBooleanDefault["default"]);
          delete data['has_body_flag'];
        }

        if (data.hasOwnProperty('sent_flag')) {
          obj['sent_flag'] = _ApiClient["default"].convertToType(data['sent_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_flag'];
        }

        if (data.hasOwnProperty('sent_from_pipedrive_flag')) {
          obj['sent_from_pipedrive_flag'] = _ApiClient["default"].convertToType(data['sent_from_pipedrive_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_from_pipedrive_flag'];
        }

        if (data.hasOwnProperty('message_time')) {
          obj['message_time'] = _ApiClient["default"].convertToType(data['message_time'], 'Date');
          delete data['message_time'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Date');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'Date');
          delete data['update_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseMailThreadMessagesAllOf;
}();
/**
 * Senders of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} from
 */


BaseMailThreadMessagesAllOf.prototype['from'] = undefined;
/**
 * Recipients of the mail thread
 * @member {Array.<module:model/MailThreadParticipant>} to
 */

BaseMailThreadMessagesAllOf.prototype['to'] = undefined;
/**
 * Participants of the Cc
 * @member {Array.<module:model/MailThreadParticipant>} cc
 */

BaseMailThreadMessagesAllOf.prototype['cc'] = undefined;
/**
 * Participants of the Bcc
 * @member {Array.<module:model/MailThreadParticipant>} bcc
 */

BaseMailThreadMessagesAllOf.prototype['bcc'] = undefined;
/**
 * A link to the mail thread message
 * @member {String} body_url
 */

BaseMailThreadMessagesAllOf.prototype['body_url'] = undefined;
/**
 * ID of the mail thread
 * @member {Number} mail_thread_id
 */

BaseMailThreadMessagesAllOf.prototype['mail_thread_id'] = undefined;
/**
 * If the mail message has a draft status then the value is the mail message object as JSON formatted string, otherwise `null`.
 * @member {String} draft
 */

BaseMailThreadMessagesAllOf.prototype['draft'] = undefined;
/**
 * Whether the mail thread message has a body
 * @member {module:model/NumberBooleanDefault0} has_body_flag
 */

BaseMailThreadMessagesAllOf.prototype['has_body_flag'] = undefined;
/**
 * Whether the mail thread message is sent
 * @member {module:model/NumberBooleanDefault0} sent_flag
 */

BaseMailThreadMessagesAllOf.prototype['sent_flag'] = undefined;
/**
 * Whether the mail thread message is sent from Pipedrive
 * @member {module:model/NumberBooleanDefault0} sent_from_pipedrive_flag
 */

BaseMailThreadMessagesAllOf.prototype['sent_from_pipedrive_flag'] = undefined;
/**
 * The time when the mail message was received or created
 * @member {Date} message_time
 */

BaseMailThreadMessagesAllOf.prototype['message_time'] = undefined;
/**
 * The time when the mail message was inserted to database
 * @member {Date} add_time
 */

BaseMailThreadMessagesAllOf.prototype['add_time'] = undefined;
/**
 * The time when the mail message was updated in database received
 * @member {Date} update_time
 */

BaseMailThreadMessagesAllOf.prototype['update_time'] = undefined;
var _default = BaseMailThreadMessagesAllOf;
exports["default"] = _default;
