// pipedrive/dist/model/BaseTeam.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

var _TeamId = _interopRequireDefault(require("./TeamId"));

var _UpdateTeamWithAdditionalProperties = _interopRequireDefault(require("./UpdateTeamWithAdditionalProperties"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The BaseTeam model module.
 * @module model/BaseTeam
 * @version 1.0.0
 */
var BaseTeam = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BaseTeam</code>.
   * @alias module:model/BaseTeam
   * @implements module:model/TeamId
   * @implements module:model/UpdateTeamWithAdditionalProperties
   * @param name {String} The team name
   * @param managerId {Number} The team manager ID
   */
  function BaseTeam(name, managerId) {
    (0, _classCallCheck2["default"])(this, BaseTeam);

    _TeamId["default"].initialize(this);

    _UpdateTeamWithAdditionalProperties["default"].initialize(this, name, managerId);

    BaseTeam.initialize(this, name, managerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(BaseTeam, null, [{
    key: "initialize",
    value: function initialize(obj, name, managerId) {
      obj['name'] = name;
      obj['manager_id'] = managerId;
    }
    /**
     * Constructs a <code>BaseTeam</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BaseTeam} obj Optional instance to populate.
     * @return {module:model/BaseTeam} The populated <code>BaseTeam</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BaseTeam();

        _TeamId["default"].constructFromObject(data, obj);

        _UpdateTeamWithAdditionalProperties["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }

        if (data.hasOwnProperty('manager_id')) {
          obj['manager_id'] = _ApiClient["default"].convertToType(data['manager_id'], 'Number');
          delete data['manager_id'];
        }

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], ['Number']);
          delete data['users'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], _NumberBoolean["default"]);
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBoolean["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('created_by_user_id')) {
          obj['created_by_user_id'] = _ApiClient["default"].convertToType(data['created_by_user_id'], 'Number');
          delete data['created_by_user_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return BaseTeam;
}();
/**
 * The team ID
 * @member {Number} id
 */


BaseTeam.prototype['id'] = undefined;
/**
 * The team name
 * @member {String} name
 */

BaseTeam.prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

BaseTeam.prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

BaseTeam.prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

BaseTeam.prototype['users'] = undefined;
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

BaseTeam.prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

BaseTeam.prototype['deleted_flag'] = undefined;
/**
 * The team creation time. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

BaseTeam.prototype['add_time'] = undefined;
/**
 * The ID of the user who created the team
 * @member {Number} created_by_user_id
 */

BaseTeam.prototype['created_by_user_id'] = undefined; // Implement TeamId interface:

/**
 * The team ID
 * @member {Number} id
 */

_TeamId["default"].prototype['id'] = undefined; // Implement UpdateTeamWithAdditionalProperties interface:

/**
 * The team name
 * @member {String} name
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['users'] = undefined;
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['deleted_flag'] = undefined;
/**
 * The team creation time. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['add_time'] = undefined;
/**
 * The ID of the user who created the team
 * @member {Number} created_by_user_id
 */

_UpdateTeamWithAdditionalProperties["default"].prototype['created_by_user_id'] = undefined;
var _default = BaseTeam;
exports["default"] = _default;
