// pipedrive/dist/model/CalculatedFields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The CalculatedFields model module.
 * @module model/CalculatedFields
 * @version 1.0.0
 */
var CalculatedFields = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CalculatedFields</code>.
   * @alias module:model/CalculatedFields
   */
  function CalculatedFields() {
    (0, _classCallCheck2["default"])(this, CalculatedFields);
    CalculatedFields.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(CalculatedFields, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CalculatedFields</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CalculatedFields} obj Optional instance to populate.
     * @return {module:model/CalculatedFields} The populated <code>CalculatedFields</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CalculatedFields();

        if (data.hasOwnProperty('calculated_type')) {
          obj['calculated_type'] = _ApiClient["default"].convertToType(data['calculated_type'], 'String');
          delete data['calculated_type'];
        }

        if (data.hasOwnProperty('calculated_related_org_id')) {
          obj['calculated_related_org_id'] = _ApiClient["default"].convertToType(data['calculated_related_org_id'], 'Number');
          delete data['calculated_related_org_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return CalculatedFields;
}();
/**
 * The calculated type of the relationship with the linked organization
 * @member {String} calculated_type
 */


CalculatedFields.prototype['calculated_type'] = undefined;
/**
 * The ID of the linked organization
 * @member {Number} calculated_related_org_id
 */

CalculatedFields.prototype['calculated_related_org_id'] = undefined;
var _default = CalculatedFields;
exports["default"] = _default;
