// pipedrive/dist/model/CommentPostPutObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The CommentPostPutObject model module.
 * @module model/CommentPostPutObject
 * @version 1.0.0
 */
var CommentPostPutObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CommentPostPutObject</code>.
   * @alias module:model/CommentPostPutObject
   * @param content {String} The content of the comment in HTML format. Subject to sanitization on the back-end.
   */
  function CommentPostPutObject(content) {
    (0, _classCallCheck2["default"])(this, CommentPostPutObject);
    CommentPostPutObject.initialize(this, content);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(CommentPostPutObject, null, [{
    key: "initialize",
    value: function initialize(obj, content) {
      obj['content'] = content;
    }
    /**
     * Constructs a <code>CommentPostPutObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CommentPostPutObject} obj Optional instance to populate.
     * @return {module:model/CommentPostPutObject} The populated <code>CommentPostPutObject</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CommentPostPutObject();

        if (data.hasOwnProperty('content')) {
          obj['content'] = _ApiClient["default"].convertToType(data['content'], 'String');
          delete data['content'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return CommentPostPutObject;
}();
/**
 * The content of the comment in HTML format. Subject to sanitization on the back-end.
 * @member {String} content
 */


CommentPostPutObject.prototype['content'] = undefined;
var _default = CommentPostPutObject;
exports["default"] = _default;
