// pipedrive/dist/model/CommonMailThread.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The CommonMailThread model module.
 * @module model/CommonMailThread
 * @version 1.0.0
 */
var CommonMailThread = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CommonMailThread</code>.
   * @alias module:model/CommonMailThread
   */
  function CommonMailThread() {
    (0, _classCallCheck2["default"])(this, CommonMailThread);
    CommonMailThread.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(CommonMailThread, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CommonMailThread</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CommonMailThread} obj Optional instance to populate.
     * @return {module:model/CommonMailThread} The populated <code>CommonMailThread</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CommonMailThread();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
          delete data['account_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
          delete data['subject'];
        }

        if (data.hasOwnProperty('snippet')) {
          obj['snippet'] = _ApiClient["default"].convertToType(data['snippet'], 'String');
          delete data['snippet'];
        }

        if (data.hasOwnProperty('read_flag')) {
          obj['read_flag'] = _ApiClient["default"].convertToType(data['read_flag'], _NumberBooleanDefault["default"]);
          delete data['read_flag'];
        }

        if (data.hasOwnProperty('mail_tracking_status')) {
          obj['mail_tracking_status'] = _ApiClient["default"].convertToType(data['mail_tracking_status'], 'String');
          delete data['mail_tracking_status'];
        }

        if (data.hasOwnProperty('has_attachments_flag')) {
          obj['has_attachments_flag'] = _ApiClient["default"].convertToType(data['has_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_attachments_flag'];
        }

        if (data.hasOwnProperty('has_inline_attachments_flag')) {
          obj['has_inline_attachments_flag'] = _ApiClient["default"].convertToType(data['has_inline_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_inline_attachments_flag'];
        }

        if (data.hasOwnProperty('has_real_attachments_flag')) {
          obj['has_real_attachments_flag'] = _ApiClient["default"].convertToType(data['has_real_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_real_attachments_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBooleanDefault["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('synced_flag')) {
          obj['synced_flag'] = _ApiClient["default"].convertToType(data['synced_flag'], _NumberBooleanDefault["default"]);
          delete data['synced_flag'];
        }

        if (data.hasOwnProperty('smart_bcc_flag')) {
          obj['smart_bcc_flag'] = _ApiClient["default"].convertToType(data['smart_bcc_flag'], _NumberBooleanDefault["default"]);
          delete data['smart_bcc_flag'];
        }

        if (data.hasOwnProperty('mail_link_tracking_enabled_flag')) {
          obj['mail_link_tracking_enabled_flag'] = _ApiClient["default"].convertToType(data['mail_link_tracking_enabled_flag'], _NumberBooleanDefault["default"]);
          delete data['mail_link_tracking_enabled_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return CommonMailThread;
}();
/**
 * ID of the mail thread
 * @member {Number} id
 */


CommonMailThread.prototype['id'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

CommonMailThread.prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

CommonMailThread.prototype['user_id'] = undefined;
/**
 * The subject
 * @member {String} subject
 */

CommonMailThread.prototype['subject'] = undefined;
/**
 * A snippet
 * @member {String} snippet
 */

CommonMailThread.prototype['snippet'] = undefined;
/**
 * Whether the mail thread is read
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

CommonMailThread.prototype['read_flag'] = undefined;
/**
 * Mail tracking status
 * @member {String} mail_tracking_status
 */

CommonMailThread.prototype['mail_tracking_status'] = undefined;
/**
 * Whether the mail thread has an attachment
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

CommonMailThread.prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail thread has inline attachments
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

CommonMailThread.prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail thread has real attachments (which are not inline)
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

CommonMailThread.prototype['has_real_attachments_flag'] = undefined;
/**
 * Whether the mail thread is deleted
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

CommonMailThread.prototype['deleted_flag'] = undefined;
/**
 * Whether the mail thread is synced
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

CommonMailThread.prototype['synced_flag'] = undefined;
/**
 * Whether one of the parties of the mail thread is Bcc
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

CommonMailThread.prototype['smart_bcc_flag'] = undefined;
/**
 * Whether the link tracking of the mail thread is enabled
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

CommonMailThread.prototype['mail_link_tracking_enabled_flag'] = undefined;
var _default = CommonMailThread;
exports["default"] = _default;
