// pipedrive/dist/model/CreateTeam.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The CreateTeam model module.
 * @module model/CreateTeam
 * @version 1.0.0
 */
var CreateTeam = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateTeam</code>.
   * @alias module:model/CreateTeam
   * @param name {String} The team name
   * @param managerId {Number} The team manager ID
   */
  function CreateTeam(name, managerId) {
    (0, _classCallCheck2["default"])(this, CreateTeam);
    CreateTeam.initialize(this, name, managerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(CreateTeam, null, [{
    key: "initialize",
    value: function initialize(obj, name, managerId) {
      obj['name'] = name;
      obj['manager_id'] = managerId;
    }
    /**
     * Constructs a <code>CreateTeam</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateTeam} obj Optional instance to populate.
     * @return {module:model/CreateTeam} The populated <code>CreateTeam</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateTeam();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }

        if (data.hasOwnProperty('manager_id')) {
          obj['manager_id'] = _ApiClient["default"].convertToType(data['manager_id'], 'Number');
          delete data['manager_id'];
        }

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], ['Number']);
          delete data['users'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return CreateTeam;
}();
/**
 * The team name
 * @member {String} name
 */


CreateTeam.prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

CreateTeam.prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

CreateTeam.prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

CreateTeam.prototype['users'] = undefined;
var _default = CreateTeam;
exports["default"] = _default;
