// pipedrive/dist/model/DealsMovementsInfoFormattedValues.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealsMovementsInfoFormattedValues model module.
 * @module model/DealsMovementsInfoFormattedValues
 * @version 1.0.0
 */
var DealsMovementsInfoFormattedValues = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealsMovementsInfoFormattedValues</code>.
   * The formatted values of the deals
   * @alias module:model/DealsMovementsInfoFormattedValues
   */
  function DealsMovementsInfoFormattedValues() {
    (0, _classCallCheck2["default"])(this, DealsMovementsInfoFormattedValues);
    DealsMovementsInfoFormattedValues.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealsMovementsInfoFormattedValues, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealsMovementsInfoFormattedValues</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealsMovementsInfoFormattedValues} obj Optional instance to populate.
     * @return {module:model/DealsMovementsInfoFormattedValues} The populated <code>DealsMovementsInfoFormattedValues</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealsMovementsInfoFormattedValues();

        if (data.hasOwnProperty('CURRENCY_ID')) {
          obj['CURRENCY_ID'] = _ApiClient["default"].convertToType(data['CURRENCY_ID'], 'String');
          delete data['CURRENCY_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealsMovementsInfoFormattedValues;
}();
/**
 * The formatted values of the deals
 * @member {String} CURRENCY_ID
 */


DealsMovementsInfoFormattedValues.prototype['CURRENCY_ID'] = undefined;
var _default = DealsMovementsInfoFormattedValues;
exports["default"] = _default;
