// pipedrive/dist/model/DealListActivitiesResponseAllOfRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RelatedDealData = _interopRequireDefault(require("./RelatedDealData"));

var _RelatedOrganizationDataWithActiveFlag = _interopRequireDefault(require("./RelatedOrganizationDataWithActiveFlag"));

var _RelatedPersonDataWithActiveFlag = _interopRequireDefault(require("./RelatedPersonDataWithActiveFlag"));

var _RelatedUserData = _interopRequireDefault(require("./RelatedUserData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealListActivitiesResponseAllOfRelatedObjects model module.
 * @module model/DealListActivitiesResponseAllOfRelatedObjects
 * @version 1.0.0
 */
var DealListActivitiesResponseAllOfRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealListActivitiesResponseAllOfRelatedObjects</code>.
   * @alias module:model/DealListActivitiesResponseAllOfRelatedObjects
   */
  function DealListActivitiesResponseAllOfRelatedObjects() {
    (0, _classCallCheck2["default"])(this, DealListActivitiesResponseAllOfRelatedObjects);
    DealListActivitiesResponseAllOfRelatedObjects.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealListActivitiesResponseAllOfRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealListActivitiesResponseAllOfRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealListActivitiesResponseAllOfRelatedObjects} obj Optional instance to populate.
     * @return {module:model/DealListActivitiesResponseAllOfRelatedObjects} The populated <code>DealListActivitiesResponseAllOfRelatedObjects</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealListActivitiesResponseAllOfRelatedObjects();

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _RelatedOrganizationDataWithActiveFlag["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('person')) {
          obj['person'] = _RelatedPersonDataWithActiveFlag["default"].constructFromObject(data['person']);
          delete data['person'];
        }

        if (data.hasOwnProperty('deal')) {
          obj['deal'] = _RelatedDealData["default"].constructFromObject(data['deal']);
          delete data['deal'];
        }

        if (data.hasOwnProperty('user')) {
          obj['user'] = _RelatedUserData["default"].constructFromObject(data['user']);
          delete data['user'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealListActivitiesResponseAllOfRelatedObjects;
}();
/**
 * @member {module:model/RelatedOrganizationDataWithActiveFlag} organization
 */


DealListActivitiesResponseAllOfRelatedObjects.prototype['organization'] = undefined;
/**
 * @member {module:model/RelatedPersonDataWithActiveFlag} person
 */

DealListActivitiesResponseAllOfRelatedObjects.prototype['person'] = undefined;
/**
 * @member {module:model/RelatedDealData} deal
 */

DealListActivitiesResponseAllOfRelatedObjects.prototype['deal'] = undefined;
/**
 * @member {module:model/RelatedUserData} user
 */

DealListActivitiesResponseAllOfRelatedObjects.prototype['user'] = undefined;
var _default = DealListActivitiesResponseAllOfRelatedObjects;
exports["default"] = _default;
