// pipedrive/dist/model/DealSummaryPerStagesSTAGEIDCURRENCYID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSummaryPerStagesSTAGEIDCURRENCYID model module.
 * @module model/DealSummaryPerStagesSTAGEIDCURRENCYID
 * @version 1.0.0
 */
var DealSummaryPerStagesSTAGEIDCURRENCYID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSummaryPerStagesSTAGEIDCURRENCYID</code>.
   * The currency summary. This parameter is dynamic and changes according to &#x60;currency_id&#x60; value.
   * @alias module:model/DealSummaryPerStagesSTAGEIDCURRENCYID
   */
  function DealSummaryPerStagesSTAGEIDCURRENCYID() {
    (0, _classCallCheck2["default"])(this, DealSummaryPerStagesSTAGEIDCURRENCYID);
    DealSummaryPerStagesSTAGEIDCURRENCYID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSummaryPerStagesSTAGEIDCURRENCYID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSummaryPerStagesSTAGEIDCURRENCYID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSummaryPerStagesSTAGEIDCURRENCYID} obj Optional instance to populate.
     * @return {module:model/DealSummaryPerStagesSTAGEIDCURRENCYID} The populated <code>DealSummaryPerStagesSTAGEIDCURRENCYID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSummaryPerStagesSTAGEIDCURRENCYID();

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('value_formatted')) {
          obj['value_formatted'] = _ApiClient["default"].convertToType(data['value_formatted'], 'String');
          delete data['value_formatted'];
        }

        if (data.hasOwnProperty('weighted_value')) {
          obj['weighted_value'] = _ApiClient["default"].convertToType(data['weighted_value'], 'Number');
          delete data['weighted_value'];
        }

        if (data.hasOwnProperty('weighted_value_formatted')) {
          obj['weighted_value_formatted'] = _ApiClient["default"].convertToType(data['weighted_value_formatted'], 'String');
          delete data['weighted_value_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSummaryPerStagesSTAGEIDCURRENCYID;
}();
/**
 * Deals count per currency
 * @member {Number} count
 */


DealSummaryPerStagesSTAGEIDCURRENCYID.prototype['count'] = undefined;
/**
 * Deals value per currency
 * @member {Number} value
 */

DealSummaryPerStagesSTAGEIDCURRENCYID.prototype['value'] = undefined;
/**
 * Deals value formatted per currency
 * @member {String} value_formatted
 */

DealSummaryPerStagesSTAGEIDCURRENCYID.prototype['value_formatted'] = undefined;
/**
 * Deals weighted value per currency
 * @member {Number} weighted_value
 */

DealSummaryPerStagesSTAGEIDCURRENCYID.prototype['weighted_value'] = undefined;
/**
 * Deals weighted value formatted per currency
 * @member {String} weighted_value_formatted
 */

DealSummaryPerStagesSTAGEIDCURRENCYID.prototype['weighted_value_formatted'] = undefined;
var _default = DealSummaryPerStagesSTAGEIDCURRENCYID;
exports["default"] = _default;
