// pipedrive/dist/model/DealStrictWithMergeIdAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealStrictWithMergeIdAllOf model module.
 * @module model/DealStrictWithMergeIdAllOf
 * @version 1.0.0
 */
var DealStrictWithMergeIdAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealStrictWithMergeIdAllOf</code>.
   * @alias module:model/DealStrictWithMergeIdAllOf
   */
  function DealStrictWithMergeIdAllOf() {
    (0, _classCallCheck2["default"])(this, DealStrictWithMergeIdAllOf);
    DealStrictWithMergeIdAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealStrictWithMergeIdAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealStrictWithMergeIdAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealStrictWithMergeIdAllOf} obj Optional instance to populate.
     * @return {module:model/DealStrictWithMergeIdAllOf} The populated <code>DealStrictWithMergeIdAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealStrictWithMergeIdAllOf();

        if (data.hasOwnProperty('merge_what_id')) {
          obj['merge_what_id'] = _ApiClient["default"].convertToType(data['merge_what_id'], 'Number');
          delete data['merge_what_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealStrictWithMergeIdAllOf;
}();
/**
 * The deal ID of the deal which the original deal was merged with
 * @member {Number} merge_what_id
 */


DealStrictWithMergeIdAllOf.prototype['merge_what_id'] = undefined;
var _default = DealStrictWithMergeIdAllOf;
exports["default"] = _default;
