// pipedrive/dist/model/DealSummary.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSummaryPerCurrency = _interopRequireDefault(require("./DealSummaryPerCurrency"));

var _DealSummaryPerCurrencyFull = _interopRequireDefault(require("./DealSummaryPerCurrencyFull"));

var _DealSummaryPerStages = _interopRequireDefault(require("./DealSummaryPerStages"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSummary model module.
 * @module model/DealSummary
 * @version 1.0.0
 */
var DealSummary = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSummary</code>.
   * Deals summary
   * @alias module:model/DealSummary
   */
  function DealSummary() {
    (0, _classCallCheck2["default"])(this, DealSummary);
    DealSummary.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSummary, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSummary</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSummary} obj Optional instance to populate.
     * @return {module:model/DealSummary} The populated <code>DealSummary</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSummary();

        if (data.hasOwnProperty('per_stages')) {
          obj['per_stages'] = _DealSummaryPerStages["default"].constructFromObject(data['per_stages']);
          delete data['per_stages'];
        }

        if (data.hasOwnProperty('per_currency')) {
          obj['per_currency'] = _DealSummaryPerCurrency["default"].constructFromObject(data['per_currency']);
          delete data['per_currency'];
        }

        if (data.hasOwnProperty('total_count')) {
          obj['total_count'] = _ApiClient["default"].convertToType(data['total_count'], 'Number');
          delete data['total_count'];
        }

        if (data.hasOwnProperty('per_currency_full')) {
          obj['per_currency_full'] = _DealSummaryPerCurrencyFull["default"].constructFromObject(data['per_currency_full']);
          delete data['per_currency_full'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSummary;
}();
/**
 * @member {module:model/DealSummaryPerStages} per_stages
 */


DealSummary.prototype['per_stages'] = undefined;
/**
 * @member {module:model/DealSummaryPerCurrency} per_currency
 */

DealSummary.prototype['per_currency'] = undefined;
/**
 * Deals count
 * @member {Number} total_count
 */

DealSummary.prototype['total_count'] = undefined;
/**
 * @member {module:model/DealSummaryPerCurrencyFull} per_currency_full
 */

DealSummary.prototype['per_currency_full'] = undefined;
var _default = DealSummary;
exports["default"] = _default;
