// pipedrive/dist/model/DealsMovementsInfoValues.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealsMovementsInfoValues model module.
 * @module model/DealsMovementsInfoValues
 * @version 1.0.0
 */
var DealsMovementsInfoValues = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealsMovementsInfoValues</code>.
   * The values of the deals
   * @alias module:model/DealsMovementsInfoValues
   */
  function DealsMovementsInfoValues() {
    (0, _classCallCheck2["default"])(this, DealsMovementsInfoValues);
    DealsMovementsInfoValues.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealsMovementsInfoValues, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealsMovementsInfoValues</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealsMovementsInfoValues} obj Optional instance to populate.
     * @return {module:model/DealsMovementsInfoValues} The populated <code>DealsMovementsInfoValues</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealsMovementsInfoValues();

        if (data.hasOwnProperty('CURRENCY_ID')) {
          obj['CURRENCY_ID'] = _ApiClient["default"].convertToType(data['CURRENCY_ID'], 'Number');
          delete data['CURRENCY_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealsMovementsInfoValues;
}();
/**
 * The value of the deals
 * @member {Number} CURRENCY_ID
 */


DealsMovementsInfoValues.prototype['CURRENCY_ID'] = undefined;
var _default = DealsMovementsInfoValues;
exports["default"] = _default;
