// pipedrive/dist/model/DealSummaryPerCurrencyFull.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSummaryPerCurrencyFullCURRENCYID = _interopRequireDefault(require("./DealSummaryPerCurrencyFullCURRENCYID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSummaryPerCurrencyFull model module.
 * @module model/DealSummaryPerCurrencyFull
 * @version 1.0.0
 */
var DealSummaryPerCurrencyFull = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSummaryPerCurrencyFull</code>.
   * Full currency summaries
   * @alias module:model/DealSummaryPerCurrencyFull
   */
  function DealSummaryPerCurrencyFull() {
    (0, _classCallCheck2["default"])(this, DealSummaryPerCurrencyFull);
    DealSummaryPerCurrencyFull.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSummaryPerCurrencyFull, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSummaryPerCurrencyFull</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSummaryPerCurrencyFull} obj Optional instance to populate.
     * @return {module:model/DealSummaryPerCurrencyFull} The populated <code>DealSummaryPerCurrencyFull</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSummaryPerCurrencyFull();

        if (data.hasOwnProperty('CURRENCY_ID')) {
          obj['CURRENCY_ID'] = _DealSummaryPerCurrencyFullCURRENCYID["default"].constructFromObject(data['CURRENCY_ID']);
          delete data['CURRENCY_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSummaryPerCurrencyFull;
}();
/**
 * @member {module:model/DealSummaryPerCurrencyFullCURRENCYID} CURRENCY_ID
 */


DealSummaryPerCurrencyFull.prototype['CURRENCY_ID'] = undefined;
var _default = DealSummaryPerCurrencyFull;
exports["default"] = _default;
