// pipedrive/dist/model/DealSearchItemItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSearchItemItemOrganization = _interopRequireDefault(require("./DealSearchItemItemOrganization"));

var _DealSearchItemItemOwner = _interopRequireDefault(require("./DealSearchItemItemOwner"));

var _DealSearchItemItemPerson = _interopRequireDefault(require("./DealSearchItemItemPerson"));

var _DealSearchItemItemStage = _interopRequireDefault(require("./DealSearchItemItemStage"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSearchItemItem model module.
 * @module model/DealSearchItemItem
 * @version 1.0.0
 */
var DealSearchItemItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSearchItemItem</code>.
   * @alias module:model/DealSearchItemItem
   */
  function DealSearchItemItem() {
    (0, _classCallCheck2["default"])(this, DealSearchItemItem);
    DealSearchItemItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSearchItemItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSearchItemItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSearchItemItem} obj Optional instance to populate.
     * @return {module:model/DealSearchItemItem} The populated <code>DealSearchItemItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSearchItemItem();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'Number');
          delete data['currency'];
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'Number');
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _DealSearchItemItemOwner["default"].constructFromObject(data['owner']);
          delete data['owner'];
        }

        if (data.hasOwnProperty('stage')) {
          obj['stage'] = _DealSearchItemItemStage["default"].constructFromObject(data['stage']);
          delete data['stage'];
        }

        if (data.hasOwnProperty('person')) {
          obj['person'] = _DealSearchItemItemPerson["default"].constructFromObject(data['person']);
          delete data['person'];
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _DealSearchItemItemOrganization["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('custom_fields')) {
          obj['custom_fields'] = _ApiClient["default"].convertToType(data['custom_fields'], 'Number');
          delete data['custom_fields'];
        }

        if (data.hasOwnProperty('notes')) {
          obj['notes'] = _ApiClient["default"].convertToType(data['notes'], ['String']);
          delete data['notes'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSearchItemItem;
}();
/**
 * The ID of the deal
 * @member {Number} id
 */


DealSearchItemItem.prototype['id'] = undefined;
/**
 * The type of the item
 * @member {String} type
 */

DealSearchItemItem.prototype['type'] = undefined;
/**
 * The title of the deal
 * @member {String} title
 */

DealSearchItemItem.prototype['title'] = undefined;
/**
 * The value of the deal
 * @member {Number} value
 */

DealSearchItemItem.prototype['value'] = undefined;
/**
 * The currency of the deal
 * @member {Number} currency
 */

DealSearchItemItem.prototype['currency'] = undefined;
/**
 * The status of the deal
 * @member {String} status
 */

DealSearchItemItem.prototype['status'] = undefined;
/**
 * The visibility of the deal
 * @member {Number} visible_to
 */

DealSearchItemItem.prototype['visible_to'] = undefined;
/**
 * @member {module:model/DealSearchItemItemOwner} owner
 */

DealSearchItemItem.prototype['owner'] = undefined;
/**
 * @member {module:model/DealSearchItemItemStage} stage
 */

DealSearchItemItem.prototype['stage'] = undefined;
/**
 * @member {module:model/DealSearchItemItemPerson} person
 */

DealSearchItemItem.prototype['person'] = undefined;
/**
 * @member {module:model/DealSearchItemItemOrganization} organization
 */

DealSearchItemItem.prototype['organization'] = undefined;
/**
 * Custom fields
 * @member {Number} custom_fields
 */

DealSearchItemItem.prototype['custom_fields'] = undefined;
/**
 * An array of notes
 * @member {Array.<String>} notes
 */

DealSearchItemItem.prototype['notes'] = undefined;
var _default = DealSearchItemItem;
exports["default"] = _default;
