// pipedrive/dist/model/DealSearchResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSearchItem = _interopRequireDefault(require("./DealSearchItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSearchResponseAllOfData model module.
 * @module model/DealSearchResponseAllOfData
 * @version 1.0.0
 */
var DealSearchResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSearchResponseAllOfData</code>.
   * @alias module:model/DealSearchResponseAllOfData
   */
  function DealSearchResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, DealSearchResponseAllOfData);
    DealSearchResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSearchResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSearchResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSearchResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/DealSearchResponseAllOfData} The populated <code>DealSearchResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSearchResponseAllOfData();

        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_DealSearchItem["default"]]);
          delete data['items'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSearchResponseAllOfData;
}();
/**
 * The array of deals
 * @member {Array.<module:model/DealSearchItem>} items
 */


DealSearchResponseAllOfData.prototype['items'] = undefined;
var _default = DealSearchResponseAllOfData;
exports["default"] = _default;
