// pipedrive/dist/model/DealListActivitiesResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataWithAdditionalData = _interopRequireDefault(require("./ActivityDistributionDataWithAdditionalData"));

var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _DealListActivitiesResponseAllOf = _interopRequireDefault(require("./DealListActivitiesResponseAllOf"));

var _DealListActivitiesResponseAllOfRelatedObjects = _interopRequireDefault(require("./DealListActivitiesResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealListActivitiesResponse model module.
 * @module model/DealListActivitiesResponse
 * @version 1.0.0
 */
var DealListActivitiesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealListActivitiesResponse</code>.
   * @alias module:model/DealListActivitiesResponse
   * @implements module:model/BaseResponse
   * @implements module:model/DealListActivitiesResponseAllOf
   */
  function DealListActivitiesResponse() {
    (0, _classCallCheck2["default"])(this, DealListActivitiesResponse);

    _BaseResponse["default"].initialize(this);

    _DealListActivitiesResponseAllOf["default"].initialize(this);

    DealListActivitiesResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealListActivitiesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealListActivitiesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealListActivitiesResponse} obj Optional instance to populate.
     * @return {module:model/DealListActivitiesResponse} The populated <code>DealListActivitiesResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealListActivitiesResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _DealListActivitiesResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ActivityResponseObject["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ActivityDistributionDataWithAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _DealListActivitiesResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealListActivitiesResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


DealListActivitiesResponse.prototype['success'] = undefined;
/**
 * The array of activities
 * @member {Array.<module:model/ActivityResponseObject>} data
 */

DealListActivitiesResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ActivityDistributionDataWithAdditionalData} additional_data
 */

DealListActivitiesResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/DealListActivitiesResponseAllOfRelatedObjects} related_objects
 */

DealListActivitiesResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement DealListActivitiesResponseAllOf interface:

/**
 * The array of activities
 * @member {Array.<module:model/ActivityResponseObject>} data
 */

_DealListActivitiesResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ActivityDistributionDataWithAdditionalData} additional_data
 */

_DealListActivitiesResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/DealListActivitiesResponseAllOfRelatedObjects} related_objects
 */

_DealListActivitiesResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = DealListActivitiesResponse;
exports["default"] = _default;
