// pipedrive/dist/model/DealSummaryPerCurrencyFullCURRENCYID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSummaryPerCurrencyFullCURRENCYID model module.
 * @module model/DealSummaryPerCurrencyFullCURRENCYID
 * @version 1.0.0
 */
var DealSummaryPerCurrencyFullCURRENCYID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSummaryPerCurrencyFullCURRENCYID</code>.
   * The currency summary. This parameter is dynamic and changes according to &#x60;currency_id&#x60; value.
   * @alias module:model/DealSummaryPerCurrencyFullCURRENCYID
   */
  function DealSummaryPerCurrencyFullCURRENCYID() {
    (0, _classCallCheck2["default"])(this, DealSummaryPerCurrencyFullCURRENCYID);
    DealSummaryPerCurrencyFullCURRENCYID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSummaryPerCurrencyFullCURRENCYID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSummaryPerCurrencyFullCURRENCYID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSummaryPerCurrencyFullCURRENCYID} obj Optional instance to populate.
     * @return {module:model/DealSummaryPerCurrencyFullCURRENCYID} The populated <code>DealSummaryPerCurrencyFullCURRENCYID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSummaryPerCurrencyFullCURRENCYID();

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSummaryPerCurrencyFullCURRENCYID;
}();
/**
 * Deals count per currency
 * @member {Number} count
 */


DealSummaryPerCurrencyFullCURRENCYID.prototype['count'] = undefined;
/**
 * Deals value per currency
 * @member {Number} value
 */

DealSummaryPerCurrencyFullCURRENCYID.prototype['value'] = undefined;
var _default = DealSummaryPerCurrencyFullCURRENCYID;
exports["default"] = _default;
