// pipedrive/dist/model/DealFlowResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _DealFlowResponseAllOfData = _interopRequireDefault(require("./DealFlowResponseAllOfData"));

var _DealFlowResponseAllOfRelatedObjects = _interopRequireDefault(require("./DealFlowResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealFlowResponseAllOf model module.
 * @module model/DealFlowResponseAllOf
 * @version 1.0.0
 */
var DealFlowResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealFlowResponseAllOf</code>.
   * @alias module:model/DealFlowResponseAllOf
   */
  function DealFlowResponseAllOf() {
    (0, _classCallCheck2["default"])(this, DealFlowResponseAllOf);
    DealFlowResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealFlowResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealFlowResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealFlowResponseAllOf} obj Optional instance to populate.
     * @return {module:model/DealFlowResponseAllOf} The populated <code>DealFlowResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealFlowResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_DealFlowResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _DealFlowResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealFlowResponseAllOf;
}();
/**
 * @member {Array.<module:model/DealFlowResponseAllOfData>} data
 */


DealFlowResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

DealFlowResponseAllOf.prototype['additional_data'] = undefined;
/**
 * @member {module:model/DealFlowResponseAllOfRelatedObjects} related_objects
 */

DealFlowResponseAllOf.prototype['related_objects'] = undefined;
var _default = DealFlowResponseAllOf;
exports["default"] = _default;
