// pipedrive/dist/model/DealPersonData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealPersonDataEmail = _interopRequireDefault(require("./DealPersonDataEmail"));

var _DealPersonDataPhone = _interopRequireDefault(require("./DealPersonDataPhone"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealPersonData model module.
 * @module model/DealPersonData
 * @version 1.0.0
 */
var DealPersonData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealPersonData</code>.
   * The person who is associated with the deal
   * @alias module:model/DealPersonData
   */
  function DealPersonData() {
    (0, _classCallCheck2["default"])(this, DealPersonData);
    DealPersonData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealPersonData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealPersonData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealPersonData} obj Optional instance to populate.
     * @return {module:model/DealPersonData} The populated <code>DealPersonData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealPersonData();

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], [_DealPersonDataEmail["default"]]);
          delete data['email'];
        }

        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], [_DealPersonDataPhone["default"]]);
          delete data['phone'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealPersonData;
}();
/**
 * Whether the associated person is active or not
 * @member {Boolean} active_flag
 */


DealPersonData.prototype['active_flag'] = undefined;
/**
 * The name of the person associated with the deal
 * @member {String} name
 */

DealPersonData.prototype['name'] = undefined;
/**
 * The emails of the person associated with the deal
 * @member {Array.<module:model/DealPersonDataEmail>} email
 */

DealPersonData.prototype['email'] = undefined;
/**
 * The phone numbers of the person associated with the deal
 * @member {Array.<module:model/DealPersonDataPhone>} phone
 */

DealPersonData.prototype['phone'] = undefined;
/**
 * The ID of the owner of the person that is associated with the deal
 * @member {Number} owner_id
 */

DealPersonData.prototype['owner_id'] = undefined;
var _default = DealPersonData;
exports["default"] = _default;
