// pipedrive/dist/model/DealUserData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealUserData model module.
 * @module model/DealUserData
 * @version 1.0.0
 */
var DealUserData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealUserData</code>.
   * The user who is associated with the deal
   * @alias module:model/DealUserData
   */
  function DealUserData() {
    (0, _classCallCheck2["default"])(this, DealUserData);
    DealUserData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealUserData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealUserData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealUserData} obj Optional instance to populate.
     * @return {module:model/DealUserData} The populated <code>DealUserData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealUserData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
          delete data['email'];
        }

        if (data.hasOwnProperty('has_pic')) {
          obj['has_pic'] = _ApiClient["default"].convertToType(data['has_pic'], 'Boolean');
          delete data['has_pic'];
        }

        if (data.hasOwnProperty('pic_hash')) {
          obj['pic_hash'] = _ApiClient["default"].convertToType(data['pic_hash'], 'String');
          delete data['pic_hash'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealUserData;
}();
/**
 * The ID of the user
 * @member {Number} id
 */


DealUserData.prototype['id'] = undefined;
/**
 * The name of the user
 * @member {String} name
 */

DealUserData.prototype['name'] = undefined;
/**
 * The email of the user
 * @member {String} email
 */

DealUserData.prototype['email'] = undefined;
/**
 * If the user has a picture or not
 * @member {Boolean} has_pic
 */

DealUserData.prototype['has_pic'] = undefined;
/**
 * The user picture hash
 * @member {String} pic_hash
 */

DealUserData.prototype['pic_hash'] = undefined;
/**
 * Whether the user is active or not
 * @member {Boolean} active_flag
 */

DealUserData.prototype['active_flag'] = undefined;
var _default = DealUserData;
exports["default"] = _default;
