// pipedrive/dist/model/DealParticipantCountInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealParticipantCountInfo model module.
 * @module model/DealParticipantCountInfo
 * @version 1.0.0
 */
var DealParticipantCountInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealParticipantCountInfo</code>.
   * @alias module:model/DealParticipantCountInfo
   */
  function DealParticipantCountInfo() {
    (0, _classCallCheck2["default"])(this, DealParticipantCountInfo);
    DealParticipantCountInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealParticipantCountInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealParticipantCountInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealParticipantCountInfo} obj Optional instance to populate.
     * @return {module:model/DealParticipantCountInfo} The populated <code>DealParticipantCountInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealParticipantCountInfo();

        if (data.hasOwnProperty('participant_open_deals_count')) {
          obj['participant_open_deals_count'] = _ApiClient["default"].convertToType(data['participant_open_deals_count'], 'Number');
          delete data['participant_open_deals_count'];
        }

        if (data.hasOwnProperty('participant_closed_deals_count')) {
          obj['participant_closed_deals_count'] = _ApiClient["default"].convertToType(data['participant_closed_deals_count'], 'Number');
          delete data['participant_closed_deals_count'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealParticipantCountInfo;
}();
/**
 * The count of open participant deals related with the item
 * @member {Number} participant_open_deals_count
 */


DealParticipantCountInfo.prototype['participant_open_deals_count'] = undefined;
/**
 * The count of closed participant deals related with the item
 * @member {Number} participant_closed_deals_count
 */

DealParticipantCountInfo.prototype['participant_closed_deals_count'] = undefined;
var _default = DealParticipantCountInfo;
exports["default"] = _default;
