// pipedrive/dist/model/DealsMovementsInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealsMovementsInfoFormattedValues = _interopRequireDefault(require("./DealsMovementsInfoFormattedValues"));

var _DealsMovementsInfoValues = _interopRequireDefault(require("./DealsMovementsInfoValues"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealsMovementsInfo model module.
 * @module model/DealsMovementsInfo
 * @version 1.0.0
 */
var DealsMovementsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealsMovementsInfo</code>.
   * Deals summary
   * @alias module:model/DealsMovementsInfo
   */
  function DealsMovementsInfo() {
    (0, _classCallCheck2["default"])(this, DealsMovementsInfo);
    DealsMovementsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealsMovementsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealsMovementsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealsMovementsInfo} obj Optional instance to populate.
     * @return {module:model/DealsMovementsInfo} The populated <code>DealsMovementsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealsMovementsInfo();

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('deals_ids')) {
          obj['deals_ids'] = _ApiClient["default"].convertToType(data['deals_ids'], ['Number']);
          delete data['deals_ids'];
        }

        if (data.hasOwnProperty('values')) {
          obj['values'] = _DealsMovementsInfoValues["default"].constructFromObject(data['values']);
          delete data['values'];
        }

        if (data.hasOwnProperty('formatted_values')) {
          obj['formatted_values'] = _DealsMovementsInfoFormattedValues["default"].constructFromObject(data['formatted_values']);
          delete data['formatted_values'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealsMovementsInfo;
}();
/**
 * The count of the deals
 * @member {Number} count
 */


DealsMovementsInfo.prototype['count'] = undefined;
/**
 * The IDs of the deals that have been moved
 * @member {Array.<Number>} deals_ids
 */

DealsMovementsInfo.prototype['deals_ids'] = undefined;
/**
 * @member {module:model/DealsMovementsInfoValues} values
 */

DealsMovementsInfo.prototype['values'] = undefined;
/**
 * @member {module:model/DealsMovementsInfoFormattedValues} formatted_values
 */

DealsMovementsInfo.prototype['formatted_values'] = undefined;
var _default = DealsMovementsInfo;
exports["default"] = _default;
