// pipedrive/dist/model/DealStrictModeFields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealStrictModeFields model module.
 * @module model/DealStrictModeFields
 * @version 1.0.0
 */
var DealStrictModeFields = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealStrictModeFields</code>.
   * @alias module:model/DealStrictModeFields
   */
  function DealStrictModeFields() {
    (0, _classCallCheck2["default"])(this, DealStrictModeFields);
    DealStrictModeFields.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealStrictModeFields, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealStrictModeFields</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealStrictModeFields} obj Optional instance to populate.
     * @return {module:model/DealStrictModeFields} The populated <code>DealStrictModeFields</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealStrictModeFields();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('creator_user_id')) {
          obj['creator_user_id'] = _ApiClient["default"].convertToType(data['creator_user_id'], 'Number');
          delete data['creator_user_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealStrictModeFields;
}();
/**
 * The ID of the deal
 * @member {Number} id
 */


DealStrictModeFields.prototype['id'] = undefined;
/**
 * The ID of the deal creator
 * @member {Number} creator_user_id
 */

DealStrictModeFields.prototype['creator_user_id'] = undefined;
/**
 * The ID of the user
 * @member {Number} user_id
 */

DealStrictModeFields.prototype['user_id'] = undefined;
/**
 * The ID of the person associated with the deal
 * @member {Number} person_id
 */

DealStrictModeFields.prototype['person_id'] = undefined;
/**
 * The ID of the organization associated with the deal
 * @member {Number} org_id
 */

DealStrictModeFields.prototype['org_id'] = undefined;
var _default = DealStrictModeFields;
exports["default"] = _default;
