// pipedrive/dist/model/DealSummaryPerStages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSummaryPerStagesSTAGEID = _interopRequireDefault(require("./DealSummaryPerStagesSTAGEID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSummaryPerStages model module.
 * @module model/DealSummaryPerStages
 * @version 1.0.0
 */
var DealSummaryPerStages = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSummaryPerStages</code>.
   * The stage objects containing deals currency information
   * @alias module:model/DealSummaryPerStages
   */
  function DealSummaryPerStages() {
    (0, _classCallCheck2["default"])(this, DealSummaryPerStages);
    DealSummaryPerStages.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSummaryPerStages, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSummaryPerStages</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSummaryPerStages} obj Optional instance to populate.
     * @return {module:model/DealSummaryPerStages} The populated <code>DealSummaryPerStages</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSummaryPerStages();

        if (data.hasOwnProperty('STAGE_ID')) {
          obj['STAGE_ID'] = _DealSummaryPerStagesSTAGEID["default"].constructFromObject(data['STAGE_ID']);
          delete data['STAGE_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSummaryPerStages;
}();
/**
 * @member {module:model/DealSummaryPerStagesSTAGEID} STAGE_ID
 */


DealSummaryPerStages.prototype['STAGE_ID'] = undefined;
var _default = DealSummaryPerStages;
exports["default"] = _default;
