// pipedrive/dist/model/DealListActivitiesResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataWithAdditionalData = _interopRequireDefault(require("./ActivityDistributionDataWithAdditionalData"));

var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));

var _DealListActivitiesResponseAllOfRelatedObjects = _interopRequireDefault(require("./DealListActivitiesResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealListActivitiesResponseAllOf model module.
 * @module model/DealListActivitiesResponseAllOf
 * @version 1.0.0
 */
var DealListActivitiesResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealListActivitiesResponseAllOf</code>.
   * @alias module:model/DealListActivitiesResponseAllOf
   */
  function DealListActivitiesResponseAllOf() {
    (0, _classCallCheck2["default"])(this, DealListActivitiesResponseAllOf);
    DealListActivitiesResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealListActivitiesResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealListActivitiesResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealListActivitiesResponseAllOf} obj Optional instance to populate.
     * @return {module:model/DealListActivitiesResponseAllOf} The populated <code>DealListActivitiesResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealListActivitiesResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ActivityResponseObject["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ActivityDistributionDataWithAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _DealListActivitiesResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealListActivitiesResponseAllOf;
}();
/**
 * The array of activities
 * @member {Array.<module:model/ActivityResponseObject>} data
 */


DealListActivitiesResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/ActivityDistributionDataWithAdditionalData} additional_data
 */

DealListActivitiesResponseAllOf.prototype['additional_data'] = undefined;
/**
 * @member {module:model/DealListActivitiesResponseAllOfRelatedObjects} related_objects
 */

DealListActivitiesResponseAllOf.prototype['related_objects'] = undefined;
var _default = DealListActivitiesResponseAllOf;
exports["default"] = _default;
