// pipedrive/dist/model/DealNonStrictWithDetailsAllOfAge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealNonStrictWithDetailsAllOfAge model module.
 * @module model/DealNonStrictWithDetailsAllOfAge
 * @version 1.0.0
 */
var DealNonStrictWithDetailsAllOfAge = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealNonStrictWithDetailsAllOfAge</code>.
   * The lifetime of the deal
   * @alias module:model/DealNonStrictWithDetailsAllOfAge
   */
  function DealNonStrictWithDetailsAllOfAge() {
    (0, _classCallCheck2["default"])(this, DealNonStrictWithDetailsAllOfAge);
    DealNonStrictWithDetailsAllOfAge.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealNonStrictWithDetailsAllOfAge, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealNonStrictWithDetailsAllOfAge</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealNonStrictWithDetailsAllOfAge} obj Optional instance to populate.
     * @return {module:model/DealNonStrictWithDetailsAllOfAge} The populated <code>DealNonStrictWithDetailsAllOfAge</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealNonStrictWithDetailsAllOfAge();

        if (data.hasOwnProperty('y')) {
          obj['y'] = _ApiClient["default"].convertToType(data['y'], 'Number');
          delete data['y'];
        }

        if (data.hasOwnProperty('m')) {
          obj['m'] = _ApiClient["default"].convertToType(data['m'], 'Number');
          delete data['m'];
        }

        if (data.hasOwnProperty('d')) {
          obj['d'] = _ApiClient["default"].convertToType(data['d'], 'Number');
          delete data['d'];
        }

        if (data.hasOwnProperty('h')) {
          obj['h'] = _ApiClient["default"].convertToType(data['h'], 'Number');
          delete data['h'];
        }

        if (data.hasOwnProperty('i')) {
          obj['i'] = _ApiClient["default"].convertToType(data['i'], 'Number');
          delete data['i'];
        }

        if (data.hasOwnProperty('s')) {
          obj['s'] = _ApiClient["default"].convertToType(data['s'], 'Number');
          delete data['s'];
        }

        if (data.hasOwnProperty('total_seconds')) {
          obj['total_seconds'] = _ApiClient["default"].convertToType(data['total_seconds'], 'Number');
          delete data['total_seconds'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealNonStrictWithDetailsAllOfAge;
}();
/**
 * Years
 * @member {Number} y
 */


DealNonStrictWithDetailsAllOfAge.prototype['y'] = undefined;
/**
 * Months
 * @member {Number} m
 */

DealNonStrictWithDetailsAllOfAge.prototype['m'] = undefined;
/**
 * Days
 * @member {Number} d
 */

DealNonStrictWithDetailsAllOfAge.prototype['d'] = undefined;
/**
 * Hours
 * @member {Number} h
 */

DealNonStrictWithDetailsAllOfAge.prototype['h'] = undefined;
/**
 * Minutes
 * @member {Number} i
 */

DealNonStrictWithDetailsAllOfAge.prototype['i'] = undefined;
/**
 * Seconds
 * @member {Number} s
 */

DealNonStrictWithDetailsAllOfAge.prototype['s'] = undefined;
/**
 * The total time in seconds
 * @member {Number} total_seconds
 */

DealNonStrictWithDetailsAllOfAge.prototype['total_seconds'] = undefined;
var _default = DealNonStrictWithDetailsAllOfAge;
exports["default"] = _default;
