// pipedrive/dist/model/DealNonStrictModeFields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealNonStrictModeFieldsCreatorUserId = _interopRequireDefault(require("./DealNonStrictModeFieldsCreatorUserId"));

var _DealOrganizationDataWithId = _interopRequireDefault(require("./DealOrganizationDataWithId"));

var _DealPersonDataWithId = _interopRequireDefault(require("./DealPersonDataWithId"));

var _DealUserDataWithId = _interopRequireDefault(require("./DealUserDataWithId"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealNonStrictModeFields model module.
 * @module model/DealNonStrictModeFields
 * @version 1.0.0
 */
var DealNonStrictModeFields = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealNonStrictModeFields</code>.
   * @alias module:model/DealNonStrictModeFields
   */
  function DealNonStrictModeFields() {
    (0, _classCallCheck2["default"])(this, DealNonStrictModeFields);
    DealNonStrictModeFields.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealNonStrictModeFields, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealNonStrictModeFields</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealNonStrictModeFields} obj Optional instance to populate.
     * @return {module:model/DealNonStrictModeFields} The populated <code>DealNonStrictModeFields</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealNonStrictModeFields();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('creator_user_id')) {
          obj['creator_user_id'] = _DealNonStrictModeFieldsCreatorUserId["default"].constructFromObject(data['creator_user_id']);
          delete data['creator_user_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _DealUserDataWithId["default"].constructFromObject(data['user_id']);
          delete data['user_id'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _DealPersonDataWithId["default"].constructFromObject(data['person_id']);
          delete data['person_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _DealOrganizationDataWithId["default"].constructFromObject(data['org_id']);
          delete data['org_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealNonStrictModeFields;
}();
/**
 * The ID of the deal
 * @member {Number} id
 */


DealNonStrictModeFields.prototype['id'] = undefined;
/**
 * @member {module:model/DealNonStrictModeFieldsCreatorUserId} creator_user_id
 */

DealNonStrictModeFields.prototype['creator_user_id'] = undefined;
/**
 * @member {module:model/DealUserDataWithId} user_id
 */

DealNonStrictModeFields.prototype['user_id'] = undefined;
/**
 * @member {module:model/DealPersonDataWithId} person_id
 */

DealNonStrictModeFields.prototype['person_id'] = undefined;
/**
 * @member {module:model/DealOrganizationDataWithId} org_id
 */

DealNonStrictModeFields.prototype['org_id'] = undefined;
var _default = DealNonStrictModeFields;
exports["default"] = _default;
