// pipedrive/dist/model/DealSearchResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _DealSearchResponseAllOf = _interopRequireDefault(require("./DealSearchResponseAllOf"));

var _DealSearchResponseAllOfData = _interopRequireDefault(require("./DealSearchResponseAllOfData"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DealSearchResponse model module.
 * @module model/DealSearchResponse
 * @version 1.0.0
 */
var DealSearchResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DealSearchResponse</code>.
   * @alias module:model/DealSearchResponse
   * @implements module:model/BaseResponse
   * @implements module:model/DealSearchResponseAllOf
   */
  function DealSearchResponse() {
    (0, _classCallCheck2["default"])(this, DealSearchResponse);

    _BaseResponse["default"].initialize(this);

    _DealSearchResponseAllOf["default"].initialize(this);

    DealSearchResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DealSearchResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DealSearchResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DealSearchResponse} obj Optional instance to populate.
     * @return {module:model/DealSearchResponse} The populated <code>DealSearchResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DealSearchResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _DealSearchResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _DealSearchResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DealSearchResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


DealSearchResponse.prototype['success'] = undefined;
/**
 * @member {module:model/DealSearchResponseAllOfData} data
 */

DealSearchResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

DealSearchResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement DealSearchResponseAllOf interface:

/**
 * @member {module:model/DealSearchResponseAllOfData} data
 */

_DealSearchResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_DealSearchResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = DealSearchResponse;
exports["default"] = _default;
