// pipedrive/dist/model/DeleteTeamUserRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeleteTeamUserRequest model module.
 * @module model/DeleteTeamUserRequest
 * @version 1.0.0
 */
var DeleteTeamUserRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteTeamUserRequest</code>.
   * @alias module:model/DeleteTeamUserRequest
   * @param users {Array.<Number>} The list of user IDs
   */
  function DeleteTeamUserRequest(users) {
    (0, _classCallCheck2["default"])(this, DeleteTeamUserRequest);
    DeleteTeamUserRequest.initialize(this, users);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeleteTeamUserRequest, null, [{
    key: "initialize",
    value: function initialize(obj, users) {
      obj['users'] = users;
    }
    /**
     * Constructs a <code>DeleteTeamUserRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteTeamUserRequest} obj Optional instance to populate.
     * @return {module:model/DeleteTeamUserRequest} The populated <code>DeleteTeamUserRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteTeamUserRequest();

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], ['Number']);
          delete data['users'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeleteTeamUserRequest;
}();
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */


DeleteTeamUserRequest.prototype['users'] = undefined;
var _default = DeleteTeamUserRequest;
exports["default"] = _default;
