// pipedrive/dist/model/DeletePersonsInBulkResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DeletePersonsInBulkResponseAllOfData = _interopRequireDefault(require("./DeletePersonsInBulkResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeletePersonsInBulkResponseAllOf model module.
 * @module model/DeletePersonsInBulkResponseAllOf
 * @version 1.0.0
 */
var DeletePersonsInBulkResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeletePersonsInBulkResponseAllOf</code>.
   * @alias module:model/DeletePersonsInBulkResponseAllOf
   */
  function DeletePersonsInBulkResponseAllOf() {
    (0, _classCallCheck2["default"])(this, DeletePersonsInBulkResponseAllOf);
    DeletePersonsInBulkResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeletePersonsInBulkResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeletePersonsInBulkResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeletePersonsInBulkResponseAllOf} obj Optional instance to populate.
     * @return {module:model/DeletePersonsInBulkResponseAllOf} The populated <code>DeletePersonsInBulkResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeletePersonsInBulkResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _DeletePersonsInBulkResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeletePersonsInBulkResponseAllOf;
}();
/**
 * @member {module:model/DeletePersonsInBulkResponseAllOfData} data
 */


DeletePersonsInBulkResponseAllOf.prototype['data'] = undefined;
var _default = DeletePersonsInBulkResponseAllOf;
exports["default"] = _default;
