// pipedrive/dist/model/DeletePersonResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _DeletePersonResponseAllOf = _interopRequireDefault(require("./DeletePersonResponseAllOf"));

var _DeletePersonResponseAllOfData = _interopRequireDefault(require("./DeletePersonResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeletePersonResponse model module.
 * @module model/DeletePersonResponse
 * @version 1.0.0
 */
var DeletePersonResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeletePersonResponse</code>.
   * @alias module:model/DeletePersonResponse
   * @implements module:model/BaseResponse
   * @implements module:model/DeletePersonResponseAllOf
   */
  function DeletePersonResponse() {
    (0, _classCallCheck2["default"])(this, DeletePersonResponse);

    _BaseResponse["default"].initialize(this);

    _DeletePersonResponseAllOf["default"].initialize(this);

    DeletePersonResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeletePersonResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeletePersonResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeletePersonResponse} obj Optional instance to populate.
     * @return {module:model/DeletePersonResponse} The populated <code>DeletePersonResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeletePersonResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _DeletePersonResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _DeletePersonResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeletePersonResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


DeletePersonResponse.prototype['success'] = undefined;
/**
 * @member {module:model/DeletePersonResponseAllOfData} data
 */

DeletePersonResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement DeletePersonResponseAllOf interface:

/**
 * @member {module:model/DeletePersonResponseAllOfData} data
 */

_DeletePersonResponseAllOf["default"].prototype['data'] = undefined;
var _default = DeletePersonResponse;
exports["default"] = _default;
