// pipedrive/dist/model/DeleteStageResponse200Data.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeleteStageResponse200Data model module.
 * @module model/DeleteStageResponse200Data
 * @version 1.0.0
 */
var DeleteStageResponse200Data = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteStageResponse200Data</code>.
   * @alias module:model/DeleteStageResponse200Data
   */
  function DeleteStageResponse200Data() {
    (0, _classCallCheck2["default"])(this, DeleteStageResponse200Data);
    DeleteStageResponse200Data.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeleteStageResponse200Data, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeleteStageResponse200Data</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteStageResponse200Data} obj Optional instance to populate.
     * @return {module:model/DeleteStageResponse200Data} The populated <code>DeleteStageResponse200Data</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteStageResponse200Data();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeleteStageResponse200Data;
}();
/**
 * Deleted stage ID
 * @member {Number} id
 */


DeleteStageResponse200Data.prototype['id'] = undefined;
var _default = DeleteStageResponse200Data;
exports["default"] = _default;
