// pipedrive/dist/model/DeletePersonsInBulkResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeletePersonsInBulkResponseAllOfData model module.
 * @module model/DeletePersonsInBulkResponseAllOfData
 * @version 1.0.0
 */
var DeletePersonsInBulkResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeletePersonsInBulkResponseAllOfData</code>.
   * @alias module:model/DeletePersonsInBulkResponseAllOfData
   */
  function DeletePersonsInBulkResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, DeletePersonsInBulkResponseAllOfData);
    DeletePersonsInBulkResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeletePersonsInBulkResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeletePersonsInBulkResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeletePersonsInBulkResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/DeletePersonsInBulkResponseAllOfData} The populated <code>DeletePersonsInBulkResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeletePersonsInBulkResponseAllOfData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], ['Number']);
          delete data['id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeletePersonsInBulkResponseAllOfData;
}();
/**
 * The list of deleted persons IDs
 * @member {Array.<Number>} id
 */


DeletePersonsInBulkResponseAllOfData.prototype['id'] = undefined;
var _default = DeletePersonsInBulkResponseAllOfData;
exports["default"] = _default;
