// pipedrive/dist/model/DeleteNote.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeleteNote model module.
 * @module model/DeleteNote
 * @version 1.0.0
 */
var DeleteNote = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteNote</code>.
   * @alias module:model/DeleteNote
   */
  function DeleteNote() {
    (0, _classCallCheck2["default"])(this, DeleteNote);
    DeleteNote.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeleteNote, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeleteNote</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteNote} obj Optional instance to populate.
     * @return {module:model/DeleteNote} The populated <code>DeleteNote</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteNote();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'Boolean');
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeleteNote;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


DeleteNote.prototype['success'] = undefined;
/**
 * If the response is successful or not
 * @member {Boolean} data
 */

DeleteNote.prototype['data'] = undefined;
var _default = DeleteNote;
exports["default"] = _default;
