// pipedrive/dist/model/DeleteRoleAssignmentAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DeleteRoleAssignmentAllOfData = _interopRequireDefault(require("./DeleteRoleAssignmentAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The DeleteRoleAssignmentAllOf model module.
 * @module model/DeleteRoleAssignmentAllOf
 * @version 1.0.0
 */
var DeleteRoleAssignmentAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteRoleAssignmentAllOf</code>.
   * @alias module:model/DeleteRoleAssignmentAllOf
   */
  function DeleteRoleAssignmentAllOf() {
    (0, _classCallCheck2["default"])(this, DeleteRoleAssignmentAllOf);
    DeleteRoleAssignmentAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(DeleteRoleAssignmentAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DeleteRoleAssignmentAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteRoleAssignmentAllOf} obj Optional instance to populate.
     * @return {module:model/DeleteRoleAssignmentAllOf} The populated <code>DeleteRoleAssignmentAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteRoleAssignmentAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _DeleteRoleAssignmentAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return DeleteRoleAssignmentAllOf;
}();
/**
 * @member {module:model/DeleteRoleAssignmentAllOfData} data
 */


DeleteRoleAssignmentAllOf.prototype['data'] = undefined;
var _default = DeleteRoleAssignmentAllOf;
exports["default"] = _default;
