// pipedrive/dist/model/Duration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The Duration model module.
 * @module model/Duration
 * @version 1.0.0
 */
var Duration = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Duration</code>.
   * The duration of the goal
   * @alias module:model/Duration
   */
  function Duration() {
    (0, _classCallCheck2["default"])(this, Duration);
    Duration.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(Duration, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Duration</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Duration} obj Optional instance to populate.
     * @return {module:model/Duration} The populated <code>Duration</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Duration();

        if (data.hasOwnProperty('start')) {
          obj['start'] = _ApiClient["default"].convertToType(data['start'], 'String');
          delete data['start'];
        }

        if (data.hasOwnProperty('end')) {
          obj['end'] = _ApiClient["default"].convertToType(data['end'], 'String');
          delete data['end'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return Duration;
}();
/**
 * The start date of the goal
 * @member {String} start
 */


Duration.prototype['start'] = undefined;
/**
 * The end date of the goal
 * @member {String} end
 */

Duration.prototype['end'] = undefined;
var _default = Duration;
exports["default"] = _default;
