// pipedrive/dist/model/EditPipeline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasePipelineWithSelectedFlag = _interopRequireDefault(require("./BasePipelineWithSelectedFlag"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _EditPipelineAllOf = _interopRequireDefault(require("./EditPipelineAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The EditPipeline model module.
 * @module model/EditPipeline
 * @version 1.0.0
 */
var EditPipeline = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EditPipeline</code>.
   * @alias module:model/EditPipeline
   * @implements module:model/BaseResponse
   * @implements module:model/EditPipelineAllOf
   */
  function EditPipeline() {
    (0, _classCallCheck2["default"])(this, EditPipeline);

    _BaseResponse["default"].initialize(this);

    _EditPipelineAllOf["default"].initialize(this);

    EditPipeline.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(EditPipeline, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>EditPipeline</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EditPipeline} obj Optional instance to populate.
     * @return {module:model/EditPipeline} The populated <code>EditPipeline</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EditPipeline();

        _BaseResponse["default"].constructFromObject(data, obj);

        _EditPipelineAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], _BasePipelineWithSelectedFlag["default"]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return EditPipeline;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


EditPipeline.prototype['success'] = undefined;
/**
 * The pipeline object
 * @member {module:model/BasePipelineWithSelectedFlag} data
 */

EditPipeline.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement EditPipelineAllOf interface:

/**
 * The pipeline object
 * @member {module:model/BasePipelineWithSelectedFlag} data
 */

_EditPipelineAllOf["default"].prototype['data'] = undefined;
var _default = EditPipeline;
exports["default"] = _default;
