// pipedrive/dist/model/EmailInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The EmailInfo model module.
 * @module model/EmailInfo
 * @version 1.0.0
 */
var EmailInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EmailInfo</code>.
   * @alias module:model/EmailInfo
   */
  function EmailInfo() {
    (0, _classCallCheck2["default"])(this, EmailInfo);
    EmailInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(EmailInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>EmailInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EmailInfo} obj Optional instance to populate.
     * @return {module:model/EmailInfo} The populated <code>EmailInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailInfo();

        if (data.hasOwnProperty('last_incoming_mail_time')) {
          obj['last_incoming_mail_time'] = _ApiClient["default"].convertToType(data['last_incoming_mail_time'], 'String');
          delete data['last_incoming_mail_time'];
        }

        if (data.hasOwnProperty('last_outgoing_mail_time')) {
          obj['last_outgoing_mail_time'] = _ApiClient["default"].convertToType(data['last_outgoing_mail_time'], 'String');
          delete data['last_outgoing_mail_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return EmailInfo;
}();
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */


EmailInfo.prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

EmailInfo.prototype['last_outgoing_mail_time'] = undefined;
var _default = EmailInfo;
exports["default"] = _default;
