// pipedrive/dist/model/ExpectedOutcome.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ExpectedOutcome model module.
 * @module model/ExpectedOutcome
 * @version 1.0.0
 */
var ExpectedOutcome = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ExpectedOutcome</code>.
   * The expected outcome of the goal
   * @alias module:model/ExpectedOutcome
   */
  function ExpectedOutcome() {
    (0, _classCallCheck2["default"])(this, ExpectedOutcome);
    ExpectedOutcome.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ExpectedOutcome, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ExpectedOutcome</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ExpectedOutcome} obj Optional instance to populate.
     * @return {module:model/ExpectedOutcome} The populated <code>ExpectedOutcome</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ExpectedOutcome();

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'Number');
          delete data['target'];
        }

        if (data.hasOwnProperty('tracking_metric')) {
          obj['tracking_metric'] = _ApiClient["default"].convertToType(data['tracking_metric'], 'String');
          delete data['tracking_metric'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ExpectedOutcome;
}();
/**
 * The numeric target of the goal
 * @member {Number} target
 */


ExpectedOutcome.prototype['target'] = undefined;
/**
 * The tracking metric of the goal
 * @member {String} tracking_metric
 */

ExpectedOutcome.prototype['tracking_metric'] = undefined;
var _default = ExpectedOutcome;
exports["default"] = _default;
