// pipedrive/dist/model/FieldResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Field = _interopRequireDefault(require("./Field"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FieldResponseAllOf model module.
 * @module model/FieldResponseAllOf
 * @version 1.0.0
 */
var FieldResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FieldResponseAllOf</code>.
   * @alias module:model/FieldResponseAllOf
   */
  function FieldResponseAllOf() {
    (0, _classCallCheck2["default"])(this, FieldResponseAllOf);
    FieldResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FieldResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FieldResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FieldResponseAllOf} obj Optional instance to populate.
     * @return {module:model/FieldResponseAllOf} The populated <code>FieldResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FieldResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _Field["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FieldResponseAllOf;
}();
/**
 * @member {module:model/Field} data
 */


FieldResponseAllOf.prototype['data'] = undefined;
var _default = FieldResponseAllOf;
exports["default"] = _default;
