// pipedrive/dist/model/FieldsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _Field = _interopRequireDefault(require("./Field"));

var _FieldsResponseAllOf = _interopRequireDefault(require("./FieldsResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FieldsResponse model module.
 * @module model/FieldsResponse
 * @version 1.0.0
 */
var FieldsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FieldsResponse</code>.
   * @alias module:model/FieldsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/FieldsResponseAllOf
   */
  function FieldsResponse() {
    (0, _classCallCheck2["default"])(this, FieldsResponse);

    _BaseResponse["default"].initialize(this);

    _FieldsResponseAllOf["default"].initialize(this);

    FieldsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FieldsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FieldsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FieldsResponse} obj Optional instance to populate.
     * @return {module:model/FieldsResponse} The populated <code>FieldsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FieldsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _FieldsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_Field["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FieldsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


FieldsResponse.prototype['success'] = undefined;
/**
 * @member {Array.<module:model/Field>} data
 */

FieldsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

FieldsResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement FieldsResponseAllOf interface:

/**
 * @member {Array.<module:model/Field>} data
 */

_FieldsResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_FieldsResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = FieldsResponse;
exports["default"] = _default;
