// pipedrive/dist/model/FieldType.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FieldTypeAsString = _interopRequireDefault(require("./FieldTypeAsString"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FieldType model module.
 * @module model/FieldType
 * @version 1.0.0
 */
var FieldType = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FieldType</code>.
   * @alias module:model/FieldType
   * @param fieldType {module:model/FieldTypeAsString} 
   */
  function FieldType(fieldType) {
    (0, _classCallCheck2["default"])(this, FieldType);
    FieldType.initialize(this, fieldType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FieldType, null, [{
    key: "initialize",
    value: function initialize(obj, fieldType) {
      obj['field_type'] = fieldType;
    }
    /**
     * Constructs a <code>FieldType</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FieldType} obj Optional instance to populate.
     * @return {module:model/FieldType} The populated <code>FieldType</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FieldType();

        if (data.hasOwnProperty('field_type')) {
          obj['field_type'] = _FieldTypeAsString["default"].constructFromObject(data['field_type']);
          delete data['field_type'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FieldType;
}();
/**
 * @member {module:model/FieldTypeAsString} field_type
 */


FieldType.prototype['field_type'] = undefined;
var _default = FieldType;
exports["default"] = _default;
