// pipedrive/dist/model/FiltersGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _FilterGetItem = _interopRequireDefault(require("./FilterGetItem"));

var _FiltersGetResponseAllOf = _interopRequireDefault(require("./FiltersGetResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FiltersGetResponse model module.
 * @module model/FiltersGetResponse
 * @version 1.0.0
 */
var FiltersGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FiltersGetResponse</code>.
   * @alias module:model/FiltersGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/FiltersGetResponseAllOf
   */
  function FiltersGetResponse() {
    (0, _classCallCheck2["default"])(this, FiltersGetResponse);

    _BaseResponse["default"].initialize(this);

    _FiltersGetResponseAllOf["default"].initialize(this);

    FiltersGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FiltersGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FiltersGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FiltersGetResponse} obj Optional instance to populate.
     * @return {module:model/FiltersGetResponse} The populated <code>FiltersGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FiltersGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _FiltersGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _FilterGetItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FiltersGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


FiltersGetResponse.prototype['success'] = undefined;
/**
 * @member {module:model/FilterGetItem} data
 */

FiltersGetResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement FiltersGetResponseAllOf interface:

/**
 * @member {module:model/FilterGetItem} data
 */

_FiltersGetResponseAllOf["default"].prototype['data'] = undefined;
var _default = FiltersGetResponse;
exports["default"] = _default;
