// pipedrive/dist/model/FiltersBulkGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _FilterGetItem = _interopRequireDefault(require("./FilterGetItem"));

var _FiltersBulkGetResponseAllOf = _interopRequireDefault(require("./FiltersBulkGetResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FiltersBulkGetResponse model module.
 * @module model/FiltersBulkGetResponse
 * @version 1.0.0
 */
var FiltersBulkGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FiltersBulkGetResponse</code>.
   * @alias module:model/FiltersBulkGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/FiltersBulkGetResponseAllOf
   */
  function FiltersBulkGetResponse() {
    (0, _classCallCheck2["default"])(this, FiltersBulkGetResponse);

    _BaseResponse["default"].initialize(this);

    _FiltersBulkGetResponseAllOf["default"].initialize(this);

    FiltersBulkGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FiltersBulkGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FiltersBulkGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FiltersBulkGetResponse} obj Optional instance to populate.
     * @return {module:model/FiltersBulkGetResponse} The populated <code>FiltersBulkGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FiltersBulkGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _FiltersBulkGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_FilterGetItem["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FiltersBulkGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


FiltersBulkGetResponse.prototype['success'] = undefined;
/**
 * The array of filters
 * @member {Array.<module:model/FilterGetItem>} data
 */

FiltersBulkGetResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement FiltersBulkGetResponseAllOf interface:

/**
 * The array of filters
 * @member {Array.<module:model/FilterGetItem>} data
 */

_FiltersBulkGetResponseAllOf["default"].prototype['data'] = undefined;
var _default = FiltersBulkGetResponse;
exports["default"] = _default;
