// pipedrive/dist/model/FiltersBulkGetResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FilterGetItem = _interopRequireDefault(require("./FilterGetItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FiltersBulkGetResponseAllOf model module.
 * @module model/FiltersBulkGetResponseAllOf
 * @version 1.0.0
 */
var FiltersBulkGetResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FiltersBulkGetResponseAllOf</code>.
   * @alias module:model/FiltersBulkGetResponseAllOf
   */
  function FiltersBulkGetResponseAllOf() {
    (0, _classCallCheck2["default"])(this, FiltersBulkGetResponseAllOf);
    FiltersBulkGetResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FiltersBulkGetResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FiltersBulkGetResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FiltersBulkGetResponseAllOf} obj Optional instance to populate.
     * @return {module:model/FiltersBulkGetResponseAllOf} The populated <code>FiltersBulkGetResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FiltersBulkGetResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_FilterGetItem["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FiltersBulkGetResponseAllOf;
}();
/**
 * The array of filters
 * @member {Array.<module:model/FilterGetItem>} data
 */


FiltersBulkGetResponseAllOf.prototype['data'] = undefined;
var _default = FiltersBulkGetResponseAllOf;
exports["default"] = _default;
