// pipedrive/dist/model/FiltersDeleteResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FiltersDeleteResponseAllOfData model module.
 * @module model/FiltersDeleteResponseAllOfData
 * @version 1.0.0
 */
var FiltersDeleteResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FiltersDeleteResponseAllOfData</code>.
   * @alias module:model/FiltersDeleteResponseAllOfData
   */
  function FiltersDeleteResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, FiltersDeleteResponseAllOfData);
    FiltersDeleteResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FiltersDeleteResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FiltersDeleteResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FiltersDeleteResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/FiltersDeleteResponseAllOfData} The populated <code>FiltersDeleteResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FiltersDeleteResponseAllOfData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FiltersDeleteResponseAllOfData;
}();
/**
 * The ID of the deleted filter
 * @member {Number} id
 */


FiltersDeleteResponseAllOfData.prototype['id'] = undefined;
var _default = FiltersDeleteResponseAllOfData;
exports["default"] = _default;
