// pipedrive/dist/model/FindProductsByNameResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FindProductsByNameResponseData = _interopRequireDefault(require("./FindProductsByNameResponseData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FindProductsByNameResponse model module.
 * @module model/FindProductsByNameResponse
 * @version 1.0.0
 */
var FindProductsByNameResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FindProductsByNameResponse</code>.
   * @alias module:model/FindProductsByNameResponse
   */
  function FindProductsByNameResponse() {
    (0, _classCallCheck2["default"])(this, FindProductsByNameResponse);
    FindProductsByNameResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FindProductsByNameResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FindProductsByNameResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FindProductsByNameResponse} obj Optional instance to populate.
     * @return {module:model/FindProductsByNameResponse} The populated <code>FindProductsByNameResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FindProductsByNameResponse();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _FindProductsByNameResponseData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ApiClient["default"].convertToType(data['additional_data'], Object);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FindProductsByNameResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


FindProductsByNameResponse.prototype['success'] = undefined;
/**
 * @member {module:model/FindProductsByNameResponseData} data
 */

FindProductsByNameResponse.prototype['data'] = undefined;
/**
 * Any additional data related to the Products
 * @member {Object} additional_data
 */

FindProductsByNameResponse.prototype['additional_data'] = undefined;
var _default = FindProductsByNameResponse;
exports["default"] = _default;
