// pipedrive/dist/model/FindProductsByNameResponseData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FindProductsByNameResponseData model module.
 * @module model/FindProductsByNameResponseData
 * @version 1.0.0
 */
var FindProductsByNameResponseData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FindProductsByNameResponseData</code>.
   * @alias module:model/FindProductsByNameResponseData
   */
  function FindProductsByNameResponseData() {
    (0, _classCallCheck2["default"])(this, FindProductsByNameResponseData);
    FindProductsByNameResponseData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FindProductsByNameResponseData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FindProductsByNameResponseData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FindProductsByNameResponseData} obj Optional instance to populate.
     * @return {module:model/FindProductsByNameResponseData} The populated <code>FindProductsByNameResponseData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FindProductsByNameResponseData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('variation_id')) {
          obj['variation_id'] = _ApiClient["default"].convertToType(data['variation_id'], 'String');
          delete data['variation_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }

        if (data.hasOwnProperty('variation_name')) {
          obj['variation_name'] = _ApiClient["default"].convertToType(data['variation_name'], 'String');
          delete data['variation_name'];
        }

        if (data.hasOwnProperty('price')) {
          obj['price'] = _ApiClient["default"].convertToType(data['price'], 'Number');
          delete data['price'];
        }

        if (data.hasOwnProperty('price_formatted')) {
          obj['price_formatted'] = _ApiClient["default"].convertToType(data['price_formatted'], 'String');
          delete data['price_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FindProductsByNameResponseData;
}();
/**
 * The ID of the Product
 * @member {Number} id
 */


FindProductsByNameResponseData.prototype['id'] = undefined;
/**
 * The ID of the price variation of the Product
 * @member {String} variation_id
 */

FindProductsByNameResponseData.prototype['variation_id'] = undefined;
/**
 * The name of the Product
 * @member {String} name
 */

FindProductsByNameResponseData.prototype['name'] = undefined;
/**
 * The title of the Product
 * @member {String} title
 */

FindProductsByNameResponseData.prototype['title'] = undefined;
/**
 * The name of the price variation of the Product
 * @member {String} variation_name
 */

FindProductsByNameResponseData.prototype['variation_name'] = undefined;
/**
 * The value corresponding to the Product price
 * @member {Number} price
 */

FindProductsByNameResponseData.prototype['price'] = undefined;
/**
 * The Product price with the currency format
 * @member {String} price_formatted
 */

FindProductsByNameResponseData.prototype['price_formatted'] = undefined;
var _default = FindProductsByNameResponseData;
exports["default"] = _default;
