// pipedrive/dist/model/FollowerDataWithID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FollowerData = _interopRequireDefault(require("./FollowerData"));

var _FollowerDataWithIDAllOf = _interopRequireDefault(require("./FollowerDataWithIDAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FollowerDataWithID model module.
 * @module model/FollowerDataWithID
 * @version 1.0.0
 */
var FollowerDataWithID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FollowerDataWithID</code>.
   * @alias module:model/FollowerDataWithID
   * @implements module:model/FollowerDataWithIDAllOf
   * @implements module:model/FollowerData
   */
  function FollowerDataWithID() {
    (0, _classCallCheck2["default"])(this, FollowerDataWithID);

    _FollowerDataWithIDAllOf["default"].initialize(this);

    _FollowerData["default"].initialize(this);

    FollowerDataWithID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FollowerDataWithID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FollowerDataWithID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FollowerDataWithID} obj Optional instance to populate.
     * @return {module:model/FollowerDataWithID} The populated <code>FollowerDataWithID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FollowerDataWithID();

        _FollowerDataWithIDAllOf["default"].constructFromObject(data, obj);

        _FollowerData["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
          delete data['email'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('pic_hash')) {
          obj['pic_hash'] = _ApiClient["default"].convertToType(data['pic_hash'], 'String');
          delete data['pic_hash'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FollowerDataWithID;
}();
/**
 * The ID of the follower associated with the item
 * @member {Number} id
 */


FollowerDataWithID.prototype['id'] = undefined;
/**
 * The name of the follower
 * @member {String} name
 */

FollowerDataWithID.prototype['name'] = undefined;
/**
 * The email of the follower
 * @member {String} email
 */

FollowerDataWithID.prototype['email'] = undefined;
/**
 * The user ID of the follower
 * @member {Number} user_id
 */

FollowerDataWithID.prototype['user_id'] = undefined;
/**
 * The follower picture hash
 * @member {String} pic_hash
 */

FollowerDataWithID.prototype['pic_hash'] = undefined; // Implement FollowerDataWithIDAllOf interface:

/**
 * The ID of the follower associated with the item
 * @member {Number} id
 */

_FollowerDataWithIDAllOf["default"].prototype['id'] = undefined; // Implement FollowerData interface:

/**
 * The name of the follower
 * @member {String} name
 */

_FollowerData["default"].prototype['name'] = undefined;
/**
 * The email of the follower
 * @member {String} email
 */

_FollowerData["default"].prototype['email'] = undefined;
/**
 * The user ID of the follower
 * @member {Number} user_id
 */

_FollowerData["default"].prototype['user_id'] = undefined;
/**
 * The follower picture hash
 * @member {String} pic_hash
 */

_FollowerData["default"].prototype['pic_hash'] = undefined;
var _default = FollowerDataWithID;
exports["default"] = _default;
