// pipedrive/dist/model/FollowerData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FollowerData model module.
 * @module model/FollowerData
 * @version 1.0.0
 */
var FollowerData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FollowerData</code>.
   * @alias module:model/FollowerData
   */
  function FollowerData() {
    (0, _classCallCheck2["default"])(this, FollowerData);
    FollowerData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FollowerData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FollowerData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FollowerData} obj Optional instance to populate.
     * @return {module:model/FollowerData} The populated <code>FollowerData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FollowerData();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
          delete data['email'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('pic_hash')) {
          obj['pic_hash'] = _ApiClient["default"].convertToType(data['pic_hash'], 'String');
          delete data['pic_hash'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FollowerData;
}();
/**
 * The name of the follower
 * @member {String} name
 */


FollowerData.prototype['name'] = undefined;
/**
 * The email of the follower
 * @member {String} email
 */

FollowerData.prototype['email'] = undefined;
/**
 * The user ID of the follower
 * @member {Number} user_id
 */

FollowerData.prototype['user_id'] = undefined;
/**
 * The follower picture hash
 * @member {String} pic_hash
 */

FollowerData.prototype['pic_hash'] = undefined;
var _default = FollowerData;
exports["default"] = _default;
