// pipedrive/dist/model/FullRole.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FullRoleAllOf = _interopRequireDefault(require("./FullRoleAllOf"));

var _SubRole = _interopRequireDefault(require("./SubRole"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FullRole model module.
 * @module model/FullRole
 * @version 1.0.0
 */
var FullRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FullRole</code>.
   * @alias module:model/FullRole
   * @implements module:model/SubRole
   * @implements module:model/FullRoleAllOf
   */
  function FullRole() {
    (0, _classCallCheck2["default"])(this, FullRole);

    _SubRole["default"].initialize(this);

    _FullRoleAllOf["default"].initialize(this);

    FullRole.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FullRole, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FullRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FullRole} obj Optional instance to populate.
     * @return {module:model/FullRole} The populated <code>FullRole</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FullRole();

        _SubRole["default"].constructFromObject(data, obj);

        _FullRoleAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('parent_role_id')) {
          obj['parent_role_id'] = _ApiClient["default"].convertToType(data['parent_role_id'], 'Number');
          delete data['parent_role_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('assignment_count')) {
          obj['assignment_count'] = _ApiClient["default"].convertToType(data['assignment_count'], 'String');
          delete data['assignment_count'];
        }

        if (data.hasOwnProperty('sub_role_count')) {
          obj['sub_role_count'] = _ApiClient["default"].convertToType(data['sub_role_count'], 'String');
          delete data['sub_role_count'];
        }

        if (data.hasOwnProperty('level')) {
          obj['level'] = _ApiClient["default"].convertToType(data['level'], 'Number');
          delete data['level'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FullRole;
}();
/**
 * The ID of the parent role
 * @member {Number} parent_role_id
 */


FullRole.prototype['parent_role_id'] = undefined;
/**
 * The name of the role
 * @member {String} name
 */

FullRole.prototype['name'] = undefined;
/**
 * The ID of the role
 * @member {Number} id
 */

FullRole.prototype['id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

FullRole.prototype['active_flag'] = undefined;
/**
 * The number of users assigned to this role
 * @member {String} assignment_count
 */

FullRole.prototype['assignment_count'] = undefined;
/**
 * The number of sub-roles
 * @member {String} sub_role_count
 */

FullRole.prototype['sub_role_count'] = undefined;
/**
 * The level of role in the role hierarchy
 * @member {Number} level
 */

FullRole.prototype['level'] = undefined; // Implement SubRole interface:

/**
 * The ID of the parent role
 * @member {Number} parent_role_id
 */

_SubRole["default"].prototype['parent_role_id'] = undefined;
/**
 * The name of the role
 * @member {String} name
 */

_SubRole["default"].prototype['name'] = undefined;
/**
 * The ID of the role
 * @member {Number} id
 */

_SubRole["default"].prototype['id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

_SubRole["default"].prototype['active_flag'] = undefined;
/**
 * The number of users assigned to this role
 * @member {String} assignment_count
 */

_SubRole["default"].prototype['assignment_count'] = undefined;
/**
 * The number of sub-roles
 * @member {String} sub_role_count
 */

_SubRole["default"].prototype['sub_role_count'] = undefined; // Implement FullRoleAllOf interface:

/**
 * The level of role in the role hierarchy
 * @member {Number} level
 */

_FullRoleAllOf["default"].prototype['level'] = undefined;
var _default = FullRole;
exports["default"] = _default;
