// pipedrive/dist/model/FullRoleAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The FullRoleAllOf model module.
 * @module model/FullRoleAllOf
 * @version 1.0.0
 */
var FullRoleAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FullRoleAllOf</code>.
   * @alias module:model/FullRoleAllOf
   */
  function FullRoleAllOf() {
    (0, _classCallCheck2["default"])(this, FullRoleAllOf);
    FullRoleAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(FullRoleAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>FullRoleAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FullRoleAllOf} obj Optional instance to populate.
     * @return {module:model/FullRoleAllOf} The populated <code>FullRoleAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FullRoleAllOf();

        if (data.hasOwnProperty('level')) {
          obj['level'] = _ApiClient["default"].convertToType(data['level'], 'Number');
          delete data['level'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return FullRoleAllOf;
}();
/**
 * The level of role in the role hierarchy
 * @member {Number} level
 */


FullRoleAllOf.prototype['level'] = undefined;
var _default = FullRoleAllOf;
exports["default"] = _default;
