// pipedrive/dist/model/GetAllPipelines.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasePipelineWithSelectedFlag = _interopRequireDefault(require("./BasePipelineWithSelectedFlag"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _GetAllPipelinesAllOf = _interopRequireDefault(require("./GetAllPipelinesAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetAllPipelines model module.
 * @module model/GetAllPipelines
 * @version 1.0.0
 */
var GetAllPipelines = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetAllPipelines</code>.
   * @alias module:model/GetAllPipelines
   * @implements module:model/BaseResponse
   * @implements module:model/GetAllPipelinesAllOf
   */
  function GetAllPipelines() {
    (0, _classCallCheck2["default"])(this, GetAllPipelines);

    _BaseResponse["default"].initialize(this);

    _GetAllPipelinesAllOf["default"].initialize(this);

    GetAllPipelines.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetAllPipelines, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetAllPipelines</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetAllPipelines} obj Optional instance to populate.
     * @return {module:model/GetAllPipelines} The populated <code>GetAllPipelines</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetAllPipelines();

        _BaseResponse["default"].constructFromObject(data, obj);

        _GetAllPipelinesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BasePipelineWithSelectedFlag["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetAllPipelines;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


GetAllPipelines.prototype['success'] = undefined;
/**
 * Pipelines array
 * @member {Array.<module:model/BasePipelineWithSelectedFlag>} data
 */

GetAllPipelines.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement GetAllPipelinesAllOf interface:

/**
 * Pipelines array
 * @member {Array.<module:model/BasePipelineWithSelectedFlag>} data
 */

_GetAllPipelinesAllOf["default"].prototype['data'] = undefined;
var _default = GetAllPipelines;
exports["default"] = _default;
