// pipedrive/dist/model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages model module.
 * @module model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages
 * @version 1.0.0
 */
var GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages</code>.
   * The moved deals average age by the stage
   * @alias module:model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages
   */
  function GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages() {
    (0, _classCallCheck2["default"])(this, GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages);
    GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages} obj Optional instance to populate.
     * @return {module:model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages} The populated <code>GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages();

        if (data.hasOwnProperty('stage_id')) {
          obj['stage_id'] = _ApiClient["default"].convertToType(data['stage_id'], 'Number');
          delete data['stage_id'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages;
}();
/**
 * The stage ID
 * @member {Number} stage_id
 */


GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages.prototype['stage_id'] = undefined;
/**
 * The average deals age in specific stage
 * @member {Number} value
 */

GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages.prototype['value'] = undefined;
var _default = GetDealsMovementsInPipelineAllOfDataAverageAgeInDaysByStages;
exports["default"] = _default;
