// pipedrive/dist/model/GetDealsSummaryData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GetDealsSummaryDataValuesTotal = _interopRequireDefault(require("./GetDealsSummaryDataValuesTotal"));

var _GetDealsSummaryDataWeightedValuesTotal = _interopRequireDefault(require("./GetDealsSummaryDataWeightedValuesTotal"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsSummaryData model module.
 * @module model/GetDealsSummaryData
 * @version 1.0.0
 */
var GetDealsSummaryData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsSummaryData</code>.
   * The summary of deals
   * @alias module:model/GetDealsSummaryData
   */
  function GetDealsSummaryData() {
    (0, _classCallCheck2["default"])(this, GetDealsSummaryData);
    GetDealsSummaryData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsSummaryData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsSummaryData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsSummaryData} obj Optional instance to populate.
     * @return {module:model/GetDealsSummaryData} The populated <code>GetDealsSummaryData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsSummaryData();

        if (data.hasOwnProperty('values_total')) {
          obj['values_total'] = _GetDealsSummaryDataValuesTotal["default"].constructFromObject(data['values_total']);
          delete data['values_total'];
        }

        if (data.hasOwnProperty('weighted_values_total')) {
          obj['weighted_values_total'] = _GetDealsSummaryDataWeightedValuesTotal["default"].constructFromObject(data['weighted_values_total']);
          delete data['weighted_values_total'];
        }

        if (data.hasOwnProperty('total_count')) {
          obj['total_count'] = _ApiClient["default"].convertToType(data['total_count'], 'Number');
          delete data['total_count'];
        }

        if (data.hasOwnProperty('total_currency_converted_value')) {
          obj['total_currency_converted_value'] = _ApiClient["default"].convertToType(data['total_currency_converted_value'], 'Number');
          delete data['total_currency_converted_value'];
        }

        if (data.hasOwnProperty('total_weighted_currency_converted_value')) {
          obj['total_weighted_currency_converted_value'] = _ApiClient["default"].convertToType(data['total_weighted_currency_converted_value'], 'Number');
          delete data['total_weighted_currency_converted_value'];
        }

        if (data.hasOwnProperty('total_currency_converted_value_formatted')) {
          obj['total_currency_converted_value_formatted'] = _ApiClient["default"].convertToType(data['total_currency_converted_value_formatted'], 'String');
          delete data['total_currency_converted_value_formatted'];
        }

        if (data.hasOwnProperty('total_weighted_currency_converted_value_formatted')) {
          obj['total_weighted_currency_converted_value_formatted'] = _ApiClient["default"].convertToType(data['total_weighted_currency_converted_value_formatted'], 'String');
          delete data['total_weighted_currency_converted_value_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsSummaryData;
}();
/**
 * @member {module:model/GetDealsSummaryDataValuesTotal} values_total
 */


GetDealsSummaryData.prototype['values_total'] = undefined;
/**
 * @member {module:model/GetDealsSummaryDataWeightedValuesTotal} weighted_values_total
 */

GetDealsSummaryData.prototype['weighted_values_total'] = undefined;
/**
 * The total number of deals
 * @member {Number} total_count
 */

GetDealsSummaryData.prototype['total_count'] = undefined;
/**
 * The total value of deals converted into the company default currency
 * @member {Number} total_currency_converted_value
 */

GetDealsSummaryData.prototype['total_currency_converted_value'] = undefined;
/**
 * The total weighted value of deals converted into the company default currency
 * @member {Number} total_weighted_currency_converted_value
 */

GetDealsSummaryData.prototype['total_weighted_currency_converted_value'] = undefined;
/**
 * The total converted value of deals formatted with the company default currency. E.g. US$5,100.96
 * @member {String} total_currency_converted_value_formatted
 */

GetDealsSummaryData.prototype['total_currency_converted_value_formatted'] = undefined;
/**
 * The total weighted value of deals formatted with the company default currency. E.g. US$5,100.96
 * @member {String} total_weighted_currency_converted_value_formatted
 */

GetDealsSummaryData.prototype['total_weighted_currency_converted_value_formatted'] = undefined;
var _default = GetDealsSummaryData;
exports["default"] = _default;
