// pipedrive/dist/model/GetDeal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealNonStrictWithDetails = _interopRequireDefault(require("./DealNonStrictWithDetails"));

var _GetDealAdditionalData = _interopRequireDefault(require("./GetDealAdditionalData"));

var _GetDealsRelatedObjects = _interopRequireDefault(require("./GetDealsRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDeal model module.
 * @module model/GetDeal
 * @version 1.0.0
 */
var GetDeal = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDeal</code>.
   * @alias module:model/GetDeal
   */
  function GetDeal() {
    (0, _classCallCheck2["default"])(this, GetDeal);
    GetDeal.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDeal, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDeal</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDeal} obj Optional instance to populate.
     * @return {module:model/GetDeal} The populated <code>GetDeal</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDeal();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _DealNonStrictWithDetails["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _GetDealAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetDealsRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDeal;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


GetDeal.prototype['success'] = undefined;
/**
 * @member {module:model/DealNonStrictWithDetails} data
 */

GetDeal.prototype['data'] = undefined;
/**
 * @member {module:model/GetDealAdditionalData} additional_data
 */

GetDeal.prototype['additional_data'] = undefined;
/**
 * @member {module:model/GetDealsRelatedObjects} related_objects
 */

GetDeal.prototype['related_objects'] = undefined;
var _default = GetDeal;
exports["default"] = _default;
