// pipedrive/dist/model/GetDealsMovementsInPipelineAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealsMovementsInfo = _interopRequireDefault(require("./DealsMovementsInfo"));

var _GetDealsMovementsInPipelineAllOfDataAverageAgeInDays = _interopRequireDefault(require("./GetDealsMovementsInPipelineAllOfDataAverageAgeInDays"));

var _GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages = _interopRequireDefault(require("./GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsMovementsInPipelineAllOfData model module.
 * @module model/GetDealsMovementsInPipelineAllOfData
 * @version 1.0.0
 */
var GetDealsMovementsInPipelineAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsMovementsInPipelineAllOfData</code>.
   * The pipeline object
   * @alias module:model/GetDealsMovementsInPipelineAllOfData
   */
  function GetDealsMovementsInPipelineAllOfData() {
    (0, _classCallCheck2["default"])(this, GetDealsMovementsInPipelineAllOfData);
    GetDealsMovementsInPipelineAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsMovementsInPipelineAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsMovementsInPipelineAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsMovementsInPipelineAllOfData} obj Optional instance to populate.
     * @return {module:model/GetDealsMovementsInPipelineAllOfData} The populated <code>GetDealsMovementsInPipelineAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsMovementsInPipelineAllOfData();

        if (data.hasOwnProperty('movements_between_stages')) {
          obj['movements_between_stages'] = _GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages["default"].constructFromObject(data['movements_between_stages']);
          delete data['movements_between_stages'];
        }

        if (data.hasOwnProperty('new_deals')) {
          obj['new_deals'] = _DealsMovementsInfo["default"].constructFromObject(data['new_deals']);
          delete data['new_deals'];
        }

        if (data.hasOwnProperty('deals_left_open')) {
          obj['deals_left_open'] = _DealsMovementsInfo["default"].constructFromObject(data['deals_left_open']);
          delete data['deals_left_open'];
        }

        if (data.hasOwnProperty('won_deals')) {
          obj['won_deals'] = _DealsMovementsInfo["default"].constructFromObject(data['won_deals']);
          delete data['won_deals'];
        }

        if (data.hasOwnProperty('lost_deals')) {
          obj['lost_deals'] = _DealsMovementsInfo["default"].constructFromObject(data['lost_deals']);
          delete data['lost_deals'];
        }

        if (data.hasOwnProperty('average_age_in_days')) {
          obj['average_age_in_days'] = _GetDealsMovementsInPipelineAllOfDataAverageAgeInDays["default"].constructFromObject(data['average_age_in_days']);
          delete data['average_age_in_days'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsMovementsInPipelineAllOfData;
}();
/**
 * @member {module:model/GetDealsMovementsInPipelineAllOfDataMovementsBetweenStages} movements_between_stages
 */


GetDealsMovementsInPipelineAllOfData.prototype['movements_between_stages'] = undefined;
/**
 * @member {module:model/DealsMovementsInfo} new_deals
 */

GetDealsMovementsInPipelineAllOfData.prototype['new_deals'] = undefined;
/**
 * @member {module:model/DealsMovementsInfo} deals_left_open
 */

GetDealsMovementsInPipelineAllOfData.prototype['deals_left_open'] = undefined;
/**
 * @member {module:model/DealsMovementsInfo} won_deals
 */

GetDealsMovementsInPipelineAllOfData.prototype['won_deals'] = undefined;
/**
 * @member {module:model/DealsMovementsInfo} lost_deals
 */

GetDealsMovementsInPipelineAllOfData.prototype['lost_deals'] = undefined;
/**
 * @member {module:model/GetDealsMovementsInPipelineAllOfDataAverageAgeInDays} average_age_in_days
 */

GetDealsMovementsInPipelineAllOfData.prototype['average_age_in_days'] = undefined;
var _default = GetDealsMovementsInPipelineAllOfData;
exports["default"] = _default;
