// pipedrive/dist/model/GetDealsSummaryDataWeightedValuesTotal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsSummaryDataWeightedValuesTotal model module.
 * @module model/GetDealsSummaryDataWeightedValuesTotal
 * @version 1.0.0
 */
var GetDealsSummaryDataWeightedValuesTotal = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsSummaryDataWeightedValuesTotal</code>.
   * The total weighted values of the deals grouped by deal currency. The weighted value is calculated as probability times deal value.
   * @alias module:model/GetDealsSummaryDataWeightedValuesTotal
   */
  function GetDealsSummaryDataWeightedValuesTotal() {
    (0, _classCallCheck2["default"])(this, GetDealsSummaryDataWeightedValuesTotal);
    GetDealsSummaryDataWeightedValuesTotal.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsSummaryDataWeightedValuesTotal, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsSummaryDataWeightedValuesTotal</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsSummaryDataWeightedValuesTotal} obj Optional instance to populate.
     * @return {module:model/GetDealsSummaryDataWeightedValuesTotal} The populated <code>GetDealsSummaryDataWeightedValuesTotal</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsSummaryDataWeightedValuesTotal();

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('value_formatted')) {
          obj['value_formatted'] = _ApiClient["default"].convertToType(data['value_formatted'], 'String');
          delete data['value_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsSummaryDataWeightedValuesTotal;
}();
/**
 * The total weighted value of the deals in the deal currency group
 * @member {Number} value
 */


GetDealsSummaryDataWeightedValuesTotal.prototype['value'] = undefined;
/**
 * The number of deals in the deal currency group
 * @member {Number} count
 */

GetDealsSummaryDataWeightedValuesTotal.prototype['count'] = undefined;
/**
 * The total weighted value of the deals formatted with deal currency. E.g. €50
 * @member {String} value_formatted
 */

GetDealsSummaryDataWeightedValuesTotal.prototype['value_formatted'] = undefined;
var _default = GetDealsSummaryDataWeightedValuesTotal;
exports["default"] = _default;
