// pipedrive/dist/model/GetDealsTimelineDataTotals.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsTimelineDataTotals model module.
 * @module model/GetDealsTimelineDataTotals
 * @version 1.0.0
 */
var GetDealsTimelineDataTotals = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsTimelineDataTotals</code>.
   * The total values of deals for the given period
   * @alias module:model/GetDealsTimelineDataTotals
   */
  function GetDealsTimelineDataTotals() {
    (0, _classCallCheck2["default"])(this, GetDealsTimelineDataTotals);
    GetDealsTimelineDataTotals.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsTimelineDataTotals, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsTimelineDataTotals</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsTimelineDataTotals} obj Optional instance to populate.
     * @return {module:model/GetDealsTimelineDataTotals} The populated <code>GetDealsTimelineDataTotals</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsTimelineDataTotals();

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('values')) {
          obj['values'] = _ApiClient["default"].convertToType(data['values'], Object);
          delete data['values'];
        }

        if (data.hasOwnProperty('weighted_values')) {
          obj['weighted_values'] = _ApiClient["default"].convertToType(data['weighted_values'], Object);
          delete data['weighted_values'];
        }

        if (data.hasOwnProperty('open_count')) {
          obj['open_count'] = _ApiClient["default"].convertToType(data['open_count'], 'Number');
          delete data['open_count'];
        }

        if (data.hasOwnProperty('open_values')) {
          obj['open_values'] = _ApiClient["default"].convertToType(data['open_values'], Object);
          delete data['open_values'];
        }

        if (data.hasOwnProperty('weighted_open_values')) {
          obj['weighted_open_values'] = _ApiClient["default"].convertToType(data['weighted_open_values'], Object);
          delete data['weighted_open_values'];
        }

        if (data.hasOwnProperty('won_count')) {
          obj['won_count'] = _ApiClient["default"].convertToType(data['won_count'], 'Number');
          delete data['won_count'];
        }

        if (data.hasOwnProperty('won_values')) {
          obj['won_values'] = _ApiClient["default"].convertToType(data['won_values'], Object);
          delete data['won_values'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsTimelineDataTotals;
}();
/**
 * The number of deals for the given period
 * @member {Number} count
 */


GetDealsTimelineDataTotals.prototype['count'] = undefined;
/**
 * The total values of deals grouped by deal currency
 * @member {Object} values
 */

GetDealsTimelineDataTotals.prototype['values'] = undefined;
/**
 * The total weighted values of deals for the given period grouped by deal currency. The weighted value of a deal is calculated as probability times deal value.
 * @member {Object} weighted_values
 */

GetDealsTimelineDataTotals.prototype['weighted_values'] = undefined;
/**
 * The number of open deals for the given period
 * @member {Number} open_count
 */

GetDealsTimelineDataTotals.prototype['open_count'] = undefined;
/**
 * The total values of open deals for the given period grouped by deal currency
 * @member {Object} open_values
 */

GetDealsTimelineDataTotals.prototype['open_values'] = undefined;
/**
 * The total weighted values of open deals for the given period grouped by deal currency. The weighted value of a deal is calculated as probability times deal value.
 * @member {Object} weighted_open_values
 */

GetDealsTimelineDataTotals.prototype['weighted_open_values'] = undefined;
/**
 * The number of won deals for the given period
 * @member {Number} won_count
 */

GetDealsTimelineDataTotals.prototype['won_count'] = undefined;
/**
 * The total values of won deals for the given period grouped by deal currency
 * @member {Object} won_values
 */

GetDealsTimelineDataTotals.prototype['won_values'] = undefined;
var _default = GetDealsTimelineDataTotals;
exports["default"] = _default;
