// pipedrive/dist/model/GetDealsTimelineData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealStrict = _interopRequireDefault(require("./DealStrict"));

var _GetDealsTimelineDataTotals = _interopRequireDefault(require("./GetDealsTimelineDataTotals"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsTimelineData model module.
 * @module model/GetDealsTimelineData
 * @version 1.0.0
 */
var GetDealsTimelineData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsTimelineData</code>.
   * Open and won deals grouped into periods by defined interval, amount and date-type dealField (&#x60;field_key&#x60;)
   * @alias module:model/GetDealsTimelineData
   */
  function GetDealsTimelineData() {
    (0, _classCallCheck2["default"])(this, GetDealsTimelineData);
    GetDealsTimelineData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsTimelineData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsTimelineData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsTimelineData} obj Optional instance to populate.
     * @return {module:model/GetDealsTimelineData} The populated <code>GetDealsTimelineData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsTimelineData();

        if (data.hasOwnProperty('period_start')) {
          obj['period_start'] = _ApiClient["default"].convertToType(data['period_start'], 'String');
          delete data['period_start'];
        }

        if (data.hasOwnProperty('period_end')) {
          obj['period_end'] = _ApiClient["default"].convertToType(data['period_end'], 'String');
          delete data['period_end'];
        }

        if (data.hasOwnProperty('deals')) {
          obj['deals'] = _ApiClient["default"].convertToType(data['deals'], [_DealStrict["default"]]);
          delete data['deals'];
        }

        if (data.hasOwnProperty('totals')) {
          obj['totals'] = _GetDealsTimelineDataTotals["default"].constructFromObject(data['totals']);
          delete data['totals'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsTimelineData;
}();
/**
 * The start date and time of the period
 * @member {String} period_start
 */


GetDealsTimelineData.prototype['period_start'] = undefined;
/**
 * The end date and time of the period
 * @member {String} period_end
 */

GetDealsTimelineData.prototype['period_end'] = undefined;
/**
 * @member {Array.<module:model/DealStrict>} deals
 */

GetDealsTimelineData.prototype['deals'] = undefined;
/**
 * @member {module:model/GetDealsTimelineDataTotals} totals
 */

GetDealsTimelineData.prototype['totals'] = undefined;
var _default = GetDealsTimelineData;
exports["default"] = _default;
