// pipedrive/dist/model/GetDealsConversionRatesInPipelineAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _StageConversions = _interopRequireDefault(require("./StageConversions"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsConversionRatesInPipelineAllOfData model module.
 * @module model/GetDealsConversionRatesInPipelineAllOfData
 * @version 1.0.0
 */
var GetDealsConversionRatesInPipelineAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsConversionRatesInPipelineAllOfData</code>.
   * The pipeline object
   * @alias module:model/GetDealsConversionRatesInPipelineAllOfData
   */
  function GetDealsConversionRatesInPipelineAllOfData() {
    (0, _classCallCheck2["default"])(this, GetDealsConversionRatesInPipelineAllOfData);
    GetDealsConversionRatesInPipelineAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsConversionRatesInPipelineAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsConversionRatesInPipelineAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsConversionRatesInPipelineAllOfData} obj Optional instance to populate.
     * @return {module:model/GetDealsConversionRatesInPipelineAllOfData} The populated <code>GetDealsConversionRatesInPipelineAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsConversionRatesInPipelineAllOfData();

        if (data.hasOwnProperty('stage_conversions')) {
          obj['stage_conversions'] = _ApiClient["default"].convertToType(data['stage_conversions'], [_StageConversions["default"]]);
          delete data['stage_conversions'];
        }

        if (data.hasOwnProperty('won_conversion')) {
          obj['won_conversion'] = _ApiClient["default"].convertToType(data['won_conversion'], 'Number');
          delete data['won_conversion'];
        }

        if (data.hasOwnProperty('lost_conversion')) {
          obj['lost_conversion'] = _ApiClient["default"].convertToType(data['lost_conversion'], 'Number');
          delete data['lost_conversion'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsConversionRatesInPipelineAllOfData;
}();
/**
 * The stage conversions
 * @member {Array.<module:model/StageConversions>} stage_conversions
 */


GetDealsConversionRatesInPipelineAllOfData.prototype['stage_conversions'] = undefined;
/**
 * The won conversion
 * @member {Number} won_conversion
 */

GetDealsConversionRatesInPipelineAllOfData.prototype['won_conversion'] = undefined;
/**
 * The lost conversion
 * @member {Number} lost_conversion
 */

GetDealsConversionRatesInPipelineAllOfData.prototype['lost_conversion'] = undefined;
var _default = GetDealsConversionRatesInPipelineAllOfData;
exports["default"] = _default;
