// pipedrive/dist/model/GetDealsSummaryDataValuesTotal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsSummaryDataValuesTotal model module.
 * @module model/GetDealsSummaryDataValuesTotal
 * @version 1.0.0
 */
var GetDealsSummaryDataValuesTotal = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsSummaryDataValuesTotal</code>.
   * The total values of the deals grouped by deal currency
   * @alias module:model/GetDealsSummaryDataValuesTotal
   */
  function GetDealsSummaryDataValuesTotal() {
    (0, _classCallCheck2["default"])(this, GetDealsSummaryDataValuesTotal);
    GetDealsSummaryDataValuesTotal.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsSummaryDataValuesTotal, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsSummaryDataValuesTotal</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsSummaryDataValuesTotal} obj Optional instance to populate.
     * @return {module:model/GetDealsSummaryDataValuesTotal} The populated <code>GetDealsSummaryDataValuesTotal</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsSummaryDataValuesTotal();

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('count')) {
          obj['count'] = _ApiClient["default"].convertToType(data['count'], 'Number');
          delete data['count'];
        }

        if (data.hasOwnProperty('value_converted')) {
          obj['value_converted'] = _ApiClient["default"].convertToType(data['value_converted'], 'Number');
          delete data['value_converted'];
        }

        if (data.hasOwnProperty('value_formatted')) {
          obj['value_formatted'] = _ApiClient["default"].convertToType(data['value_formatted'], 'String');
          delete data['value_formatted'];
        }

        if (data.hasOwnProperty('value_converted_formatted')) {
          obj['value_converted_formatted'] = _ApiClient["default"].convertToType(data['value_converted_formatted'], 'String');
          delete data['value_converted_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsSummaryDataValuesTotal;
}();
/**
 * The total value of deals in the deal currency group
 * @member {Number} value
 */


GetDealsSummaryDataValuesTotal.prototype['value'] = undefined;
/**
 * The number of deals in the deal currency group
 * @member {Number} count
 */

GetDealsSummaryDataValuesTotal.prototype['count'] = undefined;
/**
 * The total value of deals converted into the company default currency
 * @member {Number} value_converted
 */

GetDealsSummaryDataValuesTotal.prototype['value_converted'] = undefined;
/**
 * The total value of deals formatted with deal currency. E.g. €50
 * @member {String} value_formatted
 */

GetDealsSummaryDataValuesTotal.prototype['value_formatted'] = undefined;
/**
 * The value_converted formatted with deal currency. E.g. US$50.10
 * @member {String} value_converted_formatted
 */

GetDealsSummaryDataValuesTotal.prototype['value_converted_formatted'] = undefined;
var _default = GetDealsSummaryDataValuesTotal;
exports["default"] = _default;
