// pipedrive/dist/model/GetDealsMovementsInPipelineAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GetDealsMovementsInPipelineAllOfData = _interopRequireDefault(require("./GetDealsMovementsInPipelineAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsMovementsInPipelineAllOf model module.
 * @module model/GetDealsMovementsInPipelineAllOf
 * @version 1.0.0
 */
var GetDealsMovementsInPipelineAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsMovementsInPipelineAllOf</code>.
   * @alias module:model/GetDealsMovementsInPipelineAllOf
   */
  function GetDealsMovementsInPipelineAllOf() {
    (0, _classCallCheck2["default"])(this, GetDealsMovementsInPipelineAllOf);
    GetDealsMovementsInPipelineAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsMovementsInPipelineAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsMovementsInPipelineAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsMovementsInPipelineAllOf} obj Optional instance to populate.
     * @return {module:model/GetDealsMovementsInPipelineAllOf} The populated <code>GetDealsMovementsInPipelineAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsMovementsInPipelineAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _GetDealsMovementsInPipelineAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsMovementsInPipelineAllOf;
}();
/**
 * @member {module:model/GetDealsMovementsInPipelineAllOfData} data
 */


GetDealsMovementsInPipelineAllOf.prototype['data'] = undefined;
var _default = GetDealsMovementsInPipelineAllOf;
exports["default"] = _default;
