// pipedrive/dist/model/GetDealsRelatedObjects.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealOrganizationData = _interopRequireDefault(require("./DealOrganizationData"));

var _DealPersonData = _interopRequireDefault(require("./DealPersonData"));

var _DealUserData = _interopRequireDefault(require("./DealUserData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetDealsRelatedObjects model module.
 * @module model/GetDealsRelatedObjects
 * @version 1.0.0
 */
var GetDealsRelatedObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDealsRelatedObjects</code>.
   * @alias module:model/GetDealsRelatedObjects
   */
  function GetDealsRelatedObjects() {
    (0, _classCallCheck2["default"])(this, GetDealsRelatedObjects);
    GetDealsRelatedObjects.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetDealsRelatedObjects, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetDealsRelatedObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDealsRelatedObjects} obj Optional instance to populate.
     * @return {module:model/GetDealsRelatedObjects} The populated <code>GetDealsRelatedObjects</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDealsRelatedObjects();

        if (data.hasOwnProperty('user')) {
          obj['user'] = _DealUserData["default"].constructFromObject(data['user']);
          delete data['user'];
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _DealOrganizationData["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('person')) {
          obj['person'] = _DealPersonData["default"].constructFromObject(data['person']);
          delete data['person'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetDealsRelatedObjects;
}();
/**
 * @member {module:model/DealUserData} user
 */


GetDealsRelatedObjects.prototype['user'] = undefined;
/**
 * @member {module:model/DealOrganizationData} organization
 */

GetDealsRelatedObjects.prototype['organization'] = undefined;
/**
 * @member {module:model/DealPersonData} person
 */

GetDealsRelatedObjects.prototype['person'] = undefined;
var _default = GetDealsRelatedObjects;
exports["default"] = _default;
