// pipedrive/dist/model/GetLeadsResponse200.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _LeadResponse = _interopRequireDefault(require("./LeadResponse"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetLeadsResponse200 model module.
 * @module model/GetLeadsResponse200
 * @version 1.0.0
 */
var GetLeadsResponse200 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetLeadsResponse200</code>.
   * @alias module:model/GetLeadsResponse200
   */
  function GetLeadsResponse200() {
    (0, _classCallCheck2["default"])(this, GetLeadsResponse200);
    GetLeadsResponse200.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetLeadsResponse200, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetLeadsResponse200</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetLeadsResponse200} obj Optional instance to populate.
     * @return {module:model/GetLeadsResponse200} The populated <code>GetLeadsResponse200</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetLeadsResponse200();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_LeadResponse["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetLeadsResponse200;
}();
/**
 * @member {Boolean} success
 */


GetLeadsResponse200.prototype['success'] = undefined;
/**
 * @member {Array.<module:model/LeadResponse>} data
 */

GetLeadsResponse200.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

GetLeadsResponse200.prototype['additional_data'] = undefined;
var _default = GetLeadsResponse200;
exports["default"] = _default;
