// pipedrive/dist/model/GetNotes.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalDataWithPagination = _interopRequireDefault(require("./AdditionalDataWithPagination"));

var _BaseNote = _interopRequireDefault(require("./BaseNote"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetNotes model module.
 * @module model/GetNotes
 * @version 1.0.0
 */
var GetNotes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetNotes</code>.
   * @alias module:model/GetNotes
   */
  function GetNotes() {
    (0, _classCallCheck2["default"])(this, GetNotes);
    GetNotes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetNotes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetNotes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetNotes} obj Optional instance to populate.
     * @return {module:model/GetNotes} The populated <code>GetNotes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetNotes();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseNote["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalDataWithPagination["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetNotes;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


GetNotes.prototype['success'] = undefined;
/**
 * The array of notes
 * @member {Array.<module:model/BaseNote>} data
 */

GetNotes.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalDataWithPagination} additional_data
 */

GetNotes.prototype['additional_data'] = undefined;
var _default = GetNotes;
exports["default"] = _default;
