// pipedrive/dist/model/GetPersonDetailsResponseAllOfAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetPersonDetailsResponseAllOfAdditionalData model module.
 * @module model/GetPersonDetailsResponseAllOfAdditionalData
 * @version 1.0.0
 */
var GetPersonDetailsResponseAllOfAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetPersonDetailsResponseAllOfAdditionalData</code>.
   * @alias module:model/GetPersonDetailsResponseAllOfAdditionalData
   */
  function GetPersonDetailsResponseAllOfAdditionalData() {
    (0, _classCallCheck2["default"])(this, GetPersonDetailsResponseAllOfAdditionalData);
    GetPersonDetailsResponseAllOfAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetPersonDetailsResponseAllOfAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetPersonDetailsResponseAllOfAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetPersonDetailsResponseAllOfAdditionalData} obj Optional instance to populate.
     * @return {module:model/GetPersonDetailsResponseAllOfAdditionalData} The populated <code>GetPersonDetailsResponseAllOfAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetPersonDetailsResponseAllOfAdditionalData();

        if (data.hasOwnProperty('dropbox_email')) {
          obj['dropbox_email'] = _ApiClient["default"].convertToType(data['dropbox_email'], 'String');
          delete data['dropbox_email'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetPersonDetailsResponseAllOfAdditionalData;
}();
/**
 * Dropbox email for the person
 * @member {String} dropbox_email
 */


GetPersonDetailsResponseAllOfAdditionalData.prototype['dropbox_email'] = undefined;
var _default = GetPersonDetailsResponseAllOfAdditionalData;
exports["default"] = _default;
