// pipedrive/dist/model/GetRoleAssignmentsAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RoleAssignment = _interopRequireDefault(require("./RoleAssignment"));

var _RolesAdditionalData = _interopRequireDefault(require("./RolesAdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRoleAssignmentsAllOf model module.
 * @module model/GetRoleAssignmentsAllOf
 * @version 1.0.0
 */
var GetRoleAssignmentsAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRoleAssignmentsAllOf</code>.
   * @alias module:model/GetRoleAssignmentsAllOf
   */
  function GetRoleAssignmentsAllOf() {
    (0, _classCallCheck2["default"])(this, GetRoleAssignmentsAllOf);
    GetRoleAssignmentsAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRoleAssignmentsAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRoleAssignmentsAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRoleAssignmentsAllOf} obj Optional instance to populate.
     * @return {module:model/GetRoleAssignmentsAllOf} The populated <code>GetRoleAssignmentsAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRoleAssignmentsAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_RoleAssignment["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _RolesAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRoleAssignmentsAllOf;
}();
/**
 * The role assignments
 * @member {Array.<module:model/RoleAssignment>} data
 */


GetRoleAssignmentsAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/RolesAdditionalData} additional_data
 */

GetRoleAssignmentsAllOf.prototype['additional_data'] = undefined;
var _default = GetRoleAssignmentsAllOf;
exports["default"] = _default;
