// pipedrive/dist/model/GetRoleAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GetRoleAllOfAdditionalData = _interopRequireDefault(require("./GetRoleAllOfAdditionalData"));

var _SubRole = _interopRequireDefault(require("./SubRole"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRoleAllOf model module.
 * @module model/GetRoleAllOf
 * @version 1.0.0
 */
var GetRoleAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRoleAllOf</code>.
   * @alias module:model/GetRoleAllOf
   */
  function GetRoleAllOf() {
    (0, _classCallCheck2["default"])(this, GetRoleAllOf);
    GetRoleAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRoleAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRoleAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRoleAllOf} obj Optional instance to populate.
     * @return {module:model/GetRoleAllOf} The populated <code>GetRoleAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRoleAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _SubRole["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _GetRoleAllOfAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRoleAllOf;
}();
/**
 * @member {module:model/SubRole} data
 */


GetRoleAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/GetRoleAllOfAdditionalData} additional_data
 */

GetRoleAllOf.prototype['additional_data'] = undefined;
var _default = GetRoleAllOf;
exports["default"] = _default;
