// pipedrive/dist/model/GetRoleAssignments.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _GetRoleAssignmentsAllOf = _interopRequireDefault(require("./GetRoleAssignmentsAllOf"));

var _RoleAssignment = _interopRequireDefault(require("./RoleAssignment"));

var _RolesAdditionalData = _interopRequireDefault(require("./RolesAdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRoleAssignments model module.
 * @module model/GetRoleAssignments
 * @version 1.0.0
 */
var GetRoleAssignments = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRoleAssignments</code>.
   * @alias module:model/GetRoleAssignments
   * @implements module:model/BaseResponse
   * @implements module:model/GetRoleAssignmentsAllOf
   */
  function GetRoleAssignments() {
    (0, _classCallCheck2["default"])(this, GetRoleAssignments);

    _BaseResponse["default"].initialize(this);

    _GetRoleAssignmentsAllOf["default"].initialize(this);

    GetRoleAssignments.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRoleAssignments, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRoleAssignments</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRoleAssignments} obj Optional instance to populate.
     * @return {module:model/GetRoleAssignments} The populated <code>GetRoleAssignments</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRoleAssignments();

        _BaseResponse["default"].constructFromObject(data, obj);

        _GetRoleAssignmentsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_RoleAssignment["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _RolesAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRoleAssignments;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


GetRoleAssignments.prototype['success'] = undefined;
/**
 * The role assignments
 * @member {Array.<module:model/RoleAssignment>} data
 */

GetRoleAssignments.prototype['data'] = undefined;
/**
 * @member {module:model/RolesAdditionalData} additional_data
 */

GetRoleAssignments.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement GetRoleAssignmentsAllOf interface:

/**
 * The role assignments
 * @member {Array.<module:model/RoleAssignment>} data
 */

_GetRoleAssignmentsAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/RolesAdditionalData} additional_data
 */

_GetRoleAssignmentsAllOf["default"].prototype['additional_data'] = undefined;
var _default = GetRoleAssignments;
exports["default"] = _default;
