// pipedrive/dist/model/GetRecentsAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRecentsAdditionalData model module.
 * @module model/GetRecentsAdditionalData
 * @version 1.0.0
 */
var GetRecentsAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRecentsAdditionalData</code>.
   * @alias module:model/GetRecentsAdditionalData
   */
  function GetRecentsAdditionalData() {
    (0, _classCallCheck2["default"])(this, GetRecentsAdditionalData);
    GetRecentsAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRecentsAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRecentsAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRecentsAdditionalData} obj Optional instance to populate.
     * @return {module:model/GetRecentsAdditionalData} The populated <code>GetRecentsAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRecentsAdditionalData();

        if (data.hasOwnProperty('since_timestamp')) {
          obj['since_timestamp'] = _ApiClient["default"].convertToType(data['since_timestamp'], 'String');
          delete data['since_timestamp'];
        }

        if (data.hasOwnProperty('last_timestamp_on_page')) {
          obj['last_timestamp_on_page'] = _ApiClient["default"].convertToType(data['last_timestamp_on_page'], 'String');
          delete data['last_timestamp_on_page'];
        }

        if (data.hasOwnProperty('pagination')) {
          obj['pagination'] = _AdditionalData["default"].constructFromObject(data['pagination']);
          delete data['pagination'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRecentsAdditionalData;
}();
/**
 * The timestamp in UTC. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} since_timestamp
 */


GetRecentsAdditionalData.prototype['since_timestamp'] = undefined;
/**
 * The timestamp in UTC. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} last_timestamp_on_page
 */

GetRecentsAdditionalData.prototype['last_timestamp_on_page'] = undefined;
/**
 * @member {module:model/AdditionalData} pagination
 */

GetRecentsAdditionalData.prototype['pagination'] = undefined;
var _default = GetRecentsAdditionalData;
exports["default"] = _default;
