// pipedrive/dist/model/GetRoleSettings.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _GetRoleSettingsAllOf = _interopRequireDefault(require("./GetRoleSettingsAllOf"));

var _RoleSettings = _interopRequireDefault(require("./RoleSettings"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRoleSettings model module.
 * @module model/GetRoleSettings
 * @version 1.0.0
 */
var GetRoleSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRoleSettings</code>.
   * @alias module:model/GetRoleSettings
   * @implements module:model/BaseResponse
   * @implements module:model/GetRoleSettingsAllOf
   */
  function GetRoleSettings() {
    (0, _classCallCheck2["default"])(this, GetRoleSettings);

    _BaseResponse["default"].initialize(this);

    _GetRoleSettingsAllOf["default"].initialize(this);

    GetRoleSettings.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRoleSettings, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRoleSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRoleSettings} obj Optional instance to populate.
     * @return {module:model/GetRoleSettings} The populated <code>GetRoleSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRoleSettings();

        _BaseResponse["default"].constructFromObject(data, obj);

        _GetRoleSettingsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _RoleSettings["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRoleSettings;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


GetRoleSettings.prototype['success'] = undefined;
/**
 * @member {module:model/RoleSettings} data
 */

GetRoleSettings.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement GetRoleSettingsAllOf interface:

/**
 * @member {module:model/RoleSettings} data
 */

_GetRoleSettingsAllOf["default"].prototype['data'] = undefined;
var _default = GetRoleSettings;
exports["default"] = _default;
