// pipedrive/dist/model/GetRoles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _FullRole = _interopRequireDefault(require("./FullRole"));

var _GetRolesAllOf = _interopRequireDefault(require("./GetRolesAllOf"));

var _RolesAdditionalData = _interopRequireDefault(require("./RolesAdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetRoles model module.
 * @module model/GetRoles
 * @version 1.0.0
 */
var GetRoles = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRoles</code>.
   * @alias module:model/GetRoles
   * @implements module:model/BaseResponse
   * @implements module:model/GetRolesAllOf
   */
  function GetRoles() {
    (0, _classCallCheck2["default"])(this, GetRoles);

    _BaseResponse["default"].initialize(this);

    _GetRolesAllOf["default"].initialize(this);

    GetRoles.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetRoles, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRoles</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRoles} obj Optional instance to populate.
     * @return {module:model/GetRoles} The populated <code>GetRoles</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRoles();

        _BaseResponse["default"].constructFromObject(data, obj);

        _GetRolesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_FullRole["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _RolesAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetRoles;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


GetRoles.prototype['success'] = undefined;
/**
 * The array of roles
 * @member {Array.<module:model/FullRole>} data
 */

GetRoles.prototype['data'] = undefined;
/**
 * @member {module:model/RolesAdditionalData} additional_data
 */

GetRoles.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement GetRolesAllOf interface:

/**
 * The array of roles
 * @member {Array.<module:model/FullRole>} data
 */

_GetRolesAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/RolesAdditionalData} additional_data
 */

_GetRolesAllOf["default"].prototype['additional_data'] = undefined;
var _default = GetRoles;
exports["default"] = _default;
