// pipedrive/dist/model/GetStages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _StageWithPipelineInfo = _interopRequireDefault(require("./StageWithPipelineInfo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GetStages model module.
 * @module model/GetStages
 * @version 1.0.0
 */
var GetStages = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetStages</code>.
   * @alias module:model/GetStages
   */
  function GetStages() {
    (0, _classCallCheck2["default"])(this, GetStages);
    GetStages.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GetStages, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetStages</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetStages} obj Optional instance to populate.
     * @return {module:model/GetStages} The populated <code>GetStages</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetStages();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_StageWithPipelineInfo["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GetStages;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


GetStages.prototype['success'] = undefined;
/**
 * The array of stages
 * @member {Array.<module:model/StageWithPipelineInfo>} data
 */

GetStages.prototype['data'] = undefined;
var _default = GetStages;
exports["default"] = _default;
