// pipedrive/dist/model/GlobalMessageUserData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GlobalMessageUserData model module.
 * @module model/GlobalMessageUserData
 * @version 1.0.0
 */
var GlobalMessageUserData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GlobalMessageUserData</code>.
   * The global message type info
   * @alias module:model/GlobalMessageUserData
   */
  function GlobalMessageUserData() {
    (0, _classCallCheck2["default"])(this, GlobalMessageUserData);
    GlobalMessageUserData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GlobalMessageUserData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GlobalMessageUserData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GlobalMessageUserData} obj Optional instance to populate.
     * @return {module:model/GlobalMessageUserData} The populated <code>GlobalMessageUserData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GlobalMessageUserData();

        if (data.hasOwnProperty('message_type')) {
          obj['message_type'] = _ApiClient["default"].convertToType(data['message_type'], 'String');
          delete data['message_type'];
        }

        if (data.hasOwnProperty('dismissible_flag')) {
          obj['dismissible_flag'] = _ApiClient["default"].convertToType(data['dismissible_flag'], 'Boolean');
          delete data['dismissible_flag'];
        }

        if (data.hasOwnProperty('priority')) {
          obj['priority'] = _ApiClient["default"].convertToType(data['priority'], 'Number');
          delete data['priority'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GlobalMessageUserData;
}();
/**
 * The global message type for validation checks
 * @member {String} message_type
 */


GlobalMessageUserData.prototype['message_type'] = undefined;
/**
 * Whether a user can dismiss the global message or not
 * @member {Boolean} dismissible_flag
 */

GlobalMessageUserData.prototype['dismissible_flag'] = undefined;
/**
 * Sorts messages by priority DESC (from 0 to 10)
 * @member {Number} priority
 */

GlobalMessageUserData.prototype['priority'] = undefined;
var _default = GlobalMessageUserData;
exports["default"] = _default;
