// pipedrive/dist/model/GlobalMessageDeleteAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GlobalMessageData = _interopRequireDefault(require("./GlobalMessageData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GlobalMessageDeleteAllOf model module.
 * @module model/GlobalMessageDeleteAllOf
 * @version 1.0.0
 */
var GlobalMessageDeleteAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GlobalMessageDeleteAllOf</code>.
   * @alias module:model/GlobalMessageDeleteAllOf
   */
  function GlobalMessageDeleteAllOf() {
    (0, _classCallCheck2["default"])(this, GlobalMessageDeleteAllOf);
    GlobalMessageDeleteAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GlobalMessageDeleteAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GlobalMessageDeleteAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GlobalMessageDeleteAllOf} obj Optional instance to populate.
     * @return {module:model/GlobalMessageDeleteAllOf} The populated <code>GlobalMessageDeleteAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GlobalMessageDeleteAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _GlobalMessageData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('handle_time')) {
          obj['handle_time'] = _ApiClient["default"].convertToType(data['handle_time'], 'Number');
          delete data['handle_time'];
        }

        if (data.hasOwnProperty('_req_time_ctx_ms')) {
          obj['_req_time_ctx_ms'] = _ApiClient["default"].convertToType(data['_req_time_ctx_ms'], Object);
          delete data['_req_time_ctx_ms'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GlobalMessageDeleteAllOf;
}();
/**
 * @member {module:model/GlobalMessageData} data
 */


GlobalMessageDeleteAllOf.prototype['data'] = undefined;
/**
 * The total time of the controller processing
 * @member {Number} handle_time
 */

GlobalMessageDeleteAllOf.prototype['handle_time'] = undefined;
/**
 * The list of timers formatted by name and elapsed time in ms (such as `app`, `app_xss_clean`, `db_connect`)
 * @member {Object} _req_time_ctx_ms
 */

GlobalMessageDeleteAllOf.prototype['_req_time_ctx_ms'] = undefined;
var _default = GlobalMessageDeleteAllOf;
exports["default"] = _default;
