// pipedrive/dist/model/GlobalMessageData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GlobalMessageUserData = _interopRequireDefault(require("./GlobalMessageUserData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GlobalMessageData model module.
 * @module model/GlobalMessageData
 * @version 1.0.0
 */
var GlobalMessageData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GlobalMessageData</code>.
   * The global message data
   * @alias module:model/GlobalMessageData
   */
  function GlobalMessageData() {
    (0, _classCallCheck2["default"])(this, GlobalMessageData);
    GlobalMessageData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GlobalMessageData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GlobalMessageData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GlobalMessageData} obj Optional instance to populate.
     * @return {module:model/GlobalMessageData} The populated <code>GlobalMessageData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GlobalMessageData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (data.hasOwnProperty('type_info')) {
          obj['type_info'] = _GlobalMessageUserData["default"].constructFromObject(data['type_info']);
          delete data['type_info'];
        }

        if (data.hasOwnProperty('user_data')) {
          obj['user_data'] = _ApiClient["default"].convertToType(data['user_data'], Object);
          delete data['user_data'];
        }

        if (data.hasOwnProperty('html')) {
          obj['html'] = _ApiClient["default"].convertToType(data['html'], 'String');
          delete data['html'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GlobalMessageData;
}();
/**
 * The ID of the global message
 * @member {Number} id
 */


GlobalMessageData.prototype['id'] = undefined;
/**
 * The ID of the user related to the global message
 * @member {Number} user_id
 */

GlobalMessageData.prototype['user_id'] = undefined;
/**
 * The ID of the company related to the global message
 * @member {Number} company_id
 */

GlobalMessageData.prototype['company_id'] = undefined;
/**
 * @member {module:model/GlobalMessageUserData} type_info
 */

GlobalMessageData.prototype['type_info'] = undefined;
/**
 * The user data for the global message (such as `user_id`, `name`, `email`, `phone`)
 * @member {Object} user_data
 */

GlobalMessageData.prototype['user_data'] = undefined;
/**
 * HTML for each returned global message to render views
 * @member {String} html
 */

GlobalMessageData.prototype['html'] = undefined;
var _default = GlobalMessageData;
exports["default"] = _default;
