// pipedrive/dist/model/GlobalMessageDelete.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GlobalMessageBaseResponse = _interopRequireDefault(require("./GlobalMessageBaseResponse"));

var _GlobalMessageData = _interopRequireDefault(require("./GlobalMessageData"));

var _GlobalMessageDeleteAllOf = _interopRequireDefault(require("./GlobalMessageDeleteAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GlobalMessageDelete model module.
 * @module model/GlobalMessageDelete
 * @version 1.0.0
 */
var GlobalMessageDelete = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GlobalMessageDelete</code>.
   * @alias module:model/GlobalMessageDelete
   * @implements module:model/GlobalMessageBaseResponse
   * @implements module:model/GlobalMessageDeleteAllOf
   */
  function GlobalMessageDelete() {
    (0, _classCallCheck2["default"])(this, GlobalMessageDelete);

    _GlobalMessageBaseResponse["default"].initialize(this);

    _GlobalMessageDeleteAllOf["default"].initialize(this);

    GlobalMessageDelete.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GlobalMessageDelete, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GlobalMessageDelete</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GlobalMessageDelete} obj Optional instance to populate.
     * @return {module:model/GlobalMessageDelete} The populated <code>GlobalMessageDelete</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GlobalMessageDelete();

        _GlobalMessageBaseResponse["default"].constructFromObject(data, obj);

        _GlobalMessageDeleteAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _GlobalMessageData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('handle_time')) {
          obj['handle_time'] = _ApiClient["default"].convertToType(data['handle_time'], 'Number');
          delete data['handle_time'];
        }

        if (data.hasOwnProperty('_req_time_ctx_ms')) {
          obj['_req_time_ctx_ms'] = _ApiClient["default"].convertToType(data['_req_time_ctx_ms'], Object);
          delete data['_req_time_ctx_ms'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GlobalMessageDelete;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


GlobalMessageDelete.prototype['success'] = undefined;
/**
 * @member {module:model/GlobalMessageData} data
 */

GlobalMessageDelete.prototype['data'] = undefined;
/**
 * The total time of the controller processing
 * @member {Number} handle_time
 */

GlobalMessageDelete.prototype['handle_time'] = undefined;
/**
 * The list of timers formatted by name and elapsed time in ms (such as `app`, `app_xss_clean`, `db_connect`)
 * @member {Object} _req_time_ctx_ms
 */

GlobalMessageDelete.prototype['_req_time_ctx_ms'] = undefined; // Implement GlobalMessageBaseResponse interface:

/**
 * If the request was successful or not
 * @member {Boolean} success
 */

_GlobalMessageBaseResponse["default"].prototype['success'] = undefined; // Implement GlobalMessageDeleteAllOf interface:

/**
 * @member {module:model/GlobalMessageData} data
 */

_GlobalMessageDeleteAllOf["default"].prototype['data'] = undefined;
/**
 * The total time of the controller processing
 * @member {Number} handle_time
 */

_GlobalMessageDeleteAllOf["default"].prototype['handle_time'] = undefined;
/**
 * The list of timers formatted by name and elapsed time in ms (such as `app`, `app_xss_clean`, `db_connect`)
 * @member {Object} _req_time_ctx_ms
 */

_GlobalMessageDeleteAllOf["default"].prototype['_req_time_ctx_ms'] = undefined;
var _default = GlobalMessageDelete;
exports["default"] = _default;
