// pipedrive/dist/model/GoalType.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Params = _interopRequireDefault(require("./Params"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The GoalType model module.
 * @module model/GoalType
 * @version 1.0.0
 */
var GoalType = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GoalType</code>.
   * The type of the goal
   * @alias module:model/GoalType
   */
  function GoalType() {
    (0, _classCallCheck2["default"])(this, GoalType);
    GoalType.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(GoalType, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GoalType</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GoalType} obj Optional instance to populate.
     * @return {module:model/GoalType} The populated <code>GoalType</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GoalType();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('params')) {
          obj['params'] = _Params["default"].constructFromObject(data['params']);
          delete data['params'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return GoalType;
}();
/**
 * The name of the goal type
 * @member {String} name
 */


GoalType.prototype['name'] = undefined;
/**
 * @member {module:model/Params} params
 */

GoalType.prototype['params'] = undefined;
var _default = GoalType;
exports["default"] = _default;
