// pipedrive/dist/model/ItemSearchResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _ItemSearchResponseAllOf = _interopRequireDefault(require("./ItemSearchResponseAllOf"));

var _ItemSearchResponseAllOfData = _interopRequireDefault(require("./ItemSearchResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ItemSearchResponse model module.
 * @module model/ItemSearchResponse
 * @version 1.0.0
 */
var ItemSearchResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSearchResponse</code>.
   * @alias module:model/ItemSearchResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ItemSearchResponseAllOf
   */
  function ItemSearchResponse() {
    (0, _classCallCheck2["default"])(this, ItemSearchResponse);

    _BaseResponse["default"].initialize(this);

    _ItemSearchResponseAllOf["default"].initialize(this);

    ItemSearchResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ItemSearchResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSearchResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSearchResponse} obj Optional instance to populate.
     * @return {module:model/ItemSearchResponse} The populated <code>ItemSearchResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSearchResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ItemSearchResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ItemSearchResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ItemSearchResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ItemSearchResponse.prototype['success'] = undefined;
/**
 * @member {module:model/ItemSearchResponseAllOfData} data
 */

ItemSearchResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

ItemSearchResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ItemSearchResponseAllOf interface:

/**
 * @member {module:model/ItemSearchResponseAllOfData} data
 */

_ItemSearchResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_ItemSearchResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ItemSearchResponse;
exports["default"] = _default;
