// pipedrive/dist/model/ItemSearchFieldResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _ItemSearchFieldResponseAllOfData = _interopRequireDefault(require("./ItemSearchFieldResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ItemSearchFieldResponseAllOf model module.
 * @module model/ItemSearchFieldResponseAllOf
 * @version 1.0.0
 */
var ItemSearchFieldResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSearchFieldResponseAllOf</code>.
   * @alias module:model/ItemSearchFieldResponseAllOf
   */
  function ItemSearchFieldResponseAllOf() {
    (0, _classCallCheck2["default"])(this, ItemSearchFieldResponseAllOf);
    ItemSearchFieldResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ItemSearchFieldResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSearchFieldResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSearchFieldResponseAllOf} obj Optional instance to populate.
     * @return {module:model/ItemSearchFieldResponseAllOf} The populated <code>ItemSearchFieldResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSearchFieldResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ItemSearchFieldResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ItemSearchFieldResponseAllOf;
}();
/**
 * The array of results
 * @member {Array.<module:model/ItemSearchFieldResponseAllOfData>} data
 */


ItemSearchFieldResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

ItemSearchFieldResponseAllOf.prototype['additional_data'] = undefined;
var _default = ItemSearchFieldResponseAllOf;
exports["default"] = _default;
