// pipedrive/dist/model/ItemSearchResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSearchItem = _interopRequireDefault(require("./ItemSearchItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ItemSearchResponseAllOfData model module.
 * @module model/ItemSearchResponseAllOfData
 * @version 1.0.0
 */
var ItemSearchResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSearchResponseAllOfData</code>.
   * @alias module:model/ItemSearchResponseAllOfData
   */
  function ItemSearchResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ItemSearchResponseAllOfData);
    ItemSearchResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ItemSearchResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSearchResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSearchResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ItemSearchResponseAllOfData} The populated <code>ItemSearchResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSearchResponseAllOfData();

        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_ItemSearchItem["default"]]);
          delete data['items'];
        }

        if (data.hasOwnProperty('related_items')) {
          obj['related_items'] = _ApiClient["default"].convertToType(data['related_items'], [_ItemSearchItem["default"]]);
          delete data['related_items'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ItemSearchResponseAllOfData;
}();
/**
 * The array of found items
 * @member {Array.<module:model/ItemSearchItem>} items
 */


ItemSearchResponseAllOfData.prototype['items'] = undefined;
/**
 * The array of related items if `search_for_related_items` was enabled
 * @member {Array.<module:model/ItemSearchItem>} related_items
 */

ItemSearchResponseAllOfData.prototype['related_items'] = undefined;
var _default = ItemSearchResponseAllOfData;
exports["default"] = _default;
