// pipedrive/dist/model/ItemSearchFieldResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ItemSearchFieldResponseAllOfData model module.
 * @module model/ItemSearchFieldResponseAllOfData
 * @version 1.0.0
 */
var ItemSearchFieldResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSearchFieldResponseAllOfData</code>.
   * @alias module:model/ItemSearchFieldResponseAllOfData
   */
  function ItemSearchFieldResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ItemSearchFieldResponseAllOfData);
    ItemSearchFieldResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ItemSearchFieldResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSearchFieldResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSearchFieldResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ItemSearchFieldResponseAllOfData} The populated <code>ItemSearchFieldResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSearchFieldResponseAllOfData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('$field_key')) {
          obj['$field_key'] = _ApiClient["default"].convertToType(data['$field_key'], Object);
          delete data['$field_key'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ItemSearchFieldResponseAllOfData;
}();
/**
 * The ID of the item
 * @member {Number} id
 */


ItemSearchFieldResponseAllOfData.prototype['id'] = undefined;
/**
 * The value of the searched `field_key`
 * @member {Object} $field_key
 */

ItemSearchFieldResponseAllOfData.prototype['$field_key'] = undefined;
var _default = ItemSearchFieldResponseAllOfData;
exports["default"] = _default;
