// pipedrive/dist/model/ListProductAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _ItemSearchAdditionalDataPagination = _interopRequireDefault(require("./ItemSearchAdditionalDataPagination"));

var _ListProductAdditionalDataAllOf = _interopRequireDefault(require("./ListProductAdditionalDataAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListProductAdditionalData model module.
 * @module model/ListProductAdditionalData
 * @version 1.0.0
 */
var ListProductAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListProductAdditionalData</code>.
   * @alias module:model/ListProductAdditionalData
   * @implements module:model/ListProductAdditionalDataAllOf
   * @implements module:model/ItemSearchAdditionalData
   */
  function ListProductAdditionalData() {
    (0, _classCallCheck2["default"])(this, ListProductAdditionalData);

    _ListProductAdditionalDataAllOf["default"].initialize(this);

    _ItemSearchAdditionalData["default"].initialize(this);

    ListProductAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListProductAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListProductAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListProductAdditionalData} obj Optional instance to populate.
     * @return {module:model/ListProductAdditionalData} The populated <code>ListProductAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListProductAdditionalData();

        _ListProductAdditionalDataAllOf["default"].constructFromObject(data, obj);

        _ItemSearchAdditionalData["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('products_quantity_total')) {
          obj['products_quantity_total'] = _ApiClient["default"].convertToType(data['products_quantity_total'], 'Number');
          delete data['products_quantity_total'];
        }

        if (data.hasOwnProperty('products_sum_total')) {
          obj['products_sum_total'] = _ApiClient["default"].convertToType(data['products_sum_total'], 'Number');
          delete data['products_sum_total'];
        }

        if (data.hasOwnProperty('variations_enabled')) {
          obj['variations_enabled'] = _ApiClient["default"].convertToType(data['variations_enabled'], 'Boolean');
          delete data['variations_enabled'];
        }

        if (data.hasOwnProperty('products_quantity_total_formatted')) {
          obj['products_quantity_total_formatted'] = _ApiClient["default"].convertToType(data['products_quantity_total_formatted'], 'String');
          delete data['products_quantity_total_formatted'];
        }

        if (data.hasOwnProperty('products_sum_total_formatted')) {
          obj['products_sum_total_formatted'] = _ApiClient["default"].convertToType(data['products_sum_total_formatted'], 'String');
          delete data['products_sum_total_formatted'];
        }

        if (data.hasOwnProperty('pagination')) {
          obj['pagination'] = _ItemSearchAdditionalDataPagination["default"].constructFromObject(data['pagination']);
          delete data['pagination'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListProductAdditionalData;
}();
/**
 * The total quantity of the products
 * @member {Number} products_quantity_total
 */


ListProductAdditionalData.prototype['products_quantity_total'] = undefined;
/**
 * The total sum of the products
 * @member {Number} products_sum_total
 */

ListProductAdditionalData.prototype['products_sum_total'] = undefined;
/**
 * Whether variations is enabled or not
 * @member {Boolean} variations_enabled
 */

ListProductAdditionalData.prototype['variations_enabled'] = undefined;
/**
 * The total formatted quantity of the products
 * @member {String} products_quantity_total_formatted
 */

ListProductAdditionalData.prototype['products_quantity_total_formatted'] = undefined;
/**
 * The total formatted sum of the products
 * @member {String} products_sum_total_formatted
 */

ListProductAdditionalData.prototype['products_sum_total_formatted'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalDataPagination} pagination
 */

ListProductAdditionalData.prototype['pagination'] = undefined; // Implement ListProductAdditionalDataAllOf interface:

/**
 * The total quantity of the products
 * @member {Number} products_quantity_total
 */

_ListProductAdditionalDataAllOf["default"].prototype['products_quantity_total'] = undefined;
/**
 * The total sum of the products
 * @member {Number} products_sum_total
 */

_ListProductAdditionalDataAllOf["default"].prototype['products_sum_total'] = undefined;
/**
 * Whether variations is enabled or not
 * @member {Boolean} variations_enabled
 */

_ListProductAdditionalDataAllOf["default"].prototype['variations_enabled'] = undefined;
/**
 * The total formatted quantity of the products
 * @member {String} products_quantity_total_formatted
 */

_ListProductAdditionalDataAllOf["default"].prototype['products_quantity_total_formatted'] = undefined;
/**
 * The total formatted sum of the products
 * @member {String} products_sum_total_formatted
 */

_ListProductAdditionalDataAllOf["default"].prototype['products_sum_total_formatted'] = undefined; // Implement ItemSearchAdditionalData interface:

/**
 * @member {module:model/ItemSearchAdditionalDataPagination} pagination
 */

_ItemSearchAdditionalData["default"].prototype['pagination'] = undefined;
var _default = ListProductAdditionalData;
exports["default"] = _default;
