// pipedrive/dist/model/ListActivitiesResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityDistributionDataWithAdditionalData = _interopRequireDefault(require("./ActivityDistributionDataWithAdditionalData"));

var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListActivitiesResponseAllOf = _interopRequireDefault(require("./ListActivitiesResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListActivitiesResponse model module.
 * @module model/ListActivitiesResponse
 * @version 1.0.0
 */
var ListActivitiesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListActivitiesResponse</code>.
   * @alias module:model/ListActivitiesResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListActivitiesResponseAllOf
   */
  function ListActivitiesResponse() {
    (0, _classCallCheck2["default"])(this, ListActivitiesResponse);

    _BaseResponse["default"].initialize(this);

    _ListActivitiesResponseAllOf["default"].initialize(this);

    ListActivitiesResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListActivitiesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListActivitiesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListActivitiesResponse} obj Optional instance to populate.
     * @return {module:model/ListActivitiesResponse} The populated <code>ListActivitiesResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListActivitiesResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListActivitiesResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ActivityResponseObject["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ActivityDistributionDataWithAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListActivitiesResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListActivitiesResponse.prototype['success'] = undefined;
/**
 * The array of activities
 * @member {Array.<module:model/ActivityResponseObject>} data
 */

ListActivitiesResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ActivityDistributionDataWithAdditionalData} additional_data
 */

ListActivitiesResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListActivitiesResponseAllOf interface:

/**
 * The array of activities
 * @member {Array.<module:model/ActivityResponseObject>} data
 */

_ListActivitiesResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ActivityDistributionDataWithAdditionalData} additional_data
 */

_ListActivitiesResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ListActivitiesResponse;
exports["default"] = _default;
