// pipedrive/dist/model/ListProductFollowersResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _ListProductFollowersResponseAllOfData = _interopRequireDefault(require("./ListProductFollowersResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListProductFollowersResponseAllOf model module.
 * @module model/ListProductFollowersResponseAllOf
 * @version 1.0.0
 */
var ListProductFollowersResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListProductFollowersResponseAllOf</code>.
   * @alias module:model/ListProductFollowersResponseAllOf
   */
  function ListProductFollowersResponseAllOf() {
    (0, _classCallCheck2["default"])(this, ListProductFollowersResponseAllOf);
    ListProductFollowersResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListProductFollowersResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListProductFollowersResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListProductFollowersResponseAllOf} obj Optional instance to populate.
     * @return {module:model/ListProductFollowersResponseAllOf} The populated <code>ListProductFollowersResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListProductFollowersResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ListProductFollowersResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListProductFollowersResponseAllOf;
}();
/**
 * The list of followers
 * @member {Array.<module:model/ListProductFollowersResponseAllOfData>} data
 */


ListProductFollowersResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ListProductFollowersResponseAllOf.prototype['additional_data'] = undefined;
var _default = ListProductFollowersResponseAllOf;
exports["default"] = _default;
