// pipedrive/dist/model/ListDealsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _DealNonStrict = _interopRequireDefault(require("./DealNonStrict"));

var _ListDealsResponseAllOf = _interopRequireDefault(require("./ListDealsResponseAllOf"));

var _ListDealsResponseAllOfRelatedObjects = _interopRequireDefault(require("./ListDealsResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListDealsResponse model module.
 * @module model/ListDealsResponse
 * @version 1.0.0
 */
var ListDealsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListDealsResponse</code>.
   * @alias module:model/ListDealsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListDealsResponseAllOf
   */
  function ListDealsResponse() {
    (0, _classCallCheck2["default"])(this, ListDealsResponse);

    _BaseResponse["default"].initialize(this);

    _ListDealsResponseAllOf["default"].initialize(this);

    ListDealsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListDealsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListDealsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListDealsResponse} obj Optional instance to populate.
     * @return {module:model/ListDealsResponse} The populated <code>ListDealsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListDealsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListDealsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_DealNonStrict["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _ListDealsResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListDealsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListDealsResponse.prototype['success'] = undefined;
/**
 * The array of deals
 * @member {Array.<module:model/DealNonStrict>} data
 */

ListDealsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ListDealsResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListDealsResponseAllOfRelatedObjects} related_objects
 */

ListDealsResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListDealsResponseAllOf interface:

/**
 * The array of deals
 * @member {Array.<module:model/DealNonStrict>} data
 */

_ListDealsResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_ListDealsResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListDealsResponseAllOfRelatedObjects} related_objects
 */

_ListDealsResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = ListDealsResponse;
exports["default"] = _default;
