// pipedrive/dist/model/ListPersonProductsResponseAllOfDEALID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonListProduct = _interopRequireDefault(require("./PersonListProduct"));

var _ProductBaseDeal = _interopRequireDefault(require("./ProductBaseDeal"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPersonProductsResponseAllOfDEALID model module.
 * @module model/ListPersonProductsResponseAllOfDEALID
 * @version 1.0.0
 */
var ListPersonProductsResponseAllOfDEALID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPersonProductsResponseAllOfDEALID</code>.
   * @alias module:model/ListPersonProductsResponseAllOfDEALID
   */
  function ListPersonProductsResponseAllOfDEALID() {
    (0, _classCallCheck2["default"])(this, ListPersonProductsResponseAllOfDEALID);
    ListPersonProductsResponseAllOfDEALID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPersonProductsResponseAllOfDEALID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPersonProductsResponseAllOfDEALID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPersonProductsResponseAllOfDEALID} obj Optional instance to populate.
     * @return {module:model/ListPersonProductsResponseAllOfDEALID} The populated <code>ListPersonProductsResponseAllOfDEALID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPersonProductsResponseAllOfDEALID();

        if (data.hasOwnProperty('deal')) {
          obj['deal'] = _ProductBaseDeal["default"].constructFromObject(data['deal']);
          delete data['deal'];
        }

        if (data.hasOwnProperty('product')) {
          obj['product'] = _PersonListProduct["default"].constructFromObject(data['product']);
          delete data['product'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListPersonProductsResponseAllOfDEALID;
}();
/**
 * @member {module:model/ProductBaseDeal} deal
 */


ListPersonProductsResponseAllOfDEALID.prototype['deal'] = undefined;
/**
 * @member {module:model/PersonListProduct} product
 */

ListPersonProductsResponseAllOfDEALID.prototype['product'] = undefined;
var _default = ListPersonProductsResponseAllOfDEALID;
exports["default"] = _default;
