// pipedrive/dist/model/ListProductFollowersResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListProductFollowersResponseAllOf = _interopRequireDefault(require("./ListProductFollowersResponseAllOf"));

var _ListProductFollowersResponseAllOfData = _interopRequireDefault(require("./ListProductFollowersResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListProductFollowersResponse model module.
 * @module model/ListProductFollowersResponse
 * @version 1.0.0
 */
var ListProductFollowersResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListProductFollowersResponse</code>.
   * @alias module:model/ListProductFollowersResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListProductFollowersResponseAllOf
   */
  function ListProductFollowersResponse() {
    (0, _classCallCheck2["default"])(this, ListProductFollowersResponse);

    _BaseResponse["default"].initialize(this);

    _ListProductFollowersResponseAllOf["default"].initialize(this);

    ListProductFollowersResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListProductFollowersResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListProductFollowersResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListProductFollowersResponse} obj Optional instance to populate.
     * @return {module:model/ListProductFollowersResponse} The populated <code>ListProductFollowersResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListProductFollowersResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListProductFollowersResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ListProductFollowersResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListProductFollowersResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListProductFollowersResponse.prototype['success'] = undefined;
/**
 * The list of followers
 * @member {Array.<module:model/ListProductFollowersResponseAllOfData>} data
 */

ListProductFollowersResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ListProductFollowersResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListProductFollowersResponseAllOf interface:

/**
 * The list of followers
 * @member {Array.<module:model/ListProductFollowersResponseAllOfData>} data
 */

_ListProductFollowersResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_ListProductFollowersResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ListProductFollowersResponse;
exports["default"] = _default;
