// pipedrive/dist/model/ListPersonsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListPersonsResponseAllOf = _interopRequireDefault(require("./ListPersonsResponseAllOf"));

var _ListPersonsResponseAllOfRelatedObjects = _interopRequireDefault(require("./ListPersonsResponseAllOfRelatedObjects"));

var _PersonItem = _interopRequireDefault(require("./PersonItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPersonsResponse model module.
 * @module model/ListPersonsResponse
 * @version 1.0.0
 */
var ListPersonsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPersonsResponse</code>.
   * @alias module:model/ListPersonsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListPersonsResponseAllOf
   */
  function ListPersonsResponse() {
    (0, _classCallCheck2["default"])(this, ListPersonsResponse);

    _BaseResponse["default"].initialize(this);

    _ListPersonsResponseAllOf["default"].initialize(this);

    ListPersonsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPersonsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPersonsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPersonsResponse} obj Optional instance to populate.
     * @return {module:model/ListPersonsResponse} The populated <code>ListPersonsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPersonsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListPersonsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_PersonItem["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _ListPersonsResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListPersonsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListPersonsResponse.prototype['success'] = undefined;
/**
 * The array of persons
 * @member {Array.<module:model/PersonItem>} data
 */

ListPersonsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ListPersonsResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListPersonsResponseAllOfRelatedObjects} related_objects
 */

ListPersonsResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListPersonsResponseAllOf interface:

/**
 * The array of persons
 * @member {Array.<module:model/PersonItem>} data
 */

_ListPersonsResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_ListPersonsResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListPersonsResponseAllOfRelatedObjects} related_objects
 */

_ListPersonsResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = ListPersonsResponse;
exports["default"] = _default;
