// pipedrive/dist/model/ListMailMessagesResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MailMessageItemForList = _interopRequireDefault(require("./MailMessageItemForList"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListMailMessagesResponseAllOfData model module.
 * @module model/ListMailMessagesResponseAllOfData
 * @version 1.0.0
 */
var ListMailMessagesResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListMailMessagesResponseAllOfData</code>.
   * @alias module:model/ListMailMessagesResponseAllOfData
   */
  function ListMailMessagesResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ListMailMessagesResponseAllOfData);
    ListMailMessagesResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListMailMessagesResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListMailMessagesResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListMailMessagesResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ListMailMessagesResponseAllOfData} The populated <code>ListMailMessagesResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListMailMessagesResponseAllOfData();

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
          delete data['object'];
        }

        if (data.hasOwnProperty('timestamp')) {
          obj['timestamp'] = _ApiClient["default"].convertToType(data['timestamp'], 'String');
          delete data['timestamp'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _MailMessageItemForList["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListMailMessagesResponseAllOfData;
}();
/**
 * The type of the data item
 * @member {String} object
 */


ListMailMessagesResponseAllOfData.prototype['object'] = undefined;
/**
 * The date and time when the item was created
 * @member {String} timestamp
 */

ListMailMessagesResponseAllOfData.prototype['timestamp'] = undefined;
/**
 * @member {module:model/MailMessageItemForList} data
 */

ListMailMessagesResponseAllOfData.prototype['data'] = undefined;
var _default = ListMailMessagesResponseAllOfData;
exports["default"] = _default;
