// pipedrive/dist/model/ListFilesResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _FileItem = _interopRequireDefault(require("./FileItem"));

var _ListFilesResponseAllOf = _interopRequireDefault(require("./ListFilesResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListFilesResponse model module.
 * @module model/ListFilesResponse
 * @version 1.0.0
 */
var ListFilesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListFilesResponse</code>.
   * @alias module:model/ListFilesResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListFilesResponseAllOf
   */
  function ListFilesResponse() {
    (0, _classCallCheck2["default"])(this, ListFilesResponse);

    _BaseResponse["default"].initialize(this);

    _ListFilesResponseAllOf["default"].initialize(this);

    ListFilesResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListFilesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListFilesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListFilesResponse} obj Optional instance to populate.
     * @return {module:model/ListFilesResponse} The populated <code>ListFilesResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListFilesResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListFilesResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_FileItem["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListFilesResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListFilesResponse.prototype['success'] = undefined;
/**
 * The array of files
 * @member {Array.<module:model/FileItem>} data
 */

ListFilesResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ListFilesResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListFilesResponseAllOf interface:

/**
 * The array of files
 * @member {Array.<module:model/FileItem>} data
 */

_ListFilesResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

_ListFilesResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ListFilesResponse;
exports["default"] = _default;
