// pipedrive/dist/model/ListPersonProductsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _ListPersonProductsResponseAllOf = _interopRequireDefault(require("./ListPersonProductsResponseAllOf"));

var _ListPersonProductsResponseAllOfData = _interopRequireDefault(require("./ListPersonProductsResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPersonProductsResponse model module.
 * @module model/ListPersonProductsResponse
 * @version 1.0.0
 */
var ListPersonProductsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPersonProductsResponse</code>.
   * @alias module:model/ListPersonProductsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListPersonProductsResponseAllOf
   */
  function ListPersonProductsResponse() {
    (0, _classCallCheck2["default"])(this, ListPersonProductsResponse);

    _BaseResponse["default"].initialize(this);

    _ListPersonProductsResponseAllOf["default"].initialize(this);

    ListPersonProductsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPersonProductsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPersonProductsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPersonProductsResponse} obj Optional instance to populate.
     * @return {module:model/ListPersonProductsResponse} The populated <code>ListPersonProductsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPersonProductsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListPersonProductsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ListPersonProductsResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListPersonProductsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListPersonProductsResponse.prototype['success'] = undefined;
/**
 * The array of deal products
 * @member {Array.<module:model/ListPersonProductsResponseAllOfData>} data
 */

ListPersonProductsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

ListPersonProductsResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListPersonProductsResponseAllOf interface:

/**
 * The array of deal products
 * @member {Array.<module:model/ListPersonProductsResponseAllOfData>} data
 */

_ListPersonProductsResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_ListPersonProductsResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = ListPersonProductsResponse;
exports["default"] = _default;
