// pipedrive/dist/model/ListProductAdditionalDataAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListProductAdditionalDataAllOf model module.
 * @module model/ListProductAdditionalDataAllOf
 * @version 1.0.0
 */
var ListProductAdditionalDataAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListProductAdditionalDataAllOf</code>.
   * @alias module:model/ListProductAdditionalDataAllOf
   */
  function ListProductAdditionalDataAllOf() {
    (0, _classCallCheck2["default"])(this, ListProductAdditionalDataAllOf);
    ListProductAdditionalDataAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListProductAdditionalDataAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListProductAdditionalDataAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListProductAdditionalDataAllOf} obj Optional instance to populate.
     * @return {module:model/ListProductAdditionalDataAllOf} The populated <code>ListProductAdditionalDataAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListProductAdditionalDataAllOf();

        if (data.hasOwnProperty('products_quantity_total')) {
          obj['products_quantity_total'] = _ApiClient["default"].convertToType(data['products_quantity_total'], 'Number');
          delete data['products_quantity_total'];
        }

        if (data.hasOwnProperty('products_sum_total')) {
          obj['products_sum_total'] = _ApiClient["default"].convertToType(data['products_sum_total'], 'Number');
          delete data['products_sum_total'];
        }

        if (data.hasOwnProperty('variations_enabled')) {
          obj['variations_enabled'] = _ApiClient["default"].convertToType(data['variations_enabled'], 'Boolean');
          delete data['variations_enabled'];
        }

        if (data.hasOwnProperty('products_quantity_total_formatted')) {
          obj['products_quantity_total_formatted'] = _ApiClient["default"].convertToType(data['products_quantity_total_formatted'], 'String');
          delete data['products_quantity_total_formatted'];
        }

        if (data.hasOwnProperty('products_sum_total_formatted')) {
          obj['products_sum_total_formatted'] = _ApiClient["default"].convertToType(data['products_sum_total_formatted'], 'String');
          delete data['products_sum_total_formatted'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListProductAdditionalDataAllOf;
}();
/**
 * The total quantity of the products
 * @member {Number} products_quantity_total
 */


ListProductAdditionalDataAllOf.prototype['products_quantity_total'] = undefined;
/**
 * The total sum of the products
 * @member {Number} products_sum_total
 */

ListProductAdditionalDataAllOf.prototype['products_sum_total'] = undefined;
/**
 * Whether variations is enabled or not
 * @member {Boolean} variations_enabled
 */

ListProductAdditionalDataAllOf.prototype['variations_enabled'] = undefined;
/**
 * The total formatted quantity of the products
 * @member {String} products_quantity_total_formatted
 */

ListProductAdditionalDataAllOf.prototype['products_quantity_total_formatted'] = undefined;
/**
 * The total formatted sum of the products
 * @member {String} products_sum_total_formatted
 */

ListProductAdditionalDataAllOf.prototype['products_sum_total_formatted'] = undefined;
var _default = ListProductAdditionalDataAllOf;
exports["default"] = _default;
