// pipedrive/dist/model/ListFollowersResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListFollowersResponseAllOfData model module.
 * @module model/ListFollowersResponseAllOfData
 * @version 1.0.0
 */
var ListFollowersResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListFollowersResponseAllOfData</code>.
   * @alias module:model/ListFollowersResponseAllOfData
   */
  function ListFollowersResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ListFollowersResponseAllOfData);
    ListFollowersResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListFollowersResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListFollowersResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListFollowersResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ListFollowersResponseAllOfData} The populated <code>ListFollowersResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListFollowersResponseAllOfData();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListFollowersResponseAllOfData;
}();
/**
 * The ID of the user
 * @member {Number} user_id
 */


ListFollowersResponseAllOfData.prototype['user_id'] = undefined;
/**
 * The ID of the user follower
 * @member {Number} id
 */

ListFollowersResponseAllOfData.prototype['id'] = undefined;
/**
 * The ID of the person
 * @member {Number} person_id
 */

ListFollowersResponseAllOfData.prototype['person_id'] = undefined;
/**
 * The date and time when the follower was added to the person
 * @member {String} add_time
 */

ListFollowersResponseAllOfData.prototype['add_time'] = undefined;
var _default = ListFollowersResponseAllOfData;
exports["default"] = _default;
