// pipedrive/dist/model/ListProductsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListProductAdditionalData = _interopRequireDefault(require("./ListProductAdditionalData"));

var _ListProductsResponseAllOf = _interopRequireDefault(require("./ListProductsResponseAllOf"));

var _ListProductsResponseAllOfData = _interopRequireDefault(require("./ListProductsResponseAllOfData"));

var _ListProductsResponseAllOfRelatedObjects = _interopRequireDefault(require("./ListProductsResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListProductsResponse model module.
 * @module model/ListProductsResponse
 * @version 1.0.0
 */
var ListProductsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListProductsResponse</code>.
   * @alias module:model/ListProductsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListProductsResponseAllOf
   */
  function ListProductsResponse() {
    (0, _classCallCheck2["default"])(this, ListProductsResponse);

    _BaseResponse["default"].initialize(this);

    _ListProductsResponseAllOf["default"].initialize(this);

    ListProductsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListProductsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListProductsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListProductsResponse} obj Optional instance to populate.
     * @return {module:model/ListProductsResponse} The populated <code>ListProductsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListProductsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListProductsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ListProductsResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ListProductAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _ListProductsResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListProductsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListProductsResponse.prototype['success'] = undefined;
/**
 * @member {module:model/ListProductsResponseAllOfData} data
 */

ListProductsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ListProductAdditionalData} additional_data
 */

ListProductsResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListProductsResponseAllOfRelatedObjects} related_objects
 */

ListProductsResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListProductsResponseAllOf interface:

/**
 * @member {module:model/ListProductsResponseAllOfData} data
 */

_ListProductsResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ListProductAdditionalData} additional_data
 */

_ListProductsResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/ListProductsResponseAllOfRelatedObjects} related_objects
 */

_ListProductsResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = ListProductsResponse;
exports["default"] = _default;
