// pipedrive/dist/model/ListPermittedUsersResponse2.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListPermittedUsersResponse2AllOf = _interopRequireDefault(require("./ListPermittedUsersResponse2AllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPermittedUsersResponse2 model module.
 * @module model/ListPermittedUsersResponse2
 * @version 1.0.0
 */
var ListPermittedUsersResponse2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPermittedUsersResponse2</code>.
   * @alias module:model/ListPermittedUsersResponse2
   * @implements module:model/BaseResponse
   * @implements module:model/ListPermittedUsersResponse2AllOf
   */
  function ListPermittedUsersResponse2() {
    (0, _classCallCheck2["default"])(this, ListPermittedUsersResponse2);

    _BaseResponse["default"].initialize(this);

    _ListPermittedUsersResponse2AllOf["default"].initialize(this);

    ListPermittedUsersResponse2.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPermittedUsersResponse2, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPermittedUsersResponse2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPermittedUsersResponse2} obj Optional instance to populate.
     * @return {module:model/ListPermittedUsersResponse2} The populated <code>ListPermittedUsersResponse2</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPermittedUsersResponse2();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListPermittedUsersResponse2AllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], ['Number']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return ListPermittedUsersResponse2;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListPermittedUsersResponse2.prototype['success'] = undefined;
/**
 * The list of permitted Users IDs
 * @member {Array.<Number>} data
 */

ListPermittedUsersResponse2.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListPermittedUsersResponse2AllOf interface:

/**
 * The list of permitted Users IDs
 * @member {Array.<Number>} data
 */

_ListPermittedUsersResponse2AllOf["default"].prototype['data'] = undefined;
var _default = ListPermittedUsersResponse2;
exports["default"] = _default;
