// pipedrive/dist/model/ListPermittedUsersResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ListPermittedUsersResponseAllOf = _interopRequireDefault(require("./ListPermittedUsersResponseAllOf"));

var _ListPermittedUsersResponseAllOfData = _interopRequireDefault(require("./ListPermittedUsersResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPermittedUsersResponse model module.
 * @module model/ListPermittedUsersResponse
 * @version 1.0.0
 */
var ListPermittedUsersResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPermittedUsersResponse</code>.
   * @alias module:model/ListPermittedUsersResponse
   * @implements module:model/BaseResponse
   * @implements module:model/ListPermittedUsersResponseAllOf
   */
  function ListPermittedUsersResponse() {
    (0, _classCallCheck2["default"])(this, ListPermittedUsersResponse);

    _BaseResponse["default"].initialize(this);

    _ListPermittedUsersResponseAllOf["default"].initialize(this);

    ListPermittedUsersResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPermittedUsersResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPermittedUsersResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPermittedUsersResponse} obj Optional instance to populate.
     * @return {module:model/ListPermittedUsersResponse} The populated <code>ListPermittedUsersResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPermittedUsersResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _ListPermittedUsersResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ListPermittedUsersResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListPermittedUsersResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ListPermittedUsersResponse.prototype['success'] = undefined;
/**
 * @member {module:model/ListPermittedUsersResponseAllOfData} data
 */

ListPermittedUsersResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement ListPermittedUsersResponseAllOf interface:

/**
 * @member {module:model/ListPermittedUsersResponseAllOfData} data
 */

_ListPermittedUsersResponseAllOf["default"].prototype['data'] = undefined;
var _default = ListPermittedUsersResponse;
exports["default"] = _default;
