// pipedrive/dist/model/ListPersonProductsResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ListPersonProductsResponseAllOfDEALID = _interopRequireDefault(require("./ListPersonProductsResponseAllOfDEALID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ListPersonProductsResponseAllOfData model module.
 * @module model/ListPersonProductsResponseAllOfData
 * @version 1.0.0
 */
var ListPersonProductsResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListPersonProductsResponseAllOfData</code>.
   * @alias module:model/ListPersonProductsResponseAllOfData
   */
  function ListPersonProductsResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ListPersonProductsResponseAllOfData);
    ListPersonProductsResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ListPersonProductsResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListPersonProductsResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListPersonProductsResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ListPersonProductsResponseAllOfData} The populated <code>ListPersonProductsResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListPersonProductsResponseAllOfData();

        if (data.hasOwnProperty('DEAL_ID')) {
          obj['DEAL_ID'] = _ListPersonProductsResponseAllOfDEALID["default"].constructFromObject(data['DEAL_ID']);
          delete data['DEAL_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ListPersonProductsResponseAllOfData;
}();
/**
 * @member {module:model/ListPersonProductsResponseAllOfDEALID} DEAL_ID
 */


ListPersonProductsResponseAllOfData.prototype['DEAL_ID'] = undefined;
var _default = ListPersonProductsResponseAllOfData;
exports["default"] = _default;
