// pipedrive/dist/model/MailThreadDelete.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThread = _interopRequireDefault(require("./BaseMailThread"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _MailThreadDeleteAllOf = _interopRequireDefault(require("./MailThreadDeleteAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThreadDelete model module.
 * @module model/MailThreadDelete
 * @version 1.0.0
 */
var MailThreadDelete = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadDelete</code>.
   * @alias module:model/MailThreadDelete
   * @implements module:model/BaseResponse
   * @implements module:model/MailThreadDeleteAllOf
   */
  function MailThreadDelete() {
    (0, _classCallCheck2["default"])(this, MailThreadDelete);

    _BaseResponse["default"].initialize(this);

    _MailThreadDeleteAllOf["default"].initialize(this);

    MailThreadDelete.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThreadDelete, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThreadDelete</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadDelete} obj Optional instance to populate.
     * @return {module:model/MailThreadDelete} The populated <code>MailThreadDelete</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadDelete();

        _BaseResponse["default"].constructFromObject(data, obj);

        _MailThreadDeleteAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseMailThread["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThreadDelete;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailThreadDelete.prototype['success'] = undefined;
/**
 * An array of mail threads
 * @member {Array.<module:model/BaseMailThread>} data
 */

MailThreadDelete.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement MailThreadDeleteAllOf interface:

/**
 * An array of mail threads
 * @member {Array.<module:model/BaseMailThread>} data
 */

_MailThreadDeleteAllOf["default"].prototype['data'] = undefined;
var _default = MailThreadDelete;
exports["default"] = _default;
