// pipedrive/dist/model/MailThreadOne.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThread = _interopRequireDefault(require("./BaseMailThread"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _MailThreadOneAllOf = _interopRequireDefault(require("./MailThreadOneAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThreadOne model module.
 * @module model/MailThreadOne
 * @version 1.0.0
 */
var MailThreadOne = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadOne</code>.
   * @alias module:model/MailThreadOne
   * @implements module:model/BaseResponse
   * @implements module:model/MailThreadOneAllOf
   */
  function MailThreadOne() {
    (0, _classCallCheck2["default"])(this, MailThreadOne);

    _BaseResponse["default"].initialize(this);

    _MailThreadOneAllOf["default"].initialize(this);

    MailThreadOne.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThreadOne, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThreadOne</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadOne} obj Optional instance to populate.
     * @return {module:model/MailThreadOne} The populated <code>MailThreadOne</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadOne();

        _BaseResponse["default"].constructFromObject(data, obj);

        _MailThreadOneAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], _BaseMailThread["default"]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThreadOne;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailThreadOne.prototype['success'] = undefined;
/**
 * The mail thread object
 * @member {module:model/BaseMailThread} data
 */

MailThreadOne.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement MailThreadOneAllOf interface:

/**
 * The mail thread object
 * @member {module:model/BaseMailThread} data
 */

_MailThreadOneAllOf["default"].prototype['data'] = undefined;
var _default = MailThreadOne;
exports["default"] = _default;
