// pipedrive/dist/model/MailMessageData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MailParticipant = _interopRequireDefault(require("./MailParticipant"));

var _NumberBooleanDefault = _interopRequireDefault(require("./NumberBooleanDefault0"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailMessageData model module.
 * @module model/MailMessageData
 * @version 1.0.0
 */
var MailMessageData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailMessageData</code>.
   * @alias module:model/MailMessageData
   */
  function MailMessageData() {
    (0, _classCallCheck2["default"])(this, MailMessageData);
    MailMessageData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailMessageData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailMessageData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailMessageData} obj Optional instance to populate.
     * @return {module:model/MailMessageData} The populated <code>MailMessageData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailMessageData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], [_MailParticipant["default"]]);
          delete data['from'];
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], [_MailParticipant["default"]]);
          delete data['to'];
        }

        if (data.hasOwnProperty('cc')) {
          obj['cc'] = _ApiClient["default"].convertToType(data['cc'], [_MailParticipant["default"]]);
          delete data['cc'];
        }

        if (data.hasOwnProperty('bcc')) {
          obj['bcc'] = _ApiClient["default"].convertToType(data['bcc'], [_MailParticipant["default"]]);
          delete data['bcc'];
        }

        if (data.hasOwnProperty('body_url')) {
          obj['body_url'] = _ApiClient["default"].convertToType(data['body_url'], 'String');
          delete data['body_url'];
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
          delete data['account_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('mail_thread_id')) {
          obj['mail_thread_id'] = _ApiClient["default"].convertToType(data['mail_thread_id'], 'Number');
          delete data['mail_thread_id'];
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
          delete data['subject'];
        }

        if (data.hasOwnProperty('snippet')) {
          obj['snippet'] = _ApiClient["default"].convertToType(data['snippet'], 'String');
          delete data['snippet'];
        }

        if (data.hasOwnProperty('mail_tracking_status')) {
          obj['mail_tracking_status'] = _ApiClient["default"].convertToType(data['mail_tracking_status'], 'String');
          delete data['mail_tracking_status'];
        }

        if (data.hasOwnProperty('mail_link_tracking_enabled_flag')) {
          obj['mail_link_tracking_enabled_flag'] = _ApiClient["default"].convertToType(data['mail_link_tracking_enabled_flag'], _NumberBooleanDefault["default"]);
          delete data['mail_link_tracking_enabled_flag'];
        }

        if (data.hasOwnProperty('read_flag')) {
          obj['read_flag'] = _ApiClient["default"].convertToType(data['read_flag'], _NumberBooleanDefault["default"]);
          delete data['read_flag'];
        }

        if (data.hasOwnProperty('draft')) {
          obj['draft'] = _ApiClient["default"].convertToType(data['draft'], 'String');
          delete data['draft'];
        }

        if (data.hasOwnProperty('draft_flag')) {
          obj['draft_flag'] = _ApiClient["default"].convertToType(data['draft_flag'], _NumberBooleanDefault["default"]);
          delete data['draft_flag'];
        }

        if (data.hasOwnProperty('synced_flag')) {
          obj['synced_flag'] = _ApiClient["default"].convertToType(data['synced_flag'], _NumberBooleanDefault["default"]);
          delete data['synced_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBooleanDefault["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('has_body_flag')) {
          obj['has_body_flag'] = _ApiClient["default"].convertToType(data['has_body_flag'], _NumberBooleanDefault["default"]);
          delete data['has_body_flag'];
        }

        if (data.hasOwnProperty('sent_flag')) {
          obj['sent_flag'] = _ApiClient["default"].convertToType(data['sent_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_flag'];
        }

        if (data.hasOwnProperty('sent_from_pipedrive_flag')) {
          obj['sent_from_pipedrive_flag'] = _ApiClient["default"].convertToType(data['sent_from_pipedrive_flag'], _NumberBooleanDefault["default"]);
          delete data['sent_from_pipedrive_flag'];
        }

        if (data.hasOwnProperty('smart_bcc_flag')) {
          obj['smart_bcc_flag'] = _ApiClient["default"].convertToType(data['smart_bcc_flag'], _NumberBooleanDefault["default"]);
          delete data['smart_bcc_flag'];
        }

        if (data.hasOwnProperty('message_time')) {
          obj['message_time'] = _ApiClient["default"].convertToType(data['message_time'], 'Date');
          delete data['message_time'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Date');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'Date');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('has_attachments_flag')) {
          obj['has_attachments_flag'] = _ApiClient["default"].convertToType(data['has_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_attachments_flag'];
        }

        if (data.hasOwnProperty('has_inline_attachments_flag')) {
          obj['has_inline_attachments_flag'] = _ApiClient["default"].convertToType(data['has_inline_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_inline_attachments_flag'];
        }

        if (data.hasOwnProperty('has_real_attachments_flag')) {
          obj['has_real_attachments_flag'] = _ApiClient["default"].convertToType(data['has_real_attachments_flag'], _NumberBooleanDefault["default"]);
          delete data['has_real_attachments_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailMessageData;
}();
/**
 * ID of the mail message.
 * @member {Number} id
 */


MailMessageData.prototype['id'] = undefined;
/**
 * The array of mail message sender (object)
 * @member {Array.<module:model/MailParticipant>} from
 */

MailMessageData.prototype['from'] = undefined;
/**
 * The array of mail message receiver (object)
 * @member {Array.<module:model/MailParticipant>} to
 */

MailMessageData.prototype['to'] = undefined;
/**
 * The array of mail message copies (object)
 * @member {Array.<module:model/MailParticipant>} cc
 */

MailMessageData.prototype['cc'] = undefined;
/**
 * The array of mail message blind copies (object)
 * @member {Array.<module:model/MailParticipant>} bcc
 */

MailMessageData.prototype['bcc'] = undefined;
/**
 * The mail message body URL
 * @member {String} body_url
 */

MailMessageData.prototype['body_url'] = undefined;
/**
 * The connection account ID
 * @member {String} account_id
 */

MailMessageData.prototype['account_id'] = undefined;
/**
 * ID of the user whom mail message will be assigned to
 * @member {Number} user_id
 */

MailMessageData.prototype['user_id'] = undefined;
/**
 * ID of the mail message thread
 * @member {Number} mail_thread_id
 */

MailMessageData.prototype['mail_thread_id'] = undefined;
/**
 * The subject of mail message
 * @member {String} subject
 */

MailMessageData.prototype['subject'] = undefined;
/**
 * The snippet of mail message. Snippet length is up to 225 characters.
 * @member {String} snippet
 */

MailMessageData.prototype['snippet'] = undefined;
/**
 * The status of tracking mail message. Value is `null` if tracking is not enabled.
 * @member {module:model/MailMessageData.MailTrackingStatusEnum} mail_tracking_status
 */

MailMessageData.prototype['mail_tracking_status'] = undefined;
/**
 * Whether the link tracking in mail message body is enabled.
 * @member {module:model/NumberBooleanDefault0} mail_link_tracking_enabled_flag
 */

MailMessageData.prototype['mail_link_tracking_enabled_flag'] = undefined;
/**
 * Whether the mail message is read or not by the user
 * @member {module:model/NumberBooleanDefault0} read_flag
 */

MailMessageData.prototype['read_flag'] = undefined;
/**
 * If the mail message has a draft status then the value is the mail message object as JSON formatted string, otherwise `null`.
 * @member {String} draft
 */

MailMessageData.prototype['draft'] = undefined;
/**
 * Whether the mail message is a draft or not
 * @member {module:model/NumberBooleanDefault0} draft_flag
 */

MailMessageData.prototype['draft_flag'] = undefined;
/**
 * Whether the mail message is synced with the provider or not
 * @member {module:model/NumberBooleanDefault0} synced_flag
 */

MailMessageData.prototype['synced_flag'] = undefined;
/**
 * Whether the mail message is deleted or not
 * @member {module:model/NumberBooleanDefault0} deleted_flag
 */

MailMessageData.prototype['deleted_flag'] = undefined;
/**
 * Whether the mail message has a body or not
 * @member {module:model/NumberBooleanDefault0} has_body_flag
 */

MailMessageData.prototype['has_body_flag'] = undefined;
/**
 * Whether the mail message has been sent or not
 * @member {module:model/NumberBooleanDefault0} sent_flag
 */

MailMessageData.prototype['sent_flag'] = undefined;
/**
 * Whether the mail message has been sent from Pipedrive app or not
 * @member {module:model/NumberBooleanDefault0} sent_from_pipedrive_flag
 */

MailMessageData.prototype['sent_from_pipedrive_flag'] = undefined;
/**
 * Whether the mail message has been created by Smart Email BCC feature or not
 * @member {module:model/NumberBooleanDefault0} smart_bcc_flag
 */

MailMessageData.prototype['smart_bcc_flag'] = undefined;
/**
 * Creation or receival time of the mail message
 * @member {Date} message_time
 */

MailMessageData.prototype['message_time'] = undefined;
/**
 * The insertion into the database time of the mail message
 * @member {Date} add_time
 */

MailMessageData.prototype['add_time'] = undefined;
/**
 * The updating time in the database of the mail message
 * @member {Date} update_time
 */

MailMessageData.prototype['update_time'] = undefined;
/**
 * Whether the mail message has an attachment or not
 * @member {module:model/NumberBooleanDefault0} has_attachments_flag
 */

MailMessageData.prototype['has_attachments_flag'] = undefined;
/**
 * Whether the mail message has an inline attachment or not
 * @member {module:model/NumberBooleanDefault0} has_inline_attachments_flag
 */

MailMessageData.prototype['has_inline_attachments_flag'] = undefined;
/**
 * Whether the mail message has an attachment (which is not inline) or not
 * @member {module:model/NumberBooleanDefault0} has_real_attachments_flag
 */

MailMessageData.prototype['has_real_attachments_flag'] = undefined;
/**
 * Allowed values for the <code>mail_tracking_status</code> property.
 * @enum {String}
 * @readonly
 */

MailMessageData['MailTrackingStatusEnum'] = {
  /**
   * value: "opened"
   * @const
   */
  "opened": "opened",

  /**
   * value: "not opened"
   * @const
   */
  "not opened": "not opened"
};
var _default = MailMessageData;
exports["default"] = _default;
