// pipedrive/dist/model/MailParticipant.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailParticipant model module.
 * @module model/MailParticipant
 * @version 1.0.0
 */
var MailParticipant = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailParticipant</code>.
   * @alias module:model/MailParticipant
   */
  function MailParticipant() {
    (0, _classCallCheck2["default"])(this, MailParticipant);
    MailParticipant.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailParticipant, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailParticipant</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailParticipant} obj Optional instance to populate.
     * @return {module:model/MailParticipant} The populated <code>MailParticipant</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailParticipant();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('email_address')) {
          obj['email_address'] = _ApiClient["default"].convertToType(data['email_address'], 'String');
          delete data['email_address'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('linked_person_id')) {
          obj['linked_person_id'] = _ApiClient["default"].convertToType(data['linked_person_id'], 'Number');
          delete data['linked_person_id'];
        }

        if (data.hasOwnProperty('linked_person_name')) {
          obj['linked_person_name'] = _ApiClient["default"].convertToType(data['linked_person_name'], 'String');
          delete data['linked_person_name'];
        }

        if (data.hasOwnProperty('mail_message_party_id')) {
          obj['mail_message_party_id'] = _ApiClient["default"].convertToType(data['mail_message_party_id'], 'Number');
          delete data['mail_message_party_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailParticipant;
}();
/**
 * ID of the mail participant
 * @member {Number} id
 */


MailParticipant.prototype['id'] = undefined;
/**
 * Mail address of the mail participant
 * @member {String} email_address
 */

MailParticipant.prototype['email_address'] = undefined;
/**
 * Name of the mail participant
 * @member {String} name
 */

MailParticipant.prototype['name'] = undefined;
/**
 * ID of the linked person to the mail message
 * @member {Number} linked_person_id
 */

MailParticipant.prototype['linked_person_id'] = undefined;
/**
 * Name of the linked person to the mail message
 * @member {String} linked_person_name
 */

MailParticipant.prototype['linked_person_name'] = undefined;
/**
 * ID of the mail message participant
 * @member {Number} mail_message_party_id
 */

MailParticipant.prototype['mail_message_party_id'] = undefined;
var _default = MailParticipant;
exports["default"] = _default;
