// pipedrive/dist/model/MailMessage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MailMessageAllOf = _interopRequireDefault(require("./MailMessageAllOf"));

var _MailMessageData = _interopRequireDefault(require("./MailMessageData"));

var _MailServiceBaseResponse = _interopRequireDefault(require("./MailServiceBaseResponse"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailMessage model module.
 * @module model/MailMessage
 * @version 1.0.0
 */
var MailMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailMessage</code>.
   * @alias module:model/MailMessage
   * @implements module:model/MailServiceBaseResponse
   * @implements module:model/MailMessageAllOf
   */
  function MailMessage() {
    (0, _classCallCheck2["default"])(this, MailMessage);

    _MailServiceBaseResponse["default"].initialize(this);

    _MailMessageAllOf["default"].initialize(this);

    MailMessage.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailMessage, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailMessage} obj Optional instance to populate.
     * @return {module:model/MailMessage} The populated <code>MailMessage</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailMessage();

        _MailServiceBaseResponse["default"].constructFromObject(data, obj);

        _MailMessageAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
          delete data['statusCode'];
        }

        if (data.hasOwnProperty('statusText')) {
          obj['statusText'] = _ApiClient["default"].convertToType(data['statusText'], 'String');
          delete data['statusText'];
        }

        if (data.hasOwnProperty('service')) {
          obj['service'] = _ApiClient["default"].convertToType(data['service'], 'String');
          delete data['service'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _MailMessageData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailMessage;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailMessage.prototype['success'] = undefined;
/**
 * The email service specific status code and it is returned through the response body.
 * @member {Number} statusCode
 */

MailMessage.prototype['statusCode'] = undefined;
/**
 * The status text of the response.
 * @member {String} statusText
 */

MailMessage.prototype['statusText'] = undefined;
/**
 * The service name of the response.
 * @member {String} service
 */

MailMessage.prototype['service'] = undefined;
/**
 * @member {module:model/MailMessageData} data
 */

MailMessage.prototype['data'] = undefined; // Implement MailServiceBaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_MailServiceBaseResponse["default"].prototype['success'] = undefined;
/**
 * The email service specific status code and it is returned through the response body.
 * @member {Number} statusCode
 */

_MailServiceBaseResponse["default"].prototype['statusCode'] = undefined;
/**
 * The status text of the response.
 * @member {String} statusText
 */

_MailServiceBaseResponse["default"].prototype['statusText'] = undefined;
/**
 * The service name of the response.
 * @member {String} service
 */

_MailServiceBaseResponse["default"].prototype['service'] = undefined; // Implement MailMessageAllOf interface:

/**
 * @member {module:model/MailMessageData} data
 */

_MailMessageAllOf["default"].prototype['data'] = undefined;
var _default = MailMessage;
exports["default"] = _default;
