// pipedrive/dist/model/MailThreadMessages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThreadMessages = _interopRequireDefault(require("./BaseMailThreadMessages"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _MailThreadMessagesAllOf = _interopRequireDefault(require("./MailThreadMessagesAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThreadMessages model module.
 * @module model/MailThreadMessages
 * @version 1.0.0
 */
var MailThreadMessages = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadMessages</code>.
   * @alias module:model/MailThreadMessages
   * @implements module:model/BaseResponse
   * @implements module:model/MailThreadMessagesAllOf
   */
  function MailThreadMessages() {
    (0, _classCallCheck2["default"])(this, MailThreadMessages);

    _BaseResponse["default"].initialize(this);

    _MailThreadMessagesAllOf["default"].initialize(this);

    MailThreadMessages.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThreadMessages, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThreadMessages</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadMessages} obj Optional instance to populate.
     * @return {module:model/MailThreadMessages} The populated <code>MailThreadMessages</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadMessages();

        _BaseResponse["default"].constructFromObject(data, obj);

        _MailThreadMessagesAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseMailThreadMessages["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThreadMessages;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailThreadMessages.prototype['success'] = undefined;
/**
 * The array of the mail messages of the mail thread
 * @member {Array.<module:model/BaseMailThreadMessages>} data
 */

MailThreadMessages.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement MailThreadMessagesAllOf interface:

/**
 * The array of the mail messages of the mail thread
 * @member {Array.<module:model/BaseMailThreadMessages>} data
 */

_MailThreadMessagesAllOf["default"].prototype['data'] = undefined;
var _default = MailThreadMessages;
exports["default"] = _default;
