// pipedrive/dist/model/MailThreadDeleteAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThread = _interopRequireDefault(require("./BaseMailThread"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThreadDeleteAllOf model module.
 * @module model/MailThreadDeleteAllOf
 * @version 1.0.0
 */
var MailThreadDeleteAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadDeleteAllOf</code>.
   * @alias module:model/MailThreadDeleteAllOf
   */
  function MailThreadDeleteAllOf() {
    (0, _classCallCheck2["default"])(this, MailThreadDeleteAllOf);
    MailThreadDeleteAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThreadDeleteAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThreadDeleteAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadDeleteAllOf} obj Optional instance to populate.
     * @return {module:model/MailThreadDeleteAllOf} The populated <code>MailThreadDeleteAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadDeleteAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseMailThread["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThreadDeleteAllOf;
}();
/**
 * An array of mail threads
 * @member {Array.<module:model/BaseMailThread>} data
 */


MailThreadDeleteAllOf.prototype['data'] = undefined;
var _default = MailThreadDeleteAllOf;
exports["default"] = _default;
