// pipedrive/dist/model/MailThread.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseMailThread = _interopRequireDefault(require("./BaseMailThread"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _MailThreadAllOf = _interopRequireDefault(require("./MailThreadAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThread model module.
 * @module model/MailThread
 * @version 1.0.0
 */
var MailThread = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThread</code>.
   * @alias module:model/MailThread
   * @implements module:model/BaseResponse
   * @implements module:model/MailThreadAllOf
   */
  function MailThread() {
    (0, _classCallCheck2["default"])(this, MailThread);

    _BaseResponse["default"].initialize(this);

    _MailThreadAllOf["default"].initialize(this);

    MailThread.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThread, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThread</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThread} obj Optional instance to populate.
     * @return {module:model/MailThread} The populated <code>MailThread</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThread();

        _BaseResponse["default"].constructFromObject(data, obj);

        _MailThreadAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseMailThread["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThread;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailThread.prototype['success'] = undefined;
/**
 * The array of mail threads
 * @member {Array.<module:model/BaseMailThread>} data
 */

MailThread.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement MailThreadAllOf interface:

/**
 * The array of mail threads
 * @member {Array.<module:model/BaseMailThread>} data
 */

_MailThreadAllOf["default"].prototype['data'] = undefined;
var _default = MailThread;
exports["default"] = _default;
