// pipedrive/dist/model/MailServiceBaseResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailServiceBaseResponse model module.
 * @module model/MailServiceBaseResponse
 * @version 1.0.0
 */
var MailServiceBaseResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailServiceBaseResponse</code>.
   * @alias module:model/MailServiceBaseResponse
   */
  function MailServiceBaseResponse() {
    (0, _classCallCheck2["default"])(this, MailServiceBaseResponse);
    MailServiceBaseResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailServiceBaseResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailServiceBaseResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailServiceBaseResponse} obj Optional instance to populate.
     * @return {module:model/MailServiceBaseResponse} The populated <code>MailServiceBaseResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailServiceBaseResponse();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
          delete data['statusCode'];
        }

        if (data.hasOwnProperty('statusText')) {
          obj['statusText'] = _ApiClient["default"].convertToType(data['statusText'], 'String');
          delete data['statusText'];
        }

        if (data.hasOwnProperty('service')) {
          obj['service'] = _ApiClient["default"].convertToType(data['service'], 'String');
          delete data['service'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailServiceBaseResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


MailServiceBaseResponse.prototype['success'] = undefined;
/**
 * The email service specific status code and it is returned through the response body.
 * @member {Number} statusCode
 */

MailServiceBaseResponse.prototype['statusCode'] = undefined;
/**
 * The status text of the response.
 * @member {String} statusText
 */

MailServiceBaseResponse.prototype['statusText'] = undefined;
/**
 * The service name of the response.
 * @member {String} service
 */

MailServiceBaseResponse.prototype['service'] = undefined;
var _default = MailServiceBaseResponse;
exports["default"] = _default;
