// pipedrive/dist/model/MailThreadParticipant.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MailThreadParticipant model module.
 * @module model/MailThreadParticipant
 * @version 1.0.0
 */
var MailThreadParticipant = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadParticipant</code>.
   * Member of a thread
   * @alias module:model/MailThreadParticipant
   */
  function MailThreadParticipant() {
    (0, _classCallCheck2["default"])(this, MailThreadParticipant);
    MailThreadParticipant.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MailThreadParticipant, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MailThreadParticipant</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadParticipant} obj Optional instance to populate.
     * @return {module:model/MailThreadParticipant} The populated <code>MailThreadParticipant</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadParticipant();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('latest_sent')) {
          obj['latest_sent'] = _ApiClient["default"].convertToType(data['latest_sent'], 'Boolean');
          delete data['latest_sent'];
        }

        if (data.hasOwnProperty('email_address')) {
          obj['email_address'] = _ApiClient["default"].convertToType(data['email_address'], 'String');
          delete data['email_address'];
        }

        if (data.hasOwnProperty('message_time')) {
          obj['message_time'] = _ApiClient["default"].convertToType(data['message_time'], 'Number');
          delete data['message_time'];
        }

        if (data.hasOwnProperty('linked_person_id')) {
          obj['linked_person_id'] = _ApiClient["default"].convertToType(data['linked_person_id'], 'Number');
          delete data['linked_person_id'];
        }

        if (data.hasOwnProperty('linked_person_name')) {
          obj['linked_person_name'] = _ApiClient["default"].convertToType(data['linked_person_name'], 'String');
          delete data['linked_person_name'];
        }

        if (data.hasOwnProperty('mail_message_party_id')) {
          obj['mail_message_party_id'] = _ApiClient["default"].convertToType(data['mail_message_party_id'], 'Number');
          delete data['mail_message_party_id'];
        }

        if (data.hasOwnProperty('linked_organization_id')) {
          obj['linked_organization_id'] = _ApiClient["default"].convertToType(data['linked_organization_id'], 'Number');
          delete data['linked_organization_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MailThreadParticipant;
}();
/**
 * ID of the mail thread participant
 * @member {Number} id
 */


MailThreadParticipant.prototype['id'] = undefined;
/**
 * Name of the mail thread participant
 * @member {String} name
 */

MailThreadParticipant.prototype['name'] = undefined;
/**
 * Whether the mail thread participant was last to send an email
 * @member {Boolean} latest_sent
 */

MailThreadParticipant.prototype['latest_sent'] = undefined;
/**
 * Email address of the mail thread participant
 * @member {String} email_address
 */

MailThreadParticipant.prototype['email_address'] = undefined;
/**
 * Message time
 * @member {Number} message_time
 */

MailThreadParticipant.prototype['message_time'] = undefined;
/**
 * ID of the linked person
 * @member {Number} linked_person_id
 */

MailThreadParticipant.prototype['linked_person_id'] = undefined;
/**
 * Email of the linked person
 * @member {String} linked_person_name
 */

MailThreadParticipant.prototype['linked_person_name'] = undefined;
/**
 * ID of the mail message party
 * @member {Number} mail_message_party_id
 */

MailThreadParticipant.prototype['mail_message_party_id'] = undefined;
/**
 * Linked Organization ID
 * @member {Number} linked_organization_id
 */

MailThreadParticipant.prototype['linked_organization_id'] = undefined;
var _default = MailThreadParticipant;
exports["default"] = _default;
