// pipedrive/dist/model/MergeDealsRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The MergeDealsRequest model module.
 * @module model/MergeDealsRequest
 * @version 1.0.0
 */
var MergeDealsRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MergeDealsRequest</code>.
   * @alias module:model/MergeDealsRequest
   * @param mergeWithId {Number} The ID of the deal that the deal will be merged with
   */
  function MergeDealsRequest(mergeWithId) {
    (0, _classCallCheck2["default"])(this, MergeDealsRequest);
    MergeDealsRequest.initialize(this, mergeWithId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(MergeDealsRequest, null, [{
    key: "initialize",
    value: function initialize(obj, mergeWithId) {
      obj['merge_with_id'] = mergeWithId;
    }
    /**
     * Constructs a <code>MergeDealsRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MergeDealsRequest} obj Optional instance to populate.
     * @return {module:model/MergeDealsRequest} The populated <code>MergeDealsRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MergeDealsRequest();

        if (data.hasOwnProperty('merge_with_id')) {
          obj['merge_with_id'] = _ApiClient["default"].convertToType(data['merge_with_id'], 'Number');
          delete data['merge_with_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return MergeDealsRequest;
}();
/**
 * The ID of the deal that the deal will be merged with
 * @member {Number} merge_with_id
 */


MergeDealsRequest.prototype['merge_with_id'] = undefined;
var _default = MergeDealsRequest;
exports["default"] = _default;
