// pipedrive/dist/model/NewDealProduct.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasicDealProduct = _interopRequireDefault(require("./BasicDealProduct"));

var _NewDealProductAllOf = _interopRequireDefault(require("./NewDealProductAllOf"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NewDealProduct model module.
 * @module model/NewDealProduct
 * @version 1.0.0
 */
var NewDealProduct = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewDealProduct</code>.
   * @alias module:model/NewDealProduct
   * @implements module:model/BasicDealProduct
   * @implements module:model/NewDealProductAllOf
   * @param itemPrice {Number} The price at which this product will be added to the deal
   * @param quantity {Number} Quantity – e.g. how many items of this product will be added to the deal
   * @param productId {Number} The ID of the product that will be attached
   */
  function NewDealProduct(itemPrice, quantity, productId) {
    (0, _classCallCheck2["default"])(this, NewDealProduct);

    _BasicDealProduct["default"].initialize(this);

    _NewDealProductAllOf["default"].initialize(this, productId);

    NewDealProduct.initialize(this, itemPrice, quantity, productId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NewDealProduct, null, [{
    key: "initialize",
    value: function initialize(obj, itemPrice, quantity, productId) {
      obj['item_price'] = itemPrice;
      obj['quantity'] = quantity;
      obj['product_id'] = productId;
    }
    /**
     * Constructs a <code>NewDealProduct</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewDealProduct} obj Optional instance to populate.
     * @return {module:model/NewDealProduct} The populated <code>NewDealProduct</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewDealProduct();

        _BasicDealProduct["default"].constructFromObject(data, obj);

        _NewDealProductAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('item_price')) {
          obj['item_price'] = _ApiClient["default"].convertToType(data['item_price'], 'Number');
          delete data['item_price'];
        }

        if (data.hasOwnProperty('quantity')) {
          obj['quantity'] = _ApiClient["default"].convertToType(data['quantity'], 'Number');
          delete data['quantity'];
        }

        if (data.hasOwnProperty('discount_percentage')) {
          obj['discount_percentage'] = _ApiClient["default"].convertToType(data['discount_percentage'], 'Number');
          delete data['discount_percentage'];
        }

        if (data.hasOwnProperty('duration')) {
          obj['duration'] = _ApiClient["default"].convertToType(data['duration'], 'Number');
          delete data['duration'];
        }

        if (data.hasOwnProperty('product_variation_id')) {
          obj['product_variation_id'] = _ApiClient["default"].convertToType(data['product_variation_id'], 'Number');
          delete data['product_variation_id'];
        }

        if (data.hasOwnProperty('comments')) {
          obj['comments'] = _ApiClient["default"].convertToType(data['comments'], 'String');
          delete data['comments'];
        }

        if (data.hasOwnProperty('tax')) {
          obj['tax'] = _ApiClient["default"].convertToType(data['tax'], 'Number');
          delete data['tax'];
        }

        if (data.hasOwnProperty('enabled_flag')) {
          obj['enabled_flag'] = _ApiClient["default"].convertToType(data['enabled_flag'], _NumberBoolean["default"]);
          delete data['enabled_flag'];
        }

        if (data.hasOwnProperty('product_id')) {
          obj['product_id'] = _ApiClient["default"].convertToType(data['product_id'], 'Number');
          delete data['product_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NewDealProduct;
}();
/**
 * The price at which this product will be added to the deal
 * @member {Number} item_price
 */


NewDealProduct.prototype['item_price'] = undefined;
/**
 * Quantity – e.g. how many items of this product will be added to the deal
 * @member {Number} quantity
 */

NewDealProduct.prototype['quantity'] = undefined;
/**
 * The discount %. If omitted, will be set to 0.
 * @member {Number} discount_percentage
 * @default 0
 */

NewDealProduct.prototype['discount_percentage'] = 0;
/**
 * The duration of the product (when product durations are not enabled for the company or if omitted, defaults to 1)
 * @member {Number} duration
 * @default 1
 */

NewDealProduct.prototype['duration'] = 1;
/**
 * The ID of the product variation to use. When omitted, no variation will be used.
 * @member {Number} product_variation_id
 */

NewDealProduct.prototype['product_variation_id'] = undefined;
/**
 * Any textual comment associated with this product-deal attachment. Visible and editable in the application UI.
 * @member {String} comments
 */

NewDealProduct.prototype['comments'] = undefined;
/**
 * The tax percentage
 * @member {Number} tax
 * @default 0
 */

NewDealProduct.prototype['tax'] = 0;
/**
 * Whether the product is enabled on the deal or not. This makes it possible to add products to a deal with a specific price and discount criteria - but keep them disabled, which refrains them from being included in the deal price calculation. When omitted, the product will be marked as enabled by default.
 * @member {module:model/NumberBoolean} enabled_flag
 */

NewDealProduct.prototype['enabled_flag'] = undefined;
/**
 * The ID of the product that will be attached
 * @member {Number} product_id
 */

NewDealProduct.prototype['product_id'] = undefined; // Implement BasicDealProduct interface:

/**
 * The price at which this product will be added to the deal
 * @member {Number} item_price
 */

_BasicDealProduct["default"].prototype['item_price'] = undefined;
/**
 * Quantity – e.g. how many items of this product will be added to the deal
 * @member {Number} quantity
 */

_BasicDealProduct["default"].prototype['quantity'] = undefined;
/**
 * The discount %. If omitted, will be set to 0.
 * @member {Number} discount_percentage
 * @default 0
 */

_BasicDealProduct["default"].prototype['discount_percentage'] = 0;
/**
 * The duration of the product (when product durations are not enabled for the company or if omitted, defaults to 1)
 * @member {Number} duration
 * @default 1
 */

_BasicDealProduct["default"].prototype['duration'] = 1;
/**
 * The ID of the product variation to use. When omitted, no variation will be used.
 * @member {Number} product_variation_id
 */

_BasicDealProduct["default"].prototype['product_variation_id'] = undefined;
/**
 * Any textual comment associated with this product-deal attachment. Visible and editable in the application UI.
 * @member {String} comments
 */

_BasicDealProduct["default"].prototype['comments'] = undefined;
/**
 * The tax percentage
 * @member {Number} tax
 * @default 0
 */

_BasicDealProduct["default"].prototype['tax'] = 0;
/**
 * Whether the product is enabled on the deal or not. This makes it possible to add products to a deal with a specific price and discount criteria - but keep them disabled, which refrains them from being included in the deal price calculation. When omitted, the product will be marked as enabled by default.
 * @member {module:model/NumberBoolean} enabled_flag
 */

_BasicDealProduct["default"].prototype['enabled_flag'] = undefined; // Implement NewDealProductAllOf interface:

/**
 * The ID of the product that will be attached
 * @member {Number} product_id
 */

_NewDealProductAllOf["default"].prototype['product_id'] = undefined;
var _default = NewDealProduct;
exports["default"] = _default;
