// pipedrive/dist/model/NewOrganization.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasicOrganization = _interopRequireDefault(require("./BasicOrganization"));

var _NewOrganizationAllOf = _interopRequireDefault(require("./NewOrganizationAllOf"));

var _VisibleTo = _interopRequireDefault(require("./VisibleTo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NewOrganization model module.
 * @module model/NewOrganization
 * @version 1.0.0
 */
var NewOrganization = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewOrganization</code>.
   * @alias module:model/NewOrganization
   * @implements module:model/BasicOrganization
   * @implements module:model/NewOrganizationAllOf
   * @param name {String} The name of the organization
   */
  function NewOrganization(name) {
    (0, _classCallCheck2["default"])(this, NewOrganization);

    _BasicOrganization["default"].initialize(this);

    _NewOrganizationAllOf["default"].initialize(this);

    NewOrganization.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NewOrganization, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>NewOrganization</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewOrganization} obj Optional instance to populate.
     * @return {module:model/NewOrganization} The populated <code>NewOrganization</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewOrganization();

        _BasicOrganization["default"].constructFromObject(data, obj);

        _NewOrganizationAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], _VisibleTo["default"]);
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NewOrganization;
}();
/**
 * The name of the organization
 * @member {String} name
 */


NewOrganization.prototype['name'] = undefined;
/**
 * The ID of the user who will be marked as the owner of this organization. When omitted, the authorized user ID will be used.
 * @member {Number} owner_id
 */

NewOrganization.prototype['owner_id'] = undefined;
/**
 * The visibility of the organization. If omitted, the visibility will be set to the default visibility setting of this item type for the authorized user. Read more about visibility groups <a href=\"https://support.pipedrive.com/en/article/visibility-groups\" target=\"_blank\" rel=\"noopener noreferrer\">here</a>.<h4>Essential / Advanced plan</h4><table><tr><th style=\"width:40px\">Value</th><th>Description</th></tr><tr><td>`1`</td><td>Owner &amp; followers</td><tr><td>`3`</td><td>Entire company</td></tr></table><h4>Professional / Enterprise plan</h4><table><tr><th style=\"width:40px\">Value</th><th>Description</th></tr><tr><td>`1`</td><td>Owner only</td><tr><td>`3`</td><td>Owner's visibility group</td></tr><tr><td>`5`</td><td>Owner's visibility group and sub-groups</td></tr><tr><td>`7`</td><td>Entire company</td></tr></table>
 * @member {module:model/VisibleTo} visible_to
 */

NewOrganization.prototype['visible_to'] = undefined;
/**
 * The optional creation date & time of the organization in UTC. Requires admin user API token. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

NewOrganization.prototype['add_time'] = undefined; // Implement BasicOrganization interface:

/**
 * The name of the organization
 * @member {String} name
 */

_BasicOrganization["default"].prototype['name'] = undefined;
/**
 * The ID of the user who will be marked as the owner of this organization. When omitted, the authorized user ID will be used.
 * @member {Number} owner_id
 */

_BasicOrganization["default"].prototype['owner_id'] = undefined;
/**
 * The visibility of the organization. If omitted, the visibility will be set to the default visibility setting of this item type for the authorized user. Read more about visibility groups <a href=\"https://support.pipedrive.com/en/article/visibility-groups\" target=\"_blank\" rel=\"noopener noreferrer\">here</a>.<h4>Essential / Advanced plan</h4><table><tr><th style=\"width:40px\">Value</th><th>Description</th></tr><tr><td>`1`</td><td>Owner &amp; followers</td><tr><td>`3`</td><td>Entire company</td></tr></table><h4>Professional / Enterprise plan</h4><table><tr><th style=\"width:40px\">Value</th><th>Description</th></tr><tr><td>`1`</td><td>Owner only</td><tr><td>`3`</td><td>Owner's visibility group</td></tr><tr><td>`5`</td><td>Owner's visibility group and sub-groups</td></tr><tr><td>`7`</td><td>Entire company</td></tr></table>
 * @member {module:model/VisibleTo} visible_to
 */

_BasicOrganization["default"].prototype['visible_to'] = undefined; // Implement NewOrganizationAllOf interface:

/**
 * The optional creation date & time of the organization in UTC. Requires admin user API token. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

_NewOrganizationAllOf["default"].prototype['add_time'] = undefined;
var _default = NewOrganization;
exports["default"] = _default;
