// pipedrive/dist/model/NoteParams.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NoteParams model module.
 * @module model/NoteParams
 * @version 1.0.0
 */
var NoteParams = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NoteParams</code>.
   * @alias module:model/NoteParams
   */
  function NoteParams() {
    (0, _classCallCheck2["default"])(this, NoteParams);
    NoteParams.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NoteParams, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NoteParams</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NoteParams} obj Optional instance to populate.
     * @return {module:model/NoteParams} The populated <code>NoteParams</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NoteParams();

        if (data.hasOwnProperty('content')) {
          obj['content'] = _ApiClient["default"].convertToType(data['content'], 'String');
          delete data['content'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('pinned_to_lead_flag')) {
          obj['pinned_to_lead_flag'] = _ApiClient["default"].convertToType(data['pinned_to_lead_flag'], _NumberBoolean["default"]);
          delete data['pinned_to_lead_flag'];
        }

        if (data.hasOwnProperty('pinned_to_deal_flag')) {
          obj['pinned_to_deal_flag'] = _ApiClient["default"].convertToType(data['pinned_to_deal_flag'], _NumberBoolean["default"]);
          delete data['pinned_to_deal_flag'];
        }

        if (data.hasOwnProperty('pinned_to_organization_flag')) {
          obj['pinned_to_organization_flag'] = _ApiClient["default"].convertToType(data['pinned_to_organization_flag'], _NumberBoolean["default"]);
          delete data['pinned_to_organization_flag'];
        }

        if (data.hasOwnProperty('pinned_to_person_flag')) {
          obj['pinned_to_person_flag'] = _ApiClient["default"].convertToType(data['pinned_to_person_flag'], _NumberBoolean["default"]);
          delete data['pinned_to_person_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NoteParams;
}();
/**
 * The content of the note in HTML format. Subject to sanitization on the back-end.
 * @member {String} content
 */


NoteParams.prototype['content'] = undefined;
/**
 * The ID of the user who will be marked as the author of the note. Only an admin can change the author.
 * @member {Number} user_id
 */

NoteParams.prototype['user_id'] = undefined;
/**
 * The optional creation date & time of the note in UTC. Can be set in the past or in the future. Requires admin user API token. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

NoteParams.prototype['add_time'] = undefined;
/**
 * If set, the results are filtered by note to lead pinning state (`lead_id` is also required)
 * @member {module:model/NumberBoolean} pinned_to_lead_flag
 */

NoteParams.prototype['pinned_to_lead_flag'] = undefined;
/**
 * If set, the results are filtered by note to deal pinning state (`deal_id` is also required)
 * @member {module:model/NumberBoolean} pinned_to_deal_flag
 */

NoteParams.prototype['pinned_to_deal_flag'] = undefined;
/**
 * If set, the results are filtered by note to organization pinning state (`org_id` is also required)
 * @member {module:model/NumberBoolean} pinned_to_organization_flag
 */

NoteParams.prototype['pinned_to_organization_flag'] = undefined;
/**
 * If set, the results are filtered by note to person pinning state (`person_id` is also required)
 * @member {module:model/NumberBoolean} pinned_to_person_flag
 */

NoteParams.prototype['pinned_to_person_flag'] = undefined;
var _default = NoteParams;
exports["default"] = _default;
