// pipedrive/dist/model/NoteConnectToParams.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NoteConnectToParams model module.
 * @module model/NoteConnectToParams
 * @version 1.0.0
 */
var NoteConnectToParams = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NoteConnectToParams</code>.
   * @alias module:model/NoteConnectToParams
   */
  function NoteConnectToParams() {
    (0, _classCallCheck2["default"])(this, NoteConnectToParams);
    NoteConnectToParams.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NoteConnectToParams, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NoteConnectToParams</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NoteConnectToParams} obj Optional instance to populate.
     * @return {module:model/NoteConnectToParams} The populated <code>NoteConnectToParams</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NoteConnectToParams();

        if (data.hasOwnProperty('lead_id')) {
          obj['lead_id'] = _ApiClient["default"].convertToType(data['lead_id'], 'String');
          delete data['lead_id'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NoteConnectToParams;
}();
/**
 * The ID of the lead the note will be attached to
 * @member {String} lead_id
 */


NoteConnectToParams.prototype['lead_id'] = undefined;
/**
 * The ID of the deal the note will be attached to
 * @member {Number} deal_id
 */

NoteConnectToParams.prototype['deal_id'] = undefined;
/**
 * The ID of the person the note will be attached to
 * @member {Number} person_id
 */

NoteConnectToParams.prototype['person_id'] = undefined;
/**
 * The ID of the organization the note will be attached to
 * @member {Number} org_id
 */

NoteConnectToParams.prototype['org_id'] = undefined;
var _default = NoteConnectToParams;
exports["default"] = _default;
