// pipedrive/dist/model/NoteCreatorUser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NoteCreatorUser model module.
 * @module model/NoteCreatorUser
 * @version 1.0.0
 */
var NoteCreatorUser = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NoteCreatorUser</code>.
   * The user who created the note
   * @alias module:model/NoteCreatorUser
   */
  function NoteCreatorUser() {
    (0, _classCallCheck2["default"])(this, NoteCreatorUser);
    NoteCreatorUser.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NoteCreatorUser, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NoteCreatorUser</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NoteCreatorUser} obj Optional instance to populate.
     * @return {module:model/NoteCreatorUser} The populated <code>NoteCreatorUser</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NoteCreatorUser();

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
          delete data['email'];
        }

        if (data.hasOwnProperty('icon_url')) {
          obj['icon_url'] = _ApiClient["default"].convertToType(data['icon_url'], 'String');
          delete data['icon_url'];
        }

        if (data.hasOwnProperty('is_you')) {
          obj['is_you'] = _ApiClient["default"].convertToType(data['is_you'], 'Boolean');
          delete data['is_you'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NoteCreatorUser;
}();
/**
 * The email of the note creator
 * @member {String} email
 */


NoteCreatorUser.prototype['email'] = undefined;
/**
 * The URL of the note creator avatar picture
 * @member {String} icon_url
 */

NoteCreatorUser.prototype['icon_url'] = undefined;
/**
 * Whether the note is created by you or not
 * @member {Boolean} is_you
 */

NoteCreatorUser.prototype['is_you'] = undefined;
/**
 * The name of the note creator
 * @member {String} name
 */

NoteCreatorUser.prototype['name'] = undefined;
var _default = NoteCreatorUser;
exports["default"] = _default;
