// pipedrive/dist/model/NoteField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FieldTypeAsString = _interopRequireDefault(require("./FieldTypeAsString"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NoteField model module.
 * @module model/NoteField
 * @version 1.0.0
 */
var NoteField = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NoteField</code>.
   * @alias module:model/NoteField
   */
  function NoteField() {
    (0, _classCallCheck2["default"])(this, NoteField);
    NoteField.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NoteField, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NoteField</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NoteField} obj Optional instance to populate.
     * @return {module:model/NoteField} The populated <code>NoteField</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NoteField();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
          delete data['key'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('field_type')) {
          obj['field_type'] = _FieldTypeAsString["default"].constructFromObject(data['field_type']);
          delete data['field_type'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('edit_flag')) {
          obj['edit_flag'] = _ApiClient["default"].convertToType(data['edit_flag'], 'Boolean');
          delete data['edit_flag'];
        }

        if (data.hasOwnProperty('bulk_edit_allowed')) {
          obj['bulk_edit_allowed'] = _ApiClient["default"].convertToType(data['bulk_edit_allowed'], 'Boolean');
          delete data['bulk_edit_allowed'];
        }

        if (data.hasOwnProperty('mandatory_flag')) {
          obj['mandatory_flag'] = _ApiClient["default"].convertToType(data['mandatory_flag'], 'Boolean');
          delete data['mandatory_flag'];
        }

        if (data.hasOwnProperty('options')) {
          obj['options'] = _ApiClient["default"].convertToType(data['options'], [Object]);
          delete data['options'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NoteField;
}();
/**
 * The ID of the field
 * @member {Number} id
 */


NoteField.prototype['id'] = undefined;
/**
 * The key of the field
 * @member {String} key
 */

NoteField.prototype['key'] = undefined;
/**
 * The name of the field
 * @member {String} name
 */

NoteField.prototype['name'] = undefined;
/**
 * @member {module:model/FieldTypeAsString} field_type
 */

NoteField.prototype['field_type'] = undefined;
/**
 * The active flag of the field
 * @member {Boolean} active_flag
 */

NoteField.prototype['active_flag'] = undefined;
/**
 * The edit flag of the field
 * @member {Boolean} edit_flag
 */

NoteField.prototype['edit_flag'] = undefined;
/**
 * Not used
 * @member {Boolean} bulk_edit_allowed
 */

NoteField.prototype['bulk_edit_allowed'] = undefined;
/**
 * Whether or not the field is mandatory
 * @member {Boolean} mandatory_flag
 */

NoteField.prototype['mandatory_flag'] = undefined;
/**
 * The options of the field. When there are no options, `null` is returned.
 * @member {Array.<Object>} options
 */

NoteField.prototype['options'] = undefined;
var _default = NoteField;
exports["default"] = _default;
