// pipedrive/dist/model/OrganizationFollowerItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseFollowerItem = _interopRequireDefault(require("./BaseFollowerItem"));

var _OrganizationFollowerItemAllOf = _interopRequireDefault(require("./OrganizationFollowerItemAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationFollowerItem model module.
 * @module model/OrganizationFollowerItem
 * @version 1.0.0
 */
var OrganizationFollowerItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationFollowerItem</code>.
   * @alias module:model/OrganizationFollowerItem
   * @implements module:model/OrganizationFollowerItemAllOf
   * @implements module:model/BaseFollowerItem
   */
  function OrganizationFollowerItem() {
    (0, _classCallCheck2["default"])(this, OrganizationFollowerItem);

    _OrganizationFollowerItemAllOf["default"].initialize(this);

    _BaseFollowerItem["default"].initialize(this);

    OrganizationFollowerItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationFollowerItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationFollowerItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationFollowerItem} obj Optional instance to populate.
     * @return {module:model/OrganizationFollowerItem} The populated <code>OrganizationFollowerItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationFollowerItem();

        _OrganizationFollowerItemAllOf["default"].constructFromObject(data, obj);

        _BaseFollowerItem["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'Number');
          delete data['add_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationFollowerItem;
}();
/**
 * The ID of the organization
 * @member {Number} org_id
 */


OrganizationFollowerItem.prototype['org_id'] = undefined;
/**
 * The user ID of the follower related to the item
 * @member {Number} user_id
 */

OrganizationFollowerItem.prototype['user_id'] = undefined;
/**
 * The ID of the follower
 * @member {Number} id
 */

OrganizationFollowerItem.prototype['id'] = undefined;
/**
 * The date and time of adding the follower to the item
 * @member {Number} add_time
 */

OrganizationFollowerItem.prototype['add_time'] = undefined; // Implement OrganizationFollowerItemAllOf interface:

/**
 * The ID of the organization
 * @member {Number} org_id
 */

_OrganizationFollowerItemAllOf["default"].prototype['org_id'] = undefined; // Implement BaseFollowerItem interface:

/**
 * The user ID of the follower related to the item
 * @member {Number} user_id
 */

_BaseFollowerItem["default"].prototype['user_id'] = undefined;
/**
 * The ID of the follower
 * @member {Number} id
 */

_BaseFollowerItem["default"].prototype['id'] = undefined;
/**
 * The date and time of adding the follower to the item
 * @member {Number} add_time
 */

_BaseFollowerItem["default"].prototype['add_time'] = undefined;
var _default = OrganizationFollowerItem;
exports["default"] = _default;
