// pipedrive/dist/model/OrganizationRelationshipDetails.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OrganizationRelationshipWithCalculatedFields = _interopRequireDefault(require("./OrganizationRelationshipWithCalculatedFields"));

var _RelatedOrganizationName = _interopRequireDefault(require("./RelatedOrganizationName"));

var _RelationshipOrganizationInfoItem = _interopRequireDefault(require("./RelationshipOrganizationInfoItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationRelationshipDetails model module.
 * @module model/OrganizationRelationshipDetails
 * @version 1.0.0
 */
var OrganizationRelationshipDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationRelationshipDetails</code>.
   * @alias module:model/OrganizationRelationshipDetails
   * @implements module:model/OrganizationRelationshipWithCalculatedFields
   * @implements module:model/RelatedOrganizationName
   */
  function OrganizationRelationshipDetails() {
    (0, _classCallCheck2["default"])(this, OrganizationRelationshipDetails);

    _OrganizationRelationshipWithCalculatedFields["default"].initialize(this);

    _RelatedOrganizationName["default"].initialize(this);

    OrganizationRelationshipDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationRelationshipDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationRelationshipDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationRelationshipDetails} obj Optional instance to populate.
     * @return {module:model/OrganizationRelationshipDetails} The populated <code>OrganizationRelationshipDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationRelationshipDetails();

        _OrganizationRelationshipWithCalculatedFields["default"].constructFromObject(data, obj);

        _RelatedOrganizationName["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (data.hasOwnProperty('rel_owner_org_id')) {
          obj['rel_owner_org_id'] = _RelationshipOrganizationInfoItem["default"].constructFromObject(data['rel_owner_org_id']);
          delete data['rel_owner_org_id'];
        }

        if (data.hasOwnProperty('rel_linked_org_id')) {
          obj['rel_linked_org_id'] = _RelationshipOrganizationInfoItem["default"].constructFromObject(data['rel_linked_org_id']);
          delete data['rel_linked_org_id'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'String');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('calculated_type')) {
          obj['calculated_type'] = _ApiClient["default"].convertToType(data['calculated_type'], 'String');
          delete data['calculated_type'];
        }

        if (data.hasOwnProperty('calculated_related_org_id')) {
          obj['calculated_related_org_id'] = _ApiClient["default"].convertToType(data['calculated_related_org_id'], 'Number');
          delete data['calculated_related_org_id'];
        }

        if (data.hasOwnProperty('related_organization_name')) {
          obj['related_organization_name'] = _ApiClient["default"].convertToType(data['related_organization_name'], 'String');
          delete data['related_organization_name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationRelationshipDetails;
}();
/**
 * The ID of the organization relationship
 * @member {Number} id
 */


OrganizationRelationshipDetails.prototype['id'] = undefined;
/**
 * The type of the relationship
 * @member {String} type
 */

OrganizationRelationshipDetails.prototype['type'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItem} rel_owner_org_id
 */

OrganizationRelationshipDetails.prototype['rel_owner_org_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItem} rel_linked_org_id
 */

OrganizationRelationshipDetails.prototype['rel_linked_org_id'] = undefined;
/**
 * The creation date and time of the relationship
 * @member {String} add_time
 */

OrganizationRelationshipDetails.prototype['add_time'] = undefined;
/**
 * The last updated date and time of the relationship
 * @member {String} update_time
 */

OrganizationRelationshipDetails.prototype['update_time'] = undefined;
/**
 * Whether the relationship is active or not
 * @member {String} active_flag
 */

OrganizationRelationshipDetails.prototype['active_flag'] = undefined;
/**
 * The calculated type of the relationship with the linked organization
 * @member {String} calculated_type
 */

OrganizationRelationshipDetails.prototype['calculated_type'] = undefined;
/**
 * The ID of the linked organization
 * @member {Number} calculated_related_org_id
 */

OrganizationRelationshipDetails.prototype['calculated_related_org_id'] = undefined;
/**
 * The name of the linked organization
 * @member {String} related_organization_name
 */

OrganizationRelationshipDetails.prototype['related_organization_name'] = undefined; // Implement OrganizationRelationshipWithCalculatedFields interface:

/**
 * The ID of the organization relationship
 * @member {Number} id
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['id'] = undefined;
/**
 * The type of the relationship
 * @member {String} type
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['type'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItem} rel_owner_org_id
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['rel_owner_org_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItem} rel_linked_org_id
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['rel_linked_org_id'] = undefined;
/**
 * The creation date and time of the relationship
 * @member {String} add_time
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['add_time'] = undefined;
/**
 * The last updated date and time of the relationship
 * @member {String} update_time
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['update_time'] = undefined;
/**
 * Whether the relationship is active or not
 * @member {String} active_flag
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['active_flag'] = undefined;
/**
 * The calculated type of the relationship with the linked organization
 * @member {String} calculated_type
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['calculated_type'] = undefined;
/**
 * The ID of the linked organization
 * @member {Number} calculated_related_org_id
 */

_OrganizationRelationshipWithCalculatedFields["default"].prototype['calculated_related_org_id'] = undefined; // Implement RelatedOrganizationName interface:

/**
 * The name of the linked organization
 * @member {String} related_organization_name
 */

_RelatedOrganizationName["default"].prototype['related_organization_name'] = undefined;
var _default = OrganizationRelationshipDetails;
exports["default"] = _default;
