// pipedrive/dist/model/OrganizationDetailsGetResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _OrganizationDetailsGetResponseAllOfAdditionalData = _interopRequireDefault(require("./OrganizationDetailsGetResponseAllOfAdditionalData"));

var _OrganizationItem = _interopRequireDefault(require("./OrganizationItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationDetailsGetResponseAllOf model module.
 * @module model/OrganizationDetailsGetResponseAllOf
 * @version 1.0.0
 */
var OrganizationDetailsGetResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationDetailsGetResponseAllOf</code>.
   * @alias module:model/OrganizationDetailsGetResponseAllOf
   */
  function OrganizationDetailsGetResponseAllOf() {
    (0, _classCallCheck2["default"])(this, OrganizationDetailsGetResponseAllOf);
    OrganizationDetailsGetResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationDetailsGetResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationDetailsGetResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationDetailsGetResponseAllOf} obj Optional instance to populate.
     * @return {module:model/OrganizationDetailsGetResponseAllOf} The populated <code>OrganizationDetailsGetResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationDetailsGetResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _OrganizationItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _OrganizationDetailsGetResponseAllOfAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationDetailsGetResponseAllOf;
}();
/**
 * @member {module:model/OrganizationItem} data
 */


OrganizationDetailsGetResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/OrganizationDetailsGetResponseAllOfAdditionalData} additional_data
 */

OrganizationDetailsGetResponseAllOf.prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationDetailsGetResponseAllOf.prototype['related_objects'] = undefined;
var _default = OrganizationDetailsGetResponseAllOf;
exports["default"] = _default;
