// pipedrive/dist/model/OrgAndOwnerId.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Owner = _interopRequireDefault(require("./Owner"));

var _RelationshipOrganizationInfoItemWithActiveFlag = _interopRequireDefault(require("./RelationshipOrganizationInfoItemWithActiveFlag"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrgAndOwnerId model module.
 * @module model/OrgAndOwnerId
 * @version 1.0.0
 */
var OrgAndOwnerId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrgAndOwnerId</code>.
   * @alias module:model/OrgAndOwnerId
   */
  function OrgAndOwnerId() {
    (0, _classCallCheck2["default"])(this, OrgAndOwnerId);
    OrgAndOwnerId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrgAndOwnerId, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrgAndOwnerId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrgAndOwnerId} obj Optional instance to populate.
     * @return {module:model/OrgAndOwnerId} The populated <code>OrgAndOwnerId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrgAndOwnerId();

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _Owner["default"].constructFromObject(data['owner_id']);
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _RelationshipOrganizationInfoItemWithActiveFlag["default"].constructFromObject(data['org_id']);
          delete data['org_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrgAndOwnerId;
}();
/**
 * @member {module:model/Owner} owner_id
 */


OrgAndOwnerId.prototype['owner_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItemWithActiveFlag} org_id
 */

OrgAndOwnerId.prototype['org_id'] = undefined;
var _default = OrgAndOwnerId;
exports["default"] = _default;
