// pipedrive/dist/model/OrganizationFlowResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _OrganizationFlowResponseAllOfData = _interopRequireDefault(require("./OrganizationFlowResponseAllOfData"));

var _OrganizationFlowResponseAllOfRelatedObjects = _interopRequireDefault(require("./OrganizationFlowResponseAllOfRelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationFlowResponseAllOf model module.
 * @module model/OrganizationFlowResponseAllOf
 * @version 1.0.0
 */
var OrganizationFlowResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationFlowResponseAllOf</code>.
   * @alias module:model/OrganizationFlowResponseAllOf
   */
  function OrganizationFlowResponseAllOf() {
    (0, _classCallCheck2["default"])(this, OrganizationFlowResponseAllOf);
    OrganizationFlowResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationFlowResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationFlowResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationFlowResponseAllOf} obj Optional instance to populate.
     * @return {module:model/OrganizationFlowResponseAllOf} The populated <code>OrganizationFlowResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationFlowResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_OrganizationFlowResponseAllOfData["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _OrganizationFlowResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationFlowResponseAllOf;
}();
/**
 * @member {Array.<module:model/OrganizationFlowResponseAllOfData>} data
 */


OrganizationFlowResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

OrganizationFlowResponseAllOf.prototype['additional_data'] = undefined;
/**
 * @member {module:model/OrganizationFlowResponseAllOfRelatedObjects} related_objects
 */

OrganizationFlowResponseAllOf.prototype['related_objects'] = undefined;
var _default = OrganizationFlowResponseAllOf;
exports["default"] = _default;
