// pipedrive/dist/model/OrganizationDataWithIdAndActiveFlag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OrganizationDataWithId = _interopRequireDefault(require("./OrganizationDataWithId"));

var _OrganizationDataWithIdAndActiveFlagAllOf = _interopRequireDefault(require("./OrganizationDataWithIdAndActiveFlagAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationDataWithIdAndActiveFlag model module.
 * @module model/OrganizationDataWithIdAndActiveFlag
 * @version 1.0.0
 */
var OrganizationDataWithIdAndActiveFlag = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationDataWithIdAndActiveFlag</code>.
   * The ID of the organization associated with the item
   * @alias module:model/OrganizationDataWithIdAndActiveFlag
   * @implements module:model/OrganizationDataWithIdAndActiveFlagAllOf
   * @implements module:model/OrganizationDataWithId
   */
  function OrganizationDataWithIdAndActiveFlag() {
    (0, _classCallCheck2["default"])(this, OrganizationDataWithIdAndActiveFlag);

    _OrganizationDataWithIdAndActiveFlagAllOf["default"].initialize(this);

    _OrganizationDataWithId["default"].initialize(this);

    OrganizationDataWithIdAndActiveFlag.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationDataWithIdAndActiveFlag, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationDataWithIdAndActiveFlag</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationDataWithIdAndActiveFlag} obj Optional instance to populate.
     * @return {module:model/OrganizationDataWithIdAndActiveFlag} The populated <code>OrganizationDataWithIdAndActiveFlag</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationDataWithIdAndActiveFlag();

        _OrganizationDataWithIdAndActiveFlagAllOf["default"].constructFromObject(data, obj);

        _OrganizationDataWithId["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('people_count')) {
          obj['people_count'] = _ApiClient["default"].convertToType(data['people_count'], 'Number');
          delete data['people_count'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationDataWithIdAndActiveFlag;
}();
/**
 * Whether the associated organization is active or not
 * @member {Boolean} active_flag
 */


OrganizationDataWithIdAndActiveFlag.prototype['active_flag'] = undefined;
/**
 * The ID of the organization associated with the item
 * @member {Number} id
 */

OrganizationDataWithIdAndActiveFlag.prototype['id'] = undefined;
/**
 * The name of the organization associated with the item
 * @member {String} name
 */

OrganizationDataWithIdAndActiveFlag.prototype['name'] = undefined;
/**
 * The number of people connected with the organization that is associated with the item
 * @member {Number} people_count
 */

OrganizationDataWithIdAndActiveFlag.prototype['people_count'] = undefined;
/**
 * The ID of the owner of the organization that is associated with the item
 * @member {Number} owner_id
 */

OrganizationDataWithIdAndActiveFlag.prototype['owner_id'] = undefined;
/**
 * The address of the organization
 * @member {String} address
 */

OrganizationDataWithIdAndActiveFlag.prototype['address'] = undefined;
/**
 * The BCC email of the organization associated with the item
 * @member {String} cc_email
 */

OrganizationDataWithIdAndActiveFlag.prototype['cc_email'] = undefined; // Implement OrganizationDataWithIdAndActiveFlagAllOf interface:

/**
 * Whether the associated organization is active or not
 * @member {Boolean} active_flag
 */

_OrganizationDataWithIdAndActiveFlagAllOf["default"].prototype['active_flag'] = undefined; // Implement OrganizationDataWithId interface:

/**
 * The ID of the organization associated with the item
 * @member {Number} id
 */

_OrganizationDataWithId["default"].prototype['id'] = undefined;
/**
 * The name of the organization associated with the item
 * @member {String} name
 */

_OrganizationDataWithId["default"].prototype['name'] = undefined;
/**
 * The number of people connected with the organization that is associated with the item
 * @member {Number} people_count
 */

_OrganizationDataWithId["default"].prototype['people_count'] = undefined;
/**
 * The ID of the owner of the organization that is associated with the item
 * @member {Number} owner_id
 */

_OrganizationDataWithId["default"].prototype['owner_id'] = undefined;
/**
 * The address of the organization
 * @member {String} address
 */

_OrganizationDataWithId["default"].prototype['address'] = undefined;
/**
 * The BCC email of the organization associated with the item
 * @member {String} cc_email
 */

_OrganizationDataWithId["default"].prototype['cc_email'] = undefined;
var _default = OrganizationDataWithIdAndActiveFlag;
exports["default"] = _default;
