// pipedrive/dist/model/OrganizationRelationshipGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationRelationshipsGetResponseAllOfRelatedObjects"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _OrganizationRelationshipGetResponseAllOf = _interopRequireDefault(require("./OrganizationRelationshipGetResponseAllOf"));

var _OrganizationRelationshipWithCalculatedFields = _interopRequireDefault(require("./OrganizationRelationshipWithCalculatedFields"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationRelationshipGetResponse model module.
 * @module model/OrganizationRelationshipGetResponse
 * @version 1.0.0
 */
var OrganizationRelationshipGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationRelationshipGetResponse</code>.
   * @alias module:model/OrganizationRelationshipGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/OrganizationRelationshipGetResponseAllOf
   */
  function OrganizationRelationshipGetResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationRelationshipGetResponse);

    _BaseResponse["default"].initialize(this);

    _OrganizationRelationshipGetResponseAllOf["default"].initialize(this);

    OrganizationRelationshipGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationRelationshipGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationRelationshipGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationRelationshipGetResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationRelationshipGetResponse} The populated <code>OrganizationRelationshipGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationRelationshipGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _OrganizationRelationshipGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _OrganizationRelationshipWithCalculatedFields["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationRelationshipsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationRelationshipGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


OrganizationRelationshipGetResponse.prototype['success'] = undefined;
/**
 * @member {module:model/OrganizationRelationshipWithCalculatedFields} data
 */

OrganizationRelationshipGetResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationRelationshipGetResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement OrganizationRelationshipGetResponseAllOf interface:

/**
 * @member {module:model/OrganizationRelationshipWithCalculatedFields} data
 */

_OrganizationRelationshipGetResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationRelationshipsGetResponseAllOfRelatedObjects} related_objects
 */

_OrganizationRelationshipGetResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = OrganizationRelationshipGetResponse;
exports["default"] = _default;
