// pipedrive/dist/model/OrganizationPostResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _BaseOrganizationItemWithEditNameFlag = _interopRequireDefault(require("./BaseOrganizationItemWithEditNameFlag"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _OrganizationPostResponseAllOf = _interopRequireDefault(require("./OrganizationPostResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationPostResponse model module.
 * @module model/OrganizationPostResponse
 * @version 1.0.0
 */
var OrganizationPostResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationPostResponse</code>.
   * @alias module:model/OrganizationPostResponse
   * @implements module:model/BaseResponse
   * @implements module:model/OrganizationPostResponseAllOf
   */
  function OrganizationPostResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationPostResponse);

    _BaseResponse["default"].initialize(this);

    _OrganizationPostResponseAllOf["default"].initialize(this);

    OrganizationPostResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationPostResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationPostResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationPostResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationPostResponse} The populated <code>OrganizationPostResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationPostResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _OrganizationPostResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BaseOrganizationItemWithEditNameFlag["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationPostResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


OrganizationPostResponse.prototype['success'] = undefined;
/**
 * @member {module:model/BaseOrganizationItemWithEditNameFlag} data
 */

OrganizationPostResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationPostResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement OrganizationPostResponseAllOf interface:

/**
 * @member {module:model/BaseOrganizationItemWithEditNameFlag} data
 */

_OrganizationPostResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

_OrganizationPostResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = OrganizationPostResponse;
exports["default"] = _default;
