// pipedrive/dist/model/OrganizationUpdateResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _BaseOrganizationItem = _interopRequireDefault(require("./BaseOrganizationItem"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _OrganizationUpdateResponseAllOf = _interopRequireDefault(require("./OrganizationUpdateResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationUpdateResponse model module.
 * @module model/OrganizationUpdateResponse
 * @version 1.0.0
 */
var OrganizationUpdateResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationUpdateResponse</code>.
   * @alias module:model/OrganizationUpdateResponse
   * @implements module:model/BaseResponse
   * @implements module:model/OrganizationUpdateResponseAllOf
   */
  function OrganizationUpdateResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationUpdateResponse);

    _BaseResponse["default"].initialize(this);

    _OrganizationUpdateResponseAllOf["default"].initialize(this);

    OrganizationUpdateResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationUpdateResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationUpdateResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationUpdateResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationUpdateResponse} The populated <code>OrganizationUpdateResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationUpdateResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _OrganizationUpdateResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BaseOrganizationItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationUpdateResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


OrganizationUpdateResponse.prototype['success'] = undefined;
/**
 * @member {module:model/BaseOrganizationItem} data
 */

OrganizationUpdateResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationUpdateResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement OrganizationUpdateResponseAllOf interface:

/**
 * @member {module:model/BaseOrganizationItem} data
 */

_OrganizationUpdateResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

_OrganizationUpdateResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = OrganizationUpdateResponse;
exports["default"] = _default;
