// pipedrive/dist/model/OrganizationUpdateResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _BaseOrganizationItem = _interopRequireDefault(require("./BaseOrganizationItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationUpdateResponseAllOf model module.
 * @module model/OrganizationUpdateResponseAllOf
 * @version 1.0.0
 */
var OrganizationUpdateResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationUpdateResponseAllOf</code>.
   * @alias module:model/OrganizationUpdateResponseAllOf
   */
  function OrganizationUpdateResponseAllOf() {
    (0, _classCallCheck2["default"])(this, OrganizationUpdateResponseAllOf);
    OrganizationUpdateResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationUpdateResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationUpdateResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationUpdateResponseAllOf} obj Optional instance to populate.
     * @return {module:model/OrganizationUpdateResponseAllOf} The populated <code>OrganizationUpdateResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationUpdateResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BaseOrganizationItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationUpdateResponseAllOf;
}();
/**
 * @member {module:model/BaseOrganizationItem} data
 */


OrganizationUpdateResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationUpdateResponseAllOf.prototype['related_objects'] = undefined;
var _default = OrganizationUpdateResponseAllOf;
exports["default"] = _default;
