// pipedrive/dist/model/OrganizationCountAndAddressInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OrganizationAddressInfo = _interopRequireDefault(require("./OrganizationAddressInfo"));

var _OrganizationCountInfo = _interopRequireDefault(require("./OrganizationCountInfo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationCountAndAddressInfo model module.
 * @module model/OrganizationCountAndAddressInfo
 * @version 1.0.0
 */
var OrganizationCountAndAddressInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationCountAndAddressInfo</code>.
   * @alias module:model/OrganizationCountAndAddressInfo
   * @implements module:model/OrganizationCountInfo
   * @implements module:model/OrganizationAddressInfo
   */
  function OrganizationCountAndAddressInfo() {
    (0, _classCallCheck2["default"])(this, OrganizationCountAndAddressInfo);

    _OrganizationCountInfo["default"].initialize(this);

    _OrganizationAddressInfo["default"].initialize(this);

    OrganizationCountAndAddressInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationCountAndAddressInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationCountAndAddressInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationCountAndAddressInfo} obj Optional instance to populate.
     * @return {module:model/OrganizationCountAndAddressInfo} The populated <code>OrganizationCountAndAddressInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationCountAndAddressInfo();

        _OrganizationCountInfo["default"].constructFromObject(data, obj);

        _OrganizationAddressInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('people_count')) {
          obj['people_count'] = _ApiClient["default"].convertToType(data['people_count'], 'Number');
          delete data['people_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (data.hasOwnProperty('address_subpremise')) {
          obj['address_subpremise'] = _ApiClient["default"].convertToType(data['address_subpremise'], 'String');
          delete data['address_subpremise'];
        }

        if (data.hasOwnProperty('address_street_number')) {
          obj['address_street_number'] = _ApiClient["default"].convertToType(data['address_street_number'], 'String');
          delete data['address_street_number'];
        }

        if (data.hasOwnProperty('address_route')) {
          obj['address_route'] = _ApiClient["default"].convertToType(data['address_route'], 'String');
          delete data['address_route'];
        }

        if (data.hasOwnProperty('address_sublocality')) {
          obj['address_sublocality'] = _ApiClient["default"].convertToType(data['address_sublocality'], 'String');
          delete data['address_sublocality'];
        }

        if (data.hasOwnProperty('address_locality')) {
          obj['address_locality'] = _ApiClient["default"].convertToType(data['address_locality'], 'String');
          delete data['address_locality'];
        }

        if (data.hasOwnProperty('address_admin_area_level_1')) {
          obj['address_admin_area_level_1'] = _ApiClient["default"].convertToType(data['address_admin_area_level_1'], 'String');
          delete data['address_admin_area_level_1'];
        }

        if (data.hasOwnProperty('address_admin_area_level_2')) {
          obj['address_admin_area_level_2'] = _ApiClient["default"].convertToType(data['address_admin_area_level_2'], 'String');
          delete data['address_admin_area_level_2'];
        }

        if (data.hasOwnProperty('address_country')) {
          obj['address_country'] = _ApiClient["default"].convertToType(data['address_country'], 'String');
          delete data['address_country'];
        }

        if (data.hasOwnProperty('address_postal_code')) {
          obj['address_postal_code'] = _ApiClient["default"].convertToType(data['address_postal_code'], 'String');
          delete data['address_postal_code'];
        }

        if (data.hasOwnProperty('address_formatted_address')) {
          obj['address_formatted_address'] = _ApiClient["default"].convertToType(data['address_formatted_address'], 'String');
          delete data['address_formatted_address'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationCountAndAddressInfo;
}();
/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */


OrganizationCountAndAddressInfo.prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

OrganizationCountAndAddressInfo.prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

OrganizationCountAndAddressInfo.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

OrganizationCountAndAddressInfo.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

OrganizationCountAndAddressInfo.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

OrganizationCountAndAddressInfo.prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

OrganizationCountAndAddressInfo.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

OrganizationCountAndAddressInfo.prototype['followers_count'] = undefined;
/**
 * The full address of the organization
 * @member {String} address
 */

OrganizationCountAndAddressInfo.prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

OrganizationCountAndAddressInfo.prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

OrganizationCountAndAddressInfo.prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

OrganizationCountAndAddressInfo.prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

OrganizationCountAndAddressInfo.prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

OrganizationCountAndAddressInfo.prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

OrganizationCountAndAddressInfo.prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

OrganizationCountAndAddressInfo.prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

OrganizationCountAndAddressInfo.prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

OrganizationCountAndAddressInfo.prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

OrganizationCountAndAddressInfo.prototype['address_formatted_address'] = undefined; // Implement OrganizationCountInfo interface:

/**
 * The count of email messages related to the organization
 * @member {Number} email_messages_count
 */

_OrganizationCountInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of persons related to the organization
 * @member {Number} people_count
 */

_OrganizationCountInfo["default"].prototype['people_count'] = undefined;
/**
 * The count of activities related to the organization
 * @member {Number} activities_count
 */

_OrganizationCountInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the organization
 * @member {Number} done_activities_count
 */

_OrganizationCountInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the organization
 * @member {Number} undone_activities_count
 */

_OrganizationCountInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the organization
 * @member {Number} files_count
 */

_OrganizationCountInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the organization
 * @member {Number} notes_count
 */

_OrganizationCountInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the organization
 * @member {Number} followers_count
 */

_OrganizationCountInfo["default"].prototype['followers_count'] = undefined; // Implement OrganizationAddressInfo interface:

/**
 * The full address of the organization
 * @member {String} address
 */

_OrganizationAddressInfo["default"].prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */

_OrganizationAddressInfo["default"].prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */

_OrganizationAddressInfo["default"].prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */

_OrganizationAddressInfo["default"].prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */

_OrganizationAddressInfo["default"].prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */

_OrganizationAddressInfo["default"].prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */

_OrganizationAddressInfo["default"].prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */

_OrganizationAddressInfo["default"].prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */

_OrganizationAddressInfo["default"].prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */

_OrganizationAddressInfo["default"].prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */

_OrganizationAddressInfo["default"].prototype['address_formatted_address'] = undefined;
var _default = OrganizationCountAndAddressInfo;
exports["default"] = _default;
