// pipedrive/dist/model/OrganizationSearchResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _OrganizationSearchResponseAllOf = _interopRequireDefault(require("./OrganizationSearchResponseAllOf"));

var _OrganizationSearchResponseAllOfData = _interopRequireDefault(require("./OrganizationSearchResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationSearchResponse model module.
 * @module model/OrganizationSearchResponse
 * @version 1.0.0
 */
var OrganizationSearchResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationSearchResponse</code>.
   * @alias module:model/OrganizationSearchResponse
   * @implements module:model/BaseResponse
   * @implements module:model/OrganizationSearchResponseAllOf
   */
  function OrganizationSearchResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationSearchResponse);

    _BaseResponse["default"].initialize(this);

    _OrganizationSearchResponseAllOf["default"].initialize(this);

    OrganizationSearchResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationSearchResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationSearchResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationSearchResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationSearchResponse} The populated <code>OrganizationSearchResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationSearchResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _OrganizationSearchResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _OrganizationSearchResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationSearchResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


OrganizationSearchResponse.prototype['success'] = undefined;
/**
 * @member {module:model/OrganizationSearchResponseAllOfData} data
 */

OrganizationSearchResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

OrganizationSearchResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement OrganizationSearchResponseAllOf interface:

/**
 * @member {module:model/OrganizationSearchResponseAllOfData} data
 */

_OrganizationSearchResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_OrganizationSearchResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = OrganizationSearchResponse;
exports["default"] = _default;
