// pipedrive/dist/model/OrganizationFollowersListResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _OrganizationFollowerItem = _interopRequireDefault(require("./OrganizationFollowerItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationFollowersListResponse model module.
 * @module model/OrganizationFollowersListResponse
 * @version 1.0.0
 */
var OrganizationFollowersListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationFollowersListResponse</code>.
   * @alias module:model/OrganizationFollowersListResponse
   */
  function OrganizationFollowersListResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationFollowersListResponse);
    OrganizationFollowersListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationFollowersListResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationFollowersListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationFollowersListResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationFollowersListResponse} The populated <code>OrganizationFollowersListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationFollowersListResponse();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_OrganizationFollowerItem["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationFollowersListResponse;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


OrganizationFollowersListResponse.prototype['success'] = undefined;
/**
 * The array of followers
 * @member {Array.<module:model/OrganizationFollowerItem>} data
 */

OrganizationFollowersListResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

OrganizationFollowersListResponse.prototype['additional_data'] = undefined;
var _default = OrganizationFollowersListResponse;
exports["default"] = _default;
