// pipedrive/dist/model/OrganizationDetailsGetResponseAllOfAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RelatedFollowerData = _interopRequireDefault(require("./RelatedFollowerData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationDetailsGetResponseAllOfAdditionalData model module.
 * @module model/OrganizationDetailsGetResponseAllOfAdditionalData
 * @version 1.0.0
 */
var OrganizationDetailsGetResponseAllOfAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationDetailsGetResponseAllOfAdditionalData</code>.
   * @alias module:model/OrganizationDetailsGetResponseAllOfAdditionalData
   */
  function OrganizationDetailsGetResponseAllOfAdditionalData() {
    (0, _classCallCheck2["default"])(this, OrganizationDetailsGetResponseAllOfAdditionalData);
    OrganizationDetailsGetResponseAllOfAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationDetailsGetResponseAllOfAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationDetailsGetResponseAllOfAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationDetailsGetResponseAllOfAdditionalData} obj Optional instance to populate.
     * @return {module:model/OrganizationDetailsGetResponseAllOfAdditionalData} The populated <code>OrganizationDetailsGetResponseAllOfAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationDetailsGetResponseAllOfAdditionalData();

        if (data.hasOwnProperty('followers')) {
          obj['followers'] = _RelatedFollowerData["default"].constructFromObject(data['followers']);
          delete data['followers'];
        }

        if (data.hasOwnProperty('dropbox_email')) {
          obj['dropbox_email'] = _ApiClient["default"].convertToType(data['dropbox_email'], 'String');
          delete data['dropbox_email'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationDetailsGetResponseAllOfAdditionalData;
}();
/**
 * @member {module:model/RelatedFollowerData} followers
 */


OrganizationDetailsGetResponseAllOfAdditionalData.prototype['followers'] = undefined;
/**
 * Dropbox email for the organization
 * @member {String} dropbox_email
 */

OrganizationDetailsGetResponseAllOfAdditionalData.prototype['dropbox_email'] = undefined;
var _default = OrganizationDetailsGetResponseAllOfAdditionalData;
exports["default"] = _default;
