// pipedrive/dist/model/OrganizationDetailsGetResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AllOrganizationsGetResponseAllOfRelatedObjects = _interopRequireDefault(require("./AllOrganizationsGetResponseAllOfRelatedObjects"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _OrganizationDetailsGetResponseAllOf = _interopRequireDefault(require("./OrganizationDetailsGetResponseAllOf"));

var _OrganizationDetailsGetResponseAllOfAdditionalData = _interopRequireDefault(require("./OrganizationDetailsGetResponseAllOfAdditionalData"));

var _OrganizationItem = _interopRequireDefault(require("./OrganizationItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The OrganizationDetailsGetResponse model module.
 * @module model/OrganizationDetailsGetResponse
 * @version 1.0.0
 */
var OrganizationDetailsGetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationDetailsGetResponse</code>.
   * @alias module:model/OrganizationDetailsGetResponse
   * @implements module:model/BaseResponse
   * @implements module:model/OrganizationDetailsGetResponseAllOf
   */
  function OrganizationDetailsGetResponse() {
    (0, _classCallCheck2["default"])(this, OrganizationDetailsGetResponse);

    _BaseResponse["default"].initialize(this);

    _OrganizationDetailsGetResponseAllOf["default"].initialize(this);

    OrganizationDetailsGetResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(OrganizationDetailsGetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OrganizationDetailsGetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationDetailsGetResponse} obj Optional instance to populate.
     * @return {module:model/OrganizationDetailsGetResponse} The populated <code>OrganizationDetailsGetResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationDetailsGetResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _OrganizationDetailsGetResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _OrganizationItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _OrganizationDetailsGetResponseAllOfAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AllOrganizationsGetResponseAllOfRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return OrganizationDetailsGetResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


OrganizationDetailsGetResponse.prototype['success'] = undefined;
/**
 * @member {module:model/OrganizationItem} data
 */

OrganizationDetailsGetResponse.prototype['data'] = undefined;
/**
 * @member {module:model/OrganizationDetailsGetResponseAllOfAdditionalData} additional_data
 */

OrganizationDetailsGetResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

OrganizationDetailsGetResponse.prototype['related_objects'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement OrganizationDetailsGetResponseAllOf interface:

/**
 * @member {module:model/OrganizationItem} data
 */

_OrganizationDetailsGetResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/OrganizationDetailsGetResponseAllOfAdditionalData} additional_data
 */

_OrganizationDetailsGetResponseAllOf["default"].prototype['additional_data'] = undefined;
/**
 * @member {module:model/AllOrganizationsGetResponseAllOfRelatedObjects} related_objects
 */

_OrganizationDetailsGetResponseAllOf["default"].prototype['related_objects'] = undefined;
var _default = OrganizationDetailsGetResponse;
exports["default"] = _default;
