// pipedrive/dist/model/PaginationDetailsAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PaginationDetailsAllOf model module.
 * @module model/PaginationDetailsAllOf
 * @version 1.0.0
 */
var PaginationDetailsAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PaginationDetailsAllOf</code>.
   * @alias module:model/PaginationDetailsAllOf
   */
  function PaginationDetailsAllOf() {
    (0, _classCallCheck2["default"])(this, PaginationDetailsAllOf);
    PaginationDetailsAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PaginationDetailsAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PaginationDetailsAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PaginationDetailsAllOf} obj Optional instance to populate.
     * @return {module:model/PaginationDetailsAllOf} The populated <code>PaginationDetailsAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaginationDetailsAllOf();

        if (data.hasOwnProperty('next_start')) {
          obj['next_start'] = _ApiClient["default"].convertToType(data['next_start'], 'Number');
          delete data['next_start'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PaginationDetailsAllOf;
}();
/**
 * Next pagination start
 * @member {Number} next_start
 */


PaginationDetailsAllOf.prototype['next_start'] = undefined;
var _default = PaginationDetailsAllOf;
exports["default"] = _default;
