// pipedrive/dist/model/PaginationDetails.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _PaginationDetailsAllOf = _interopRequireDefault(require("./PaginationDetailsAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PaginationDetails model module.
 * @module model/PaginationDetails
 * @version 1.0.0
 */
var PaginationDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PaginationDetails</code>.
   * Pagination details of the list
   * @alias module:model/PaginationDetails
   * @implements module:model/AdditionalData
   * @implements module:model/PaginationDetailsAllOf
   */
  function PaginationDetails() {
    (0, _classCallCheck2["default"])(this, PaginationDetails);

    _AdditionalData["default"].initialize(this);

    _PaginationDetailsAllOf["default"].initialize(this);

    PaginationDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PaginationDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PaginationDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PaginationDetails} obj Optional instance to populate.
     * @return {module:model/PaginationDetails} The populated <code>PaginationDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaginationDetails();

        _AdditionalData["default"].constructFromObject(data, obj);

        _PaginationDetailsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('start')) {
          obj['start'] = _ApiClient["default"].convertToType(data['start'], 'Number');
          delete data['start'];
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
          delete data['limit'];
        }

        if (data.hasOwnProperty('more_items_in_collection')) {
          obj['more_items_in_collection'] = _ApiClient["default"].convertToType(data['more_items_in_collection'], 'Boolean');
          delete data['more_items_in_collection'];
        }

        if (data.hasOwnProperty('next_start')) {
          obj['next_start'] = _ApiClient["default"].convertToType(data['next_start'], 'Number');
          delete data['next_start'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PaginationDetails;
}();
/**
 * Pagination start
 * @member {Number} start
 */


PaginationDetails.prototype['start'] = undefined;
/**
 * Items shown per page
 * @member {Number} limit
 */

PaginationDetails.prototype['limit'] = undefined;
/**
 * If there are more list items in the collection than displayed or not
 * @member {Boolean} more_items_in_collection
 */

PaginationDetails.prototype['more_items_in_collection'] = undefined;
/**
 * Next pagination start
 * @member {Number} next_start
 */

PaginationDetails.prototype['next_start'] = undefined; // Implement AdditionalData interface:

/**
 * Pagination start
 * @member {Number} start
 */

_AdditionalData["default"].prototype['start'] = undefined;
/**
 * Items shown per page
 * @member {Number} limit
 */

_AdditionalData["default"].prototype['limit'] = undefined;
/**
 * If there are more list items in the collection than displayed or not
 * @member {Boolean} more_items_in_collection
 */

_AdditionalData["default"].prototype['more_items_in_collection'] = undefined; // Implement PaginationDetailsAllOf interface:

/**
 * Next pagination start
 * @member {Number} next_start
 */

_PaginationDetailsAllOf["default"].prototype['next_start'] = undefined;
var _default = PaginationDetails;
exports["default"] = _default;
