// pipedrive/dist/model/Params.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The Params model module.
 * @module model/Params
 * @version 1.0.0
 */
var Params = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Params</code>.
   * The parameters that accompany the goal type
   * @alias module:model/Params
   */
  function Params() {
    (0, _classCallCheck2["default"])(this, Params);
    Params.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(Params, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Params</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Params} obj Optional instance to populate.
     * @return {module:model/Params} The populated <code>Params</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Params();

        if (data.hasOwnProperty('pipeline_id')) {
          obj['pipeline_id'] = _ApiClient["default"].convertToType(data['pipeline_id'], 'String');
          delete data['pipeline_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return Params;
}();
/**
 * The pipeline ID of the goal
 * @member {String} pipeline_id
 */


Params.prototype['pipeline_id'] = undefined;
var _default = Params;
exports["default"] = _default;
