// pipedrive/dist/model/PaymentsResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PaymentsResponseAllOfData = _interopRequireDefault(require("./PaymentsResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PaymentsResponseAllOf model module.
 * @module model/PaymentsResponseAllOf
 * @version 1.0.0
 */
var PaymentsResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PaymentsResponseAllOf</code>.
   * @alias module:model/PaymentsResponseAllOf
   */
  function PaymentsResponseAllOf() {
    (0, _classCallCheck2["default"])(this, PaymentsResponseAllOf);
    PaymentsResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PaymentsResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PaymentsResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PaymentsResponseAllOf} obj Optional instance to populate.
     * @return {module:model/PaymentsResponseAllOf} The populated <code>PaymentsResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaymentsResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _PaymentsResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PaymentsResponseAllOf;
}();
/**
 * @member {module:model/PaymentsResponseAllOfData} data
 */


PaymentsResponseAllOf.prototype['data'] = undefined;
var _default = PaymentsResponseAllOf;
exports["default"] = _default;
