// pipedrive/dist/model/PermissionSets.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _PermissionSetsAllOf = _interopRequireDefault(require("./PermissionSetsAllOf"));

var _PermissionSetsItem = _interopRequireDefault(require("./PermissionSetsItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PermissionSets model module.
 * @module model/PermissionSets
 * @version 1.0.0
 */
var PermissionSets = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PermissionSets</code>.
   * @alias module:model/PermissionSets
   * @implements module:model/BaseResponse
   * @implements module:model/PermissionSetsAllOf
   */
  function PermissionSets() {
    (0, _classCallCheck2["default"])(this, PermissionSets);

    _BaseResponse["default"].initialize(this);

    _PermissionSetsAllOf["default"].initialize(this);

    PermissionSets.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PermissionSets, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PermissionSets</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PermissionSets} obj Optional instance to populate.
     * @return {module:model/PermissionSets} The populated <code>PermissionSets</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PermissionSets();

        _BaseResponse["default"].constructFromObject(data, obj);

        _PermissionSetsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_PermissionSetsItem["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PermissionSets;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


PermissionSets.prototype['success'] = undefined;
/**
 * The array of permission set
 * @member {Array.<module:model/PermissionSetsItem>} data
 */

PermissionSets.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement PermissionSetsAllOf interface:

/**
 * The array of permission set
 * @member {Array.<module:model/PermissionSetsItem>} data
 */

_PermissionSetsAllOf["default"].prototype['data'] = undefined;
var _default = PermissionSets;
exports["default"] = _default;
