// pipedrive/dist/model/PersonSearchItemItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonSearchItemItemOrganization = _interopRequireDefault(require("./PersonSearchItemItemOrganization"));

var _PersonSearchItemItemOwner = _interopRequireDefault(require("./PersonSearchItemItemOwner"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonSearchItemItem model module.
 * @module model/PersonSearchItemItem
 * @version 1.0.0
 */
var PersonSearchItemItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonSearchItemItem</code>.
   * @alias module:model/PersonSearchItemItem
   */
  function PersonSearchItemItem() {
    (0, _classCallCheck2["default"])(this, PersonSearchItemItem);
    PersonSearchItemItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonSearchItemItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonSearchItemItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonSearchItemItem} obj Optional instance to populate.
     * @return {module:model/PersonSearchItemItem} The populated <code>PersonSearchItemItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonSearchItemItem();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('phones')) {
          obj['phones'] = _ApiClient["default"].convertToType(data['phones'], ['String']);
          delete data['phones'];
        }

        if (data.hasOwnProperty('emails')) {
          obj['emails'] = _ApiClient["default"].convertToType(data['emails'], ['String']);
          delete data['emails'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'Number');
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _PersonSearchItemItemOwner["default"].constructFromObject(data['owner']);
          delete data['owner'];
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _PersonSearchItemItemOrganization["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('custom_fields')) {
          obj['custom_fields'] = _ApiClient["default"].convertToType(data['custom_fields'], 'Number');
          delete data['custom_fields'];
        }

        if (data.hasOwnProperty('notes')) {
          obj['notes'] = _ApiClient["default"].convertToType(data['notes'], ['String']);
          delete data['notes'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonSearchItemItem;
}();
/**
 * The ID of the person
 * @member {Number} id
 */


PersonSearchItemItem.prototype['id'] = undefined;
/**
 * The type of the item
 * @member {String} type
 */

PersonSearchItemItem.prototype['type'] = undefined;
/**
 * The name of the person
 * @member {String} name
 */

PersonSearchItemItem.prototype['name'] = undefined;
/**
 * An array of phone numbers
 * @member {Array.<String>} phones
 */

PersonSearchItemItem.prototype['phones'] = undefined;
/**
 * An array of email addresses
 * @member {Array.<String>} emails
 */

PersonSearchItemItem.prototype['emails'] = undefined;
/**
 * The visibility of the person
 * @member {Number} visible_to
 */

PersonSearchItemItem.prototype['visible_to'] = undefined;
/**
 * @member {module:model/PersonSearchItemItemOwner} owner
 */

PersonSearchItemItem.prototype['owner'] = undefined;
/**
 * @member {module:model/PersonSearchItemItemOrganization} organization
 */

PersonSearchItemItem.prototype['organization'] = undefined;
/**
 * Custom fields
 * @member {Number} custom_fields
 */

PersonSearchItemItem.prototype['custom_fields'] = undefined;
/**
 * An array of notes
 * @member {Array.<String>} notes
 */

PersonSearchItemItem.prototype['notes'] = undefined;
var _default = PersonSearchItemItem;
exports["default"] = _default;
