// pipedrive/dist/model/PersonCountEmailDealAndActivityInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealCountAndActivityInfo = _interopRequireDefault(require("./DealCountAndActivityInfo"));

var _PersonCountAndEmailInfo = _interopRequireDefault(require("./PersonCountAndEmailInfo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonCountEmailDealAndActivityInfo model module.
 * @module model/PersonCountEmailDealAndActivityInfo
 * @version 1.0.0
 */
var PersonCountEmailDealAndActivityInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonCountEmailDealAndActivityInfo</code>.
   * @alias module:model/PersonCountEmailDealAndActivityInfo
   * @implements module:model/PersonCountAndEmailInfo
   * @implements module:model/DealCountAndActivityInfo
   */
  function PersonCountEmailDealAndActivityInfo() {
    (0, _classCallCheck2["default"])(this, PersonCountEmailDealAndActivityInfo);

    _PersonCountAndEmailInfo["default"].initialize(this);

    _DealCountAndActivityInfo["default"].initialize(this);

    PersonCountEmailDealAndActivityInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonCountEmailDealAndActivityInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonCountEmailDealAndActivityInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonCountEmailDealAndActivityInfo} obj Optional instance to populate.
     * @return {module:model/PersonCountEmailDealAndActivityInfo} The populated <code>PersonCountEmailDealAndActivityInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonCountEmailDealAndActivityInfo();

        _PersonCountAndEmailInfo["default"].constructFromObject(data, obj);

        _DealCountAndActivityInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('last_incoming_mail_time')) {
          obj['last_incoming_mail_time'] = _ApiClient["default"].convertToType(data['last_incoming_mail_time'], 'String');
          delete data['last_incoming_mail_time'];
        }

        if (data.hasOwnProperty('last_outgoing_mail_time')) {
          obj['last_outgoing_mail_time'] = _ApiClient["default"].convertToType(data['last_outgoing_mail_time'], 'String');
          delete data['last_outgoing_mail_time'];
        }

        if (data.hasOwnProperty('open_deals_count')) {
          obj['open_deals_count'] = _ApiClient["default"].convertToType(data['open_deals_count'], 'Number');
          delete data['open_deals_count'];
        }

        if (data.hasOwnProperty('related_open_deals_count')) {
          obj['related_open_deals_count'] = _ApiClient["default"].convertToType(data['related_open_deals_count'], 'Number');
          delete data['related_open_deals_count'];
        }

        if (data.hasOwnProperty('closed_deals_count')) {
          obj['closed_deals_count'] = _ApiClient["default"].convertToType(data['closed_deals_count'], 'Number');
          delete data['closed_deals_count'];
        }

        if (data.hasOwnProperty('related_closed_deals_count')) {
          obj['related_closed_deals_count'] = _ApiClient["default"].convertToType(data['related_closed_deals_count'], 'Number');
          delete data['related_closed_deals_count'];
        }

        if (data.hasOwnProperty('won_deals_count')) {
          obj['won_deals_count'] = _ApiClient["default"].convertToType(data['won_deals_count'], 'Number');
          delete data['won_deals_count'];
        }

        if (data.hasOwnProperty('related_won_deals_count')) {
          obj['related_won_deals_count'] = _ApiClient["default"].convertToType(data['related_won_deals_count'], 'Number');
          delete data['related_won_deals_count'];
        }

        if (data.hasOwnProperty('lost_deals_count')) {
          obj['lost_deals_count'] = _ApiClient["default"].convertToType(data['lost_deals_count'], 'Number');
          delete data['lost_deals_count'];
        }

        if (data.hasOwnProperty('related_lost_deals_count')) {
          obj['related_lost_deals_count'] = _ApiClient["default"].convertToType(data['related_lost_deals_count'], 'Number');
          delete data['related_lost_deals_count'];
        }

        if (data.hasOwnProperty('next_activity_date')) {
          obj['next_activity_date'] = _ApiClient["default"].convertToType(data['next_activity_date'], 'String');
          delete data['next_activity_date'];
        }

        if (data.hasOwnProperty('next_activity_time')) {
          obj['next_activity_time'] = _ApiClient["default"].convertToType(data['next_activity_time'], 'String');
          delete data['next_activity_time'];
        }

        if (data.hasOwnProperty('next_activity_id')) {
          obj['next_activity_id'] = _ApiClient["default"].convertToType(data['next_activity_id'], 'Number');
          delete data['next_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_id')) {
          obj['last_activity_id'] = _ApiClient["default"].convertToType(data['last_activity_id'], 'Number');
          delete data['last_activity_id'];
        }

        if (data.hasOwnProperty('last_activity_date')) {
          obj['last_activity_date'] = _ApiClient["default"].convertToType(data['last_activity_date'], 'String');
          delete data['last_activity_date'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonCountEmailDealAndActivityInfo;
}();
/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */


PersonCountEmailDealAndActivityInfo.prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

PersonCountEmailDealAndActivityInfo.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

PersonCountEmailDealAndActivityInfo.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

PersonCountEmailDealAndActivityInfo.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

PersonCountEmailDealAndActivityInfo.prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

PersonCountEmailDealAndActivityInfo.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

PersonCountEmailDealAndActivityInfo.prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

PersonCountEmailDealAndActivityInfo.prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

PersonCountEmailDealAndActivityInfo.prototype['last_outgoing_mail_time'] = undefined;
/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

PersonCountEmailDealAndActivityInfo.prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

PersonCountEmailDealAndActivityInfo.prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

PersonCountEmailDealAndActivityInfo.prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

PersonCountEmailDealAndActivityInfo.prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

PersonCountEmailDealAndActivityInfo.prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

PersonCountEmailDealAndActivityInfo.prototype['last_activity_date'] = undefined; // Implement PersonCountAndEmailInfo interface:

/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

_PersonCountAndEmailInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

_PersonCountAndEmailInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

_PersonCountAndEmailInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

_PersonCountAndEmailInfo["default"].prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

_PersonCountAndEmailInfo["default"].prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

_PersonCountAndEmailInfo["default"].prototype['last_outgoing_mail_time'] = undefined; // Implement DealCountAndActivityInfo interface:

/**
 * The count of open deals related with the item
 * @member {Number} open_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['open_deals_count'] = undefined;
/**
 * The count of related open deals related with the item
 * @member {Number} related_open_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['related_open_deals_count'] = undefined;
/**
 * The count of closed deals related with the item
 * @member {Number} closed_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['closed_deals_count'] = undefined;
/**
 * The count of related closed deals related with the item
 * @member {Number} related_closed_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['related_closed_deals_count'] = undefined;
/**
 * The count of won deals related with the item
 * @member {Number} won_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['won_deals_count'] = undefined;
/**
 * The count of related won deals related with the item
 * @member {Number} related_won_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['related_won_deals_count'] = undefined;
/**
 * The count of lost deals related with the item
 * @member {Number} lost_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['lost_deals_count'] = undefined;
/**
 * The count of related lost deals related with the item
 * @member {Number} related_lost_deals_count
 */

_DealCountAndActivityInfo["default"].prototype['related_lost_deals_count'] = undefined;
/**
 * The date of the next activity associated with the deal
 * @member {String} next_activity_date
 */

_DealCountAndActivityInfo["default"].prototype['next_activity_date'] = undefined;
/**
 * The time of the next activity associated with the deal
 * @member {String} next_activity_time
 */

_DealCountAndActivityInfo["default"].prototype['next_activity_time'] = undefined;
/**
 * The ID of the next activity associated with the deal
 * @member {Number} next_activity_id
 */

_DealCountAndActivityInfo["default"].prototype['next_activity_id'] = undefined;
/**
 * The ID of the last activity associated with the deal
 * @member {Number} last_activity_id
 */

_DealCountAndActivityInfo["default"].prototype['last_activity_id'] = undefined;
/**
 * The date of the last activity associated with the deal
 * @member {String} last_activity_date
 */

_DealCountAndActivityInfo["default"].prototype['last_activity_date'] = undefined;
var _default = PersonCountEmailDealAndActivityInfo;
exports["default"] = _default;
