// pipedrive/dist/model/PersonData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonDataEmail = _interopRequireDefault(require("./PersonDataEmail"));

var _PersonDataPhone = _interopRequireDefault(require("./PersonDataPhone"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonData model module.
 * @module model/PersonData
 * @version 1.0.0
 */
var PersonData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonData</code>.
   * @alias module:model/PersonData
   */
  function PersonData() {
    (0, _classCallCheck2["default"])(this, PersonData);
    PersonData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonData} obj Optional instance to populate.
     * @return {module:model/PersonData} The populated <code>PersonData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], [_PersonDataEmail["default"]]);
          delete data['email'];
        }

        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], [_PersonDataPhone["default"]]);
          delete data['phone'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonData;
}();
/**
 * The ID of the person associated with the item
 * @member {Number} id
 */


PersonData.prototype['id'] = undefined;
/**
 * The name of the person associated with the item
 * @member {String} name
 */

PersonData.prototype['name'] = undefined;
/**
 * The emails of the person associated with the item
 * @member {Array.<module:model/PersonDataEmail>} email
 */

PersonData.prototype['email'] = undefined;
/**
 * The phone numbers of the person associated with the item
 * @member {Array.<module:model/PersonDataPhone>} phone
 */

PersonData.prototype['phone'] = undefined;
/**
 * The ID of the owner of the person that is associated with the item
 * @member {Number} owner_id
 */

PersonData.prototype['owner_id'] = undefined;
var _default = PersonData;
exports["default"] = _default;
