// pipedrive/dist/model/PersonNameInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonNameInfo model module.
 * @module model/PersonNameInfo
 * @version 1.0.0
 */
var PersonNameInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonNameInfo</code>.
   * @alias module:model/PersonNameInfo
   */
  function PersonNameInfo() {
    (0, _classCallCheck2["default"])(this, PersonNameInfo);
    PersonNameInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonNameInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonNameInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonNameInfo} obj Optional instance to populate.
     * @return {module:model/PersonNameInfo} The populated <code>PersonNameInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonNameInfo();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
          delete data['first_name'];
        }

        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
          delete data['last_name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonNameInfo;
}();
/**
 * The name of the person
 * @member {String} name
 */


PersonNameInfo.prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

PersonNameInfo.prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

PersonNameInfo.prototype['last_name'] = undefined;
var _default = PersonNameInfo;
exports["default"] = _default;
