// pipedrive/dist/model/PersonSearchItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonSearchItemItem = _interopRequireDefault(require("./PersonSearchItemItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonSearchItem model module.
 * @module model/PersonSearchItem
 * @version 1.0.0
 */
var PersonSearchItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonSearchItem</code>.
   * @alias module:model/PersonSearchItem
   */
  function PersonSearchItem() {
    (0, _classCallCheck2["default"])(this, PersonSearchItem);
    PersonSearchItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonSearchItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonSearchItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonSearchItem} obj Optional instance to populate.
     * @return {module:model/PersonSearchItem} The populated <code>PersonSearchItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonSearchItem();

        if (data.hasOwnProperty('result_score')) {
          obj['result_score'] = _ApiClient["default"].convertToType(data['result_score'], 'Number');
          delete data['result_score'];
        }

        if (data.hasOwnProperty('item')) {
          obj['item'] = _PersonSearchItemItem["default"].constructFromObject(data['item']);
          delete data['item'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonSearchItem;
}();
/**
 * Search result relevancy
 * @member {Number} result_score
 */


PersonSearchItem.prototype['result_score'] = undefined;
/**
 * @member {module:model/PersonSearchItemItem} item
 */

PersonSearchItem.prototype['item'] = undefined;
var _default = PersonSearchItem;
exports["default"] = _default;
