// pipedrive/dist/model/PersonNameCountAndEmailInfo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonCountAndEmailInfo = _interopRequireDefault(require("./PersonCountAndEmailInfo"));

var _PersonNameInfo = _interopRequireDefault(require("./PersonNameInfo"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonNameCountAndEmailInfo model module.
 * @module model/PersonNameCountAndEmailInfo
 * @version 1.0.0
 */
var PersonNameCountAndEmailInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonNameCountAndEmailInfo</code>.
   * @alias module:model/PersonNameCountAndEmailInfo
   * @implements module:model/PersonNameInfo
   * @implements module:model/PersonCountAndEmailInfo
   */
  function PersonNameCountAndEmailInfo() {
    (0, _classCallCheck2["default"])(this, PersonNameCountAndEmailInfo);

    _PersonNameInfo["default"].initialize(this);

    _PersonCountAndEmailInfo["default"].initialize(this);

    PersonNameCountAndEmailInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonNameCountAndEmailInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonNameCountAndEmailInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonNameCountAndEmailInfo} obj Optional instance to populate.
     * @return {module:model/PersonNameCountAndEmailInfo} The populated <code>PersonNameCountAndEmailInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonNameCountAndEmailInfo();

        _PersonNameInfo["default"].constructFromObject(data, obj);

        _PersonCountAndEmailInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
          delete data['first_name'];
        }

        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
          delete data['last_name'];
        }

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('last_incoming_mail_time')) {
          obj['last_incoming_mail_time'] = _ApiClient["default"].convertToType(data['last_incoming_mail_time'], 'String');
          delete data['last_incoming_mail_time'];
        }

        if (data.hasOwnProperty('last_outgoing_mail_time')) {
          obj['last_outgoing_mail_time'] = _ApiClient["default"].convertToType(data['last_outgoing_mail_time'], 'String');
          delete data['last_outgoing_mail_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonNameCountAndEmailInfo;
}();
/**
 * The name of the person
 * @member {String} name
 */


PersonNameCountAndEmailInfo.prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

PersonNameCountAndEmailInfo.prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

PersonNameCountAndEmailInfo.prototype['last_name'] = undefined;
/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

PersonNameCountAndEmailInfo.prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

PersonNameCountAndEmailInfo.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

PersonNameCountAndEmailInfo.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

PersonNameCountAndEmailInfo.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

PersonNameCountAndEmailInfo.prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

PersonNameCountAndEmailInfo.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

PersonNameCountAndEmailInfo.prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

PersonNameCountAndEmailInfo.prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

PersonNameCountAndEmailInfo.prototype['last_outgoing_mail_time'] = undefined; // Implement PersonNameInfo interface:

/**
 * The name of the person
 * @member {String} name
 */

_PersonNameInfo["default"].prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

_PersonNameInfo["default"].prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

_PersonNameInfo["default"].prototype['last_name'] = undefined; // Implement PersonCountAndEmailInfo interface:

/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

_PersonCountAndEmailInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

_PersonCountAndEmailInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

_PersonCountAndEmailInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

_PersonCountAndEmailInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

_PersonCountAndEmailInfo["default"].prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

_PersonCountAndEmailInfo["default"].prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

_PersonCountAndEmailInfo["default"].prototype['last_outgoing_mail_time'] = undefined;
var _default = PersonNameCountAndEmailInfo;
exports["default"] = _default;
