// pipedrive/dist/model/PersonFlowResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonFlowResponseAllOfData model module.
 * @module model/PersonFlowResponseAllOfData
 * @version 1.0.0
 */
var PersonFlowResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonFlowResponseAllOfData</code>.
   * @alias module:model/PersonFlowResponseAllOfData
   */
  function PersonFlowResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, PersonFlowResponseAllOfData);
    PersonFlowResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonFlowResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonFlowResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonFlowResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/PersonFlowResponseAllOfData} The populated <code>PersonFlowResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonFlowResponseAllOfData();

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
          delete data['object'];
        }

        if (data.hasOwnProperty('timestamp')) {
          obj['timestamp'] = _ApiClient["default"].convertToType(data['timestamp'], 'String');
          delete data['timestamp'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], Object);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonFlowResponseAllOfData;
}();
/**
 * The type of the person update. (Possible object types - personChange, note, activity, file)
 * @member {String} object
 */


PersonFlowResponseAllOfData.prototype['object'] = undefined;
/**
 * The creation date and time of the update
 * @member {String} timestamp
 */

PersonFlowResponseAllOfData.prototype['timestamp'] = undefined;
/**
 * The data related to the update
 * @member {Object} data
 */

PersonFlowResponseAllOfData.prototype['data'] = undefined;
var _default = PersonFlowResponseAllOfData;
exports["default"] = _default;
