// pipedrive/dist/model/PersonDataPhone.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonDataPhone model module.
 * @module model/PersonDataPhone
 * @version 1.0.0
 */
var PersonDataPhone = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonDataPhone</code>.
   * @alias module:model/PersonDataPhone
   */
  function PersonDataPhone() {
    (0, _classCallCheck2["default"])(this, PersonDataPhone);
    PersonDataPhone.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonDataPhone, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonDataPhone</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonDataPhone} obj Optional instance to populate.
     * @return {module:model/PersonDataPhone} The populated <code>PersonDataPhone</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonDataPhone();

        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'String');
          delete data['label'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
          delete data['value'];
        }

        if (data.hasOwnProperty('primary')) {
          obj['primary'] = _ApiClient["default"].convertToType(data['primary'], 'Boolean');
          delete data['primary'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonDataPhone;
}();
/**
 * The type of the phone number
 * @member {String} label
 */


PersonDataPhone.prototype['label'] = undefined;
/**
 * The phone number of the person associated with the item
 * @member {String} value
 */

PersonDataPhone.prototype['value'] = undefined;
/**
 * Whether this is the primary phone number or not
 * @member {Boolean} primary
 */

PersonDataPhone.prototype['primary'] = undefined;
var _default = PersonDataPhone;
exports["default"] = _default;
