// pipedrive/dist/model/PersonNameInfoWithOrgAndOwnerId.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OrgAndOwnerId = _interopRequireDefault(require("./OrgAndOwnerId"));

var _Owner = _interopRequireDefault(require("./Owner"));

var _PersonNameInfo = _interopRequireDefault(require("./PersonNameInfo"));

var _RelationshipOrganizationInfoItemWithActiveFlag = _interopRequireDefault(require("./RelationshipOrganizationInfoItemWithActiveFlag"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonNameInfoWithOrgAndOwnerId model module.
 * @module model/PersonNameInfoWithOrgAndOwnerId
 * @version 1.0.0
 */
var PersonNameInfoWithOrgAndOwnerId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonNameInfoWithOrgAndOwnerId</code>.
   * @alias module:model/PersonNameInfoWithOrgAndOwnerId
   * @implements module:model/OrgAndOwnerId
   * @implements module:model/PersonNameInfo
   */
  function PersonNameInfoWithOrgAndOwnerId() {
    (0, _classCallCheck2["default"])(this, PersonNameInfoWithOrgAndOwnerId);

    _OrgAndOwnerId["default"].initialize(this);

    _PersonNameInfo["default"].initialize(this);

    PersonNameInfoWithOrgAndOwnerId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonNameInfoWithOrgAndOwnerId, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonNameInfoWithOrgAndOwnerId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonNameInfoWithOrgAndOwnerId} obj Optional instance to populate.
     * @return {module:model/PersonNameInfoWithOrgAndOwnerId} The populated <code>PersonNameInfoWithOrgAndOwnerId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonNameInfoWithOrgAndOwnerId();

        _OrgAndOwnerId["default"].constructFromObject(data, obj);

        _PersonNameInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _Owner["default"].constructFromObject(data['owner_id']);
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _RelationshipOrganizationInfoItemWithActiveFlag["default"].constructFromObject(data['org_id']);
          delete data['org_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
          delete data['first_name'];
        }

        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
          delete data['last_name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonNameInfoWithOrgAndOwnerId;
}();
/**
 * @member {module:model/Owner} owner_id
 */


PersonNameInfoWithOrgAndOwnerId.prototype['owner_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItemWithActiveFlag} org_id
 */

PersonNameInfoWithOrgAndOwnerId.prototype['org_id'] = undefined;
/**
 * The name of the person
 * @member {String} name
 */

PersonNameInfoWithOrgAndOwnerId.prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

PersonNameInfoWithOrgAndOwnerId.prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

PersonNameInfoWithOrgAndOwnerId.prototype['last_name'] = undefined; // Implement OrgAndOwnerId interface:

/**
 * @member {module:model/Owner} owner_id
 */

_OrgAndOwnerId["default"].prototype['owner_id'] = undefined;
/**
 * @member {module:model/RelationshipOrganizationInfoItemWithActiveFlag} org_id
 */

_OrgAndOwnerId["default"].prototype['org_id'] = undefined; // Implement PersonNameInfo interface:

/**
 * The name of the person
 * @member {String} name
 */

_PersonNameInfo["default"].prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

_PersonNameInfo["default"].prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

_PersonNameInfo["default"].prototype['last_name'] = undefined;
var _default = PersonNameInfoWithOrgAndOwnerId;
exports["default"] = _default;
