// pipedrive/dist/model/PersonDataWithActiveFlagAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonDataWithActiveFlagAllOf model module.
 * @module model/PersonDataWithActiveFlagAllOf
 * @version 1.0.0
 */
var PersonDataWithActiveFlagAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonDataWithActiveFlagAllOf</code>.
   * @alias module:model/PersonDataWithActiveFlagAllOf
   */
  function PersonDataWithActiveFlagAllOf() {
    (0, _classCallCheck2["default"])(this, PersonDataWithActiveFlagAllOf);
    PersonDataWithActiveFlagAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonDataWithActiveFlagAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonDataWithActiveFlagAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonDataWithActiveFlagAllOf} obj Optional instance to populate.
     * @return {module:model/PersonDataWithActiveFlagAllOf} The populated <code>PersonDataWithActiveFlagAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonDataWithActiveFlagAllOf();

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonDataWithActiveFlagAllOf;
}();
/**
 * Whether the associated person is active or not
 * @member {Boolean} active_flag
 */


PersonDataWithActiveFlagAllOf.prototype['active_flag'] = undefined;
var _default = PersonDataWithActiveFlagAllOf;
exports["default"] = _default;
