// pipedrive/dist/model/PersonDataWithActiveFlag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonData = _interopRequireDefault(require("./PersonData"));

var _PersonDataEmail = _interopRequireDefault(require("./PersonDataEmail"));

var _PersonDataPhone = _interopRequireDefault(require("./PersonDataPhone"));

var _PersonDataWithActiveFlagAllOf = _interopRequireDefault(require("./PersonDataWithActiveFlagAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonDataWithActiveFlag model module.
 * @module model/PersonDataWithActiveFlag
 * @version 1.0.0
 */
var PersonDataWithActiveFlag = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonDataWithActiveFlag</code>.
   * The ID of the person associated with the item
   * @alias module:model/PersonDataWithActiveFlag
   * @implements module:model/PersonDataWithActiveFlagAllOf
   * @implements module:model/PersonData
   */
  function PersonDataWithActiveFlag() {
    (0, _classCallCheck2["default"])(this, PersonDataWithActiveFlag);

    _PersonDataWithActiveFlagAllOf["default"].initialize(this);

    _PersonData["default"].initialize(this);

    PersonDataWithActiveFlag.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonDataWithActiveFlag, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonDataWithActiveFlag</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonDataWithActiveFlag} obj Optional instance to populate.
     * @return {module:model/PersonDataWithActiveFlag} The populated <code>PersonDataWithActiveFlag</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonDataWithActiveFlag();

        _PersonDataWithActiveFlagAllOf["default"].constructFromObject(data, obj);

        _PersonData["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], [_PersonDataEmail["default"]]);
          delete data['email'];
        }

        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], [_PersonDataPhone["default"]]);
          delete data['phone'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonDataWithActiveFlag;
}();
/**
 * Whether the associated person is active or not
 * @member {Boolean} active_flag
 */


PersonDataWithActiveFlag.prototype['active_flag'] = undefined;
/**
 * The ID of the person associated with the item
 * @member {Number} id
 */

PersonDataWithActiveFlag.prototype['id'] = undefined;
/**
 * The name of the person associated with the item
 * @member {String} name
 */

PersonDataWithActiveFlag.prototype['name'] = undefined;
/**
 * The emails of the person associated with the item
 * @member {Array.<module:model/PersonDataEmail>} email
 */

PersonDataWithActiveFlag.prototype['email'] = undefined;
/**
 * The phone numbers of the person associated with the item
 * @member {Array.<module:model/PersonDataPhone>} phone
 */

PersonDataWithActiveFlag.prototype['phone'] = undefined;
/**
 * The ID of the owner of the person that is associated with the item
 * @member {Number} owner_id
 */

PersonDataWithActiveFlag.prototype['owner_id'] = undefined; // Implement PersonDataWithActiveFlagAllOf interface:

/**
 * Whether the associated person is active or not
 * @member {Boolean} active_flag
 */

_PersonDataWithActiveFlagAllOf["default"].prototype['active_flag'] = undefined; // Implement PersonData interface:

/**
 * The ID of the person associated with the item
 * @member {Number} id
 */

_PersonData["default"].prototype['id'] = undefined;
/**
 * The name of the person associated with the item
 * @member {String} name
 */

_PersonData["default"].prototype['name'] = undefined;
/**
 * The emails of the person associated with the item
 * @member {Array.<module:model/PersonDataEmail>} email
 */

_PersonData["default"].prototype['email'] = undefined;
/**
 * The phone numbers of the person associated with the item
 * @member {Array.<module:model/PersonDataPhone>} phone
 */

_PersonData["default"].prototype['phone'] = undefined;
/**
 * The ID of the owner of the person that is associated with the item
 * @member {Number} owner_id
 */

_PersonData["default"].prototype['owner_id'] = undefined;
var _default = PersonDataWithActiveFlag;
exports["default"] = _default;
