// pipedrive/dist/model/PersonNameCountAndEmailInfoWithIds.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonNameCountAndEmailInfo = _interopRequireDefault(require("./PersonNameCountAndEmailInfo"));

var _PersonNameCountAndEmailInfoWithIdsAllOf = _interopRequireDefault(require("./PersonNameCountAndEmailInfoWithIdsAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonNameCountAndEmailInfoWithIds model module.
 * @module model/PersonNameCountAndEmailInfoWithIds
 * @version 1.0.0
 */
var PersonNameCountAndEmailInfoWithIds = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonNameCountAndEmailInfoWithIds</code>.
   * @alias module:model/PersonNameCountAndEmailInfoWithIds
   * @implements module:model/PersonNameCountAndEmailInfoWithIdsAllOf
   * @implements module:model/PersonNameCountAndEmailInfo
   */
  function PersonNameCountAndEmailInfoWithIds() {
    (0, _classCallCheck2["default"])(this, PersonNameCountAndEmailInfoWithIds);

    _PersonNameCountAndEmailInfoWithIdsAllOf["default"].initialize(this);

    _PersonNameCountAndEmailInfo["default"].initialize(this);

    PersonNameCountAndEmailInfoWithIds.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonNameCountAndEmailInfoWithIds, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonNameCountAndEmailInfoWithIds</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonNameCountAndEmailInfoWithIds} obj Optional instance to populate.
     * @return {module:model/PersonNameCountAndEmailInfoWithIds} The populated <code>PersonNameCountAndEmailInfoWithIds</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonNameCountAndEmailInfoWithIds();

        _PersonNameCountAndEmailInfoWithIdsAllOf["default"].constructFromObject(data, obj);

        _PersonNameCountAndEmailInfo["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (data.hasOwnProperty('merge_what_id')) {
          obj['merge_what_id'] = _ApiClient["default"].convertToType(data['merge_what_id'], 'Number');
          delete data['merge_what_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
          delete data['first_name'];
        }

        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
          delete data['last_name'];
        }

        if (data.hasOwnProperty('email_messages_count')) {
          obj['email_messages_count'] = _ApiClient["default"].convertToType(data['email_messages_count'], 'Number');
          delete data['email_messages_count'];
        }

        if (data.hasOwnProperty('activities_count')) {
          obj['activities_count'] = _ApiClient["default"].convertToType(data['activities_count'], 'Number');
          delete data['activities_count'];
        }

        if (data.hasOwnProperty('done_activities_count')) {
          obj['done_activities_count'] = _ApiClient["default"].convertToType(data['done_activities_count'], 'Number');
          delete data['done_activities_count'];
        }

        if (data.hasOwnProperty('undone_activities_count')) {
          obj['undone_activities_count'] = _ApiClient["default"].convertToType(data['undone_activities_count'], 'Number');
          delete data['undone_activities_count'];
        }

        if (data.hasOwnProperty('files_count')) {
          obj['files_count'] = _ApiClient["default"].convertToType(data['files_count'], 'Number');
          delete data['files_count'];
        }

        if (data.hasOwnProperty('notes_count')) {
          obj['notes_count'] = _ApiClient["default"].convertToType(data['notes_count'], 'Number');
          delete data['notes_count'];
        }

        if (data.hasOwnProperty('followers_count')) {
          obj['followers_count'] = _ApiClient["default"].convertToType(data['followers_count'], 'Number');
          delete data['followers_count'];
        }

        if (data.hasOwnProperty('last_incoming_mail_time')) {
          obj['last_incoming_mail_time'] = _ApiClient["default"].convertToType(data['last_incoming_mail_time'], 'String');
          delete data['last_incoming_mail_time'];
        }

        if (data.hasOwnProperty('last_outgoing_mail_time')) {
          obj['last_outgoing_mail_time'] = _ApiClient["default"].convertToType(data['last_outgoing_mail_time'], 'String');
          delete data['last_outgoing_mail_time'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonNameCountAndEmailInfoWithIds;
}();
/**
 * The ID of the owner related to the person
 * @member {Number} owner_id
 */


PersonNameCountAndEmailInfoWithIds.prototype['owner_id'] = undefined;
/**
 * The ID of the organization related to the person
 * @member {Number} org_id
 */

PersonNameCountAndEmailInfoWithIds.prototype['org_id'] = undefined;
/**
 * The ID of the person with what the main person was merged
 * @member {Number} merge_what_id
 */

PersonNameCountAndEmailInfoWithIds.prototype['merge_what_id'] = undefined;
/**
 * The name of the person
 * @member {String} name
 */

PersonNameCountAndEmailInfoWithIds.prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

PersonNameCountAndEmailInfoWithIds.prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

PersonNameCountAndEmailInfoWithIds.prototype['last_name'] = undefined;
/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

PersonNameCountAndEmailInfoWithIds.prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

PersonNameCountAndEmailInfoWithIds.prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

PersonNameCountAndEmailInfoWithIds.prototype['last_outgoing_mail_time'] = undefined; // Implement PersonNameCountAndEmailInfoWithIdsAllOf interface:

/**
 * The ID of the owner related to the person
 * @member {Number} owner_id
 */

_PersonNameCountAndEmailInfoWithIdsAllOf["default"].prototype['owner_id'] = undefined;
/**
 * The ID of the organization related to the person
 * @member {Number} org_id
 */

_PersonNameCountAndEmailInfoWithIdsAllOf["default"].prototype['org_id'] = undefined;
/**
 * The ID of the person with what the main person was merged
 * @member {Number} merge_what_id
 */

_PersonNameCountAndEmailInfoWithIdsAllOf["default"].prototype['merge_what_id'] = undefined; // Implement PersonNameCountAndEmailInfo interface:

/**
 * The name of the person
 * @member {String} name
 */

_PersonNameCountAndEmailInfo["default"].prototype['name'] = undefined;
/**
 * The first name of the person
 * @member {String} first_name
 */

_PersonNameCountAndEmailInfo["default"].prototype['first_name'] = undefined;
/**
 * The last name of the person
 * @member {String} last_name
 */

_PersonNameCountAndEmailInfo["default"].prototype['last_name'] = undefined;
/**
 * The count of email messages related to the person
 * @member {Number} email_messages_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['email_messages_count'] = undefined;
/**
 * The count of activities related to the person
 * @member {Number} activities_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['activities_count'] = undefined;
/**
 * The count of done activities related to the person
 * @member {Number} done_activities_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['done_activities_count'] = undefined;
/**
 * The count of undone activities related to the person
 * @member {Number} undone_activities_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['undone_activities_count'] = undefined;
/**
 * The count of files related to the person
 * @member {Number} files_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['files_count'] = undefined;
/**
 * The count of notes related to the person
 * @member {Number} notes_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['notes_count'] = undefined;
/**
 * The count of followers related to the person
 * @member {Number} followers_count
 */

_PersonNameCountAndEmailInfo["default"].prototype['followers_count'] = undefined;
/**
 * The date and time of the last incoming email associated with the person
 * @member {String} last_incoming_mail_time
 */

_PersonNameCountAndEmailInfo["default"].prototype['last_incoming_mail_time'] = undefined;
/**
 * The date and time of the last outgoing email associated with the person
 * @member {String} last_outgoing_mail_time
 */

_PersonNameCountAndEmailInfo["default"].prototype['last_outgoing_mail_time'] = undefined;
var _default = PersonNameCountAndEmailInfoWithIds;
exports["default"] = _default;
