// pipedrive/dist/model/PersonSearchResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonSearchItem = _interopRequireDefault(require("./PersonSearchItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PersonSearchResponseAllOfData model module.
 * @module model/PersonSearchResponseAllOfData
 * @version 1.0.0
 */
var PersonSearchResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonSearchResponseAllOfData</code>.
   * @alias module:model/PersonSearchResponseAllOfData
   */
  function PersonSearchResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, PersonSearchResponseAllOfData);
    PersonSearchResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PersonSearchResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PersonSearchResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonSearchResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/PersonSearchResponseAllOfData} The populated <code>PersonSearchResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonSearchResponseAllOfData();

        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_PersonSearchItem["default"]]);
          delete data['items'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PersonSearchResponseAllOfData;
}();
/**
 * The array of found items
 * @member {Array.<module:model/PersonSearchItem>} items
 */


PersonSearchResponseAllOfData.prototype['items'] = undefined;
var _default = PersonSearchResponseAllOfData;
exports["default"] = _default;
