// pipedrive/dist/model/PipelineDetailsAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DealSummary = _interopRequireDefault(require("./DealSummary"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PipelineDetailsAllOf model module.
 * @module model/PipelineDetailsAllOf
 * @version 1.0.0
 */
var PipelineDetailsAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PipelineDetailsAllOf</code>.
   * @alias module:model/PipelineDetailsAllOf
   */
  function PipelineDetailsAllOf() {
    (0, _classCallCheck2["default"])(this, PipelineDetailsAllOf);
    PipelineDetailsAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PipelineDetailsAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PipelineDetailsAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PipelineDetailsAllOf} obj Optional instance to populate.
     * @return {module:model/PipelineDetailsAllOf} The populated <code>PipelineDetailsAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PipelineDetailsAllOf();

        if (data.hasOwnProperty('deals_summary')) {
          obj['deals_summary'] = _DealSummary["default"].constructFromObject(data['deals_summary']);
          delete data['deals_summary'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PipelineDetailsAllOf;
}();
/**
 * @member {module:model/DealSummary} deals_summary
 */


PipelineDetailsAllOf.prototype['deals_summary'] = undefined;
var _default = PipelineDetailsAllOf;
exports["default"] = _default;
