// pipedrive/dist/model/Pipeline.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The Pipeline model module.
 * @module model/Pipeline
 * @version 1.0.0
 */
var Pipeline = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Pipeline</code>.
   * @alias module:model/Pipeline
   */
  function Pipeline() {
    (0, _classCallCheck2["default"])(this, Pipeline);
    Pipeline.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(Pipeline, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Pipeline</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Pipeline} obj Optional instance to populate.
     * @return {module:model/Pipeline} The populated <code>Pipeline</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Pipeline();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('deal_probability')) {
          obj['deal_probability'] = _ApiClient["default"].convertToType(data['deal_probability'], _NumberBoolean["default"]);
          delete data['deal_probability'];
        }

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], _NumberBoolean["default"]);
          delete data['active'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return Pipeline;
}();
/**
 * The name of the pipeline
 * @member {String} name
 */


Pipeline.prototype['name'] = undefined;
/**
 * Whether deal probability is disabled or enabled for this pipeline
 * @member {module:model/NumberBoolean} deal_probability
 */

Pipeline.prototype['deal_probability'] = undefined;
/**
 * Defines the order of pipelines. First order (`order_nr=0`) is the default pipeline.
 * @member {Number} order_nr
 */

Pipeline.prototype['order_nr'] = undefined;
/**
 * Whether this pipeline will be made inactive (hidden) or active
 * @member {module:model/NumberBoolean} active
 */

Pipeline.prototype['active'] = undefined;
var _default = Pipeline;
exports["default"] = _default;
