// pipedrive/dist/model/PipelineDetails.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BasePipelineWithSelectedFlag = _interopRequireDefault(require("./BasePipelineWithSelectedFlag"));

var _DealSummary = _interopRequireDefault(require("./DealSummary"));

var _PipelineDetailsAllOf = _interopRequireDefault(require("./PipelineDetailsAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PipelineDetails model module.
 * @module model/PipelineDetails
 * @version 1.0.0
 */
var PipelineDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PipelineDetails</code>.
   * The pipeline object
   * @alias module:model/PipelineDetails
   * @implements module:model/BasePipelineWithSelectedFlag
   * @implements module:model/PipelineDetailsAllOf
   */
  function PipelineDetails() {
    (0, _classCallCheck2["default"])(this, PipelineDetails);

    _BasePipelineWithSelectedFlag["default"].initialize(this);

    _PipelineDetailsAllOf["default"].initialize(this);

    PipelineDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PipelineDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PipelineDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PipelineDetails} obj Optional instance to populate.
     * @return {module:model/PipelineDetails} The populated <code>PipelineDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PipelineDetails();

        _BasePipelineWithSelectedFlag["default"].constructFromObject(data, obj);

        _PipelineDetailsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('url_title')) {
          obj['url_title'] = _ApiClient["default"].convertToType(data['url_title'], 'String');
          delete data['url_title'];
        }

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
          delete data['active'];
        }

        if (data.hasOwnProperty('deal_probability')) {
          obj['deal_probability'] = _ApiClient["default"].convertToType(data['deal_probability'], 'Boolean');
          delete data['deal_probability'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }

        if (data.hasOwnProperty('selected')) {
          obj['selected'] = _ApiClient["default"].convertToType(data['selected'], 'Boolean');
          delete data['selected'];
        }

        if (data.hasOwnProperty('deals_summary')) {
          obj['deals_summary'] = _DealSummary["default"].constructFromObject(data['deals_summary']);
          delete data['deals_summary'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PipelineDetails;
}();
/**
 * The ID of the pipeline
 * @member {Number} id
 */


PipelineDetails.prototype['id'] = undefined;
/**
 * The name of the pipeline
 * @member {String} name
 */

PipelineDetails.prototype['name'] = undefined;
/**
 * The pipeline title displayed in the URL
 * @member {String} url_title
 */

PipelineDetails.prototype['url_title'] = undefined;
/**
 * Defines the order of pipelines. First order (`order_nr=0`) is the default pipeline.
 * @member {Number} order_nr
 */

PipelineDetails.prototype['order_nr'] = undefined;
/**
 * Whether this pipeline will be made inactive (hidden) or active
 * @member {Boolean} active
 */

PipelineDetails.prototype['active'] = undefined;
/**
 * Whether deal probability is disabled or enabled for this pipeline
 * @member {Boolean} deal_probability
 */

PipelineDetails.prototype['deal_probability'] = undefined;
/**
 * The pipeline creation time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} add_time
 */

PipelineDetails.prototype['add_time'] = undefined;
/**
 * The pipeline update time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} update_time
 */

PipelineDetails.prototype['update_time'] = undefined;
/**
 * A boolean that shows if the pipeline is selected from a filter or not
 * @member {Boolean} selected
 */

PipelineDetails.prototype['selected'] = undefined;
/**
 * @member {module:model/DealSummary} deals_summary
 */

PipelineDetails.prototype['deals_summary'] = undefined; // Implement BasePipelineWithSelectedFlag interface:

/**
 * The ID of the pipeline
 * @member {Number} id
 */

_BasePipelineWithSelectedFlag["default"].prototype['id'] = undefined;
/**
 * The name of the pipeline
 * @member {String} name
 */

_BasePipelineWithSelectedFlag["default"].prototype['name'] = undefined;
/**
 * The pipeline title displayed in the URL
 * @member {String} url_title
 */

_BasePipelineWithSelectedFlag["default"].prototype['url_title'] = undefined;
/**
 * Defines the order of pipelines. First order (`order_nr=0`) is the default pipeline.
 * @member {Number} order_nr
 */

_BasePipelineWithSelectedFlag["default"].prototype['order_nr'] = undefined;
/**
 * Whether this pipeline will be made inactive (hidden) or active
 * @member {Boolean} active
 */

_BasePipelineWithSelectedFlag["default"].prototype['active'] = undefined;
/**
 * Whether deal probability is disabled or enabled for this pipeline
 * @member {Boolean} deal_probability
 */

_BasePipelineWithSelectedFlag["default"].prototype['deal_probability'] = undefined;
/**
 * The pipeline creation time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} add_time
 */

_BasePipelineWithSelectedFlag["default"].prototype['add_time'] = undefined;
/**
 * The pipeline update time. Format: YYYY-MM-DD HH:MM:SS.
 * @member {String} update_time
 */

_BasePipelineWithSelectedFlag["default"].prototype['update_time'] = undefined;
/**
 * A boolean that shows if the pipeline is selected from a filter or not
 * @member {Boolean} selected
 */

_BasePipelineWithSelectedFlag["default"].prototype['selected'] = undefined; // Implement PipelineDetailsAllOf interface:

/**
 * @member {module:model/DealSummary} deals_summary
 */

_PipelineDetailsAllOf["default"].prototype['deals_summary'] = undefined;
var _default = PipelineDetails;
exports["default"] = _default;
