// pipedrive/dist/model/PostRoleSettingsAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PostRoleSettingsAllO;

/**
 * The PostRoleSettingsAllOfData model module.
 * @module model/PostRoleSettingsAllOfData
 * @version 1.0.0
 */
var PostRoleSettingsAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostRoleSettingsAllOfData</code>.
   * The response data
   * @alias module:model/PostRoleSettingsAllOfData
   */
  function PostRoleSettingsAllOfData() {
    (0, _classCallCheck2["default"])(this, PostRoleSettingsAllOfData);
    PostRoleSettingsAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PostRoleSettingsAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PostRoleSettingsAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostRoleSettingsAllOfData} obj Optional instance to populate.
     * @return {module:model/PostRoleSettingsAllOfData} The populated <code>PostRoleSettingsAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostRoleSettingsAllOfData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('deal_default_visibility')) {
          obj['deal_default_visibility'] = _ApiClient["default"].convertToType(data['deal_default_visibility'], 'Number');
          delete data['deal_default_visibility'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PostRoleSettingsAllOfData;
}();
/**
 * The ID of the role
 * @member {Number} id
 */


PostRoleSettingsAllOfData.prototype['id'] = undefined;
/**
 * The setting
 * @member {module:model/PostRoleSettingsAllOfData.DealDefaultVisibilityEnum} deal_default_visibility
 */

PostRoleSettingsAllOfData.prototype['deal_default_visibility'] = undefined;
/**
 * Allowed values for the <code>deal_default_visibility</code> property.
 * @enum {Number}
 * @readonly
 */

PostRoleSettingsAllOfData['DealDefaultVisibilityEnum'] = (_PostRoleSettingsAllO = {
  /**
   * value: null
   * @const
   */
  "null": null
}, (0, _defineProperty2["default"])(_PostRoleSettingsAllO, "null", null), (0, _defineProperty2["default"])(_PostRoleSettingsAllO, "null", null), (0, _defineProperty2["default"])(_PostRoleSettingsAllO, "null", null), (0, _defineProperty2["default"])(_PostRoleSettingsAllO, "null", null), _PostRoleSettingsAllO);
var _default = PostRoleSettingsAllOfData;
exports["default"] = _default;
