// pipedrive/dist/model/PostGoalResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GoalsResponseComponent = _interopRequireDefault(require("./GoalsResponseComponent"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PostGoalResponse model module.
 * @module model/PostGoalResponse
 * @version 1.0.0
 */
var PostGoalResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostGoalResponse</code>.
   * @alias module:model/PostGoalResponse
   */
  function PostGoalResponse() {
    (0, _classCallCheck2["default"])(this, PostGoalResponse);
    PostGoalResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PostGoalResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PostGoalResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostGoalResponse} obj Optional instance to populate.
     * @return {module:model/PostGoalResponse} The populated <code>PostGoalResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostGoalResponse();

        if (data.hasOwnProperty('goal')) {
          obj['goal'] = _GoalsResponseComponent["default"].constructFromObject(data['goal']);
          delete data['goal'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PostGoalResponse;
}();
/**
 * @member {module:model/GoalsResponseComponent} goal
 */


PostGoalResponse.prototype['goal'] = undefined;
var _default = PostGoalResponse;
exports["default"] = _default;
