// pipedrive/dist/model/PostDealParticipants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GetDealsRelatedObjects = _interopRequireDefault(require("./GetDealsRelatedObjects"));

var _PersonItem = _interopRequireDefault(require("./PersonItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PostDealParticipants model module.
 * @module model/PostDealParticipants
 * @version 1.0.0
 */
var PostDealParticipants = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostDealParticipants</code>.
   * @alias module:model/PostDealParticipants
   */
  function PostDealParticipants() {
    (0, _classCallCheck2["default"])(this, PostDealParticipants);
    PostDealParticipants.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PostDealParticipants, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PostDealParticipants</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostDealParticipants} obj Optional instance to populate.
     * @return {module:model/PostDealParticipants} The populated <code>PostDealParticipants</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostDealParticipants();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], _PersonItem["default"]);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetDealsRelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PostDealParticipants;
}();
/**
 * If the request was successful or not
 * @member {Boolean} success
 */


PostDealParticipants.prototype['success'] = undefined;
/**
 * The object of participant
 * @member {module:model/PersonItem} data
 */

PostDealParticipants.prototype['data'] = undefined;
/**
 * @member {module:model/GetDealsRelatedObjects} related_objects
 */

PostDealParticipants.prototype['related_objects'] = undefined;
var _default = PostDealParticipants;
exports["default"] = _default;
