// pipedrive/dist/model/PostRolesAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PostRolesAllOfData model module.
 * @module model/PostRolesAllOfData
 * @version 1.0.0
 */
var PostRolesAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostRolesAllOfData</code>.
   * The response data
   * @alias module:model/PostRolesAllOfData
   */
  function PostRolesAllOfData() {
    (0, _classCallCheck2["default"])(this, PostRolesAllOfData);
    PostRolesAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PostRolesAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PostRolesAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostRolesAllOfData} obj Optional instance to populate.
     * @return {module:model/PostRolesAllOfData} The populated <code>PostRolesAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostRolesAllOfData();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PostRolesAllOfData;
}();
/**
 * The ID of the created role
 * @member {Number} id
 */


PostRolesAllOfData.prototype['id'] = undefined;
var _default = PostRolesAllOfData;
exports["default"] = _default;
