// pipedrive/dist/model/PostRoleAssignmentAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The PostRoleAssignmentAllOfData model module.
 * @module model/PostRoleAssignmentAllOfData
 * @version 1.0.0
 */
var PostRoleAssignmentAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PostRoleAssignmentAllOfData</code>.
   * The response data
   * @alias module:model/PostRoleAssignmentAllOfData
   */
  function PostRoleAssignmentAllOfData() {
    (0, _classCallCheck2["default"])(this, PostRoleAssignmentAllOfData);
    PostRoleAssignmentAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(PostRoleAssignmentAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PostRoleAssignmentAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PostRoleAssignmentAllOfData} obj Optional instance to populate.
     * @return {module:model/PostRoleAssignmentAllOfData} The populated <code>PostRoleAssignmentAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PostRoleAssignmentAllOfData();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], 'Number');
          delete data['role_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return PostRoleAssignmentAllOfData;
}();
/**
 * The ID of the user that was added to the role
 * @member {Number} user_id
 */


PostRoleAssignmentAllOfData.prototype['user_id'] = undefined;
/**
 * The ID of the role the user was added to
 * @member {Number} role_id
 */

PostRoleAssignmentAllOfData.prototype['role_id'] = undefined;
var _default = PostRoleAssignmentAllOfData;
exports["default"] = _default;
