// pipedrive/dist/model/ProductSearchResponseAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ProductSearchItem = _interopRequireDefault(require("./ProductSearchItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ProductSearchResponseAllOfData model module.
 * @module model/ProductSearchResponseAllOfData
 * @version 1.0.0
 */
var ProductSearchResponseAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductSearchResponseAllOfData</code>.
   * @alias module:model/ProductSearchResponseAllOfData
   */
  function ProductSearchResponseAllOfData() {
    (0, _classCallCheck2["default"])(this, ProductSearchResponseAllOfData);
    ProductSearchResponseAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ProductSearchResponseAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductSearchResponseAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductSearchResponseAllOfData} obj Optional instance to populate.
     * @return {module:model/ProductSearchResponseAllOfData} The populated <code>ProductSearchResponseAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductSearchResponseAllOfData();

        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_ProductSearchItem["default"]]);
          delete data['items'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ProductSearchResponseAllOfData;
}();
/**
 * The array of found items
 * @member {Array.<module:model/ProductSearchItem>} items
 */


ProductSearchResponseAllOfData.prototype['items'] = undefined;
var _default = ProductSearchResponseAllOfData;
exports["default"] = _default;
