// pipedrive/dist/model/ProductSearchResponseAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _ProductSearchResponseAllOfData = _interopRequireDefault(require("./ProductSearchResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ProductSearchResponseAllOf model module.
 * @module model/ProductSearchResponseAllOf
 * @version 1.0.0
 */
var ProductSearchResponseAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductSearchResponseAllOf</code>.
   * @alias module:model/ProductSearchResponseAllOf
   */
  function ProductSearchResponseAllOf() {
    (0, _classCallCheck2["default"])(this, ProductSearchResponseAllOf);
    ProductSearchResponseAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ProductSearchResponseAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductSearchResponseAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductSearchResponseAllOf} obj Optional instance to populate.
     * @return {module:model/ProductSearchResponseAllOf} The populated <code>ProductSearchResponseAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductSearchResponseAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ProductSearchResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ProductSearchResponseAllOf;
}();
/**
 * @member {module:model/ProductSearchResponseAllOfData} data
 */


ProductSearchResponseAllOf.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

ProductSearchResponseAllOf.prototype['additional_data'] = undefined;
var _default = ProductSearchResponseAllOf;
exports["default"] = _default;
