// pipedrive/dist/model/ProductSearchItemItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ProductSearchItemItemOwner = _interopRequireDefault(require("./ProductSearchItemItemOwner"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ProductSearchItemItem model module.
 * @module model/ProductSearchItemItem
 * @version 1.0.0
 */
var ProductSearchItemItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductSearchItemItem</code>.
   * @alias module:model/ProductSearchItemItem
   */
  function ProductSearchItemItem() {
    (0, _classCallCheck2["default"])(this, ProductSearchItemItem);
    ProductSearchItemItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ProductSearchItemItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductSearchItemItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductSearchItemItem} obj Optional instance to populate.
     * @return {module:model/ProductSearchItemItem} The populated <code>ProductSearchItemItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductSearchItemItem();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'Number');
          delete data['code'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'Number');
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _ProductSearchItemItemOwner["default"].constructFromObject(data['owner']);
          delete data['owner'];
        }

        if (data.hasOwnProperty('custom_fields')) {
          obj['custom_fields'] = _ApiClient["default"].convertToType(data['custom_fields'], 'Number');
          delete data['custom_fields'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ProductSearchItemItem;
}();
/**
 * The ID of the product
 * @member {Number} id
 */


ProductSearchItemItem.prototype['id'] = undefined;
/**
 * The type of the item
 * @member {String} type
 */

ProductSearchItemItem.prototype['type'] = undefined;
/**
 * The name of the product
 * @member {String} name
 */

ProductSearchItemItem.prototype['name'] = undefined;
/**
 * The code of the product
 * @member {Number} code
 */

ProductSearchItemItem.prototype['code'] = undefined;
/**
 * The visibility of the product
 * @member {Number} visible_to
 */

ProductSearchItemItem.prototype['visible_to'] = undefined;
/**
 * @member {module:model/ProductSearchItemItemOwner} owner
 */

ProductSearchItemItem.prototype['owner'] = undefined;
/**
 * The custom fields
 * @member {Number} custom_fields
 */

ProductSearchItemItem.prototype['custom_fields'] = undefined;
var _default = ProductSearchItemItem;
exports["default"] = _default;
