// pipedrive/dist/model/ProductsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));

var _GetActivitiesResponse200RelatedObjects = _interopRequireDefault(require("./GetActivitiesResponse200RelatedObjects"));

var _ProductResponse = _interopRequireDefault(require("./ProductResponse"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ProductsResponse model module.
 * @module model/ProductsResponse
 * @version 1.0.0
 */
var ProductsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductsResponse</code>.
   * @alias module:model/ProductsResponse
   */
  function ProductsResponse() {
    (0, _classCallCheck2["default"])(this, ProductsResponse);
    ProductsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ProductsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductsResponse} obj Optional instance to populate.
     * @return {module:model/ProductsResponse} The populated <code>ProductsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductsResponse();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ProductResponse["default"]]);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetActivitiesResponse200RelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ProductsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


ProductsResponse.prototype['success'] = undefined;
/**
 * Array containing data for all products
 * @member {Array.<module:model/ProductResponse>} data
 */

ProductsResponse.prototype['data'] = undefined;
/**
 * @member {module:model/AdditionalData} additional_data
 */

ProductsResponse.prototype['additional_data'] = undefined;
/**
 * @member {module:model/GetActivitiesResponse200RelatedObjects} related_objects
 */

ProductsResponse.prototype['related_objects'] = undefined;
var _default = ProductsResponse;
exports["default"] = _default;
