// pipedrive/dist/model/ProductAttachementFields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ProductAttachementFields model module.
 * @module model/ProductAttachementFields
 * @version 1.0.0
 */
var ProductAttachementFields = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductAttachementFields</code>.
   * @alias module:model/ProductAttachementFields
   */
  function ProductAttachementFields() {
    (0, _classCallCheck2["default"])(this, ProductAttachementFields);
    ProductAttachementFields.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ProductAttachementFields, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductAttachementFields</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductAttachementFields} obj Optional instance to populate.
     * @return {module:model/ProductAttachementFields} The populated <code>ProductAttachementFields</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductAttachementFields();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('product_id')) {
          obj['product_id'] = _ApiClient["default"].convertToType(data['product_id'], 'Number');
          delete data['product_id'];
        }

        if (data.hasOwnProperty('duration_unit')) {
          obj['duration_unit'] = _ApiClient["default"].convertToType(data['duration_unit'], 'String');
          delete data['duration_unit'];
        }

        if (data.hasOwnProperty('sum_no_discount')) {
          obj['sum_no_discount'] = _ApiClient["default"].convertToType(data['sum_no_discount'], 'Number');
          delete data['sum_no_discount'];
        }

        if (data.hasOwnProperty('sum')) {
          obj['sum'] = _ApiClient["default"].convertToType(data['sum'], 'Number');
          delete data['sum'];
        }

        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'String');
          delete data['currency'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('last_edit')) {
          obj['last_edit'] = _ApiClient["default"].convertToType(data['last_edit'], 'String');
          delete data['last_edit'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('tax')) {
          obj['tax'] = _ApiClient["default"].convertToType(data['tax'], 'Number');
          delete data['tax'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ProductAttachementFields;
}();
/**
 * The ID of the deal-product (the ID of the product attached to the deal)
 * @member {Number} id
 */


ProductAttachementFields.prototype['id'] = undefined;
/**
 * The ID of the company
 * @member {Number} company_id
 */

ProductAttachementFields.prototype['company_id'] = undefined;
/**
 * The ID of the deal
 * @member {Number} deal_id
 */

ProductAttachementFields.prototype['deal_id'] = undefined;
/**
 * The ID of the product
 * @member {Number} product_id
 */

ProductAttachementFields.prototype['product_id'] = undefined;
/**
 * The type of the duration. (For example hourly, daily, etc.)
 * @member {String} duration_unit
 */

ProductAttachementFields.prototype['duration_unit'] = undefined;
/**
 * The product sum without the discount
 * @member {Number} sum_no_discount
 */

ProductAttachementFields.prototype['sum_no_discount'] = undefined;
/**
 * The sum of all the products attached to the deal
 * @member {Number} sum
 */

ProductAttachementFields.prototype['sum'] = undefined;
/**
 * The currency associated with the deal product
 * @member {String} currency
 */

ProductAttachementFields.prototype['currency'] = undefined;
/**
 * The date and time when the product was added to the deal
 * @member {String} add_time
 */

ProductAttachementFields.prototype['add_time'] = undefined;
/**
 * The date and time when the deal product was last edited
 * @member {String} last_edit
 */

ProductAttachementFields.prototype['last_edit'] = undefined;
/**
 * Whether the product is active or not
 * @member {Boolean} active_flag
 */

ProductAttachementFields.prototype['active_flag'] = undefined;
/**
 * The product tax
 * @member {Number} tax
 */

ProductAttachementFields.prototype['tax'] = undefined;
/**
 * The product name
 * @member {String} name
 */

ProductAttachementFields.prototype['name'] = undefined;
var _default = ProductAttachementFields;
exports["default"] = _default;
