// pipedrive/dist/model/RecentsUser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseUser = _interopRequireDefault(require("./BaseUser"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RecentsUser model module.
 * @module model/RecentsUser
 * @version 1.0.0
 */
var RecentsUser = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RecentsUser</code>.
   * @alias module:model/RecentsUser
   */
  function RecentsUser() {
    (0, _classCallCheck2["default"])(this, RecentsUser);
    RecentsUser.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RecentsUser, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RecentsUser</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RecentsUser} obj Optional instance to populate.
     * @return {module:model/RecentsUser} The populated <code>RecentsUser</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RecentsUser();

        if (data.hasOwnProperty('item')) {
          obj['item'] = _ApiClient["default"].convertToType(data['item'], 'String');
          delete data['item'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BaseUser["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RecentsUser;
}();
/**
 * @member {module:model/RecentsUser.ItemEnum} item
 */


RecentsUser.prototype['item'] = undefined;
/**
 * @member {Number} id
 */

RecentsUser.prototype['id'] = undefined;
/**
 * @member {module:model/BaseUser} data
 */

RecentsUser.prototype['data'] = undefined;
/**
 * Allowed values for the <code>item</code> property.
 * @enum {String}
 * @readonly
 */

RecentsUser['ItemEnum'] = {
  /**
   * value: "user"
   * @const
   */
  "user": "user"
};
var _default = RecentsUser;
exports["default"] = _default;
