// pipedrive/dist/model/RelatedUserData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UserDataWithId = _interopRequireDefault(require("./UserDataWithId"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelatedUserData model module.
 * @module model/RelatedUserData
 * @version 1.0.0
 */
var RelatedUserData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelatedUserData</code>.
   * @alias module:model/RelatedUserData
   */
  function RelatedUserData() {
    (0, _classCallCheck2["default"])(this, RelatedUserData);
    RelatedUserData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelatedUserData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelatedUserData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelatedUserData} obj Optional instance to populate.
     * @return {module:model/RelatedUserData} The populated <code>RelatedUserData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelatedUserData();

        if (data.hasOwnProperty('USER_ID')) {
          obj['USER_ID'] = _UserDataWithId["default"].constructFromObject(data['USER_ID']);
          delete data['USER_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelatedUserData;
}();
/**
 * @member {module:model/UserDataWithId} USER_ID
 */


RelatedUserData.prototype['USER_ID'] = undefined;
var _default = RelatedUserData;
exports["default"] = _default;
