// pipedrive/dist/model/RelatedPictureData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PictureDataWithID = _interopRequireDefault(require("./PictureDataWithID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelatedPictureData model module.
 * @module model/RelatedPictureData
 * @version 1.0.0
 */
var RelatedPictureData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelatedPictureData</code>.
   * The picture that is associated with the item
   * @alias module:model/RelatedPictureData
   */
  function RelatedPictureData() {
    (0, _classCallCheck2["default"])(this, RelatedPictureData);
    RelatedPictureData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelatedPictureData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelatedPictureData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelatedPictureData} obj Optional instance to populate.
     * @return {module:model/RelatedPictureData} The populated <code>RelatedPictureData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelatedPictureData();

        if (data.hasOwnProperty('PICTURE_ID')) {
          obj['PICTURE_ID'] = _ApiClient["default"].convertToType(data['PICTURE_ID'], _PictureDataWithID["default"]);
          delete data['PICTURE_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelatedPictureData;
}();
/**
 * The ID of the picture
 * @member {module:model/PictureDataWithID} PICTURE_ID
 */


RelatedPictureData.prototype['PICTURE_ID'] = undefined;
var _default = RelatedPictureData;
exports["default"] = _default;
