// pipedrive/dist/model/RelatedPersonDataWithActiveFlag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PersonDataWithActiveFlag = _interopRequireDefault(require("./PersonDataWithActiveFlag"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelatedPersonDataWithActiveFlag model module.
 * @module model/RelatedPersonDataWithActiveFlag
 * @version 1.0.0
 */
var RelatedPersonDataWithActiveFlag = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelatedPersonDataWithActiveFlag</code>.
   * @alias module:model/RelatedPersonDataWithActiveFlag
   */
  function RelatedPersonDataWithActiveFlag() {
    (0, _classCallCheck2["default"])(this, RelatedPersonDataWithActiveFlag);
    RelatedPersonDataWithActiveFlag.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelatedPersonDataWithActiveFlag, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelatedPersonDataWithActiveFlag</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelatedPersonDataWithActiveFlag} obj Optional instance to populate.
     * @return {module:model/RelatedPersonDataWithActiveFlag} The populated <code>RelatedPersonDataWithActiveFlag</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelatedPersonDataWithActiveFlag();

        if (data.hasOwnProperty('PERSON_ID')) {
          obj['PERSON_ID'] = _PersonDataWithActiveFlag["default"].constructFromObject(data['PERSON_ID']);
          delete data['PERSON_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelatedPersonDataWithActiveFlag;
}();
/**
 * @member {module:model/PersonDataWithActiveFlag} PERSON_ID
 */


RelatedPersonDataWithActiveFlag.prototype['PERSON_ID'] = undefined;
var _default = RelatedPersonDataWithActiveFlag;
exports["default"] = _default;
