// pipedrive/dist/model/RelatedFollowerData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FollowerDataWithID = _interopRequireDefault(require("./FollowerDataWithID"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelatedFollowerData model module.
 * @module model/RelatedFollowerData
 * @version 1.0.0
 */
var RelatedFollowerData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelatedFollowerData</code>.
   * The follower that is associated with the item
   * @alias module:model/RelatedFollowerData
   */
  function RelatedFollowerData() {
    (0, _classCallCheck2["default"])(this, RelatedFollowerData);
    RelatedFollowerData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelatedFollowerData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelatedFollowerData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelatedFollowerData} obj Optional instance to populate.
     * @return {module:model/RelatedFollowerData} The populated <code>RelatedFollowerData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelatedFollowerData();

        if (data.hasOwnProperty('FOLLOWER_USER_ID')) {
          obj['FOLLOWER_USER_ID'] = _ApiClient["default"].convertToType(data['FOLLOWER_USER_ID'], _FollowerDataWithID["default"]);
          delete data['FOLLOWER_USER_ID'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelatedFollowerData;
}();
/**
 * The user ID of the follower
 * @member {module:model/FollowerDataWithID} FOLLOWER_USER_ID
 */


RelatedFollowerData.prototype['FOLLOWER_USER_ID'] = undefined;
var _default = RelatedFollowerData;
exports["default"] = _default;
