// pipedrive/dist/model/RelationshipOrganizationInfoItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OrganizationData = _interopRequireDefault(require("./OrganizationData"));

var _RelationshipOrganizationInfoItemAllOf = _interopRequireDefault(require("./RelationshipOrganizationInfoItemAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelationshipOrganizationInfoItem model module.
 * @module model/RelationshipOrganizationInfoItem
 * @version 1.0.0
 */
var RelationshipOrganizationInfoItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelationshipOrganizationInfoItem</code>.
   * @alias module:model/RelationshipOrganizationInfoItem
   * @implements module:model/OrganizationData
   * @implements module:model/RelationshipOrganizationInfoItemAllOf
   */
  function RelationshipOrganizationInfoItem() {
    (0, _classCallCheck2["default"])(this, RelationshipOrganizationInfoItem);

    _OrganizationData["default"].initialize(this);

    _RelationshipOrganizationInfoItemAllOf["default"].initialize(this);

    RelationshipOrganizationInfoItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelationshipOrganizationInfoItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelationshipOrganizationInfoItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelationshipOrganizationInfoItem} obj Optional instance to populate.
     * @return {module:model/RelationshipOrganizationInfoItem} The populated <code>RelationshipOrganizationInfoItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelationshipOrganizationInfoItem();

        _OrganizationData["default"].constructFromObject(data, obj);

        _RelationshipOrganizationInfoItemAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('people_count')) {
          obj['people_count'] = _ApiClient["default"].convertToType(data['people_count'], 'Number');
          delete data['people_count'];
        }

        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }

        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelationshipOrganizationInfoItem;
}();
/**
 * The name of the organization associated with the item
 * @member {String} name
 */


RelationshipOrganizationInfoItem.prototype['name'] = undefined;
/**
 * The number of people connected with the organization that is associated with the item
 * @member {Number} people_count
 */

RelationshipOrganizationInfoItem.prototype['people_count'] = undefined;
/**
 * The ID of the owner of the organization that is associated with the item
 * @member {Number} owner_id
 */

RelationshipOrganizationInfoItem.prototype['owner_id'] = undefined;
/**
 * The address of the organization
 * @member {String} address
 */

RelationshipOrganizationInfoItem.prototype['address'] = undefined;
/**
 * The BCC email of the organization associated with the item
 * @member {String} cc_email
 */

RelationshipOrganizationInfoItem.prototype['cc_email'] = undefined;
/**
 * The ID of the organization
 * @member {Number} value
 */

RelationshipOrganizationInfoItem.prototype['value'] = undefined; // Implement OrganizationData interface:

/**
 * The name of the organization associated with the item
 * @member {String} name
 */

_OrganizationData["default"].prototype['name'] = undefined;
/**
 * The number of people connected with the organization that is associated with the item
 * @member {Number} people_count
 */

_OrganizationData["default"].prototype['people_count'] = undefined;
/**
 * The ID of the owner of the organization that is associated with the item
 * @member {Number} owner_id
 */

_OrganizationData["default"].prototype['owner_id'] = undefined;
/**
 * The address of the organization
 * @member {String} address
 */

_OrganizationData["default"].prototype['address'] = undefined;
/**
 * The BCC email of the organization associated with the item
 * @member {String} cc_email
 */

_OrganizationData["default"].prototype['cc_email'] = undefined; // Implement RelationshipOrganizationInfoItemAllOf interface:

/**
 * The ID of the organization
 * @member {Number} value
 */

_RelationshipOrganizationInfoItemAllOf["default"].prototype['value'] = undefined;
var _default = RelationshipOrganizationInfoItem;
exports["default"] = _default;
