// pipedrive/dist/model/RelatedDealDataDEALID.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RelatedDealDataDEALID model module.
 * @module model/RelatedDealDataDEALID
 * @version 1.0.0
 */
var RelatedDealDataDEALID = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelatedDealDataDEALID</code>.
   * The ID of the deal which is associated with the item
   * @alias module:model/RelatedDealDataDEALID
   */
  function RelatedDealDataDEALID() {
    (0, _classCallCheck2["default"])(this, RelatedDealDataDEALID);
    RelatedDealDataDEALID.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RelatedDealDataDEALID, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RelatedDealDataDEALID</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelatedDealDataDEALID} obj Optional instance to populate.
     * @return {module:model/RelatedDealDataDEALID} The populated <code>RelatedDealDataDEALID</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelatedDealDataDEALID();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'String');
          delete data['currency'];
        }

        if (data.hasOwnProperty('stage_id')) {
          obj['stage_id'] = _ApiClient["default"].convertToType(data['stage_id'], 'Number');
          delete data['stage_id'];
        }

        if (data.hasOwnProperty('pipeline_id')) {
          obj['pipeline_id'] = _ApiClient["default"].convertToType(data['pipeline_id'], 'Number');
          delete data['pipeline_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RelatedDealDataDEALID;
}();
/**
 * The ID of the deal associated with the item
 * @member {Number} id
 */


RelatedDealDataDEALID.prototype['id'] = undefined;
/**
 * The title of the deal associated with the item
 * @member {String} title
 */

RelatedDealDataDEALID.prototype['title'] = undefined;
/**
 * The status of the deal associated with the item
 * @member {String} status
 */

RelatedDealDataDEALID.prototype['status'] = undefined;
/**
 * The value of the deal that is associated with the item
 * @member {Number} value
 */

RelatedDealDataDEALID.prototype['value'] = undefined;
/**
 * The currency of the deal value
 * @member {String} currency
 */

RelatedDealDataDEALID.prototype['currency'] = undefined;
/**
 * The ID of the stage the deal is currently at
 * @member {Number} stage_id
 */

RelatedDealDataDEALID.prototype['stage_id'] = undefined;
/**
 * The ID of the pipeline the deal is in
 * @member {Number} pipeline_id
 */

RelatedDealDataDEALID.prototype['pipeline_id'] = undefined;
var _default = RelatedDealDataDEALID;
exports["default"] = _default;
