// pipedrive/dist/model/ResponseCallLogObjectAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ResponseCallLogObjectAllOf model module.
 * @module model/ResponseCallLogObjectAllOf
 * @version 1.0.0
 */
var ResponseCallLogObjectAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResponseCallLogObjectAllOf</code>.
   * @alias module:model/ResponseCallLogObjectAllOf
   */
  function ResponseCallLogObjectAllOf() {
    (0, _classCallCheck2["default"])(this, ResponseCallLogObjectAllOf);
    ResponseCallLogObjectAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ResponseCallLogObjectAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ResponseCallLogObjectAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResponseCallLogObjectAllOf} obj Optional instance to populate.
     * @return {module:model/ResponseCallLogObjectAllOf} The populated <code>ResponseCallLogObjectAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResponseCallLogObjectAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
          delete data['id'];
        }

        if (data.hasOwnProperty('has_recording')) {
          obj['has_recording'] = _ApiClient["default"].convertToType(data['has_recording'], 'Boolean');
          delete data['has_recording'];
        }

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ResponseCallLogObjectAllOf;
}();
/**
 * The call log ID, generated when the call log was created
 * @member {String} id
 */


ResponseCallLogObjectAllOf.prototype['id'] = undefined;
/**
 * If the call log has an audio recording attached, the value should be true
 * @member {Boolean} has_recording
 */

ResponseCallLogObjectAllOf.prototype['has_recording'] = undefined;
/**
 * The company ID of the owner of the call log
 * @member {Number} company_id
 */

ResponseCallLogObjectAllOf.prototype['company_id'] = undefined;
var _default = ResponseCallLogObjectAllOf;
exports["default"] = _default;
