// pipedrive/dist/model/ResponseCallLogObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CallLogObject = _interopRequireDefault(require("./CallLogObject"));

var _ResponseCallLogObjectAllOf = _interopRequireDefault(require("./ResponseCallLogObjectAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The ResponseCallLogObject model module.
 * @module model/ResponseCallLogObject
 * @version 1.0.0
 */
var ResponseCallLogObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResponseCallLogObject</code>.
   * @alias module:model/ResponseCallLogObject
   * @implements module:model/CallLogObject
   * @implements module:model/ResponseCallLogObjectAllOf
   * @param outcome {module:model/ResponseCallLogObject.OutcomeEnum} Describes the outcome of the call
   * @param toPhoneNumber {String} The number called
   * @param startTime {Date} The date and time of the start of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
   * @param endTime {Date} The date and time of the end of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
   */
  function ResponseCallLogObject(outcome, toPhoneNumber, startTime, endTime) {
    (0, _classCallCheck2["default"])(this, ResponseCallLogObject);

    _CallLogObject["default"].initialize(this, outcome, toPhoneNumber, startTime, endTime);

    _ResponseCallLogObjectAllOf["default"].initialize(this);

    ResponseCallLogObject.initialize(this, outcome, toPhoneNumber, startTime, endTime);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(ResponseCallLogObject, null, [{
    key: "initialize",
    value: function initialize(obj, outcome, toPhoneNumber, startTime, endTime) {
      obj['outcome'] = outcome;
      obj['to_phone_number'] = toPhoneNumber;
      obj['start_time'] = startTime;
      obj['end_time'] = endTime;
    }
    /**
     * Constructs a <code>ResponseCallLogObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResponseCallLogObject} obj Optional instance to populate.
     * @return {module:model/ResponseCallLogObject} The populated <code>ResponseCallLogObject</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResponseCallLogObject();

        _CallLogObject["default"].constructFromObject(data, obj);

        _ResponseCallLogObjectAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('activity_id')) {
          obj['activity_id'] = _ApiClient["default"].convertToType(data['activity_id'], 'Number');
          delete data['activity_id'];
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
          delete data['subject'];
        }

        if (data.hasOwnProperty('duration')) {
          obj['duration'] = _ApiClient["default"].convertToType(data['duration'], 'String');
          delete data['duration'];
        }

        if (data.hasOwnProperty('outcome')) {
          obj['outcome'] = _ApiClient["default"].convertToType(data['outcome'], 'String');
          delete data['outcome'];
        }

        if (data.hasOwnProperty('from_phone_number')) {
          obj['from_phone_number'] = _ApiClient["default"].convertToType(data['from_phone_number'], 'String');
          delete data['from_phone_number'];
        }

        if (data.hasOwnProperty('to_phone_number')) {
          obj['to_phone_number'] = _ApiClient["default"].convertToType(data['to_phone_number'], 'String');
          delete data['to_phone_number'];
        }

        if (data.hasOwnProperty('start_time')) {
          obj['start_time'] = _ApiClient["default"].convertToType(data['start_time'], 'Date');
          delete data['start_time'];
        }

        if (data.hasOwnProperty('end_time')) {
          obj['end_time'] = _ApiClient["default"].convertToType(data['end_time'], 'Date');
          delete data['end_time'];
        }

        if (data.hasOwnProperty('person_id')) {
          obj['person_id'] = _ApiClient["default"].convertToType(data['person_id'], 'Number');
          delete data['person_id'];
        }

        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('note')) {
          obj['note'] = _ApiClient["default"].convertToType(data['note'], 'String');
          delete data['note'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
          delete data['id'];
        }

        if (data.hasOwnProperty('has_recording')) {
          obj['has_recording'] = _ApiClient["default"].convertToType(data['has_recording'], 'Boolean');
          delete data['has_recording'];
        }

        if (data.hasOwnProperty('company_id')) {
          obj['company_id'] = _ApiClient["default"].convertToType(data['company_id'], 'Number');
          delete data['company_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return ResponseCallLogObject;
}();
/**
 * The ID of the owner of the call log
 * @member {Number} user_id
 */


ResponseCallLogObject.prototype['user_id'] = undefined;
/**
 * If specified, this activity will be converted into a call log, with the information provided. When this field is used, you don't need to specify `deal_id`, `person_id` or `org_id`, as they will be ignored in favor of the values already available in the activity.
 * @member {Number} activity_id
 */

ResponseCallLogObject.prototype['activity_id'] = undefined;
/**
 * The name of the activity this call is attached to
 * @member {String} subject
 */

ResponseCallLogObject.prototype['subject'] = undefined;
/**
 * The duration of the call in seconds
 * @member {String} duration
 */

ResponseCallLogObject.prototype['duration'] = undefined;
/**
 * Describes the outcome of the call
 * @member {module:model/ResponseCallLogObject.OutcomeEnum} outcome
 */

ResponseCallLogObject.prototype['outcome'] = undefined;
/**
 * The number that made the call
 * @member {String} from_phone_number
 */

ResponseCallLogObject.prototype['from_phone_number'] = undefined;
/**
 * The number called
 * @member {String} to_phone_number
 */

ResponseCallLogObject.prototype['to_phone_number'] = undefined;
/**
 * The date and time of the start of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {Date} start_time
 */

ResponseCallLogObject.prototype['start_time'] = undefined;
/**
 * The date and time of the end of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {Date} end_time
 */

ResponseCallLogObject.prototype['end_time'] = undefined;
/**
 * The ID of the person this call is associated with
 * @member {Number} person_id
 */

ResponseCallLogObject.prototype['person_id'] = undefined;
/**
 * The ID of the organization this call is associated with
 * @member {Number} org_id
 */

ResponseCallLogObject.prototype['org_id'] = undefined;
/**
 * The ID of the deal this call is associated with
 * @member {Number} deal_id
 */

ResponseCallLogObject.prototype['deal_id'] = undefined;
/**
 * The note for the call log in HTML format
 * @member {String} note
 */

ResponseCallLogObject.prototype['note'] = undefined;
/**
 * The call log ID, generated when the call log was created
 * @member {String} id
 */

ResponseCallLogObject.prototype['id'] = undefined;
/**
 * If the call log has an audio recording attached, the value should be true
 * @member {Boolean} has_recording
 */

ResponseCallLogObject.prototype['has_recording'] = undefined;
/**
 * The company ID of the owner of the call log
 * @member {Number} company_id
 */

ResponseCallLogObject.prototype['company_id'] = undefined; // Implement CallLogObject interface:

/**
 * The ID of the owner of the call log
 * @member {Number} user_id
 */

_CallLogObject["default"].prototype['user_id'] = undefined;
/**
 * If specified, this activity will be converted into a call log, with the information provided. When this field is used, you don't need to specify `deal_id`, `person_id` or `org_id`, as they will be ignored in favor of the values already available in the activity.
 * @member {Number} activity_id
 */

_CallLogObject["default"].prototype['activity_id'] = undefined;
/**
 * The name of the activity this call is attached to
 * @member {String} subject
 */

_CallLogObject["default"].prototype['subject'] = undefined;
/**
 * The duration of the call in seconds
 * @member {String} duration
 */

_CallLogObject["default"].prototype['duration'] = undefined;
/**
 * Describes the outcome of the call
 * @member {module:model/CallLogObject.OutcomeEnum} outcome
 */

_CallLogObject["default"].prototype['outcome'] = undefined;
/**
 * The number that made the call
 * @member {String} from_phone_number
 */

_CallLogObject["default"].prototype['from_phone_number'] = undefined;
/**
 * The number called
 * @member {String} to_phone_number
 */

_CallLogObject["default"].prototype['to_phone_number'] = undefined;
/**
 * The date and time of the start of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {Date} start_time
 */

_CallLogObject["default"].prototype['start_time'] = undefined;
/**
 * The date and time of the end of the call in UTC. Format: YYYY-MM-DD HH:MM:SS.
 * @member {Date} end_time
 */

_CallLogObject["default"].prototype['end_time'] = undefined;
/**
 * The ID of the person this call is associated with
 * @member {Number} person_id
 */

_CallLogObject["default"].prototype['person_id'] = undefined;
/**
 * The ID of the organization this call is associated with
 * @member {Number} org_id
 */

_CallLogObject["default"].prototype['org_id'] = undefined;
/**
 * The ID of the deal this call is associated with
 * @member {Number} deal_id
 */

_CallLogObject["default"].prototype['deal_id'] = undefined;
/**
 * The note for the call log in HTML format
 * @member {String} note
 */

_CallLogObject["default"].prototype['note'] = undefined; // Implement ResponseCallLogObjectAllOf interface:

/**
 * The call log ID, generated when the call log was created
 * @member {String} id
 */

_ResponseCallLogObjectAllOf["default"].prototype['id'] = undefined;
/**
 * If the call log has an audio recording attached, the value should be true
 * @member {Boolean} has_recording
 */

_ResponseCallLogObjectAllOf["default"].prototype['has_recording'] = undefined;
/**
 * The company ID of the owner of the call log
 * @member {Number} company_id
 */

_ResponseCallLogObjectAllOf["default"].prototype['company_id'] = undefined;
/**
 * Allowed values for the <code>outcome</code> property.
 * @enum {String}
 * @readonly
 */

ResponseCallLogObject['OutcomeEnum'] = {
  /**
   * value: "connected"
   * @const
   */
  "connected": "connected",

  /**
   * value: "no_answer"
   * @const
   */
  "no_answer": "no_answer",

  /**
   * value: "left_message"
   * @const
   */
  "left_message": "left_message",

  /**
   * value: "left_voicemail"
   * @const
   */
  "left_voicemail": "left_voicemail",

  /**
   * value: "wrong_number"
   * @const
   */
  "wrong_number": "wrong_number",

  /**
   * value: "busy"
   * @const
   */
  "busy": "busy"
};
var _default = ResponseCallLogObject;
exports["default"] = _default;
