// pipedrive/dist/model/RolesAdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RolesAdditionalDataPagination = _interopRequireDefault(require("./RolesAdditionalDataPagination"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RolesAdditionalData model module.
 * @module model/RolesAdditionalData
 * @version 1.0.0
 */
var RolesAdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RolesAdditionalData</code>.
   * The additional data in the role list
   * @alias module:model/RolesAdditionalData
   */
  function RolesAdditionalData() {
    (0, _classCallCheck2["default"])(this, RolesAdditionalData);
    RolesAdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RolesAdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RolesAdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RolesAdditionalData} obj Optional instance to populate.
     * @return {module:model/RolesAdditionalData} The populated <code>RolesAdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RolesAdditionalData();

        if (data.hasOwnProperty('pagination')) {
          obj['pagination'] = _RolesAdditionalDataPagination["default"].constructFromObject(data['pagination']);
          delete data['pagination'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RolesAdditionalData;
}();
/**
 * @member {module:model/RolesAdditionalDataPagination} pagination
 */


RolesAdditionalData.prototype['pagination'] = undefined;
var _default = RolesAdditionalData;
exports["default"] = _default;
