// pipedrive/dist/model/RoleAssignmentAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The RoleAssignmentAllOf model module.
 * @module model/RoleAssignmentAllOf
 * @version 1.0.0
 */
var RoleAssignmentAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RoleAssignmentAllOf</code>.
   * @alias module:model/RoleAssignmentAllOf
   */
  function RoleAssignmentAllOf() {
    (0, _classCallCheck2["default"])(this, RoleAssignmentAllOf);
    RoleAssignmentAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(RoleAssignmentAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RoleAssignmentAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RoleAssignmentAllOf} obj Optional instance to populate.
     * @return {module:model/RoleAssignmentAllOf} The populated <code>RoleAssignmentAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RoleAssignmentAllOf();

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], 'Number');
          delete data['role_id'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return RoleAssignmentAllOf;
}();
/**
 * The user ID
 * @member {Number} user_id
 */


RoleAssignmentAllOf.prototype['user_id'] = undefined;
/**
 * The role ID
 * @member {Number} role_id
 */

RoleAssignmentAllOf.prototype['role_id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

RoleAssignmentAllOf.prototype['active_flag'] = undefined;
/**
 * The assignment type
 * @member {String} type
 */

RoleAssignmentAllOf.prototype['type'] = undefined;
var _default = RoleAssignmentAllOf;
exports["default"] = _default;
