// pipedrive/dist/model/SinglePermissionSetsItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PermissionSetsItem = _interopRequireDefault(require("./PermissionSetsItem"));

var _SinglePermissionSetsItemAllOf = _interopRequireDefault(require("./SinglePermissionSetsItemAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SinglePermissionSetsItem model module.
 * @module model/SinglePermissionSetsItem
 * @version 1.0.0
 */
var SinglePermissionSetsItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SinglePermissionSetsItem</code>.
   * @alias module:model/SinglePermissionSetsItem
   * @implements module:model/PermissionSetsItem
   * @implements module:model/SinglePermissionSetsItemAllOf
   */
  function SinglePermissionSetsItem() {
    (0, _classCallCheck2["default"])(this, SinglePermissionSetsItem);

    _PermissionSetsItem["default"].initialize(this);

    _SinglePermissionSetsItemAllOf["default"].initialize(this);

    SinglePermissionSetsItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SinglePermissionSetsItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SinglePermissionSetsItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SinglePermissionSetsItem} obj Optional instance to populate.
     * @return {module:model/SinglePermissionSetsItem} The populated <code>SinglePermissionSetsItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SinglePermissionSetsItem();

        _PermissionSetsItem["default"].constructFromObject(data, obj);

        _SinglePermissionSetsItemAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('assignment_count')) {
          obj['assignment_count'] = _ApiClient["default"].convertToType(data['assignment_count'], 'Number');
          delete data['assignment_count'];
        }

        if (data.hasOwnProperty('contents')) {
          obj['contents'] = _ApiClient["default"].convertToType(data['contents'], ['String']);
          delete data['contents'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SinglePermissionSetsItem;
}();
/**
 * The ID of user permission set
 * @member {Number} id
 */


SinglePermissionSetsItem.prototype['id'] = undefined;
/**
 * The name of the permission set
 * @member {String} name
 */

SinglePermissionSetsItem.prototype['name'] = undefined;
/**
 * The number of users assigned to this permission set
 * @member {Number} assignment_count
 */

SinglePermissionSetsItem.prototype['assignment_count'] = undefined;
/**
 * A permission assigned to this permission set
 * @member {Array.<String>} contents
 */

SinglePermissionSetsItem.prototype['contents'] = undefined; // Implement PermissionSetsItem interface:

/**
 * The ID of user permission set
 * @member {Number} id
 */

_PermissionSetsItem["default"].prototype['id'] = undefined;
/**
 * The name of the permission set
 * @member {String} name
 */

_PermissionSetsItem["default"].prototype['name'] = undefined;
/**
 * The number of users assigned to this permission set
 * @member {Number} assignment_count
 */

_PermissionSetsItem["default"].prototype['assignment_count'] = undefined; // Implement SinglePermissionSetsItemAllOf interface:

/**
 * A permission assigned to this permission set
 * @member {Array.<String>} contents
 */

_SinglePermissionSetsItemAllOf["default"].prototype['contents'] = undefined;
var _default = SinglePermissionSetsItem;
exports["default"] = _default;
