// pipedrive/dist/model/StageConversions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The StageConversions model module.
 * @module model/StageConversions
 * @version 1.0.0
 */
var StageConversions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StageConversions</code>.
   * @alias module:model/StageConversions
   */
  function StageConversions() {
    (0, _classCallCheck2["default"])(this, StageConversions);
    StageConversions.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(StageConversions, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>StageConversions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StageConversions} obj Optional instance to populate.
     * @return {module:model/StageConversions} The populated <code>StageConversions</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StageConversions();

        if (data.hasOwnProperty('from_stage_id')) {
          obj['from_stage_id'] = _ApiClient["default"].convertToType(data['from_stage_id'], 'Number');
          delete data['from_stage_id'];
        }

        if (data.hasOwnProperty('to_stage_id')) {
          obj['to_stage_id'] = _ApiClient["default"].convertToType(data['to_stage_id'], 'Number');
          delete data['to_stage_id'];
        }

        if (data.hasOwnProperty('conversion_rate')) {
          obj['conversion_rate'] = _ApiClient["default"].convertToType(data['conversion_rate'], 'Number');
          delete data['conversion_rate'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return StageConversions;
}();
/**
 * The stage ID from where conversion starts
 * @member {Number} from_stage_id
 */


StageConversions.prototype['from_stage_id'] = undefined;
/**
 * The stage ID to where conversion ends
 * @member {Number} to_stage_id
 */

StageConversions.prototype['to_stage_id'] = undefined;
/**
 * The conversion rate
 * @member {Number} conversion_rate
 */

StageConversions.prototype['conversion_rate'] = undefined;
var _default = StageConversions;
exports["default"] = _default;
