// pipedrive/dist/model/StageWithPipelineInfoAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The StageWithPipelineInfoAllOf model module.
 * @module model/StageWithPipelineInfoAllOf
 * @version 1.0.0
 */
var StageWithPipelineInfoAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StageWithPipelineInfoAllOf</code>.
   * @alias module:model/StageWithPipelineInfoAllOf
   */
  function StageWithPipelineInfoAllOf() {
    (0, _classCallCheck2["default"])(this, StageWithPipelineInfoAllOf);
    StageWithPipelineInfoAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(StageWithPipelineInfoAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>StageWithPipelineInfoAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StageWithPipelineInfoAllOf} obj Optional instance to populate.
     * @return {module:model/StageWithPipelineInfoAllOf} The populated <code>StageWithPipelineInfoAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StageWithPipelineInfoAllOf();

        if (data.hasOwnProperty('pipeline_name')) {
          obj['pipeline_name'] = _ApiClient["default"].convertToType(data['pipeline_name'], 'String');
          delete data['pipeline_name'];
        }

        if (data.hasOwnProperty('pipeline_deal_probability')) {
          obj['pipeline_deal_probability'] = _ApiClient["default"].convertToType(data['pipeline_deal_probability'], 'Boolean');
          delete data['pipeline_deal_probability'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return StageWithPipelineInfoAllOf;
}();
/**
 * The name of the pipeline
 * @member {String} pipeline_name
 */


StageWithPipelineInfoAllOf.prototype['pipeline_name'] = undefined;
/**
 * The pipeline deal probability. When `true`, overrides the stage probability.
 * @member {Boolean} pipeline_deal_probability
 */

StageWithPipelineInfoAllOf.prototype['pipeline_deal_probability'] = undefined;
var _default = StageWithPipelineInfoAllOf;
exports["default"] = _default;
