// pipedrive/dist/model/SubRoleAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubRoleAllOf model module.
 * @module model/SubRoleAllOf
 * @version 1.0.0
 */
var SubRoleAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubRoleAllOf</code>.
   * @alias module:model/SubRoleAllOf
   */
  function SubRoleAllOf() {
    (0, _classCallCheck2["default"])(this, SubRoleAllOf);
    SubRoleAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubRoleAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SubRoleAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubRoleAllOf} obj Optional instance to populate.
     * @return {module:model/SubRoleAllOf} The populated <code>SubRoleAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubRoleAllOf();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('assignment_count')) {
          obj['assignment_count'] = _ApiClient["default"].convertToType(data['assignment_count'], 'String');
          delete data['assignment_count'];
        }

        if (data.hasOwnProperty('sub_role_count')) {
          obj['sub_role_count'] = _ApiClient["default"].convertToType(data['sub_role_count'], 'String');
          delete data['sub_role_count'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubRoleAllOf;
}();
/**
 * The ID of the role
 * @member {Number} id
 */


SubRoleAllOf.prototype['id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

SubRoleAllOf.prototype['active_flag'] = undefined;
/**
 * The number of users assigned to this role
 * @member {String} assignment_count
 */

SubRoleAllOf.prototype['assignment_count'] = undefined;
/**
 * The number of sub-roles
 * @member {String} sub_role_count
 */

SubRoleAllOf.prototype['sub_role_count'] = undefined;
var _default = SubRoleAllOf;
exports["default"] = _default;
