// pipedrive/dist/model/SubRole.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseRole = _interopRequireDefault(require("./BaseRole"));

var _SubRoleAllOf = _interopRequireDefault(require("./SubRoleAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubRole model module.
 * @module model/SubRole
 * @version 1.0.0
 */
var SubRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubRole</code>.
   * The details of the sub-role
   * @alias module:model/SubRole
   * @implements module:model/BaseRole
   * @implements module:model/SubRoleAllOf
   */
  function SubRole() {
    (0, _classCallCheck2["default"])(this, SubRole);

    _BaseRole["default"].initialize(this);

    _SubRoleAllOf["default"].initialize(this);

    SubRole.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubRole, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SubRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubRole} obj Optional instance to populate.
     * @return {module:model/SubRole} The populated <code>SubRole</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubRole();

        _BaseRole["default"].constructFromObject(data, obj);

        _SubRoleAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('parent_role_id')) {
          obj['parent_role_id'] = _ApiClient["default"].convertToType(data['parent_role_id'], 'Number');
          delete data['parent_role_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('assignment_count')) {
          obj['assignment_count'] = _ApiClient["default"].convertToType(data['assignment_count'], 'String');
          delete data['assignment_count'];
        }

        if (data.hasOwnProperty('sub_role_count')) {
          obj['sub_role_count'] = _ApiClient["default"].convertToType(data['sub_role_count'], 'String');
          delete data['sub_role_count'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubRole;
}();
/**
 * The ID of the parent role
 * @member {Number} parent_role_id
 */


SubRole.prototype['parent_role_id'] = undefined;
/**
 * The name of the role
 * @member {String} name
 */

SubRole.prototype['name'] = undefined;
/**
 * The ID of the role
 * @member {Number} id
 */

SubRole.prototype['id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

SubRole.prototype['active_flag'] = undefined;
/**
 * The number of users assigned to this role
 * @member {String} assignment_count
 */

SubRole.prototype['assignment_count'] = undefined;
/**
 * The number of sub-roles
 * @member {String} sub_role_count
 */

SubRole.prototype['sub_role_count'] = undefined; // Implement BaseRole interface:

/**
 * The ID of the parent role
 * @member {Number} parent_role_id
 */

_BaseRole["default"].prototype['parent_role_id'] = undefined;
/**
 * The name of the role
 * @member {String} name
 */

_BaseRole["default"].prototype['name'] = undefined; // Implement SubRoleAllOf interface:

/**
 * The ID of the role
 * @member {Number} id
 */

_SubRoleAllOf["default"].prototype['id'] = undefined;
/**
 * Whether the role is active or not
 * @member {Boolean} active_flag
 */

_SubRoleAllOf["default"].prototype['active_flag'] = undefined;
/**
 * The number of users assigned to this role
 * @member {String} assignment_count
 */

_SubRoleAllOf["default"].prototype['assignment_count'] = undefined;
/**
 * The number of sub-roles
 * @member {String} sub_role_count
 */

_SubRoleAllOf["default"].prototype['sub_role_count'] = undefined;
var _default = SubRole;
exports["default"] = _default;
