// pipedrive/dist/model/SubscriptionsIdResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _SubscriptionItem = _interopRequireDefault(require("./SubscriptionItem"));

var _SubscriptionsIdResponseAllOf = _interopRequireDefault(require("./SubscriptionsIdResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubscriptionsIdResponse model module.
 * @module model/SubscriptionsIdResponse
 * @version 1.0.0
 */
var SubscriptionsIdResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubscriptionsIdResponse</code>.
   * @alias module:model/SubscriptionsIdResponse
   * @implements module:model/BaseResponse
   * @implements module:model/SubscriptionsIdResponseAllOf
   */
  function SubscriptionsIdResponse() {
    (0, _classCallCheck2["default"])(this, SubscriptionsIdResponse);

    _BaseResponse["default"].initialize(this);

    _SubscriptionsIdResponseAllOf["default"].initialize(this);

    SubscriptionsIdResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubscriptionsIdResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SubscriptionsIdResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubscriptionsIdResponse} obj Optional instance to populate.
     * @return {module:model/SubscriptionsIdResponse} The populated <code>SubscriptionsIdResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubscriptionsIdResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _SubscriptionsIdResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _SubscriptionItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubscriptionsIdResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


SubscriptionsIdResponse.prototype['success'] = undefined;
/**
 * @member {module:model/SubscriptionItem} data
 */

SubscriptionsIdResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement SubscriptionsIdResponseAllOf interface:

/**
 * @member {module:model/SubscriptionItem} data
 */

_SubscriptionsIdResponseAllOf["default"].prototype['data'] = undefined;
var _default = SubscriptionsIdResponse;
exports["default"] = _default;
