// pipedrive/dist/model/SubscriptionRecurringCancelRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubscriptionRecurringCancelRequest model module.
 * @module model/SubscriptionRecurringCancelRequest
 * @version 1.0.0
 */
var SubscriptionRecurringCancelRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubscriptionRecurringCancelRequest</code>.
   * @alias module:model/SubscriptionRecurringCancelRequest
   */
  function SubscriptionRecurringCancelRequest() {
    (0, _classCallCheck2["default"])(this, SubscriptionRecurringCancelRequest);
    SubscriptionRecurringCancelRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubscriptionRecurringCancelRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SubscriptionRecurringCancelRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubscriptionRecurringCancelRequest} obj Optional instance to populate.
     * @return {module:model/SubscriptionRecurringCancelRequest} The populated <code>SubscriptionRecurringCancelRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubscriptionRecurringCancelRequest();

        if (data.hasOwnProperty('end_date')) {
          obj['end_date'] = _ApiClient["default"].convertToType(data['end_date'], 'Date');
          delete data['end_date'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubscriptionRecurringCancelRequest;
}();
/**
 * The subscription termination date. All payments after specified date will be deleted. Default value is the current date.
 * @member {Date} end_date
 */


SubscriptionRecurringCancelRequest.prototype['end_date'] = undefined;
var _default = SubscriptionRecurringCancelRequest;
exports["default"] = _default;
