// pipedrive/dist/model/SubscriptionInstallmentCreateRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubscriptionInstallmentCreateRequest model module.
 * @module model/SubscriptionInstallmentCreateRequest
 * @version 1.0.0
 */
var SubscriptionInstallmentCreateRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubscriptionInstallmentCreateRequest</code>.
   * @alias module:model/SubscriptionInstallmentCreateRequest
   * @param dealId {Number} The ID of the deal this installment subscription is associated with
   * @param currency {String} The currency of the installment subscription. Accepts a 3-character currency code.
   * @param payments {Array.<Object>} Array of payments. It requires a minimum structure as follows: [{ amount:SUM, description:DESCRIPTION, due_at:PAYMENT_DATE }]. Replace SUM with a payment amount, DESCRIPTION with an explanation string, PAYMENT_DATE with a date (format YYYY-MM-DD).
   */
  function SubscriptionInstallmentCreateRequest(dealId, currency, payments) {
    (0, _classCallCheck2["default"])(this, SubscriptionInstallmentCreateRequest);
    SubscriptionInstallmentCreateRequest.initialize(this, dealId, currency, payments);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubscriptionInstallmentCreateRequest, null, [{
    key: "initialize",
    value: function initialize(obj, dealId, currency, payments) {
      obj['deal_id'] = dealId;
      obj['currency'] = currency;
      obj['payments'] = payments;
    }
    /**
     * Constructs a <code>SubscriptionInstallmentCreateRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubscriptionInstallmentCreateRequest} obj Optional instance to populate.
     * @return {module:model/SubscriptionInstallmentCreateRequest} The populated <code>SubscriptionInstallmentCreateRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubscriptionInstallmentCreateRequest();

        if (data.hasOwnProperty('deal_id')) {
          obj['deal_id'] = _ApiClient["default"].convertToType(data['deal_id'], 'Number');
          delete data['deal_id'];
        }

        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'String');
          delete data['currency'];
        }

        if (data.hasOwnProperty('payments')) {
          obj['payments'] = _ApiClient["default"].convertToType(data['payments'], [Object]);
          delete data['payments'];
        }

        if (data.hasOwnProperty('update_deal_value')) {
          obj['update_deal_value'] = _ApiClient["default"].convertToType(data['update_deal_value'], 'Boolean');
          delete data['update_deal_value'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubscriptionInstallmentCreateRequest;
}();
/**
 * The ID of the deal this installment subscription is associated with
 * @member {Number} deal_id
 */


SubscriptionInstallmentCreateRequest.prototype['deal_id'] = undefined;
/**
 * The currency of the installment subscription. Accepts a 3-character currency code.
 * @member {String} currency
 */

SubscriptionInstallmentCreateRequest.prototype['currency'] = undefined;
/**
 * Array of payments. It requires a minimum structure as follows: [{ amount:SUM, description:DESCRIPTION, due_at:PAYMENT_DATE }]. Replace SUM with a payment amount, DESCRIPTION with an explanation string, PAYMENT_DATE with a date (format YYYY-MM-DD).
 * @member {Array.<Object>} payments
 */

SubscriptionInstallmentCreateRequest.prototype['payments'] = undefined;
/**
 * Indicates that the deal value must be set to the installment subscription's total value
 * @member {Boolean} update_deal_value
 */

SubscriptionInstallmentCreateRequest.prototype['update_deal_value'] = undefined;
var _default = SubscriptionInstallmentCreateRequest;
exports["default"] = _default;
