// pipedrive/dist/model/Teams.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _BaseTeam = _interopRequireDefault(require("./BaseTeam"));

var _TeamsAllOf = _interopRequireDefault(require("./TeamsAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The Teams model module.
 * @module model/Teams
 * @version 1.0.0
 */
var Teams = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Teams</code>.
   * @alias module:model/Teams
   * @implements module:model/BaseResponse
   * @implements module:model/TeamsAllOf
   */
  function Teams() {
    (0, _classCallCheck2["default"])(this, Teams);

    _BaseResponse["default"].initialize(this);

    _TeamsAllOf["default"].initialize(this);

    Teams.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(Teams, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Teams</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Teams} obj Optional instance to populate.
     * @return {module:model/Teams} The populated <code>Teams</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Teams();

        _BaseResponse["default"].constructFromObject(data, obj);

        _TeamsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseTeam["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return Teams;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


Teams.prototype['success'] = undefined;
/**
 * @member {Array.<module:model/BaseTeam>} data
 */

Teams.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement TeamsAllOf interface:

/**
 * @member {Array.<module:model/BaseTeam>} data
 */

_TeamsAllOf["default"].prototype['data'] = undefined;
var _default = Teams;
exports["default"] = _default;
