// pipedrive/dist/model/UpdateProductResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GetActivitiesResponse200RelatedObjects = _interopRequireDefault(require("./GetActivitiesResponse200RelatedObjects"));

var _ProductWithObjectPrices = _interopRequireDefault(require("./ProductWithObjectPrices"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateProductResponse model module.
 * @module model/UpdateProductResponse
 * @version 1.0.0
 */
var UpdateProductResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateProductResponse</code>.
   * @alias module:model/UpdateProductResponse
   */
  function UpdateProductResponse() {
    (0, _classCallCheck2["default"])(this, UpdateProductResponse);
    UpdateProductResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateProductResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateProductResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateProductResponse} obj Optional instance to populate.
     * @return {module:model/UpdateProductResponse} The populated <code>UpdateProductResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateProductResponse();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ProductWithObjectPrices["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _GetActivitiesResponse200RelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateProductResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


UpdateProductResponse.prototype['success'] = undefined;
/**
 * @member {module:model/ProductWithObjectPrices} data
 */

UpdateProductResponse.prototype['data'] = undefined;
/**
 * @member {module:model/GetActivitiesResponse200RelatedObjects} related_objects
 */

UpdateProductResponse.prototype['related_objects'] = undefined;
var _default = UpdateProductResponse;
exports["default"] = _default;
