// pipedrive/dist/model/UpdateUserRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateUserRequest model module.
 * @module model/UpdateUserRequest
 * @version 1.0.0
 */
var UpdateUserRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateUserRequest</code>.
   * @alias module:model/UpdateUserRequest
   * @param activeFlag {Boolean} Whether the user is active or not. `false` = Not activated, `true` = Activated
   */
  function UpdateUserRequest(activeFlag) {
    (0, _classCallCheck2["default"])(this, UpdateUserRequest);
    UpdateUserRequest.initialize(this, activeFlag);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateUserRequest, null, [{
    key: "initialize",
    value: function initialize(obj, activeFlag) {
      obj['active_flag'] = activeFlag || true;
    }
    /**
     * Constructs a <code>UpdateUserRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateUserRequest} obj Optional instance to populate.
     * @return {module:model/UpdateUserRequest} The populated <code>UpdateUserRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateUserRequest();

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }

        if (Object.keys(data).length > 0) {
          obj['extra'] = data;
        }
      }

      return obj;
    }
  }]);
  return UpdateUserRequest;
}();
/**
 * Whether the user is active or not. `false` = Not activated, `true` = Activated
 * @member {Boolean} active_flag
 * @default true
 */


UpdateUserRequest.prototype['active_flag'] = true;
var _default = UpdateUserRequest;
exports["default"] = _default;
