// pipedrive/dist/model/UpdateStageRequestAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateStageRequestAllOf model module.
 * @module model/UpdateStageRequestAllOf
 * @version 1.0.0
 */
var UpdateStageRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateStageRequestAllOf</code>.
   * @alias module:model/UpdateStageRequestAllOf
   */
  function UpdateStageRequestAllOf() {
    (0, _classCallCheck2["default"])(this, UpdateStageRequestAllOf);
    UpdateStageRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateStageRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateStageRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateStageRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateStageRequestAllOf} The populated <code>UpdateStageRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateStageRequestAllOf();

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateStageRequestAllOf;
}();
/**
 * An order number for this stage. Order numbers should be used to order the stages in the pipeline.
 * @member {Number} order_nr
 */


UpdateStageRequestAllOf.prototype['order_nr'] = undefined;
var _default = UpdateStageRequestAllOf;
exports["default"] = _default;
