// pipedrive/dist/model/UpdateActivityResponse200.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActivityResponseObject = _interopRequireDefault(require("./ActivityResponseObject"));

var _AddActivityResponse200RelatedObjects = _interopRequireDefault(require("./AddActivityResponse200RelatedObjects"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateActivityResponse200 model module.
 * @module model/UpdateActivityResponse200
 * @version 1.0.0
 */
var UpdateActivityResponse200 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateActivityResponse200</code>.
   * @alias module:model/UpdateActivityResponse200
   */
  function UpdateActivityResponse200() {
    (0, _classCallCheck2["default"])(this, UpdateActivityResponse200);
    UpdateActivityResponse200.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateActivityResponse200, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateActivityResponse200</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateActivityResponse200} obj Optional instance to populate.
     * @return {module:model/UpdateActivityResponse200} The populated <code>UpdateActivityResponse200</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateActivityResponse200();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ActivityResponseObject["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('related_objects')) {
          obj['related_objects'] = _AddActivityResponse200RelatedObjects["default"].constructFromObject(data['related_objects']);
          delete data['related_objects'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateActivityResponse200;
}();
/**
 * @member {Boolean} success
 */


UpdateActivityResponse200.prototype['success'] = undefined;
/**
 * @member {module:model/ActivityResponseObject} data
 */

UpdateActivityResponse200.prototype['data'] = undefined;
/**
 * @member {module:model/AddActivityResponse200RelatedObjects} related_objects
 */

UpdateActivityResponse200.prototype['related_objects'] = undefined;
var _default = UpdateActivityResponse200;
exports["default"] = _default;
