// pipedrive/dist/model/UpdateTeamWithAdditionalProperties.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseTeamAdditionalProperties = _interopRequireDefault(require("./BaseTeamAdditionalProperties"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

var _UpdateTeam = _interopRequireDefault(require("./UpdateTeam"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateTeamWithAdditionalProperties model module.
 * @module model/UpdateTeamWithAdditionalProperties
 * @version 1.0.0
 */
var UpdateTeamWithAdditionalProperties = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTeamWithAdditionalProperties</code>.
   * @alias module:model/UpdateTeamWithAdditionalProperties
   * @implements module:model/UpdateTeam
   * @implements module:model/BaseTeamAdditionalProperties
   * @param name {String} The team name
   * @param managerId {Number} The team manager ID
   */
  function UpdateTeamWithAdditionalProperties(name, managerId) {
    (0, _classCallCheck2["default"])(this, UpdateTeamWithAdditionalProperties);

    _UpdateTeam["default"].initialize(this, name, managerId);

    _BaseTeamAdditionalProperties["default"].initialize(this);

    UpdateTeamWithAdditionalProperties.initialize(this, name, managerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateTeamWithAdditionalProperties, null, [{
    key: "initialize",
    value: function initialize(obj, name, managerId) {
      obj['name'] = name;
      obj['manager_id'] = managerId;
    }
    /**
     * Constructs a <code>UpdateTeamWithAdditionalProperties</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTeamWithAdditionalProperties} obj Optional instance to populate.
     * @return {module:model/UpdateTeamWithAdditionalProperties} The populated <code>UpdateTeamWithAdditionalProperties</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTeamWithAdditionalProperties();

        _UpdateTeam["default"].constructFromObject(data, obj);

        _BaseTeamAdditionalProperties["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }

        if (data.hasOwnProperty('manager_id')) {
          obj['manager_id'] = _ApiClient["default"].convertToType(data['manager_id'], 'Number');
          delete data['manager_id'];
        }

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], ['Number']);
          delete data['users'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], _NumberBoolean["default"]);
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBoolean["default"]);
          delete data['deleted_flag'];
        }

        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }

        if (data.hasOwnProperty('created_by_user_id')) {
          obj['created_by_user_id'] = _ApiClient["default"].convertToType(data['created_by_user_id'], 'Number');
          delete data['created_by_user_id'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateTeamWithAdditionalProperties;
}();
/**
 * The team name
 * @member {String} name
 */


UpdateTeamWithAdditionalProperties.prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

UpdateTeamWithAdditionalProperties.prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

UpdateTeamWithAdditionalProperties.prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

UpdateTeamWithAdditionalProperties.prototype['users'] = undefined;
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

UpdateTeamWithAdditionalProperties.prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

UpdateTeamWithAdditionalProperties.prototype['deleted_flag'] = undefined;
/**
 * The team creation time. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

UpdateTeamWithAdditionalProperties.prototype['add_time'] = undefined;
/**
 * The ID of the user who created the team
 * @member {Number} created_by_user_id
 */

UpdateTeamWithAdditionalProperties.prototype['created_by_user_id'] = undefined; // Implement UpdateTeam interface:

/**
 * The team name
 * @member {String} name
 */

_UpdateTeam["default"].prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

_UpdateTeam["default"].prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

_UpdateTeam["default"].prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

_UpdateTeam["default"].prototype['users'] = undefined;
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

_UpdateTeam["default"].prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

_UpdateTeam["default"].prototype['deleted_flag'] = undefined; // Implement BaseTeamAdditionalProperties interface:

/**
 * The team creation time. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */

_BaseTeamAdditionalProperties["default"].prototype['add_time'] = undefined;
/**
 * The ID of the user who created the team
 * @member {Number} created_by_user_id
 */

_BaseTeamAdditionalProperties["default"].prototype['created_by_user_id'] = undefined;
var _default = UpdateTeamWithAdditionalProperties;
exports["default"] = _default;
