// pipedrive/dist/model/UpdateTeam.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateTeam = _interopRequireDefault(require("./CreateTeam"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

var _UpdateTeamAllOf = _interopRequireDefault(require("./UpdateTeamAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateTeam model module.
 * @module model/UpdateTeam
 * @version 1.0.0
 */
var UpdateTeam = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTeam</code>.
   * @alias module:model/UpdateTeam
   * @implements module:model/CreateTeam
   * @implements module:model/UpdateTeamAllOf
   * @param name {String} The team name
   * @param managerId {Number} The team manager ID
   */
  function UpdateTeam(name, managerId) {
    (0, _classCallCheck2["default"])(this, UpdateTeam);

    _CreateTeam["default"].initialize(this, name, managerId);

    _UpdateTeamAllOf["default"].initialize(this);

    UpdateTeam.initialize(this, name, managerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateTeam, null, [{
    key: "initialize",
    value: function initialize(obj, name, managerId) {
      obj['name'] = name;
      obj['manager_id'] = managerId;
    }
    /**
     * Constructs a <code>UpdateTeam</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTeam} obj Optional instance to populate.
     * @return {module:model/UpdateTeam} The populated <code>UpdateTeam</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTeam();

        _CreateTeam["default"].constructFromObject(data, obj);

        _UpdateTeamAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
          delete data['description'];
        }

        if (data.hasOwnProperty('manager_id')) {
          obj['manager_id'] = _ApiClient["default"].convertToType(data['manager_id'], 'Number');
          delete data['manager_id'];
        }

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], ['Number']);
          delete data['users'];
        }

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], _NumberBoolean["default"]);
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBoolean["default"]);
          delete data['deleted_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateTeam;
}();
/**
 * The team name
 * @member {String} name
 */


UpdateTeam.prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

UpdateTeam.prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

UpdateTeam.prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

UpdateTeam.prototype['users'] = undefined;
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

UpdateTeam.prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

UpdateTeam.prototype['deleted_flag'] = undefined; // Implement CreateTeam interface:

/**
 * The team name
 * @member {String} name
 */

_CreateTeam["default"].prototype['name'] = undefined;
/**
 * The team description
 * @member {String} description
 */

_CreateTeam["default"].prototype['description'] = undefined;
/**
 * The team manager ID
 * @member {Number} manager_id
 */

_CreateTeam["default"].prototype['manager_id'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} users
 */

_CreateTeam["default"].prototype['users'] = undefined; // Implement UpdateTeamAllOf interface:

/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */

_UpdateTeamAllOf["default"].prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

_UpdateTeamAllOf["default"].prototype['deleted_flag'] = undefined;
var _default = UpdateTeam;
exports["default"] = _default;
