// pipedrive/dist/model/UpdateStageRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Stage = _interopRequireDefault(require("./Stage"));

var _UpdateStageRequestAllOf = _interopRequireDefault(require("./UpdateStageRequestAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateStageRequest model module.
 * @module model/UpdateStageRequest
 * @version 1.0.0
 */
var UpdateStageRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateStageRequest</code>.
   * @alias module:model/UpdateStageRequest
   * @implements module:model/Stage
   * @implements module:model/UpdateStageRequestAllOf
   */
  function UpdateStageRequest() {
    (0, _classCallCheck2["default"])(this, UpdateStageRequest);

    _Stage["default"].initialize(this);

    _UpdateStageRequestAllOf["default"].initialize(this);

    UpdateStageRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateStageRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateStageRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateStageRequest} obj Optional instance to populate.
     * @return {module:model/UpdateStageRequest} The populated <code>UpdateStageRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateStageRequest();

        _Stage["default"].constructFromObject(data, obj);

        _UpdateStageRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (data.hasOwnProperty('pipeline_id')) {
          obj['pipeline_id'] = _ApiClient["default"].convertToType(data['pipeline_id'], 'Number');
          delete data['pipeline_id'];
        }

        if (data.hasOwnProperty('deal_probability')) {
          obj['deal_probability'] = _ApiClient["default"].convertToType(data['deal_probability'], 'Number');
          delete data['deal_probability'];
        }

        if (data.hasOwnProperty('rotten_flag')) {
          obj['rotten_flag'] = _ApiClient["default"].convertToType(data['rotten_flag'], 'Number');
          delete data['rotten_flag'];
        }

        if (data.hasOwnProperty('rotten_days')) {
          obj['rotten_days'] = _ApiClient["default"].convertToType(data['rotten_days'], 'Number');
          delete data['rotten_days'];
        }

        if (data.hasOwnProperty('order_nr')) {
          obj['order_nr'] = _ApiClient["default"].convertToType(data['order_nr'], 'Number');
          delete data['order_nr'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateStageRequest;
}();
/**
 * The name of the stage
 * @member {String} name
 */


UpdateStageRequest.prototype['name'] = undefined;
/**
 * The ID of the pipeline to add stage to
 * @member {Number} pipeline_id
 */

UpdateStageRequest.prototype['pipeline_id'] = undefined;
/**
 * The success probability percentage of the deal. Used/shown when deal weighted values are used.
 * @member {Number} deal_probability
 */

UpdateStageRequest.prototype['deal_probability'] = undefined;
/**
 * Whether deals in this stage can become rotten
 * @member {module:model/UpdateStageRequest.RottenFlagEnum} rotten_flag
 */

UpdateStageRequest.prototype['rotten_flag'] = undefined;
/**
 * The number of days the deals not updated in this stage would become rotten. Applies only if the `rotten_flag` is set.
 * @member {Number} rotten_days
 */

UpdateStageRequest.prototype['rotten_days'] = undefined;
/**
 * An order number for this stage. Order numbers should be used to order the stages in the pipeline.
 * @member {Number} order_nr
 */

UpdateStageRequest.prototype['order_nr'] = undefined; // Implement Stage interface:

/**
 * The name of the stage
 * @member {String} name
 */

_Stage["default"].prototype['name'] = undefined;
/**
 * The ID of the pipeline to add stage to
 * @member {Number} pipeline_id
 */

_Stage["default"].prototype['pipeline_id'] = undefined;
/**
 * The success probability percentage of the deal. Used/shown when deal weighted values are used.
 * @member {Number} deal_probability
 */

_Stage["default"].prototype['deal_probability'] = undefined;
/**
 * Whether deals in this stage can become rotten
 * @member {module:model/Stage.RottenFlagEnum} rotten_flag
 */

_Stage["default"].prototype['rotten_flag'] = undefined;
/**
 * The number of days the deals not updated in this stage would become rotten. Applies only if the `rotten_flag` is set.
 * @member {Number} rotten_days
 */

_Stage["default"].prototype['rotten_days'] = undefined; // Implement UpdateStageRequestAllOf interface:

/**
 * An order number for this stage. Order numbers should be used to order the stages in the pipeline.
 * @member {Number} order_nr
 */

_UpdateStageRequestAllOf["default"].prototype['order_nr'] = undefined;
/**
 * Allowed values for the <code>rotten_flag</code> property.
 * @enum {Number}
 * @readonly
 */

UpdateStageRequest['RottenFlagEnum'] = {
  /**
   * value: 0
   * @const
   */
  "0": 0,

  /**
   * value: 1
   * @const
   */
  "1": 1
};
var _default = UpdateStageRequest;
exports["default"] = _default;
