// pipedrive/dist/model/UpdateTeamAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NumberBoolean = _interopRequireDefault(require("./NumberBoolean"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UpdateTeamAllOf model module.
 * @module model/UpdateTeamAllOf
 * @version 1.0.0
 */
var UpdateTeamAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTeamAllOf</code>.
   * @alias module:model/UpdateTeamAllOf
   */
  function UpdateTeamAllOf() {
    (0, _classCallCheck2["default"])(this, UpdateTeamAllOf);
    UpdateTeamAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UpdateTeamAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateTeamAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTeamAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateTeamAllOf} The populated <code>UpdateTeamAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTeamAllOf();

        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], _NumberBoolean["default"]);
          delete data['active_flag'];
        }

        if (data.hasOwnProperty('deleted_flag')) {
          obj['deleted_flag'] = _ApiClient["default"].convertToType(data['deleted_flag'], _NumberBoolean["default"]);
          delete data['deleted_flag'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UpdateTeamAllOf;
}();
/**
 * Flag that indicates whether the team is active
 * @member {module:model/NumberBoolean} active_flag
 */


UpdateTeamAllOf.prototype['active_flag'] = undefined;
/**
 * Flag that indicates whether the team is deleted
 * @member {module:model/NumberBoolean} deleted_flag
 */

UpdateTeamAllOf.prototype['deleted_flag'] = undefined;
var _default = UpdateTeamAllOf;
exports["default"] = _default;
