// pipedrive/dist/model/UserSettingsItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserSettingsItem model module.
 * @module model/UserSettingsItem
 * @version 1.0.0
 */
var UserSettingsItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserSettingsItem</code>.
   * @alias module:model/UserSettingsItem
   */
  function UserSettingsItem() {
    (0, _classCallCheck2["default"])(this, UserSettingsItem);
    UserSettingsItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserSettingsItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserSettingsItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserSettingsItem} obj Optional instance to populate.
     * @return {module:model/UserSettingsItem} The populated <code>UserSettingsItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserSettingsItem();

        if (data.hasOwnProperty('marketplace_allow_custom_install_url')) {
          obj['marketplace_allow_custom_install_url'] = _ApiClient["default"].convertToType(data['marketplace_allow_custom_install_url'], 'Boolean');
          delete data['marketplace_allow_custom_install_url'];
        }

        if (data.hasOwnProperty('marketplace_app_extensions_vendor')) {
          obj['marketplace_app_extensions_vendor'] = _ApiClient["default"].convertToType(data['marketplace_app_extensions_vendor'], 'Boolean');
          delete data['marketplace_app_extensions_vendor'];
        }

        if (data.hasOwnProperty('marketplace_team')) {
          obj['marketplace_team'] = _ApiClient["default"].convertToType(data['marketplace_team'], 'Boolean');
          delete data['marketplace_team'];
        }

        if (data.hasOwnProperty('list_limit')) {
          obj['list_limit'] = _ApiClient["default"].convertToType(data['list_limit'], 'Number');
          delete data['list_limit'];
        }

        if (data.hasOwnProperty('beta_app')) {
          obj['beta_app'] = _ApiClient["default"].convertToType(data['beta_app'], 'Boolean');
          delete data['beta_app'];
        }

        if (data.hasOwnProperty('prevent_salesphone_callto_override')) {
          obj['prevent_salesphone_callto_override'] = _ApiClient["default"].convertToType(data['prevent_salesphone_callto_override'], 'Boolean');
          delete data['prevent_salesphone_callto_override'];
        }

        if (data.hasOwnProperty('file_upload_destination')) {
          obj['file_upload_destination'] = _ApiClient["default"].convertToType(data['file_upload_destination'], 'String');
          delete data['file_upload_destination'];
        }

        if (data.hasOwnProperty('callto_link_syntax')) {
          obj['callto_link_syntax'] = _ApiClient["default"].convertToType(data['callto_link_syntax'], 'String');
          delete data['callto_link_syntax'];
        }

        if (data.hasOwnProperty('autofill_deal_expected_close_date')) {
          obj['autofill_deal_expected_close_date'] = _ApiClient["default"].convertToType(data['autofill_deal_expected_close_date'], 'Boolean');
          delete data['autofill_deal_expected_close_date'];
        }

        if (data.hasOwnProperty('person_duplicate_condition')) {
          obj['person_duplicate_condition'] = _ApiClient["default"].convertToType(data['person_duplicate_condition'], 'String');
          delete data['person_duplicate_condition'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserSettingsItem;
}();
/**
 * If the vendors are allowed to install custom Marketplace apps with functionality and content in the Pipedrive UI or not
 * @member {Boolean} marketplace_allow_custom_install_url
 */


UserSettingsItem.prototype['marketplace_allow_custom_install_url'] = undefined;
/**
 * If the vendors are allowed to extend their Marketplace apps with functionality and content in the Pipedrive UI or not
 * @member {Boolean} marketplace_app_extensions_vendor
 */

UserSettingsItem.prototype['marketplace_app_extensions_vendor'] = undefined;
/**
 * If the vendors are allowed to be part of the Marketplace team or not
 * @member {Boolean} marketplace_team
 */

UserSettingsItem.prototype['marketplace_team'] = undefined;
/**
 * The number of results shown in list by default
 * @member {Number} list_limit
 */

UserSettingsItem.prototype['list_limit'] = undefined;
/**
 * Whether beta app is enabled
 * @member {Boolean} beta_app
 */

UserSettingsItem.prototype['beta_app'] = undefined;
/**
 * Prevent salesphone call to override
 * @member {Boolean} prevent_salesphone_callto_override
 */

UserSettingsItem.prototype['prevent_salesphone_callto_override'] = undefined;
/**
 * The destination of file upload
 * @member {String} file_upload_destination
 */

UserSettingsItem.prototype['file_upload_destination'] = undefined;
/**
 * The call to link syntax
 * @member {String} callto_link_syntax
 */

UserSettingsItem.prototype['callto_link_syntax'] = undefined;
/**
 * Whether the expected close date of the deal is filled automatically or not
 * @member {Boolean} autofill_deal_expected_close_date
 */

UserSettingsItem.prototype['autofill_deal_expected_close_date'] = undefined;
/**
 * Allow the vendors to duplicate a person
 * @member {String} person_duplicate_condition
 */

UserSettingsItem.prototype['person_duplicate_condition'] = undefined;
var _default = UserSettingsItem;
exports["default"] = _default;
