// pipedrive/dist/model/UserConnectionsAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserConnectionsAllOfData model module.
 * @module model/UserConnectionsAllOfData
 * @version 1.0.0
 */
var UserConnectionsAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserConnectionsAllOfData</code>.
   * The object of UserConnections
   * @alias module:model/UserConnectionsAllOfData
   */
  function UserConnectionsAllOfData() {
    (0, _classCallCheck2["default"])(this, UserConnectionsAllOfData);
    UserConnectionsAllOfData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserConnectionsAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserConnectionsAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserConnectionsAllOfData} obj Optional instance to populate.
     * @return {module:model/UserConnectionsAllOfData} The populated <code>UserConnectionsAllOfData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserConnectionsAllOfData();

        if (data.hasOwnProperty('google')) {
          obj['google'] = _ApiClient["default"].convertToType(data['google'], 'String');
          delete data['google'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserConnectionsAllOfData;
}();
/**
 * The third party ID or false in case the ID is not found
 * @member {String} google
 */


UserConnectionsAllOfData.prototype['google'] = undefined;
var _default = UserConnectionsAllOfData;
exports["default"] = _default;
