// pipedrive/dist/model/UserAssignmentToPermissionSet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserAssignmentToPermissionSet model module.
 * @module model/UserAssignmentToPermissionSet
 * @version 1.0.0
 */
var UserAssignmentToPermissionSet = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserAssignmentToPermissionSet</code>.
   * @alias module:model/UserAssignmentToPermissionSet
   */
  function UserAssignmentToPermissionSet() {
    (0, _classCallCheck2["default"])(this, UserAssignmentToPermissionSet);
    UserAssignmentToPermissionSet.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserAssignmentToPermissionSet, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserAssignmentToPermissionSet</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserAssignmentToPermissionSet} obj Optional instance to populate.
     * @return {module:model/UserAssignmentToPermissionSet} The populated <code>UserAssignmentToPermissionSet</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserAssignmentToPermissionSet();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }

        if (data.hasOwnProperty('permission_set_id')) {
          obj['permission_set_id'] = _ApiClient["default"].convertToType(data['permission_set_id'], 'Number');
          delete data['permission_set_id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserAssignmentToPermissionSet;
}();
/**
 * The ID of the assignment of the permission set
 * @member {Number} id
 */


UserAssignmentToPermissionSet.prototype['id'] = undefined;
/**
 * The ID of the user in the permission set
 * @member {Number} user_id
 */

UserAssignmentToPermissionSet.prototype['user_id'] = undefined;
/**
 * The ID of the permission set
 * @member {Number} permission_set_id
 */

UserAssignmentToPermissionSet.prototype['permission_set_id'] = undefined;
/**
 * The name of the permission set
 * @member {String} name
 */

UserAssignmentToPermissionSet.prototype['name'] = undefined;
var _default = UserAssignmentToPermissionSet;
exports["default"] = _default;
