// pipedrive/dist/model/UserAssignmentsToPermissionSet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _UserAssignmentToPermissionSet = _interopRequireDefault(require("./UserAssignmentToPermissionSet"));

var _UserAssignmentsToPermissionSetAllOf = _interopRequireDefault(require("./UserAssignmentsToPermissionSetAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserAssignmentsToPermissionSet model module.
 * @module model/UserAssignmentsToPermissionSet
 * @version 1.0.0
 */
var UserAssignmentsToPermissionSet = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserAssignmentsToPermissionSet</code>.
   * @alias module:model/UserAssignmentsToPermissionSet
   * @implements module:model/BaseResponse
   * @implements module:model/UserAssignmentsToPermissionSetAllOf
   */
  function UserAssignmentsToPermissionSet() {
    (0, _classCallCheck2["default"])(this, UserAssignmentsToPermissionSet);

    _BaseResponse["default"].initialize(this);

    _UserAssignmentsToPermissionSetAllOf["default"].initialize(this);

    UserAssignmentsToPermissionSet.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserAssignmentsToPermissionSet, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserAssignmentsToPermissionSet</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserAssignmentsToPermissionSet} obj Optional instance to populate.
     * @return {module:model/UserAssignmentsToPermissionSet} The populated <code>UserAssignmentsToPermissionSet</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserAssignmentsToPermissionSet();

        _BaseResponse["default"].constructFromObject(data, obj);

        _UserAssignmentsToPermissionSetAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_UserAssignmentToPermissionSet["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserAssignmentsToPermissionSet;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


UserAssignmentsToPermissionSet.prototype['success'] = undefined;
/**
 * An array of the assignments of the user
 * @member {Array.<module:model/UserAssignmentToPermissionSet>} data
 */

UserAssignmentsToPermissionSet.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement UserAssignmentsToPermissionSetAllOf interface:

/**
 * An array of the assignments of the user
 * @member {Array.<module:model/UserAssignmentToPermissionSet>} data
 */

_UserAssignmentsToPermissionSetAllOf["default"].prototype['data'] = undefined;
var _default = UserAssignmentsToPermissionSet;
exports["default"] = _default;
