// pipedrive/dist/model/UserPermissionsAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UserPermissionsItem = _interopRequireDefault(require("./UserPermissionsItem"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserPermissionsAllOf model module.
 * @module model/UserPermissionsAllOf
 * @version 1.0.0
 */
var UserPermissionsAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserPermissionsAllOf</code>.
   * @alias module:model/UserPermissionsAllOf
   */
  function UserPermissionsAllOf() {
    (0, _classCallCheck2["default"])(this, UserPermissionsAllOf);
    UserPermissionsAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserPermissionsAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserPermissionsAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserPermissionsAllOf} obj Optional instance to populate.
     * @return {module:model/UserPermissionsAllOf} The populated <code>UserPermissionsAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserPermissionsAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _UserPermissionsItem["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserPermissionsAllOf;
}();
/**
 * @member {module:model/UserPermissionsItem} data
 */


UserPermissionsAllOf.prototype['data'] = undefined;
var _default = UserPermissionsAllOf;
exports["default"] = _default;
