// pipedrive/dist/model/UserIDs.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _UserIDsAllOf = _interopRequireDefault(require("./UserIDsAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The UserIDs model module.
 * @module model/UserIDs
 * @version 1.0.0
 */
var UserIDs = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserIDs</code>.
   * @alias module:model/UserIDs
   * @implements module:model/BaseResponse
   * @implements module:model/UserIDsAllOf
   */
  function UserIDs() {
    (0, _classCallCheck2["default"])(this, UserIDs);

    _BaseResponse["default"].initialize(this);

    _UserIDsAllOf["default"].initialize(this);

    UserIDs.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(UserIDs, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserIDs</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserIDs} obj Optional instance to populate.
     * @return {module:model/UserIDs} The populated <code>UserIDs</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserIDs();

        _BaseResponse["default"].constructFromObject(data, obj);

        _UserIDsAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], ['Number']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return UserIDs;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


UserIDs.prototype['success'] = undefined;
/**
 * The list of user IDs
 * @member {Array.<Number>} data
 */

UserIDs.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement UserIDsAllOf interface:

/**
 * The list of user IDs
 * @member {Array.<Number>} data
 */

_UserIDsAllOf["default"].prototype['data'] = undefined;
var _default = UserIDs;
exports["default"] = _default;
