// pipedrive/dist/model/WebhooksDeleteForbiddenSchema.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _WebhooksDeleteForbiddenSchemaAllOf = _interopRequireDefault(require("./WebhooksDeleteForbiddenSchemaAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The WebhooksDeleteForbiddenSchema model module.
 * @module model/WebhooksDeleteForbiddenSchema
 * @version 1.0.0
 */
var WebhooksDeleteForbiddenSchema = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhooksDeleteForbiddenSchema</code>.
   * @alias module:model/WebhooksDeleteForbiddenSchema
   * @implements module:model/BaseResponse
   * @implements module:model/WebhooksDeleteForbiddenSchemaAllOf
   */
  function WebhooksDeleteForbiddenSchema() {
    (0, _classCallCheck2["default"])(this, WebhooksDeleteForbiddenSchema);

    _BaseResponse["default"].initialize(this);

    _WebhooksDeleteForbiddenSchemaAllOf["default"].initialize(this);

    WebhooksDeleteForbiddenSchema.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(WebhooksDeleteForbiddenSchema, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebhooksDeleteForbiddenSchema</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebhooksDeleteForbiddenSchema} obj Optional instance to populate.
     * @return {module:model/WebhooksDeleteForbiddenSchema} The populated <code>WebhooksDeleteForbiddenSchema</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhooksDeleteForbiddenSchema();

        _BaseResponse["default"].constructFromObject(data, obj);

        _WebhooksDeleteForbiddenSchemaAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
          delete data['message'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return WebhooksDeleteForbiddenSchema;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


WebhooksDeleteForbiddenSchema.prototype['success'] = undefined;
/**
 * The error message
 * @member {String} message
 */

WebhooksDeleteForbiddenSchema.prototype['message'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement WebhooksDeleteForbiddenSchemaAllOf interface:

/**
 * The error message
 * @member {String} message
 */

_WebhooksDeleteForbiddenSchemaAllOf["default"].prototype['message'] = undefined;
var _default = WebhooksDeleteForbiddenSchema;
exports["default"] = _default;
