// pipedrive/dist/model/WebhookBadRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponseWithStatus = _interopRequireDefault(require("./BaseResponseWithStatus"));

var _WebhookBadRequestAllOf = _interopRequireDefault(require("./WebhookBadRequestAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The WebhookBadRequest model module.
 * @module model/WebhookBadRequest
 * @version 1.0.0
 */
var WebhookBadRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhookBadRequest</code>.
   * @alias module:model/WebhookBadRequest
   * @implements module:model/BaseResponseWithStatus
   * @implements module:model/WebhookBadRequestAllOf
   */
  function WebhookBadRequest() {
    (0, _classCallCheck2["default"])(this, WebhookBadRequest);

    _BaseResponseWithStatus["default"].initialize(this);

    _WebhookBadRequestAllOf["default"].initialize(this);

    WebhookBadRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(WebhookBadRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebhookBadRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebhookBadRequest} obj Optional instance to populate.
     * @return {module:model/WebhookBadRequest} The populated <code>WebhookBadRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookBadRequest();

        _BaseResponseWithStatus["default"].constructFromObject(data, obj);

        _WebhookBadRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], Object);
          delete data['errors'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return WebhookBadRequest;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


WebhookBadRequest.prototype['success'] = undefined;
/**
 * The status of the response
 * @member {String} status
 */

WebhookBadRequest.prototype['status'] = undefined;
/**
 * List of errors
 * @member {Object} errors
 */

WebhookBadRequest.prototype['errors'] = undefined; // Implement BaseResponseWithStatus interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponseWithStatus["default"].prototype['success'] = undefined;
/**
 * The status of the response
 * @member {String} status
 */

_BaseResponseWithStatus["default"].prototype['status'] = undefined; // Implement WebhookBadRequestAllOf interface:

/**
 * List of errors
 * @member {Object} errors
 */

_WebhookBadRequestAllOf["default"].prototype['errors'] = undefined;
var _default = WebhookBadRequest;
exports["default"] = _default;
