// pipedrive/dist/model/WebhookAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseWebhook = _interopRequireDefault(require("./BaseWebhook"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The WebhookAllOf model module.
 * @module model/WebhookAllOf
 * @version 1.0.0
 */
var WebhookAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhookAllOf</code>.
   * @alias module:model/WebhookAllOf
   */
  function WebhookAllOf() {
    (0, _classCallCheck2["default"])(this, WebhookAllOf);
    WebhookAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(WebhookAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebhookAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebhookAllOf} obj Optional instance to populate.
     * @return {module:model/WebhookAllOf} The populated <code>WebhookAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookAllOf();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _BaseWebhook["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return WebhookAllOf;
}();
/**
 * @member {module:model/BaseWebhook} data
 */


WebhookAllOf.prototype['data'] = undefined;
var _default = WebhookAllOf;
exports["default"] = _default;
