// pipedrive/dist/model/WebhookBadRequestAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The WebhookBadRequestAllOf model module.
 * @module model/WebhookBadRequestAllOf
 * @version 1.0.0
 */
var WebhookBadRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhookBadRequestAllOf</code>.
   * @alias module:model/WebhookBadRequestAllOf
   */
  function WebhookBadRequestAllOf() {
    (0, _classCallCheck2["default"])(this, WebhookBadRequestAllOf);
    WebhookBadRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(WebhookBadRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebhookBadRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebhookBadRequestAllOf} obj Optional instance to populate.
     * @return {module:model/WebhookBadRequestAllOf} The populated <code>WebhookBadRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookBadRequestAllOf();

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], Object);
          delete data['errors'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return WebhookBadRequestAllOf;
}();
/**
 * List of errors
 * @member {Object} errors
 */


WebhookBadRequestAllOf.prototype['errors'] = undefined;
var _default = WebhookBadRequestAllOf;
exports["default"] = _default;
