// pipedrive/dist/model/Webhooks.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponseWithStatus = _interopRequireDefault(require("./BaseResponseWithStatus"));

var _BaseWebhook = _interopRequireDefault(require("./BaseWebhook"));

var _WebhooksAllOf = _interopRequireDefault(require("./WebhooksAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The Webhooks model module.
 * @module model/Webhooks
 * @version 1.0.0
 */
var Webhooks = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Webhooks</code>.
   * @alias module:model/Webhooks
   * @implements module:model/BaseResponseWithStatus
   * @implements module:model/WebhooksAllOf
   */
  function Webhooks() {
    (0, _classCallCheck2["default"])(this, Webhooks);

    _BaseResponseWithStatus["default"].initialize(this);

    _WebhooksAllOf["default"].initialize(this);

    Webhooks.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(Webhooks, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Webhooks</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Webhooks} obj Optional instance to populate.
     * @return {module:model/Webhooks} The populated <code>Webhooks</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Webhooks();

        _BaseResponseWithStatus["default"].constructFromObject(data, obj);

        _WebhooksAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
          delete data['status'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_BaseWebhook["default"]]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return Webhooks;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


Webhooks.prototype['success'] = undefined;
/**
 * The status of the response
 * @member {String} status
 */

Webhooks.prototype['status'] = undefined;
/**
 * The array of Webhooks
 * @member {Array.<module:model/BaseWebhook>} data
 */

Webhooks.prototype['data'] = undefined; // Implement BaseResponseWithStatus interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponseWithStatus["default"].prototype['success'] = undefined;
/**
 * The status of the response
 * @member {String} status
 */

_BaseResponseWithStatus["default"].prototype['status'] = undefined; // Implement WebhooksAllOf interface:

/**
 * The array of Webhooks
 * @member {Array.<module:model/BaseWebhook>} data
 */

_WebhooksAllOf["default"].prototype['data'] = undefined;
var _default = Webhooks;
exports["default"] = _default;
