// pipedrive/dist/api/GlobalMessagesApi.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GlobalMessageDelete = _interopRequireDefault(require("../model/GlobalMessageDelete"));

var _GlobalMessageGet = _interopRequireDefault(require("../model/GlobalMessageGet"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

/**
* GlobalMessages service.
* @module api/GlobalMessagesApi
* @version 1.0.0
*/
var GlobalMessagesApi = /*#__PURE__*/function () {
  /**
  * Constructs a new GlobalMessagesApi. 
  * @alias module:api/GlobalMessagesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function GlobalMessagesApi(apiClient) {
    (0, _classCallCheck2["default"])(this, GlobalMessagesApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Dismiss a global message
   * Removes a global message from being shown if the message is dismissible.
   * @param {Number} id The ID of the global message to be dismissed
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/GlobalMessageDelete} and HTTP response
   */


  (0, _createClass2["default"])(GlobalMessagesApi, [{
    key: "deleteGlobalMessageWithHttpInfo",
    value: function deleteGlobalMessageWithHttpInfo(id) {
      var opts = {};
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling deleteGlobalMessage");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var formParamArray = [];
      var contentTypes = [];
      var isURLEncoded = contentTypes.includes('application/x-www-form-urlencoded');
      var isJSON = contentTypes.includes('application/json');

      if (isJSON) {
        postBody = _objectSpread(_objectSpread({}, postBody), opts);
      } else if (isURLEncoded) {
        for (var key in opts) {
          if (opts.hasOwnProperty(key) && !formParamArray.includes(key)) {
            formParams[key] = opts[key];
          }
        }
      }

      var authNames = ['api_key'];
      var accepts = ['application/json'];
      var returnType = _GlobalMessageDelete["default"];
      return this.apiClient.callApi('/globalMessages/{id}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * Dismiss a global message
     * Removes a global message from being shown if the message is dismissible.
     * @param {Number} id The ID of the global message to be dismissed
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/GlobalMessageDelete}
     */

  }, {
    key: "deleteGlobalMessage",
    value: function deleteGlobalMessage(id) {
      return this.deleteGlobalMessageWithHttpInfo(id).then(function (response_and_data) {
        return response_and_data;
      });
    }
    /**
     * Get global messages
     * Returns data about the global messages to display for the authorized user.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.limit The number of messages to get from 1 to 100. The message number 1 is returned by default. (default to 1)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/GlobalMessageGet} and HTTP response
     */

  }, {
    key: "getGlobalMessagesWithHttpInfo",
    value: function getGlobalMessagesWithHttpInfo(opts) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'limit': opts['limit']
      };
      var headerParams = {};
      var formParams = {};
      var formParamArray = [];
      var contentTypes = [];
      var isURLEncoded = contentTypes.includes('application/x-www-form-urlencoded');
      var isJSON = contentTypes.includes('application/json');

      if (isJSON) {
        postBody = _objectSpread(_objectSpread({}, postBody), opts);
      } else if (isURLEncoded) {
        for (var key in opts) {
          if (opts.hasOwnProperty(key) && !formParamArray.includes(key)) {
            formParams[key] = opts[key];
          }
        }
      }

      var authNames = ['api_key'];
      var accepts = ['application/json'];
      var returnType = _GlobalMessageGet["default"];
      return this.apiClient.callApi('/globalMessages', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * Get global messages
     * Returns data about the global messages to display for the authorized user.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.limit The number of messages to get from 1 to 100. The message number 1 is returned by default. (default to 1)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/GlobalMessageGet}
     */

  }, {
    key: "getGlobalMessages",
    value: function getGlobalMessages(opts) {
      return this.getGlobalMessagesWithHttpInfo(opts).then(function (response_and_data) {
        return response_and_data;
      });
    }
  }]);
  return GlobalMessagesApi;
}();

exports["default"] = GlobalMessagesApi;
